<HTML> <HEAD> </HEAD>
<TITLE>MMC Monte Carlo program guide</TITLE>
<BODY BGCOLOR="#FFFFFF">
<CENTER>Users' Guide to the Monte Carlo Program MMC</CENTER>
<p>
<CENTER>for Computer Simulation of Molecular Solutions.</CENTER>
<p>
<CENTER>Mihaly Mezei</CENTER>
<p>
<CENTER>Department of Pharmacological Sciences</CENTER>
<p>
<CENTER>Icahn School of Medicine at Mount Sinai</CENTER>
<p>
<CENTER>E-mail address: <A HREF="mailto:Mihaly.Mezei@mssm.edu">
Mihaly.Mezei@mssm.edu</A></CENTER>
<p>
<CENTER>April 04, 2022</CENTER><BR>
<p>
<CENTER><A NAME="Contents">TABLE OF CONTENTS<BR></A></CENTER>
<UL>
<LI>I.&nbsp;
<A HREF="#General">General description</A><BR>
<UL>
<LI>I.1.&nbsp;<A HREF="#General_systems">Systems modeled</A><BR>
<LI>I.2.&nbsp;<A HREF="#General_properties">Properties
calculated during a simulation</A><BR>
<LI>I.3.&nbsp;<A HREF="#General_potential">Potential
functions</A><BR>
<LI>I.4.&nbsp;<A HREF="#General_sampling">Sampling
techniques</A><BR>
<LI>I.5.&nbsp;<A HREF="#General_free">Free energy methods</A><BR>
<LI>I.6.&nbsp;<A HREF="#General_solute">Solute move
selection</A><BR>
<LI>I.7.&nbsp;<A HREF="#General_initial">Initial
configuration</A><BR>
<LI>I.8.&nbsp;<A HREF="#General_information">Information needed
for a simulation</A><BR>
<LI>I.9.&nbsp;<A HREF="#General_self">Self test</A><BR>
<LI>I.10.&nbsp;<A HREF="#General_anal">General analysis</A><BR>
</UL>
<LI>II.&nbsp;<A HREF="#File_org">File Organization</A><BR>
<LI>III.&nbsp;<A HREF="#Input">Input description</A><BR>
<UL>
<LI>III.01. <A HREF="#Input_FILES">File handling keywords</A><BR>
<LI>III.02. <A HREF="#Input_SYSTEM">System descriptor
keywords</A><BR>
<LI>III.03. <A HREF="#Input_FREE">Free-energy calculation
keywords</A><BR>
<LI>III.04. <A HREF="#Input_MOLEC">Molecule descriptor
keywords</A><BR>
<LI>III.05. <A HREF="#Input_PF">Potential descriptor
keywords</A><BR>
<LI>III.06. <A HREF="#Input_SAMP">Sampling-related keywords</A><BR>
<LI>III.07. <A HREF="#Input_FLOW">Calculation flow keywords</A><BR>
<LI>III.08. <A HREF="#Input_OUTPUT">Output related keywords</A><BR>
<LI>III.09. <A HREF="#Input_ANAL">General analysis keywords</A><BR>
<LI>III.10. <A HREF="#Input_PROXI">Proximity analysis
keywords</A><BR>
<LI>III.11. <A HREF="#Input_NUM">Numerical precision related
keywords</A><BR>
<LI>III.12. <A HREF="#Input_DEVELOP">Developpers' keywords</A><BR>
<LI>III.13. <A HREF="#Input_SUMMARY">Keyword summary</A><BR>
</UL>
<LI>IV.&nbsp;<A HREF="#Examples">Examples</A><BR>
<LI>V.&nbsp;<A HREF="#Brief">Brief description of the
output</A><BR>
<LI>VI.&nbsp;<A HREF="#Compilation">Compilation of the
program</A><BR>
<LI>VIII.&nbsp;<A HREF="#Testing">Testing</A><BR>
<LI>IX.&nbsp;<A HREF="#References">References</A> <BR>
</UL>
<p>
<!-- HardPage --><p>
<A NAME="General">I</A>. GENERAL DESCRIPTION
<p>
The present document describes the Force-Biased Metropolis
Monte Carlo program for computer simulation studies of molecular
liquids and solutions in the canonical, isobaric-isothermal
and grand-canonical
ensembles and the analysis of such calculations based on the
<A HREF="#Mehrotra">Proximity Criterion</A>.
This section provides an overview of the
capabilities of the program.
<p>
<A NAME="General_systems">I.1.</A><b> Systems modeled</b>
<p>
The present program was originally written for simulations on
infinitely dilute solutions (one solute in (<B>numsol</B>)
solvent molecules).
As a result of this origin, the program has developed two
different treatments of molecules.
<i>Solvent molecules have to be rigid and be of the same type</i>
(this allows the use of special, rapid energy calculation code).
The collection of <i>all other molecules</i> (i.e., the one(s) not
declared to be solvents) <i>are called by the program the
solute.</i>
Thus the solute can consist of a single rigid
molecule, a collection of freely translating and rotating rigid
molecules or a collection of freely translating and rotating
molecules with certain (user specified) torsion angles freely
rotating.
<p>
The program establishes the <i>connectivity</i> of the solute
molecule(s)
based on interatomic distances and built-in threshold values.
By default H-H bonds and bonds between an H and a heavy atom on different
groups (residues) are ignored, but this can be changed
(see the keys <B><A HREF="#BDHH">BDHH</A></B> and
<B><A HREF="#BDRH">BDRH</A></B>).
Also, the bond thresholds can be changed with the
<B><A HREF="#MODA">MODA</A></B> key.
This sometimes leads to missing or extra bonds, especally if the
input structure (given by the <B><A HREF="#SLTA">SLTA</A></B> key)
is obtained from a molecular dynamics simulation.
Such defects can be identified by using the
<B><A HREF="#FCGA">FCGA</A></B> and
<B><A HREF="#BNDL">BNDL</A></B> keys and corrected by the keys
<B><A HREF="#MAKB">MAKB</A></B> or
<B><A HREF="#BRKB">BRKB</A></B> or forced to be ignored (ok for molecules
without active torsions) with the <B><A HREF="#MOLD">MOLD</A></B> key.
<p>
<i>Free energy simulations</i> can be performed on selected
molecules
of the solute (see Sec. <A HREF="#General_free">I.5.</A>).
<p>
<i>The torsion angles</i> to be changed on the solute are specified
by the two atoms defining the bond of the torsion
(with the key <B><A HREF="#TORD">TORD</A></B>).
The torsions can be grouped and
it is the group of torsions that is selected for change. The main
reason for this is that if a torsion affects atoms that are on an
other rotatable bond then it may be efficient to move the
affected bond as well. These moves, however, are not correlated
(i.e., each is generated by a different random number).
There is also an option of changing the torsion angles of a
loop segment (with the key <B><A HREF="#LOOP">LOOP</A></B>).
The torsion potential parameters can be obtained from a CHARMM or
AMBER parameter file.
<p>
It is possible to impose <i>periodicity on the solute</i>,
either along the X axis (with the key <B><A HREF="#SLTA">SLTA</A>
POLY</B>)
i.e., with period equal to the edge of the cell in the X direction
- an infinite polymer,
or in all three directions
(with the key <B><A HREF="#SLTA">SLTA</A> XTAL</B>) with its
unit cell (or a multiple of a unit cell) coinciding with the
periodic cell of the simulation (i.e., a crystal).
The periodic extension of the solute outside the
periodic cell is done automatically by the program.
<p>
The intermolecular interactions are assumed to be of the usual 12-6-1 type,
i.e., Lennard-Jones terms for the repulsion and dispersion plus
electrostatics based on partial charges on atoms (or pseudo atoms).
The parameters can be obtained from libraries stores by the program
or defined via input.
The interactions can be truncated
(see keys <B><A HREF="#SVVC">SVVC</A></B>,
<B><A HREF="#SUVC">SUVC</A></B>, <B><A HREF="#SUUC">SUUC</A></B>)
either by a spherical cutoff
(each atom interacts with all others within a certain distance)
or by the minimum image convention
(each atom interacts with one periodic image of all the other atoms).
For the purpose of cutoff or image calculation, distances usually refer
to molecule-molecule or residue-residue distances.
<p>
When spherical truncation is used, the long-range electrostatic energy
can be estimated by the reaction field formula
(provided all residues are neutral) - see key
<B><A HREF="#RFCR">RFCR</A></B>.
<p>
<A NAME="General_properties">I.2.</A> <b>Properties
calculated during a simulation</b>
<p>
The major outputs of a Monte Carlo simulation (besides the
configurations saved for further analysis by the
<B><A HREF="#TRAJ">TRAJ</A></B> key) are as follows.
<UL>
<LI>Macroscopic thermodynamic properties of the system:
internal energy, pressure, heat capacity (constant-volume
or pressure, depending on the ensemble used), the density far from
the solute (see key <B><A HREF="#MOND">MOND</A></B>). Simulations
in the grand-canonical ensemble (see key <B><A
HREF="#GCEN">GCEN</A></B>) also provide the density, excess
chemical potential, isothermal compressibility, and
expansivity. Simulations in the isothermal isobaric ensemble
(see key <B><A HREF="#IBEN">IBEN</A></B>) also provide the density
and the thermal expansion coefficient.
<LI>Several methods to calculate free energy have been
implemented (see key <B><A HREF="#FREE">FREE</A></B>) by relating
it to either some computed energy values or to some distributions.
<LI>Bulk solvent-solvent and solute-solvent distribution functions
(see key <B><A HREF="#DSTC">DSTC</A></B>):
radial distribution function based on center of mass (COM),
quasi-component distribution functions for coordination number,
binding energy and pair energy, and dipole correlation functions.
<LI>The derivative of the solute and total energy with respect
to the solute potential parameters can also be calculated as an
ensemble average (see key <B><A HREF="#SENS">SENS</A></B>). This
may be used for the development of
potential parameters and for the sensitivity of the calculated
energy to the potential parameters.
<LI>Calculation of the mean forces exerted by the solvent on the solute atoms
(see key <B><A HREF="#SENS">SENS</A></B>) and the electric field
gradients at the solute and solvent sites (see key
<B><A HREF="#FLDG">FLDG</A></B>) can also be requested.
<LI>For systems with variable torsion angles the torsion
angle distribution can be calculated
(see key <B><A HREF="#TAND">TAND</A></B>)
<LI>Proximity analysis, i.e., distributions based on the
<A HREF="#Mehrotra">original form</A> of the Proximity Criterion
or its <A HREF="#Mezei_88">enhanced version</A> can also be
performed. This analysis partitions the solvent according to the
proximal solute atoms and calculates distributions for each
partition.
The solute atoms are partitioned into functional
groups either defined by their chemical identity
(see key <B><A HREF="#FCGA">FCGA</A></B>) or into alternative
functional groups specified by the user
(see key <B><A HREF="#FCGD">FCGD</A></B>).
The proximity analysis results are presented for individual solute
atoms, functional groups, and solute residues (see key
<B><A HREF="#SLTA">SLTA</A></B>) and also are
averaged over functional groups of the same kind.
</UL>
<A NAME="General_potential">I.3.</A> <b>Potential
functions</b>
<p>
Several potential functions are built into the program. For
the solute-solvent nonbonded interactions
(specified with the <B><A HREF="#SUPT">SUPT</A></B> key).
Clementi et al.'s
library, EPEN/QPEN, Kollman et al.'s libraries (AMBER-94 and
AMBER-2002), CHARMM22 library, Berendsen-Van Gunsteren library
(GROMOS87 and GROMOS96);
and Jorgensen's OPLS library.
Note that for the GROMOS force field only the combination
rule can be used - the non-bonded parameters for pairs
are not implemented.
Except for the EPEN form, parameters are stored internally, but
they can be modified from input and allowance is made for
inputting new atom types besides the stored ones. As these
parameter libraries are continually updated and extended, it is
advisable to check if the value stored is the best one
(using the <B><A HREF="#PLBP">PLBP</A></B> key).
For aqueous solutions, the solvent-solvent interaction
(specified with the <B><A HREF="#SUPT">SUPT</A></B> key)
can be either
in the form of MCY-CI type or TIPS/TIP3P/SPC type or
TIPS2/TIP4PB-F type or TIP5P type potential. For the MCY-CI type the
parameters of both the MCY-CI-II and the Yoon-Morokuma-Davidson
parametrizations are built in. For TIPS2/TIP4P/B-F types, the A, B
and q parameters may also be provided by the user, but the TIPS,
SPC, TIP3P, Bernal-Fowler, TIPS2, TIP4P and TIP5P water model parameters
are built in. For solvent other than water, 12-6-1 type potential
can be used with parameters from the AMBER, GROMOS or OPLS
libraries.
<p>
There is a framework for adding field-dependent potentials to the
currently implemented atom-atom potentials, but this requires some
additional coding.
<p>
<A NAME="General_sampling">I.4.</A> <b>Sampling techniques</b>
<p>
Various sampling techniques can be selected with the
<B><A HREF="#MOVE">MOVE</A></B> key.
Translations and rotations can be sampled with the original
<A HREF="#Metropolis">Metropolis Method</A> or with the
Force-Biased algorithm of
<A HREF="#Pangali">Pangali</A>, Rao and Berne (not implemented for
multimolecular solute moves, though).
Force biasing, of course, necessitates the
recalculation of energies and forces at every accepted step,
thereby significantly increasing the computer time required
to perform a move. The improvement in convergence, however, more
than outweighs the added cost. By default a lambda value of
<A HREF="#Rossky">1/2</A>
is used but the scaled force-bias technique where lambda is
<A HREF="#Mezei_90">scaled down near the solute</A> has also been
implemented.
Novel torsion-angle sampling techniques include the
<A HREF="#EXBIref">extension-biased torsion moves</a> and
the
<A HREF="#LOOPref">local moves with the reverse proximity criterion</a>.
<p>
Molecules to be moved can be selected randomly, in a cycle
or in a
<A HREF="#Mezei_81">shuffled cycle</A> or using the preferential
sampling scheme of
<A HREF="#Owicki">J.C. Owicki</A>, which leads to improved
statistics for the local solution environment of the solute (see
key <B><A HREF="#MOVE">MOVE</A> PRF*</B>).
The <A HREF="#Mezei_87">cavity-biased insertion</A> technique for
grand-canonical ensemble simulations and the
<A HREF="#Mezei_83">virial-biased volume-change</A>
for the isobaric ensemble have also been implemented.
<p>
<A NAME="General_free">I.5.</A> <b>Free energy methods</b>
<p>
Free energy simulations can be performed using a coupling
parameter lambda (CP in the followings) in the energy expression.
There are two
<A HREF="#Mezei_86_NYAS">fundamentally different approaches</A>
to generate a path: (A) simultaneous
creation/annihilation and (B) continuous deformation. In the
first the Hamiltonians of the initial and final states are used
in unchanged and the coupling parameter is used to combine the
two; in the second case there is a single hybrid Hamiltonian that
is based on a parametrization that is a mixture of the two
systems. For multimolecular solute, the free energy simulation
can be applied to just one or two solute molecules, depending on
the type of coupling.
<p>
I.5.A. <i>Creation/annihilation path</i>
<p>
I.5.A.1. Creation/annihilation path to be used in
thermodynamic integration (<B><A HREF="#FREE2">FREE&nbsp;TICA</A></B>).
Here
<p>
E(CP)=CP<sup>k</sup>*E<sub>1</sub>+(1-CP)<sup>k</sup>*E<sub>o</sub>
<p>
where E<sub>o</sub> and E<sub>1</sub> represent the
Hamiltonians initial and final
state, respectively. k=1 represents linear coupling, k&gt;1 has been
referred to as or 'almost linear' coupling (e.g., k=4 for
1/r<sup>12</sup>
repulsion). It is also possible to use different k values for
the different types of energy terms
(1/r<sup>12</sup>, 1/r<sup>6</sup>, and 1/r) with the
so-called <A HREF="#Resat">polynomial TI</A>.
E<sub>o</sub> and E<sub>1</sub> can differ in conformation,
partial charges, and chemical identity (the present version
requires equal number of atoms in the two states - this can
always be achieved with he use of dummy atoms. It is also
possible to provide two more coupling parameter values
CP<sub>o</sub> and CP<sub>1</sub> such that
CP<sub>o</sub> &lt; CP &lt; CP<sub>1</sub>.
The program will compute the free energy
difference between states represented by E(CP<sub>o</sub>) and
E(CP<sub>1</sub>) with the
perturbation method as the sum of two free energy differences:
A<sub>1</sub>-A<sub>o</sub> =
-kT&nbsp;ln&lt;exp((E(CP<sub>1</sub>)-E(CP))/kT)&gt;<sub>CP</sub>
+ kT&nbsp;ln&lt;exp((E(CP<sub>o</sub>)-E(CP))/kT)&gt;<sub>CP</sub>.
&lt;E(CP<sub>o</sub>)&gt;<sub>CPo</sub> and
&lt;E(CP<sub>1</sub>)&gt;<sub>CP1</sub> are also computed for
consistency check.
The two molecules have to contain the same number of residues
(groups), but can have different number of atoms.
<p>
I.5.A.2. Perturbation method based on a sequence of
intermediate states (<B><A HREF="#FREE">FREE&nbsp;TILI</A></B>). This
option performs a
perturbation method calculation between two states generated
using the deformation-type path described below in I.5.B.
The reference state will be
(E(CP<sub>1</sub>)-E(CP<sub>o</sub>))/2. In this case, the
inital and final states have to have the same number of atoms also.
<p>
I.5.B. <i>Deformation path</i>
<p>
I.5.B.1. The program also provides option for a
fundamentally nonlinear coupling where the coupling parameter
changes both the solute coordinates and the potential parameters
to generate intermediate states (continuous deformation).
Besides creating intermediate E<sub>o</sub> and E<sub>1</sub>
states for the linear or
'almost linear' couplings described above in I.5.A.2, this
coupling can be used with the probability ratio method
(<B><A HREF="#FREE5">FREE&nbsp;PMF1</A></B>) or with the overlap ratio
method (<B><A HREF="#FREE">FREE&nbsp;PMNL</A></B>), as
described
below in I.5.B.2 and I.5.B.3, respectively. The initial and final
states may differ in position, partial charge or identity of
atoms (there are limitations for EPEN).
<p>
In each of the following cases, the path involves one or two
molecules of the solute. The following paths are supported:
<p>
I.5.B.1.1. Linear combination of two conformations of a
single molecule (<B><A HREF="#FREE5">FREE&nbsp;PMF1</A> GENL</B>). In
this case all atoms of the molecule may
change positions independently. The configurations at the
beginning and at the end of the path are given on input and the
intermediate configurations are convex linear combinations (i.e.,
R(CP)=CP*R<sub>1</sub>+(1-CP)*R<sub>o</sub>) of these two.
<p>
I.5.B.1.2. Moving two molecules of the solute in a
correlated fashion (<B><A HREF="#FREE5">FREE&nbsp;PMF1</A> WRMM</B>). In
this case, the
intramolecular distances within the two parts should remain
constant. Note, that if the orientation of either part is
different in the initial and final states then at the
intermediate states the molecule will be distorted.
<p>
I.5.B.1.3. Varyig the distance between two solute molecules
of the solute but keeping the one of them fixed
(<B><A HREF="#FREE5">FREE&nbsp;PMF1</A> WRFM</B>).
In this case the configurations at the two ends of the path
differ only in the conformation of the moving part of the solute.
This option may be more economical when one part is large and the
other is small.
<p>
I.5.B.1.4. Moving and rotating one molecule while keeping
the other one fixed (<B><A HREF="#FREE5">FREE&nbsp;PMF1</A> WRFR</B>).
In this case the center of
mass of the first part is moved between the center of masses at
the two endpoints and the orientation of the moving part is also
changed to transform it from the initial to the final
orientation. This option differs from I.5.B.1.3 in that at each
point of the path the moving molecule has the
same intramolecular conformation.
<p>
I.5.B.1.5. Correlated rotation of a set of torsion angles
(<B><A HREF="#FREE5">FREE&nbsp;PMF1</A> TORS</B>). In this case a set of
torsion angles are
changed in a user-specified interval using a common CP parameter:
CP=0 sets all the selected angles to the beginning of their
interval and CP=1 sets all of them to the end of their interval.
<p>
I.5.B.2. The solvent contribution to the free energy change
between two solute conformations can also be determined by
computing the ratio of the Boltzmann probabilities as sampled
along a line in the conformation space of the solute (described
by any of the paths described in Sec. I.5.B.1,
<B><A HREF="#FREE5">FREE&nbsp;PMF1</A> **** *USS</B>).
This approach is called the probability ratio method.
Calculations of this type usually require umbrella sampling
techniques. The program allows either the use of a harmonic
potential to be given from input or the self-consistent
determination of a tabulated weighing function using the
<A HREF="#Mezei_87_AUS">Adaptive Umbrella Sampling Method</A>.
<p>
I.5.B.2. Any of the paths descibed in Sec. I.5.B.1 can be
used to estimate free energy changes simultaneosly using the
overlap ratio method of
<A HREF="#Jacucci">Jacucci and Quirke</A>] and the
perturbation method (<B><A HREF="#FREE6">FREE&nbsp;PMNL</A></B>). Here
the calculations
are run at a given fixed E(CP) and two more coupling parameter
values, CP<sub>o</sub> and CP<sub>1</sub> are specified,
CP<sub>o</sub> &lt; CP &lt; CP<sub>1</sub>. For the overlap
ratio method, the program will calculate the CP averaged
distribution of &lt;E(CP)-E(CP<sub>o</sub>)&gt;<sub>CP</sub>
and &lt;E(CP<sub>1</sub>)-E(CP)&gt;<sub>CP</sub>. Separate
calculations at CP<sub>o</sub> and CP<sub>1</sub> will give the
needed CP<sub>o</sub> and CP<sub>1</sub> averaged
distributions to be used with the overlap ratio method.
Simultaneously, the program also calculates the perturbation
method differences between the states CP<sub>o</sub> and CP as
well as CP and CP<sub>1</sub>. Notice that this kind of
perturbation method calculation is
different from the calculation described with the
creation/annihilation path above: there the middle state E(CP) was
a linear combination of the initial and final states,
E(CP<sub>o</sub>) and E(CP<sub>1</sub>).
For consistency check, (similarly to the perturbation
method calculations described above) the CP<sub>o</sub> average
of E(CP<sub>o</sub>) and
the CP<sub>1</sub> average of E(CP<sub>1</sub>) are also
calculated.
<p>
I.5.C. <i>Chemical potential</i>.
<p>
Free energy can be also obtained from the chemical potential.
The excess chemical potential can be obtained either from
grand-canonical
ensemble simulations (key <B><A HREF="#GCEN">GCEN</A></B>),
from calculating the excess free energy
of changing a molecule to ideal gas (with one of the methods
described
above, e.g., with key <B><A HREF="#FREE2">FREE&nbsp;TICA</A></B>)
or by the Widom insertion method
(key <B><A HREF="#FREE1">FREE&nbsp;WIDO</A></B>).
The program implements a version that is based on
<A HREF="#WIDOref">insertions into cavities</a>.
<p>
<A NAME="General_solute">I.6.</A> <b>Solute move selection</b>
<p>
The sampling of the moves is specified as follows. The
overall frequency of solute moves is specified by input provided
with the <B><A HREF="#STEP">STEP</A></B> key. As discussed above,
the whole solute can be a) rotated and displaced;
b) one constituent molecule of the solute
can be rotated and displaced; c) two constituent molecules can be
swapped; d) a group of torsion angles can be changed; and e) the
coupling parameter used in a potential of mean force calculation
can be changed. If moves of type b, c, d, and e are required,
then the program will request a relative frequency (weight) of making that
move. Regular solute moves and coupling parameter changes are
both given a weight of 1.0 . Thus giving a weight of 2.0 for type
3 moves, out of 4 solute move attempts, 1 will change the solute
position, 1 will change the coupling parameter, and 2 will change
one of the randomly selected torsion angle groups (on the
average, of course). Clearly, the more types of moves on is using,
the more frequently has the solute to be moved.
<p>
For the free energy options, the treatment of moves of type 2
or 3 depends of the type of path. For the creation/annihilation
path using TI (<B><A HREF="#FREE2">FREE&nbsp;TICA</A></B>)
the torsions are independently sampled on
both copies, since these can be, and generally are, different
molecules. For the continuous path
(<B><A HREF="#FREE6">FREE&nbsp;PMF1</A></B> or
<B><A HREF="#FREE6">FREE&nbsp;PMNL</A></B>) or
the cration/annihilation PM (<B><A HREF="#FREE">FREE&nbsp;PMLI</A></B>),
the torsions are treated
identically in each copy since here it is assumed that the
molecule is changing smoothly. This means, that for a
creation/annihilation TI these moves have to be sampled twice as
frequently.
<p>
<A NAME="General_initial">I.7.</A> <b>Initial
configuration</b>
<p>
The starting configuration for the Monte Carlo
(see key <B><A HREF="#CNFG">CNFG</A></B>) is either read from a
previously prepared file, generated from the solute's description
(see key <B><A HREF="#SLTA">SLTA</A></B>) and randomly placed solvents
or from a configuration of another system through various
transformations.
Grand-canonical ensemble simulation
(see key <B><A HREF="#GCEN">GCEN</A></B>) is also an efficient way
of obtaining a random initial configuration.
<p>
<A NAME="General_information">I.8.</A>
<b>Information needed for a simulation</b>
<p>
Several steps in the preparation of the input to and the analysis of
the results form MMC can be helped by the program
<A HREF="https://mezeim01.u.hpc.mssm.edu/simulaid">Simulaid</a>.
The list of its features are described  in a
<A HREF="#SIMref">J. Comp. Chem. paper</a>.
<p>
I.9.1. <B>T</B>, temperature at which the simulations are to be
run (specified with the <B><A HREF="#TEMP">TEMP</A></B> key).
<p>
I.9.2. The coordinates of the nuclei, electrons and
pseudoatoms (if any) in a local coordinate framework, and the
charges associated with these centers for the solute and
(optionally) for the solvent
(specified with the <B><A HREF="#SLTA">SLTA</A></B> key).
The program
<A HREF="https://mezeim01.u.hpc.mssm.edu/simulaid">Simulaid</a>
can help convert structures in different file formats
(e.g., PDB, CHARMM .CRD) to MMC format and can
optimize the orientation of a solute within a periodic box
(to maximize the smallest inter-image distance and thereby minimize
the artifactual effect of the solute images), within a boundig box,
or optimally center a solute in a sphere (for droplet-type
calculations).
<p>
I.9.3. Potential specifications.
For each solute atom the potential type has to be specified from
input (except for EPEN where for each atom and pseudoatom
the A, B, and C parameters have to be specified individually).
The program
<A HREF="https://mezeim01.u.hpc.mssm.edu/simulaid">Simulaid</a>
can help to convert from PDB, CHARMM, AMBER or Macromodel
input files to MMC format.
Several water potentials are implemented with specific codes,
but solvent can be defined similarly to the solute as well.
<p>
I.9.4. <B>numsol</B>, the number of solvent molecules used in the
simulation,
specified with the <B><A HREF="#NSLV">NSLV</A></B> key.
<p>
I.9.5. Size and shape of the periodic cell
(specified with the <B><A HREF="#PBCN">PBCN</A></B> key).
<p>
I.9.6. When generating a new configuration the program can
determine <B>nmolec</B> and the simulation cell size parameters
from the partial molar volumes of the solute and solvent and
the required number of solvent shells around the solute.
<p>
I.9.7. For partial solute displacements the groups of solute
atoms (independent molecules) to be moved separately
(specified with the <B><A HREF="#PARD">PARD</A></B> key).
<p>
I.9.8. For solute torsions
(specified with the <B><A HREF="#TORD">TORD</A></B> key).
the atoms defining the torsion
angle and the torsion potential parameters.
<p>
<A NAME="General_self">I.9.</A>
<b>Self testing</b>
<p>
There is an extensive
<A HREF="#Mezei_86">self testing</A> routine in the program that
tests the integrity/consistency of the data.
It protects against both corruption of data and program errors.
It is invoked at several ways:
<UL>
<LI>Automatically at startup
<LI>Periodically, as specified by the user with the
<B><A HREF="#SLFT">SLFT</A></B> key.
<LI>Optionally at the end of the run, as specified by the user with
the <B><A HREF="#STOP">STOP</A></B> key.
<LI>Automatically at every 2,500,000th step (but not at the last
one).
</UL>
<A NAME="General_anal">I.10. </A><B>General analysis</B>
<P>
Except for the the bulk distributions and most thermodynamic
variables,
all the the properties discussed above in
<A HREF="General_properties">Sec. I.2.</A>, can be calculated from
the history of a simulation as well. Note, that such analyses can
be done
on simulation trajectories generated by other programs, e.g., by
CHARMM or AMBER. In addition to the analyses described above,
several other types of analyses are implemented as summarized
below.
<UL>
<LI><A HREF="#Gensite">Generic solvent sites</a> can be calculated
(see key <B><A HREF="#GENS">GENS</A></B>)
<LI>Two configurations can be compared and the changes in the
solute molecules can be calculated
(see key <B><A HREF="#CMPC">CMPC</A></B>)
<LI>Cavities in the solute can be visualized by a grid technique
(see key <B><A HREF="#PRTG">PRTG</A></B>) and their volume be
estimated (see key <B><A HREF="#STVG">STVG</A></B>)
<LI>The solute-solvent and solute solute energies can be separated
into terms involving each solute atom
(see key <B><A HREF="#ENGL">ENGL</A></B>)
<LI>The distribution of the solvents around the solute can be
visulaized by creating a configuration with the solvent coordinates
from a set of selected configurations
(see key <B><A HREF="#DENF">DENF</A></B>)
<LI>A simulation history can also be filtered according to various
criteria, e.g., energetic, geometric or simply eliminating segments
of the system (see key <B><A HREF="#FILT">FILT</A></B>) </UL>
<!-- HardPage -->
<p>
<A NAME="File_org">II.</A> FILE ORGANIZATION.
<p>
Beside then standard input (Fortran unit 5) - for the information
needed
for the run - and the standard output file (unit 6) - printed
output, diagnostics, described in Section
<A HREF="#Brief">V.</A> - the program uses a number of different
files. In the followings,
writing to the standard output will be refererred to as printing.
<p>
File names are <B>limited to 80 characters</B> and
are structured the following way:
<p><B>&lt;jobname&gt;[.&lt;numrun&gt;][_&lt;version&gt;].&lt;file
type&gt;</B>.
<p>
where
<UL>
<LI><B>&lt;jobname&gt;</B> is any legitimate filename, common to
all files generated by the run (specified by the
<B><A HREF="#FILE">FILE</A></B> key);
<LI><B>&lt;numrun&gt;</B> is a number
greater than 1 and less than 99 (i.e., the omit
the run number from the file name for <B>&lt;numrun&gt;</B>=1
and limit the number of runs to 98);
After each <B><A HREF="#RUNS">RUNS</A></B> command,
<B>&lt;numrun&gt;</B> gets incremented by one.
This allows the execution of several different MC runs
(equilibrations, free energy runs with different coupling
parameters, etc.) within a single job.
Also, <B>&lt;numrun&gt;</B> can be modified
by the <B><A HREF="#FILE">FILE</A></B> command.
<LI><B>&lt;version&gt;</B> is an integer between 2 and 9999
(i.e., the omit
the version number from the file name for <B>&lt;version&gt;</B>=1.
It is used whenever a given runnumber deals with several files of the
same type during a run
(e.g., <B><A HREF="#FILT">FILT</A></B>,
<B><A HREF="#TRAJ">TRAJ</A></B>
<B><A HREF="#CHKP">CHKP</A></B> or
the <B><A HREF="#SCKP">SCKP</A></B> keys)
<LI><B>&lt;file type&gt;</B>, as explained below, is one of
<p>
 {<B>ckp</B>, <B>crd</B>, <B>hst</B>, <B>dst</B>, <B>idl</B>,
<B>pxc</B>, <B>pxp</B>, <B>pxi</B>, <B>slt</B>, <B>wmp</B>,
<B>dat</B>, <B>grd</B>, <B>pxv</B>, <B>wmp</B>,
<B>pdb</B>, <B>fgd</B>, <B>CRD</B>}
<p>
<UL>
<LI>File type=<B>ckp</B>: Checkpoint file storing the
complete program status (checkpoint). It is created automatically
when the <B><A HREF="#RUNS">RUNS</A></B>,
<B><A HREF="#WCKP">WCKP</A></B> or
<B><A HREF="#SCKP">SCKP</A></B> keys are used.
It allows the resumption of the
calculation with the <B><A HREF="#RCKP">RCKP</A></B> command by
restoring all common
blocks to their value prior to the last write.
It starts with the
coordinates of the current configuration, in the same (binary)
format as on file <B>.crd</B>
(except for the coupling parameter that is stored differently).<br>
Checpoint file is written before a selftest
(see key <B><A HREF="#SLFT">SLFT</A></B>),
at every <B>nplt</b>-th step
(before printing the distributions
- see key <B><A HREF="#RUNS">RUNS</A></B>)
and at every <b>nrecd</b> step.
By default, <b>nrecd</b> is set
to the smaller of <B>nplt</b> or
1000000, 750000, 500000, 200000, 100000 for systems with
number of atoms &lt;3000, &lt;5000, &lt;7500, &lt;10000, <u>&gt;</u>10000,
respectively but it can be set to any value with the key
<B><A HREF="#CHKP">CHKP</A></B>.
<LI>File type=<B>crd</B>: The coordinates of the initial
configuration are stored here.
Additional records may store the coupling parameter
(see key <B><A HREF="#FREE">FREE</A></B>),
the number of molecules
(see key <B><A HREF="#GCEN">GCEN</A></B>)
and/or the simulation cell sizes
(see key <B><A HREF="#IBEN">IBEN</A></B>)
It is read by the <B><A HREF="#CNFG">CNFG</A></B> command,
and can be written by the <B><A HREF="#WCNF">WCNF</A></B> command
<LI>File type=<B>hst</B>: Trajectory file where the simulation
history is (to be) stored. It's use is activated by the
<B><A HREF="#TRAJ">TRAJ</A></B> command.
<LI>File type=<B>dst</B>: The bulk solvent distribution functions
calculated during the simulation may
be saved after every <B>nplt</B> steps (the g(R), various QCDF's
(<A HREF="#Ben-Naim">Quasi-component distribution
functions</A>) on this file. It can later
be used with plot programs for graphical displays. Examples for
QCDF's are the distribution of coordination numbers or binding
energies.
<LI>File type=<B>idl</B>: The log of insertions and deletions are
stored here, as described for the command
<B><A HREF="#GCEN">GCEN</A></B>.
<LI>File type=<B>pxc</B>: The checkpoint file for proximity
analys is generated when keys
<B><A HREF="#PXAN">PXAN</A></B> or
<B><A HREF="#SCAN">SCAN</A></B> are used.
<LI>File type=<B>pxp</B>: The file containing the proximity
analysis distributions for further plotting, written when the
<B><A HREF="#PXPL">PXPL</A></B> key is present.
<LI>File type=<B>pxi</B>: The file containing the proximity
information (generated by <B><A HREF="#PXWR">PXWR</A></B>)
for each configuration analyzed.
<LI>File type=<B>slt</B>: The file containing the solute
description (read after the key <B><A HREF="#SLTA">SLTA</A></B>).
<LI>File type=<B>wmp</B>: The file of matched PMF segments prepared
for plotting by the <B><A HREF="#WMAT">WMAT</A></B> command.
<LI>File type=<B>dat</B>: Configuration file written for InsightII
input (generated by <B><A HREF="#WCNF">WCNF</A> SVAN</B>).
<LI>File type=<B>grd</B>: Grid point file written for cavity
analysis, generated by <B><A HREF="#PRTG">PRTG</A></B>.
<LI>File type=<B>pxv</B>: Proximity region visualization file,
generated by <B><A HREF="#GENV">GENV</A></B> <B>WRIT</B>.
<LI>File type=<B>pdb</B>: Configuration written in
<A HREF="https://www.rcsb.org/pdb/">PDB</A> format.
<LI>File type=<B>fgd</B>: Field-gradient file, generated by the
<B><A HREF="#FLDG">FLDG</A></B> key.
<LI>File type=<B>CRD</B>: Coordinate file in CHARMM CRD format
(with or without the CHARMM headers).
</UL>
</UL>
<!-- HardPage --><p>
<A NAME="Input">III</A>. INPUT DESCRIPTION
<p>
The program input is based on keyword driven commands.
The structure of a command is as follows:
<UL>
<LI>Leading keyword
<LI>Auxiliary keyword(s)
<LI>Unformatted data
<LI>Formatted data
</UL>
Of these only the leading keyword appears all the time.
Each keyword is a four-character, upper case string.
Keywords and unformatted data have to be on the same line.
Commands requiring
more than one line require the continuation character "~" at the
end of the line that is not the end of the command.
Several
commands can be put on the same line if they are separated by the
separator character "|". Text after the comment character "!" is
neglected (i.e., considered just comment).
<p>
The type of items required are defined by the leading keyword.
Auxiliary keywords and data may have default values.
The order of keywords and data is fixed, so if there is a needed
auxiliary keyword or data, all other items have to be specified
that precede the needed one.
Formatted data (when required) always start in a line after the
keywords and it can not be omitted, but for items with
default value, can be secified as zero.
<p>
While there is no rigid order of the leading keywords, there
are certain rules. Certain leading keywords have
<em u>prerequisite</em>
keywords (i.e., the prerequisite keywords all have to be inputted
earlier) and <em u>potential precursor</em> keywords that, if
used, must be read before the current one or after the next
<B><A HREF="#RUNS">RUNS</A></B>) or
<B><A HREF="#SCAN">SCAN</A></B>).
If KEY1 is a <em u>potential precursor</em> of KEY2 then Key2 is a
<em u>potential successor</em> of KEY1.
If the same leading keyword is used
more than once before a <B><A HREF="#RUNS">RUNS</A></B> command, a
warning is given and the last occurrence will be used (except for
<B><A HREF="#TITL">TITL</A></B>).
<p>
During input and initializations the data read is cheked
for consistency and for current program limits.
This may result in messages prefixed by
<b>&gt;&gt;&gt;&gt;&gt;&nbsp;OVERRIDE</b>,
<b>-----&nbsp;WARNING</b>, <b>=====&nbsp;STRONG WARNING</b> or
<b>*****&nbsp;ERROR</b>.
At the conclusion of the run, the number of each type of messages
is printed.
<b>ERROR</b> messages prevent a simulation or an analysis run
but don't abort immediately the input.
When the input has more than one <b>ERROR</b> messages,
some of the additional <b>ERROR</b> messages may
only be the consequence of the first one.
<b>WARNING</b> message indicates
that under some circumstances the input may be incorrect
and <b>STRONG WARNING</b> message indicates
that the input is likely but not necessarily incorrect or that the
error is not fatal.
<b>OVERRIDE</b> messages are generally harmless, but they can
indicate that the input was not what the user intended.
<p>
This section describes the input for each leading keyword
and specifies their prerequisites and potential precursors.
The description of the commands is given by leading keywords.
For example
<B><A HREF="#CNFG">CNFG</A></B> Key1 [Key2 Key3 Data] Fdata
means that the
leading keyword <B><A HREF="#CNFG">CNFG</A></B> has to be followed
by at least one other auxiliary
keyword and optionally two more ("Key 2 Key 3"),
followed by free-format numerical data ("Data")
(all on the line containing the leading keyword), followed by
formatted data, started in a new line.
Note also that the type of (or the need for) auxiliary keys,
free-format and formatted
data may depend on the actual choice of auxiliary keys.
 
The possible keywords to be used for
Key1, Key2 and Key3 and their effects are explained subsequently,
as well as the nature and (for Fdata) the format of the data.
Keywords are always given with upper case <B>BOLD</B>
characters while variable (data) names are given in lower case
<B>bold</B> characters. Input item that is optional is given in
[ brackets ] and the default values are given in { braces }.
The default value for an auxiliary keyword
is the first one on the list.
<p>
A record type number (RECTYPE) is assigned
to every formatted data record. It is used in error messages
on the program's output to
identifying the type of data read when the error was encountered.
<p>
<!-- HardPage --><p>
<A NAME="Input_FILES"></A>
III.1. <B>File handling keywords</B>
<OL>
<LI><TT><B><A HREF="#FILE">FILE</A></B>: </TT>Job name<BR>
<LI><TT><B><A HREF="#WCNF">WCNF</A></B>: </TT>Configuration
saving<BR>
<LI><TT><B><A HREF="#WCKP">WCKP</A></B>: </TT>Creation of a
checkpoint file<BR>
<LI><TT><B><A HREF="#SCKP">SCKP</A></B>: </TT>Periodic archiving of
checkpoint files<BR>
<LI><TT><B><A HREF="#RMCK">RMCK</A></B>: </TT>Checkpoint file
removal<BR>
<LI><TT><B><A HREF="#TRAJ">TRAJ</A></B>: </TT>History file
specification<BR>
<LI><TT><B><A HREF="#MINE">MINE</A></B>: </TT>Minimum energy extraction into
history file
<LI><TT><B><A HREF="#IDAC">IDAC</A></B>: </TT>Proximity
insertion/deletion acceptance rate
<LI><TT><B><A HREF="#IDLG">IDLG</A></B>: </TT>Insertion/deletion
log file specification<BR>
<LI><TT><B><A HREF="#PXPL">PXPL</A></B>: </TT>Proximity analysis
distribution plot file writing<BR>
<LI><TT><B><A HREF="#PXWR">PXWR</A></B>: </TT>Proximity information
file specification<BR>
<LI><TT><B><A HREF="#WTRA">WTRA</A></B>: </TT>Trajectory
conversions<BR>
<LI><TT><B><A HREF="#STIR">STIR</A></B>: </TT>Contact elimination
<LI><TT><B><A HREF="#LPAT">LPAT</A></B>: </TT>Path of local disk directory<BR>
<LI><TT><B><A HREF="#BUFF">BUFF</A></B>: </TT>History file buffer
save<BR>
</OL>
<P>
<A NAME="FILE"></A>
III.01.01. <em u>Job name</em>
<P>
<B>FILE</B> Data
<p>
<U>Data</U>: <B>jobname</B>[, <B>numrun</B>]
<UL>
<LI><B>jobname</B> - Each
<A HREF="#File_org">file name</A> will start with <B>jobname.</B>
<LI><B>numrun</B> - if present, the
<A HREF="#File_org">run number</A>.
</UL>
<A NAME="WCNF"></A>
III.01.02. <em u>Configuration saving</em>
<p>
<B>WCNF</B>
<p>
Prerequisite: <B><A HREF="#CNFG">CNFG</A></B>,
<B><A HREF="#SLTA">SLTA</A></B><BR>
Related key:
<B><A HREF="#PDBT">PDBT</A></B>.
<p>
<u>Key1</u> (coordinate file format):
<UL>
<LI><B>BNRY</B> - Configuration is saved in binary<BR>
<LI><B>ASCI</B> - Configuration is saved in ASCII, (3f15.5)<BR>
<LI><B>ASAN</B> - Configuration is saved in ASCII, and in , each
atom's coordinates are preceded by its atomic symbol and followed
by the group number and charge (a4,1x,3f15.5i5,f10.5)<BR>
<LI><B>PDB </B> - Configuration is exported to
<A HREF="https://www.rcsb.org/pdb/">PDB</A> format;
last two colums contain the atomic radius and charge<BR>
<LI><B>PDBO</B> - Like Key1=<B>PDB </B>, but the
last two colums of solvent atoms contain
the fractional occupancies and RMS's determined by
<B><A HREF="#GENS">GENS</A></B> or read by
<B><A HREF="#CNFG">CNFG</A></B>.
<A HREF="https://www.rcsb.org/pdb/">PDB</A> format;
<LI><B>PDBQ</B> - Save the configuration in GRASP PDB format #1
(last two colums contain the atomic radius and charge)
<LI><B>CHRM</B> - Configuration is exported to CHARMM CRD format<BR>
<u>Key2</u> (possible manipulations) - for any of the keys above:
<UL>
<LI><B>UNCH</B> - Save the configuration unchanged<BR>
<LI><B>FULL</B> - When the <B><A HREF="#FLXR">FLXR</A></B> key
is present, this option will save the reassembled full structure
since otherwise only the moving atoms are saved.<BR>
<LI><B>CENT</B> - The solute is centered in the simulation cell
before saving.<BR>
<LI><B>FIXC</B> - Change all solute molecules to conform to the
geometry defined by the <B><A HREF="#SLTA">SLTA</A></B> key.<BR>
<LI><B>SW23</B> - Switch between 2-copy and 3-copy free energy
solutes before saving the configuration:
<UL>
<LI>For <B><A HREF="#FREE2">FREE&nbsp;TICA</A></B> or
<B><A HREF="#FREE">FREE PM**</A></B> runs a third solute copy is
added (to prepare an input for <B><A HREF="#FREE5">FREE
PMF1</A></B> run)<BR>
<LI>For <B><A HREF="#FREE5">FREE PMF1</A></B> runs the third
solute copy is removed
</UL>
<LI><B>MLCT</B> - Extract from the configuration the centers of the
solute molecules and discard the rest (including the solvent)
before saving<BR>
<LI><B>REPL</B> - Extend the system with its periodic replicas
as specified by the <B><A HREF="#PBCN">PBCN</A></B> key.
If extensions required only in a few directions then the
required replica cell centers should be read with the
<B><A HREF="#PBCN">PBCN</A> READ</B> (cell center coordinates
can be printed with the <B><A HREF="#PRPB">PRPB</A></B> key).
All solvent molecules are printed after all the solute images and
the solute atoms affected by the key
<B><A HREF="#PARD">PARD</A></B>
are also printed contiguously.
<LI><B>REP0</B> - Extend the system with its periodic replicas
but don't replicate the solute. This option is useful for enlarging an
existing system.
<LI><B>NOSV</B> - Don't write the solvents
<LI><B>PRET</B> - Apply the pre-transformation that has been specified
by the solute data header read with the
<B><A HREF="#SLTA">SLTA</A></B> key or defined by the
<B><A HREF="#CNFG">CNFG</A> RANC</B> key<BR>
<U>Data</U>: <B>numrunr</B>
<UL>
<LI><B>numrunr</B> - runnumber to use on the new file
</UL>
<LI><B>TRRT</B> - Translate and rotate the system before saving<BR>
<LI><B>RTTR</B> - Rotate and translate the system before saving<BR>
<U>Data</U> (for <B>TRRT</B> or <B>RTTR</B>): <B>numrunr</B>,
<b>dx</b>, <b>dy</b>, <b>dz</b>,
<b>R11</b>,<b>R21</b>,<b>R31</b>,
<b>R12</b>,<b>R22</b>,<b>R32</b>,
<b>R13</b>,<b>R23</b>,<b>R33</b>
<UL>
<LI><B>numrunr</B> - runnumber to use
<LI><b>dx</b>, <b>dy</b>, <b>dz</b> - the components of the
displacement vector
<LI><B>R</B> - the rotation matrix, specified column-wise
</UL>
<LI><B>PXSR</B> - Rearrange the solvents so that the ones non-proximal
to selected solute atoms will be put first.<BR>
<U>Data</U> (for <B>PXSR</B>): <B>numrunr</B>,
<B>nasltrange</b>, <B>iasltrange1</b>, <B>iasltrange2</b>, ...
<UL>
<LI><B>numrunr</B> - runnumber to use
<LI> <B>nasltrange</b>, <B>iasltrange1</b>, <B>iasltrange2</b>, ...
The selected solute atoms fall into <B>nasltrange</b> segments and
the limits of each segment are
<B>iasltrange<i>i</i></b> and <B>iasltrange<i>i</i></b>.
</UL>
<LI><B>SPSR</B> - Rearrange the solvents so that the ones outside a
user-defined spehere will be put first.<BR>
<U>Data</U> (for <B>SPSR</B>): <B>numrunr</B>,
<B>x</B>, <B>y</b>, <B>z</b>, <B>rsph</b>.
<UL>
<LI><B>numrunr</B> - runnumber to use
<LI><B>x</B>, <B>y</b>, <B>z</b> - center of the sphere
<LI> <B>rsph</b> - radius of the sphere
</UL>
</UL>
<LI><B>PDBD</B> - Save the configuration in
<A HREF="https://www.rcsb.org/pdb/">PDB</A> format
(last two colums contain general data in free format
- see Key2 and Key3 below and key <B><A HREF="#KMNP">KMNP</A></B>)
<LI><B>PDBG</B> - Save the configuration in
<A HREF="">Grasp</A> PDB format #2
(last two colums contain general data in free format
- see Key2 and Key3 below and key <B><A HREF="#KMNP">KMNP</A></B>)
<LI><B>PDTD</B> - Same as <B>PDBD</B> except the data in the two
data colums will come from the difference between two datasets,
generated by the <B><A HREF="#PXDT">PXDT</A></B> key.
<LI><B>PDTG</B> - Same as <B>PDBG</B> except the data in the two
data colums will come from the difference between two datasets,
generated by the <B><A HREF="#PXDT">PXDT</A></B> key.
<LI><B>CHRD</B> - Save the configuration in CHARMM CRD format. The
content of the data column after the coordinates will be determined
 
by Key2 as described below. <BR>
<u>Key2</u> (for Key1=<B>CHRD</B> only);
<u>Key2</u>, <u>Key3</u> (for Key1=<B>PD*D</B> or Key1=<B>PD*G</B>
only):
Any (two) of the following keywords, specifying the
two quantities to be printed in the last (two) data column(s)
of the atom records
<UL>
<LI><B>VFSH</B> Volume of the first solvation shell<BR>
<LI><B>V2SH</B> Volume of the first two solvation shells<BR>
<LI><B>NFSH</B> Number of solvents in the first solvation shell<BR>
<LI><B>N2SH</B> Number of solvents in the first and second
solvation shells
<LI><B>DFSH</B> Solvent density in the first solvation shell<BR>
<LI><B>DSSH</B> Solvent density in the first and second solvation
shell<BR
<LI><B>EBFS</B> Solute-solvent binding energy in the first
solvation shell<BR>
<LI><B>EPFS</B> Solute-solvent binding energy in the first
solvation shell<BR>
<LI><B>TOBE</B> Total binding energy<BR>
<LI><B>TOTN</B> Total number of solvents<BR>
<LI><B>NFVV</B> Solvent-solvent coordination number<BR>
<LI><B>EPVV</B> Solvent-solvent pair energy<BR>
<LI><B>EBVV</B> Solvent-solvent binding energy<BR>
<LI><B>MRDF</B> First minimum of the primary RDF<BR>
<LI><B>KRDF</B> Coordination number using the actual
first minimum of the primary RDF<BR>
<LI><B>VRDF</B> First shell volume using the actual
first minimum of the primary RDF<BR>
<LI><B>DRDF</B> First shell density using the actual
first minimum of the primary RDF
<LI><B>CRCV</B> Circular variance, based on all solute atoms<BR>
<LI><B>CRCH</B> Circular variance, based on the solute heavy atoms only
<LI><B>INAC</B> insertion acceptance rate (saved with the
<B><A HREF="#IDAC">IDAC</A></B> key)
<LI><B>DLAC</B> deletion acceptance rate<BR>
<U>Data</U>: <B>numrunr</B>, <B>rcrcv</b>
<UL>
<LI><B>numrunr</B> {<B>numrun</B>} - The configuration is saved
to the file <B>&lt;job&nbsp;name&gt;.&lt;numrunr&gt;.dat</B> when
Key1 is <B>SVAN</B>,
to the file <B>&lt;job&nbsp;name&gt;.&lt;numrunr&gt;.pdb</B> when
Key1 is <B>PDB*</B>,
otherwise to file <B>&lt;job&nbsp;name&gt;.&lt;numrunr&gt;.crd</B>
When Key2 or Key3 = <B>CRC*</B>:
<LI><B>rcrcv</B> {6.0} - The
circular variance will be calculated
using solute atoms within a radius of <B>rcrcv</B>.
 <A NAME="CRCV"></A>
The circular variance for a point <B>R</b> is defined as<BR>
CV = 1.0 - [ SUM (<B>&lt;R&gt;</B>-<B>r<sub>i</sub></B>)
/ SUM (|<B>&lt;R&gt;</B>-<B>r<sub>i</sub></B>| ]<BR>
where <B>r<sub>i</sub></B> are the coordinates of solute atoms
within
<B>rcrcv</b>. When <B>R</B> is inside the solute, CV is near one,
and
when <B>R</B> is outside the solute, CV is near zero.
</UL>
</UL>
</UL>
<A NAME="WCKP"></A>
III.01.03. <em u>Creation of a checkpoint file</em>
<p>
<B>WCKP</B> [Data]
<p>
Prerequisite: <B><A HREF="#FILE">FILE</A></B>
<p>
<U>Data</U>: <B>numrunw</B>
<UL><LI><B>numrunw</B> {<B>numrun</B>} - Checkpoint file
<B>&lt;jobname&gt;.numrun.ckp</B> is to be written.
</UL>
<A NAME="SCKP"></A>
III.01.04. <em u>Periodic archiving of checkpoint files</em>
<p>
<B>SCKP</B> [Data]
<p>
<U>Data</U>: <B>nsavckpf</B>
<UL><LI><B>nsvckpf</B> - Frequency of writing a separate checkpoint
file.
</UL>
Each checkpoint file will have the runnumber of the current run
and consecutive version numbers, starting at 2.
This is option allows the later calculation of proximity analysis
averages over different segments of the run (see key
<B><A HREF="#CMPC">CMPC</A></B>),
provides a backup procedure in case the program
crashes while writing the checkpoint file, and is
useful for debugging.
<p>
<A NAME="RMCK"></A>
III.01.05. <em u>Checkpoint file removal</em>
<p>
<B>RMCK</B> [Data]
<P>
<U>Data</U>: <B>numrunp</B>
<UL>
<LI><B>numrunp</B> {&lt;current runnumber&gt; -1} - if positive,
the checkpoint file(s) for
<A HREF="#File_org">run number</A>.
<B>numrunp</B> will be removed
(by default, the checkpoint file created by the last
<B><A HREF="#RUNS">RUNS</A></B> or <B><A HREF="#SCAN">SCAN</A></B>
will be removed).
If <B>numrunp</B> is negative then the checkpoint file(s) for
the &lt;current runnumber&gt; - |<B>numrunp</B>| will be removed.
</UL>
This option is useful for preventing the disk to become too full
when the calculation is broken up into several
<B><A HREF="#RUNS">RUNS</A></B> or <B><A HREF="#SCAN">SCAN</A></B>
steps.
<P>
<A NAME="TRAJ"></A>
III.01.06. <em u>History file specification</em>
<p>
<B>TRAJ</B> Key1, [Key2,Key3] [Data]
<p>
Can not be followed by: <B><A HREF="#MINE">MINE</A></B><BR>
Potential successors:
<B><A HREF="#RCKP">RCKP</A></B><BR>
Prerequisite of:
<B><A HREF="#DENF">DENF</A></B>, <B><A HREF="#WTRA">WTRA</A></B><BR>
Variable to set to 'T' in the <A HREF="#PREP">preprocessor</A>: <B>ME</B><BR>
Related keys:
<B><A HREF="#MINE">MINE</A></B>,
<B><A HREF="#WTRA">WTRA</A></B>,
<B><A HREF="#BUFF">BUFF</A></B>.
<p>
For MC runs this key sets up the saving or reading
of the run history on
file <B>&lt;jobname&gt;.&lt;runnum&gt;_&lt;runvers&gt;.hst</B>
(For <B>runnum</B>=1 the run number part is omitted and for
<B>runvers</B>=1 the version part is omitted).
See also the key <B><A HREF="#BUFF">BUFF</A></B>.
<p>
For analysis (<B><A HREF="#SCAN">SCAN</A></B>) or
filtering (<B><A HREF="#FILT">FILT</A> **** TRAJ</B>)
this command
specifies the format of the history file. An input
history can consist of several files with the same runnumber and
with consecutive versionnumbers (i.e., the
second, third, etc. piece should have names
<B>&lt;jobname&gt;.&lt;runnum&gt;_&lt;runvers+1&gt;.hst</B> ,
<B>&lt;jobname&gt;.&lt;runnum&gt;_&lt;runvers+2&gt;.hst</B> , etc.).
Note, that this naming convention also applies to
non-MMC formats that have customary extensions
(e.g., <b>.pdb</b>, <b>.DCD</b>) requiring either renaming
these files or setting up aliases for them.
<p>
<u>Key1</u> (trajectory file format - if any):
<UL>
<LI><B>NONE</B> - No saving of run history.
<LI><B>ALLE</B> -
The following information is stored at every accepted step in
blocks of 250 when no variable coupling parameter is applied:
<UL>
<LI><B>estac</B> - Total energy of the configuration
<LI><B>cst</B> - Coordinates of the first three atoms of the
moved molecule <B>istc</B>.
<LI><B>istc</B> - Number (label) of the solvent molecule moved
(0 when the whole solute was moved).
<LI><B>nmcst</B> - Step number in the MC walk.
<LI><B>bngst</B> - Binding energy of the molecule moved. When
variable coupling parameter is used (i.e.,
<B><A HREF="#FREE5">FREE&nbsp;PMF1</A></B>
run), the coupling parameter value is saved instead of the binding
energy.
</UL>
This option allows the full reconstruction of the
run's history but works only when neither the key
<B><A HREF="#PARD">PARD</A></B> nor the key
<B><A HREF="#PART">PART</A></B> is in use and
only in the canonical ensemble.
<LI><B>ALLV</B> - Same as <B>ALLE</B>, but the total viral sum is
saved instead of the binding energy.
Same limitations apply as for <B>ALLE</B>.
<LI><B>ALLC</B> - Trajectory file contains full configurations
in formatted (ASCII) records as follows:
<UL>
<LI>1st line: number of atoms written (<B>natomp</B>), number of
solvent molecules+1, total number of atoms, first solute atom written
out, last solute atom written out, MC stepnumber, coupling
parameter in format(i6,9x,i6,8x,i6,7x,2i6,5x,i9,4x,f8.0)
<LI>Next <B>natomp</B> lines: x,y,z coordinates (in &%197;)
(3f15.5) format
<LI>Last -1 line: ins/del stepnumber, total energy, number of
accepted insertions and deletions, umbrella sampling weight
factor for the configuration
<LI>Last line: solute-solvent energy and its contributions for
each interaction term. Important: for water solvents only
the first three atoms of each solvent molecule are written out.<BR>
For (T,P,N) ensemble runs the three edge parameters (see key
<B><A HREF="#PBCN">PBCN</A></B>) are also saved
in an additional record in (3f10.5) format.
</UL>
<LI><B>ALLA</B> - Trajectory file contains full configurations
file in formatted (ASCII) records as follows:
<UL>
<LI>1st line: number of atoms written (<B>natomp</B>), number of
solvent molecules+1, total number of atoms, first solute atom written
out,i last solute atom written out, MC stepnumber, coupling
parameter in format(i6,9x,i6,8x,i6,7x,2i6,5x,i9,4x,f8.0)
<LI>Next <B>natomp</B> lines: atomic symbol, x,y,z coordinates
(in &#197;), group (residue) number, partial charge with
(a4,1x,3f15.5,i5,f10.5) format
<LI>Last -1 line: ins/del stepnumber, total energy, number of
accepted insertions and deletions, umbrella sampling weight
factor for the configuration
<LI>Last line: solute-solvent energy and its contributions for
each interaction term. Important: for water solvents only
the first three atoms of each solvent molecule are written out.
<LI>For (T,P,N) ensemble runs the three edge parameters (see key
<B><A HREF="#PBCN">PBCN</A></B>) are also saved in an additional
record.
</UL>
<LI><B>ALLB</B> - Trajectory file contains full configurations
in binary format:
<UL>
<LI>First record: number of solvent
molecules, number of atoms, umbrella sampling weight factor for the
configuration, MC stepnumber, insertion/deletion stepnumber, number
of accepted insertions, number of accepted deletions, first atom
written out, total energy, solute-solvent energy and its
contributions for each interaction term
<LI>Second record:
the whole configuration
<LI> For free energy runs the coupling parameter is written as an
additional record. Again, for water solvents only the
first three solvent atoms are written out.<BR>
<LI>For (T,P,N) ensemble runs the three edge parameters (see key
<B><A HREF="#PBCN">PBCN</A></B>) are also saved in an additional
record.
</UL>
<LI><B>ALLP</B> - Trajectory file contains full configurations
in <A HREF="https://www.rcsb.org/pdb/">PDB</A> format.
The first record is a REMARK giving the same information as the
first line of the <B>ALLA</B> format in
format(9x,i6,3x,i6,5x,i6,7x,2i6,5x,i9,4x,f8.0) and the next REMARK
may give the total energy and solute solvent energy in the format
(9x,e16.8,17x,e13.6); additional REMARKs may follow.
<LI><B>ALLH</B> - Trajectory file contains full configurations
in CHARMM CRD format.
The first title line gives the same information as the
first line of the <B>ALLA</B> format in
format(9x,i6,3x,i6,5x,i6,7x,2i6,5x,i9,4x,f8.0) and the next title line
may give the total energy and solute solvent energy in the format
(9x,e16.8,17x,e13.6); additional title lines may follow.
<LI><B>CHRM</B> - Trajectory file contains full configurations
in CHARMM binary trajectory format.
For <b>runnum</b>=1 the <b>.DCD</b> extension is also allowed.
<LI><B>AMBR</B> - Trajectory file contains full configurations
in AMBER format. The program will
check if the system is centered around (0,0,0) or
(Ex/2,Ey/2,Ez/2).<BR>
<u>Key2</u> (level of solute changes):
<UL>
<LI><B>RGFX</B> - The solute is rigid and is not moved during the
simulation<BR>
<LI><B>RGMO</B> - The solute is rigid and but it is allowed to
move during the simulation<BR>
<LI><B>FLEX</B> - The solute is flexible<BR>
<u>Key3</u> after Key1=<B>ALL*</B>
(level of solute atom saving):
<UL>
<LI><B>ALST</B> - All solute atoms are saved<BR>
<LI><B>MVST</B> - Only the solute atoms affected by partial
solute moves (<B><A HREF="#PARD">PARD</A> </B>key) are saved<BR>
<LI><B>NOST</B> - Solute atoms are not saved
<LI><B>NOSV</B> - Solvent atoms are not saved
</UL>
<u>Key3</u> after Key1=<B>AMBR</B> (PBC information):
<UL>
<LI><B>NOBX</B> - No box-size information after configurations<BR>
<LI><B>BOX</B> - After each configuration the simulation cell
edges are written in a new line<BR>
<LI><B>BOXX</B> - Read the box description after each
configuration but keep using the values inputted with the
<B><A HREF="#NSLV">NSLV</A></B> command<BR>
<U>Data</U>: <B>natskip</B>, <B>runnum</B>, <B>runvers</B>,
<b>tfilename</b>, <B>incrun</B>, <B>alttrajext</B>
<UL>
<LI><B>natskip</B> {0} - When <B>natskip</B> &gt; 0 then
the last <B>natskip</B> atoms of each frame will be skipped.
When <B>natskip</B> &lt; 0 then
the first <B>natskip</B> atoms of each frame will be skipped.
Ignored with MMC trajectories.<BR>
<LI><B>runnum</B> {current run number} - The
<A HREF="#File_org">run number</A>
of the trajectory file.
<LI><B>runvers</B> {1} - The version number of the (initial)
trajectory file.
<LI><b>tfilename</b> {<b>jobname</b>} - The trajectory file
is named <b>tfilename</b>.<B>runnum</B>_<B>runvers</B>.hst
<LI><B>incrun</B> {0} - If zero, additional trajectory segments will be searched
with increasing the <i>version</i> number; if one, with increasing the <i>run number</i>.
<LI><B>alttrajext</B> - the trajectory extension (instead of hst).
It has to be exactly three characters.
</UL>
</UL>
</UL>
</UL>
When running a MC simulation, full configurations
(Key1=<B>ALL*</B>)
will be written on the history file
at every <B>nmcrec</B>-th MC step
(read with the <B><A HREF="#RUNS">RUNS</A></B> key).
<P>
<A NAME="MINE"></A>
III.01.07. <em u>Minimum energy extraction</em>
<p>
<B>MINE</B> Key1
<p>
Related key:
<B><A HREF="#TRAJ">TRAJ</A></B>
<p>
<u>Key1</u> (trajectory or overall):
<UL>
<LI><B>NONE</B> - Don't extract the minimum energy
<LI><B>TRAJ</B> - extract the minimum energy
over every <b>nmcrep</b> MC step segment and
save it on the trajectory (<b>.hst</b>) file.
Requires the <B><A HREF="#TRAJ">TRAJ</A></B> key.
Also, save the overall lowest energy and write it on a PDB file
at the end of the run defined by the
<B><A HREF="#RUNS">RUNS</A></B> key.
<LI><B>ALLC</B> - Only extract the minimum energy and write it
on a PDB file at the end of the run defined by the
<B><A HREF="#RUNS">RUNS</A></B> key.
<LI><B>WANN</B> - Besides the function of <B>ALLC</B>
the configuration at each temperature change is written to a file
<b>jobname</b>_annh.<B>runnum</B>.pdb
<LI><B>WANN</B> - Besides the function of <B>ALLC</B>
the reassembled full configuration at each temperature change is written to a file
<b>jobname</b>_annfh.<B>runnum</B>.pdb (valid only when the 
<B><A HREF="#TRAJ">TRAJ</A></B> key is present).
</UL>
<p>
<A NAME="IDAC"></A>
III.01.08. <em u>Proximity insertion/deletion acceptance rate</em>
<p>
<B>IDAC</B> [Data]
<p>
<U>Data</U>: <B>rpxidlim</B>, <b>nrange</b>, <b>irangestart</b>, <b>irangestop</
(<b>nrange</b> times)
<UL>
<LI><B>rpxidlim</B> {4.0} - Insertion and deletion acceptance
rates within <B>rpxidlim</B> of the nearest solute heavy atom will be
calculated for each solute heavy atom.
<LI><b>nrange</b> {0} - Number of solute atom index ranges
<LI><b>irangestart</b>, <b>irangestop</b> (<b>nrange</b> times) - proximity
index calculation will be restricted to heavy atoms in the rage(s)
<b>irangestart</b> - <b>irangestop</b>
</UL>
</UL>
<p>
<A NAME="IDLG"></A>
III.01.09. <em u>Insertion/deletion log file specification</em>
<p>
<B>IDLG</B> Key1 [Data]
<p>
Prerequisite: <B><A HREF="#GCEN">GCEN</A></B>.<BR>
Prerequisite of:
<B><A HREF="#IDAG">IDAG</A></B><BR>
Related key: <B><A HREF="#IDAG">IDAG</A></B>.
<p>
<u>Key1</u> (ins/del log file creation):
<UL>
<LI><B>OFF </B> - No log is written.
<LI><B>ASCI</B> - For each accepted insertion and deletion the
program writes +1 or -1 (for insertion or deletion), the solvent
molecule index, the number of solvent molecules after the
insertion/deletion, the MC
stepnumber and the coordinates of the center of mass of the
molecule inserted/deleted on the <B>.idl</B> file.
<LI><B>ASPX</B> - Same as <B>ASCI</B>, but the atom number
of the nearest solute atom to the insertion/deletion site as well as
the distance between them are also printed.<BR>
<U>Data</U>: <B>rpxidlim</B>
<UL>
<LI><B>rpxidlim</B> {4.0} - Solute atoms with insertion and deletion sites
within <B>rpxidlim</B> will be listed with the respective counts.
</UL>
</UL>
<A NAME="PXPL"></A>
III.01.10. <em u>Proximity analysis distribution plot file
writing</em>
<p>
<B>PXPL</B>
If this keyword is present, the program will write an ASCII
file containing all the calculated proximity distributions to the
<B>.pxp</B> file.
The distributions written on the <B>.pxp</B> file can be plotted
with
the companion (Fortran-77) program <b>pxps.f</b>.
<b>pxps.f</b> produces Postscipt files of plots, 8 plots per page.
It can put two distributions on the same plot, but both have to use
the
same dependent variable.
The right-side vertical axis will belong to the second plot.
<p>
A typical run of <b>pxps.f</b> is as follows (user input is
displayed in <b>bold face</b>):
<p>
<pre>
% pxps
 
Proximity plot generator Version:/07/22/98
Data file name root=<b>wt1</b>
 File wt1.pxp opened                             &lt;-  the file wt1.pxp
                                                     was generated by
                                                     the PXPL key
Use @ for: @f=Greek fnt, @n=Normal fnt, @+=superscript
 @-=subscript, @m=newline, @#n#=font size n, @@=@symbol
Title of graph page:
<b>GnRH Wild Type, Extended Conformation</b>
Do you want grid lines drawn (y/n)? <b>n</b>   <- If yes, only one plot
                                                  can be plotted
Plot types:
 1  Solute-solvent radial distribution functions       &lt;- from key PXGR
 2  Solute-solvent pair energy distribution functions  &lt;- from key PXBE
 3  Mean solvent orientation as a function of R        &lt;- from key PXDP
 4  Distribution of first shell solvent orientations   &lt;- from key PXDP
 5  Solvent-solvent radial distribution functions      &lt;- from key PXWW
Plot type number=<b>1</b>
Data colums: 1 R        2 gpx(R)   3 Kpx(R)   4 gt(R)    5 Kt(R)
Data column number=<b>2</b>
Do you want a second function also plotted (y/n)? <b>y</b>
Plot types:
 1  Solute-solvent radial distribution functions
 2  Solute-solvent pair energy distribution functions
 3  Mean solvent orientation as a function of R
 4  Distribution of first shell solvent orientations
 5  Solvent-solvent radial distribution functions
Plot type number=<b>1</b>
Data colums: 1 R        2 gpx(R)   3 Kpx(R)   4 gt(R)    5 Kt(R)
Data column number=<b>3</b>
Number of graphs to print (0: will ask for range)=<b>0</b>
First and last distribution function=<b>1,18</b>
Data set not found (skipped): gkr is=   1    &lt;- empty proximity
regions
Data set not found (skipped): gkr is=  10
 File wt1_2_9_gpx_Kpx.ps opened
 File wt1_11_18_gpx_Kpx.ps opened
More plots (y/n)? <b>n</b>   &lt;- allows to plot other distributions
%
</pre>
<p>
The format of the <B>.pxp</B> file is as follows:
<UL>
<LI>Header:
<UL>
<LI><B>nsltpx</B>, <b>nsltpxgr</b> (2i5)<BR>
<LI><B>ianslt</B>  (20i4)<BR>
<LI><B>resnam,atnam</B>  (20a4)<BR>
<UL>
<LI><b>nsltpx</b> - Number of solute atoms analysed
<LI><b>nsltpxgr</b> - Number of rdf's generated
<LI><B>ianslt</B> - atomic numbers of all solute atoms analysed
<LI><B>resnam,atnam</B> - 8-character label of all solute atoms
analysed
</UL>
</UL>
<LI>Radial distribution functions (<B><A HREF="#PXGR">PXGR</A></B>
key was present). For each distribution function the program writes
<UL>
<LI><B>is</B>, <B>mxgrd</B> (' is=',i4,' ngrids=',i4,' gkr')
<LI>(<B>grd</B>(i), <B>gr</B>(i), <B>rcoord</B>(i),
<B>grt</B>(i), <B>rcoort</B>(i), i=1,mxgrd) (5f8.3)
<UL>
<LI><B>is</B> - rdf number
<LI><B>mxgrd</B> - Number of gridpoins
<LI><B>grd</B>(i) - Radius of the i-th gridpoint
<LI><B>gr</B>(i) - Primary rdf
<LI><B>rcoord</B>(i) - Primary running coordination number
<LI><B>grt</B>(i) - Total rdf
<LI><B>rcoort</B>(i) - Total running coordination number
</UL>
</UL>
<LI>Primary solute pair energy distributions
(<B><A HREF="#PXBE">PXBE</A></B> key was present).
For each distribution function the program writes
<UL>
<LI><B>is</B>, <B>mxgrd</B> (' is=',i4,' ngrids=',i4,' xpe')
<LI>(<B>egr</B>(i),<B>sltep</B>(i),<B>rcoore</B>(i),i=1,mxgrd)
(10f8.4)
<UL>
<LI><B>is</B> - rdf number
<LI><B>mxgrd</B> - Number of gridpoins
<LI><B>grd</B>(i) - Energy of the i-th gridpoint
<LI><B>sltep</B>(i) - Primary pair energy distribution value
<LI><B>rcoord</B>(i) - Primary running coordination number
</UL>
</UL>
<LI>Primary solvent-solvent rdfs
(<B><A HREF="#PXWW">PXWW</A></B> key was present).
For each distribution function the program writes
<UL>
<LI><B>is</B>, <B>mxgrd</B>, <B>nijgvvp</B>
(' is=',i4,' ngrids=',i4,' ngvv=',i2,' gvv')
<LI>mxgrd lines: <B>grd</B>(i),(<B>groo</B>(i,iv),v=1,nijgvvp)
(10f8.4)
<UL>
<LI><B>is</B> - rdf number
<LI><B>mxgrd</B> - Number of gridpoins
<LI><B>nijgvvp</B> - Number of rdf types (between different
solvent centers)
<LI><B>grd</B>(i) - Radius of the i-th gridpoint
<LI><B>groo</B>(i,iv) - Primary solvent-solvent rdf
</UL>
</UL>
<LI>Solvent mean orientation
(<B><A HREF="#PXDP">PXDP</A></B> key was present).
For each distribution function the program writes
<UL>
<LI><B>is</B>, <B>mxgrd</B> (' is=',i4,' ngrids=',i4,' tav')
<LI>mxgrd lines: (<B>grd</B>(i),<B>tav</B>(i),i=1,mxgrd) (10f8.4)
<UL>
<LI><B>is</B> - rdf number
<LI><B>mxgrd</B> - Number of gridpoins centers)
<LI><B>grd</B>(i) - Radius of the i-th gridpoint
<LI><B>tav</B>(i) - Primary mean solvent-solute angle
</UL>
</UL>
<LI>Solvent orientation rdf (first shell)
(<B><A HREF="#PXDP">PXDP</A></B> key was present).
For each distribution function the program writes
<UL>
<LI><B>is</B>, <B>mxgrd</B> (' is=',i4,' ngrids=',i4,' xtd')
<LI>mxgrd lines: (<B>grd</B>(i),<B>xtd</B>(i),i=1,mxgrd) (2f9.4)
<UL>
<LI><B>is</B> - rdf number
<LI><B>mxgrd</B> - Number of gridpoins
centers)
<LI><B>grd</B>(i) - Angle THETA of the i-th gridpoint
<LI><B>xtd</B>(i) - Primary solvent-solute angle distribution
</UL>
</UL>
</UL>
<p>
<A NAME="PXWR"></A>
III.01.11. <em u>Proximity information file specification</em>
<p>
<B>PXWR</B> Key1 [Key2 Data]
<p>
Prerequisite: <B><A HREF="#PXCR">PXCR</A></B>,
<p>
<u>Key1</u> (file format - if any):
<UL>
<LI><B>OFF </B> - Proximity information writing is turned off.
<LI><B>ASCI</B> - The proximity information file is an annotated
ASCI file containing the following data for each configuration
analyzed:<BR> (nsv,nmc,ncnfpx,
(iproxi(i),resname(i),resno(i),atnam(i),rmin(i),esvst(i),i=1,nsv)
<BR>
where
<UL>
<LI>nsv is the number of solvent molecules
<LI>nmc is the MC stepnumber
<LI>ncnfpx is the number of configurations analyzed so far
<LI>iproxi(i) is the solute atom nearest to solvent i
<LI>rmins(i) is the corresponding minimum distance (in &#197;)
<LI>esvst(i) is the interaction energy of solvent i with the solute
(in kcal/mol).
</UL>
<LI><B>BNRY</B> - The proximity information file is a
binary file containing one record for each configuration analyzed:
<BR>
(nsv,nmc,ncnfpx,(iproxi(i),rmin(i)<sup>2</sup>,esvst(i),i=1,nsv)
<BR>
<u>Important</u>: iproxi is stored as INTEGER*2 to save space.
Note, that the size of a file created with the <B>BNRY</B> option
is about the tenth of the size of a file
created with the <B>ASCI</B> option.
<LI><B>BNRR</B> - The proximity information file is a
binary file containing one record for each configuration analyzed:
<BR>
(nsv,nmc,ncnfpx,(iproxi(i),rmin(i)<sup>2</sup>,i=1,nsv) .
<BR>
This file is about 2/3rd of the size of the file written with
the <B>BNRY</B> option.<BR>
<u>Key2</u> (read or write):
<UL>
<LI><B>WRIT</B> - Write the proximity information
<LI><B>READ</B> - Read the proximity information from an existing
.pxi file<BR>
<U>Data</U>:
<UL>
<LI><B>numrunpxi</B> - The
<A HREF="#File_org">run number</A>
of the .pxi file to read from
(only read with Key2=<B>READ</B>)
</UL>
</UL>
</UL>
The proximity information file has extension <B>.pxi</B>.
<p>
<A NAME="WTRA"></A>
III.01.12. <em u>Converting/shifting a trajectory</em>
<p>
<B>WTRA</B> Key1 Key2 [Data]
<P>
Prerequisite: <B><A HREF="#TRAJ">TRAJ</A></B>
<p>
<u>Key1</u> (transformation type):
<UL>
<LI><B>UNCH</B> - Don't change coordinates
<LI><B>STCT</B> - Shift the system to put the solute's center of
mass at the cell center
<LI><B>SHCT</B> - Shift the system to move the origin from the
lower left corner of the box to the center (update PBC images
accordingly)<BR>
<u>Key2</u> (trajectory format):
<UL>
<LI> One of <B>ALLC</B>, <B>ALLA</B>, <B>ALLB</B>, <B>ALLP</B> -
see key <B><A HREF="#TRAJ">TRAJ</A></B>.
Note, that <B>ALLB</B> is the most compact.<BR>
<U>Data</U>: <B>nmcmax</B>, <B>nmcfreq</B>, <B>nmcskip</B>,
<B>incvers</B>, <B>nmcmaxseg</B><BR>
<UL>
<LI><B>nmcmax</B> {100000000} - gather configurations until the
<B>nmcmax</B>-th step.
<LI><B>nmcfreq</B> {1} - use configuration saved at every
<B>nmcfreq</B>-th simulation step
<LI><B>nmcskip</B> {0} - skip the first <B>nmcskip</B> simulation steps
<LI><B>incvers</B> {10} - The converted trajectory will
have a version number that is incremented from the initial
by <B>incvers</B>
<LI><B>nmcmaxseg</B> {0} - When &gt; 0, after <B>nmcmaxseg</B> steps,
open a new output file with version number incremented by one.
</UL>
</UL>
</UL>
This key allows the conversion of the input trajectory to a new
format
and gives the option of centering it in two different ways.
<p>
<A NAME="STIR"></A>
III.01.13. <em u>Contact elimination</em>
<p>
<B>STIR</B> [Data]
<P>
Prerequisite: <B><A HREF="#CNFG">CNFG</A></B>
<p>
<U>Data</U>: <B>eijmin</B>, <B>disp</B>, <B>dispa</B>
<UL>
<LI><B>eijmin</B> {1000.0}, <B>disp</B> {3.0}
- solute molecule pairs with energy above
<B>eijmin</B> kcal/mol will be pulled apart by <B>disp</B> &#197;
<LI><B>dispa</B> {90.0} - solute torsion group members with intramolecular
energy exceeding <B>eijmin</B> will be incremented by <B>dispa</B>
degrees.
</UL>
This option is useful for 'cleanng up' a starting configuration
generated with the RANC option.
<P>
<A NAME="LPAT"></A>
III.01.14. <em u>Local disk directory</em>
<p>
<B>LPAT</B> [Data]
<p>
<U>Data</U>: <B>scratchpath</B>
<UL>
<LI><B>scratchpath</B> - The full path to the directory where temporary files
and checkpoint files are written during the run.
The checkpoint files wil be copied to the originating directory only
at the end of the run.
This option allows the reduction of network traffic.
</UL>
<P>
<A NAME="Input_SYSTEM"></A>
III.02. <B>System descriptor keywords</B>
<OL>
<LI><TT><B><A HREF="#NSLV">NSLV</A></B>: </TT>Number of
solvent molecules<BR>
<LI><TT><B><A HREF="#TEMP">TEMP</A></B>: </TT>Simulation
temperature<BR>
<LI><TT><B><A HREF="#TITL">TITL</A></B>: </TT>Description<BR>
<LI><TT><B><A HREF="#PBCN">PBCN</A></B>: </TT>Periodic boundary
conditions<BR>
<LI><TT><B><A HREF="#GCEN">GCEN</A></B>: </TT>Grand-canonical (T,V,mu)
ensemble<BR>
<LI><TT><B><A HREF="#STPX">STPX</A></B>: </TT>Stop GCE run at
nx<BR>
<LI><TT><B><A HREF="#BTUN">BTUN</A></B>: </TT>Tune GCE B
parameter<BR>
<LI><TT><B><A HREF="#LIMG">LIMG</A></B>: </TT>Limit the cavity
grid<BR>
<LI><TT><B><A HREF="#IBEN">IBEN</A></B>: </TT>Isothermal isobaric
(T,P,N) ensemble<BR>
<LI><TT><B><A HREF="#LCMP">LCMP</A></B>: </TT>Virial biasing factor
<LI><TT><B><A HREF="#CNFG">CNFG</A></B>: </TT>Initial
configuration<BR>
<LI><TT><B><A HREF="#SCRM">SCRM</A></B>: </TT>Scramble torsion angles
<LI><TT><B><A HREF="#SCAL">SCAL</A></B>: </TT>Simulation cell
rescaling<BR>
<LI><TT><B><A HREF="#OVST">OVST</A></B>: </TT>Overlap of the second
solute copy with the first<BR>
<LI><TT><B><A HREF="#SPRD">SPRD</A></B>: </TT>Spread out solute
molecules<BR>
<LI><TT><B><A HREF="#PBGR">PBGR</A></B>: </TT>Solute reset with
grids<BR>
<LI><TT><B><A HREF="#LIGA">LIGA</A></B>: </TT>Multi-copy ligand<BR>
<LI><TT><B><A HREF="#STUN">STUN</A></B>: </TT>Stepsize tuning<BR>
<LI><TT><B><A HREF="#TAC0">TAC0</A></B>: </TT>Stepsize accumulator reset<BR>
<LI><TT><B><A HREF="#SANN">SANN</A></B>: </TT>Simulated annealing<BR>
<LI><TT><B><A HREF="#SACP">SACP</A></B>: </TT>Simulated annealing of the chemical potential<BR>
</OL>
<P>
<A NAME="NSLV"></A>
III.02.01. <em u>System size</em>
<p>
<B>NSLV</B> Data
<p>
Can not be followed by:
<B><A HREF="#HRDW">HRDW</A></B><BR>
Potential successors:
<B><A HREF="#CNFG">CNFG</A></B>, <B><A HREF="#RCKP">RCKP</A></B><BR>
<p>
<U>Data</U>: <B>numsol</B>, <B>nsolfix</B>
<UL>
<LI><B>numsol</B> - Number of solvent molecules.
For grand-canonical ensemble runs (<B><A HREF="#GCEN">GCEN</A></B> key)
using <B><A HREF="#CNFG">CNFG</A> RAN*</B> will generate
<B>numsol</B> randomly placed solvents to start with.
<LI><B>nsolfix</B> {0} - The first <B>nsolfix</B> solvent molecules
will be kept fixed during the simulations.
</UL>
<B>nsolfix</B> &gt; 0 requires that the solute does not undergo
translation/rotation (<B>cedslt</B> and <B>rtxslt</B> should both be
0.0 - see the <B><A HREF="#STEP">STEP</A></B> key).
<p>
If the input has no <B>NSLV</B> key then the number of solvent molecules
will be established from the <b>.crd</b> or <b>.hst</b> files.
<p>
<A NAME="TEMP"></A>
III.02.02. <em u>Simulation temperature</em>
<p>
<B>TEMP</B> Data
<p>
Prerequisite of:
<B><A HREF="#RUNS">RUNS</A></B>, <B><A HREF="#SANN">SANN</A></B><BR>
<p>
<U>Data</U>: T
<UL>
<LI><B>T</B> - the simulation temperature in Kelvin
</UL>
<A NAME="TITL"></A>
III.02.03. <em u>Title</em>
<p>
<B>TITL</B> Data
<p>
Potential successor:
<B><A HREF="#RCKP">RCKP</A></B><BR>
<p>
<U>Data</U>: description of the calculation.
<p>
The program is prepared to read at most two title lines.
<p>
<A NAME="PBCN"></A>
III.02.04. <em u>Periodic boundary conditions and cell
dimension.</em>
<p>
<B>PBCN</B>&nbsp;Key1 [Key2 Key3] Data [Fdata]
<p>
Prerequisites:
<B><A HREF="#SUVC">SUVC</A></B>, <B><A HREF="#SVVC">SVVC</A></B><BR>
Potential successor:
<B><A HREF="#RCKP">RCKP</A></B><BR>
Prerequisite of:
<B><A HREF="#SLTA">SLTA</A></B>, <B><A HREF="#GCEN">GCEN</A></B>,
<B><A HREF="#MOND">MOND</A></B>, <B><A HREF="#LIMG">LIMG</A></B>,
<B><A HREF="#VORO">VORO</A></B>, <B><A HREF="#FREE">FREE</A></B>,
<B><A HREF="#PRPB">PRPB</A></B>, <B><A HREF="#GENS">GENS</A></B><BR>
<p>
<u>Key1</u> (cell shape):
<UL>
<LI><B>RECT</B> - Rectangular (cubic) boundary conditions (SC).<BR>
<U>Data</U>: <B>edgex</B> [<B>edgey</B>, <B>edgez</B>] (in &#197;).
<UL>
<LI><B>edgex</B>, <B>edgey</B> {<B>edgex</B>}, <B>edgez</B>
{<B>edgex</B>} - The X,Y,Z edges of the rectangle.
<LI>Vol=<B>edgex</B>*<B>edgey</B>*<B>edgez</B> &#197;<sup>3</sup>,
R<sub>insc</sub>=min(<B>edgex</B>,<B>edgey</B>,<B>edgez</B>)/2
</UL>
<LI><B>FCC </B> - Face-centered cubic boundary conditions
(FCC).<BR>
<U>Data</U>: <B>edg</B>
<UL>
<LI><B>edg</B>- The edge of the rombohedral dodecahedron is
<B>edg</B>*sqrt(3)/2
<LI>Vol=2*<B>edg</B><sup>3</sup>&#197;<sup>3</sup>,
R<sub>insc</sub>=<B>edg</B>/sqrt(2)
</UL>
<LI><B>TOCT</B> - Truncated octahedron boundary conditions.<BR>
<U>Data</U>: <B>edg</B>
<UL>
<LI><B>edg</B> - distance of the squrre face from the center
<LI>Vol=4*<B>edg</B><sup>3</sup>&#197;<sup>3</sup>,
R<sub>insc</sub>=<B>edg</B>(sqrt(3)/2)<BR>
The coordinate axes go through the center of the square faces.
</UL>
<LI><B>HCP </B> - Hexagonal close-packed boundary conditions.<BR>
<U>Data</U>: <B>edg</B>
<UL>
<LI><B>edg</B> - diameter of the inscribed sphere
<LI>Vol=<B>edg</B><sup>3</sup>/sqrt(2)&#197;<sup>3</sup>,
<LI>R<sub>insc</sub>=<B>edg</B>/2
</UL>
<LI><B>HEXG</B> - Hexagonal prism boundary conditions (HP).<BR>
<U>Data</U>: <B>plen</B>, <B>pedge</B>
<UL>
<LI><B>plen</B> - The length of the prism
<LI><B>pedge</B> - The edge of the hexagon.
<LI>Vol=<B>plen</B>*<B>pedge</B><sup>2</sup>*3*sqrt(3)
&#197;<sup>3</sup>,
R<sub>insc</sub>=min(<B>plen</B>, <B>pedge</B>*sqrt(3)/2.
<LI>The axis of the prism is the X axis and the vertex of the
hexagon goes through the Z axis.
For analyzing trajectories with different axis conventions, use the
<B><A HREF="#PXYZ">PXYZ</A></B> key.
</UL>
<LI><B>SPHR</B> - Sphere boundary conditions (SPH)<BR>
<U>Data</U>: <B>rsph</B>
<UL>
<LI><B>rsph</B>-radius of the sphere.
</UL>
<A NAME="PHS"></A>
<LI><B>PHS </B> - Primary hydration shell
(<A HREF="#PHSref">Beglov &amp; Roux</a>)<BR>
Variable to set to 'T' in the <A HREF="#PREP">preprocessor</A>: <B>PH</B><BR>
<u>Key2</u> (solute atom radius source):
<UL>
<LI><B>RVDW</B> - Standard Van der Waals radii will be used for
the solute atom radius.<BR>
<LI><B>RSIG</B> - Lennard-Jones SIG/2 values will be used for the
solute atom radius.<BR>
<u>Key3</u> (target energy definition):
<UL>
<LI><B>UTTO</B> - <B>uphsref</B> read is the
<i>total</i> restraining energy
<LI><B>UTPM</B> - <B>uphsref</B> read is the
<i>per molecule</i> restraining energy
<LI><B>UPML</B> - <B>uphsref</B> read is the
<i>per molecule</i> restraining energy, but solvents farther than
a threshold are ignored for the reference shell energy calculation
<UL>
<U>Data</U>: <B>uphsref</B>, <B>phsk</B>, <B>rphs</B>,
<B>rphsmin</B>, <B>rphsmax</B>, <B>denphs</B>, <B>nmcphs</B>,
<B>ranboxedge</B>,
<B>rphswid</B>
(only for Key3=<B>UPML</B>!),
<B>naphsrange</b>, <B>iaphsrange1</b>, <B>iaphsrange2</b>, ...
<UL>
<LI><B>uphsref</B> - reference hydration shell energy
<LI><B>phsk</B> - force constant of the restraining potential
<LI><B>rsph</B>, <B>rphsmin</B>, <B>rphsmax</B> - initial,
minimum and maximum radius of the shell, resp.
<LI><B>denphs</B> - solvent bulk density (used to calculate g(r)'s)
<LI><B>nmcphs</B> - frequency (in MC steps) of shell radius updates
<LI><B>ranboxedge</B> {1000.0} - edge of cube within which random
points will be generated for proximity RDFs
<LI><B>rsphwid</B> {<B>rphsmax</B>} - width of the
restraining shell contributing to
the per molecule restraining energy (used only with Key3=<B>UPML</B>).
<LI><B>naphsrange</b> {0} - the number of solute atom ranges the
primary hydration shell is restricted to (when <B>naphsrange</b>=0, all
solute heavy atoms will be used)
<LI><B>iaphsrange1</b>, <B>iaphsrange2</b> (<B>naphsrange</b> times)
- the first and last solute atoms of each range to use
</UL>
</UL>
</UL>
</UL>
<LI><B>READ</B> - Input boundary conditions.<BR>
<U>Fdata:</U>
<UL>
<LI>RECTYPE 29: <B>rinscr</B>, <B>vol</B>, <B> rmax</B> (3F10.0)
<LI><B>rinscr</B> - Radius of the largest inscribed sphere into
the cell.
<LI><B>vol</B> - Volume of the unit cell (in <sup>3</sup>)
<LI><B>rmax</B> - For minimum image convention the cutoff
radius used is <B>rmax</B>.
<LI>RECTYPE 39: <B>ncell</B> (I5)
<LI><B>ncell</B> - Number of cell-center coordinates to be read.
<LI>RECTYPE 38's: (<B>cic(</B>k,i),k=1,3) (3F15.0)
<LI><B>ncell</B> times repeated.
<LI><B>cic</B> - The <em u>negatives</em> of the
coordinates of center i (i.e., -X, -Y, -Z).
</UL>
</UL>
Note: Input boundary conditions work only if the image cell
can be found by comparing the distances of a point from the
various cell centers, the smallest distance will belong tot he
wanted cell number. It works when the periodic system
can be described by an orthogonal crystal coordinate
system. For other cases, more complicated calculations are
required - to start with, a transformation to
non-orthogonal coordinate system - but it is not
implemented yet.
<p>
For pure liquids, <B>FCC</B> is the best since this maximizes the
distance of a molecule from its image. On the other hand,
<B>RECT</B> is conceptually the simplest. Furthermore, for
<B>RECT</B> the dimensions of the cell can be different along the
three axes, allowing for various types of crystals. Solutes that
are far from spherical can also be wrapped around by waters more
efficiently.
(In this case, however, the solute rotation should be prevented
- see key <B><A HREF="#STEP">STEP</A></B>.) <B>HEXG</B> have been
developed for long
solutes or solutes representing an infinite polymer chain, to
maximize the distance of the chain from its image. <B>SPHR</B>
simply encloses the system into a sphere of inputted radius. Note
that the <B>FCC</B> calculations are only partially vectorized.
<p>
For the rectangular, face-centered cubic, hexagonal prism
and sphere boundary conditions there are special algorithms in
the program, one for each. When boundary conditions are inputted,
the cell number of the periodic image cell where a molecule is
located is determined simply by comparing the distance of the
molecule from the various cell centers read in.
<p>
The unit cell parameters are determined from the bulk
density or molar volume of the solution, N<sub>mol</sub>, and the
choice of boundary conditions. If the density of the system is d
g/ml , then
<p>
d (g/ml) = weight in g of N<sub>mol</sub> / volume of
N<sub>mol</sub> in ml =
<p>
(N<sub>mol</sub>/L)*Mol.wt / (Vol/10<sup>24</sup>)
<p>
where L is Avogadro's number. The cell parameter(s) can be
obtained from the resulting Vol value.
<p>
<A NAME="GCEN"></A>
III.02.05. <em u>Grand-canonical ensemble selection</em>
<p>
<B>GCEN</B>[ Key1 Key2 Key3 Key4 Key5 Data ]
Fdata
<P>
Prequisite: <B><A HREF="#PBCN">PBCN</A></B><BR>
Can not be followed by: <B><A HREF="#LIMG">LIMG</A></B><BR>
Prerequisite of:
<B><A HREF="#PRTG">PRTG</A></B>, <B><A HREF="#PRFI">PRFI</A></B>,
<B><A HREF="#BTUN">BTUN</A></B>, <B><A HREF="#STVG">STVG</A></B>,
<B><A HREF="#STPX">STPX</A></B>, <B><A HREF="#SACP">SACP</A></B><BR>
Related keys:
<B><A HREF="#IDLG">IDLG</A></B>,
<B><A HREF="#IDAG">IDAG</A></B><BR>
Variable to set to 'T' in the <A HREF="#PREP">preprocessor</A>: <B>CB</B><BR>
<p>
Grand-canonical ensemble simulation using the cavity-biased
method with <A HREF="#Mezei_87">grid-insertion</A> or the original,
<A HREF="#Mezei_82">random insertion method</A>.
<p>
<u>Key1</u> (GCE insertion strategy):
<UL>
<LI><B>OFF </B> - Return to canonical ensemble.<BR>
<LI><B>CAVB</B> - Cavity biased insertions at gridpoints<BR>
<LI><B>CVBF</B> - Cavity biased insertions at a rectangle around
the gridpoints<BR>
<LI><B>UNBI</B> - Unbiased (random insertions)<BR>
<u>Key2</u> (solute atom radius source):
<UL>
<LI><B>RVDW</B> - Standard Van der Waals radii will be used for
the solute atom radius.<BR>
<LI><B>RSIG</B> - Lennard-Jones SIG/2 values will be used for the
solute atom radius.<BR>
<LI><B>RSGH</B> - Like <B>RSIG</B>, but hydrogens with SIGMA=0 will
be assigned SIGMA=1.<BR>
<u>Key3</u> (insertion/deletion selection strategy):
<UL>
<LI><B>ALTI</B> - Alternating insertion and deletions attempts<BR>
<LI><B>RANI</B> - Randomly decided insertion and deletions
attempts<BR>
<u>Key4</u> (alternative Pcav definitions):
<UL>
<LI><B>ALLG</B> - Pcav is defined as the number of free grids /
total number of grids<BR>
<LI><B>SLVG</B> - Pcav is defined as the number of free grids /
number of grids not covered by the solute<BR>
<u>Key5</u> (strategy to use electrostatic term
for insertion/deletion steps):
<UL>
<LI><B>LJQI</B> - All potential terms are used for the I/D
decision.<BR>
<LI><B>LJNC</B> - The electrostatic contributions are
neglected for the insertion/deletion decision. Ensemble averages
for the thermodynamical quantities are not corrected
for the applied approximation.<BR>
<LI><B>LJCN</B> - As in <B>LJNC</B>, but the ensemble
averages for the thermodynamic quantities are corrected for by
utilizing the applied umbrella sampling.<BR>
<LI><B>LJCU</B> - As in <B>LJCN</B>, but the umbrella
sampling is applied to both the insertion/deletion and the regular
moves, and thermodynamic quantities are
corrected for the biasing.<BR>
<U>Data</U>: <B>ucspma</B>, <B>usmixp</B>.
<UL>
<LI><B>ucspma</B> - Average solute-solvent electrostatic
energy(in kcal/mole). If the supplied average electrostatic
solute-solvent energy is less than -1000 kcal/mole,
then the program uses the starting configuration's
value for the average solute-solvent electrostatic
energy. Umbrella sampling bias is E<sub>w</sub> =
<B>usmixp</B>*(<B>us1</B>-<B>n</B>*<B>ucspma</B>) where
<B>us1</B>=solute-solvent electrostatic
energy, and <B>n</B>=number of solvent molecules at a given
configuration.<BR>
<U>Fdata:</U>
<UL>
<LI>RECTYPE 13: <A NAME="GCENBA"><B>ba</B></a>,
<B>diamslv</B>,
<B>pmvslt</B>, <B>ngridx</B>, <B>ngridy</B>, <B>ngridz</B>,
<B>nslvxp</B>,
<B>nmolfx</B>, <B>idfreq</B>, <B>idrepf</B>, <B>nmcransh</B>
(3F10.0,6I5,2I10)
<LI><B>ba</B> - The B parameter of
<A HREF="#Adams">Adams</A>. The larger it is, the larger will the
chemical potential be.
<LI><B>diamslv</B> - The solvent diameter in , to be used as the cavity radius.
<LI><B>pmvslt</B> - The molar volume of the solute (to
be used in the density calculation).
<LI><B>ngridx</B>, <B>ngridy</B>, <B>ngridz</B> - The
number of gridpoints in the X, Y and Z directions (not used for
<B><A HREF="#GCEN">GCEN</A> UNBI</B> runs).
By default, the whole simulation cell is covered by the grid,
but a smaller portion can also be specified with the
<B><A HREF="#LIMG">LIMG</A></B> key.
<LI><B>nslvxp</B> - The targeted number of solvent molecules
referred to the key <B><A HREF="#STPX">STPX</A></B>
<LI><B>nmolfx</B> {0}- The number of solvent molecules that are
not allowed to be deleted.
<LI><B>idfreq</B> - The frequency of insertion/deletion
attempts (in MC steps).
<LI><B>idrepf</B> - The frequency of summary line (in
Ins/Del steps)
<LI><B>nmcransh</B> - The inital grid will be
regenerated after every <B>nmcransh</B>-th MC step with a random
shift in its position.
</UL>
</UL>
</UL>
</UL>
</UL>
</UL>
</UL>
Other aspects of a GCE simulation are controlled by the keys
<A HREF="#PRFI">PRFI</A></B>, <A HREF="#IDLG">IDLG</A></B>,
<A HREF="#IDAG">IDAG</A></B>, <A HREF="#STPX">STPX</A></B>,
<A HREF="#LIMG">LIMG</A></B>, and <A HREF="#BTUN">BTUN</A></B>.
<P>
<A NAME="STPX"></A>
III.02.06. <em u>Stop GCE run at nx</em>
<p>
<B>STPX</B> Data
<p>
Prerequisite: <B><A HREF="#GCEN">GCEN</A></B>,
<p>
<U>Data</U> <B>nslvxp</B>
<UL>
<LI><B>nslvxp</B> {<B>nslvxp</B> read with the
<B><A HREF="#GCEN">GCEN</A></B> key}
- target number of solvents to stop at.
</UL>
<p>
When this key is present the run will stop if the number of
solvent molecules reached <B>nslvxp</B>.
<p>
<A NAME="BTUN"></A>
III.02.07. <em u>Tune the GCE B parameter</em>
<p>
<B>BTUN</B> Key1 [Data]
<P>
Prerequisite: <B><A HREF="#GCEN">GCEN</A></B>,
<B><A HREF="#MOND">MOND</A></B> or <B><A HREF="#BLKW">BLKW</A></B><BR>
<P>
<u>Key1</u> (tuning algortithm)
<UL>
<LI><B>NONE</B> - Turn off tuning, continue with the last B value
(not recommended)
<LI><B>AVRG</B> - Turn off tuning, continue with the average B value
over the last run
<LI><B>ADPT</B> - Adaptive algorithm
<LI><B>FLUC</B> - Fluctuation-based algorithm
<LI><B>PICL</B> - <A HREF="#BTUNEref">Process control algorithm</a><BR>
<U>Data</U> for Key1=<B>ADPT</B> and Key1=<B>FLUC</B>:
<B>targetden</B>, <B>nmctunave</B>, <B>nmctunskip</B>,
<B>chabmax</B>, <B>dentol</B>, <B>inorout</B>
<UL>
<LI><B>targetden</B> {0.997} - The targeted density in the bulk
region as defined by the <B><A HREF="#MOND">MOND</A></B> key.
<LI><B>nmctunave</B> {100000} - The number of MC steps to use for
collecting data before each B parameter change.
<LI><B>nmctunskip</B> {<B>nmctunave</B>} - The number of MC steps
to skip (as equilibration) after each B parameter change.
<LI><B>chabmax</B> {1.0} - The maximum absolute change allowed for
the B parameter.
<LI><B>dentol</B> {0.01} - The density range around
<B>targetden</B> within which the tuning is considered converged.
<LI><B>inorout</B> {2} - when =1, the region inside the cube defined
by the <B><A HREF="#MOND">MOND</A></B> key will be assumed bulk;
when =2, the region outside the cube is assumed bulk.
<LI><B>targetn</B> {-50.0} - the targeted number of solvent molecules
in the region selected by <B>inorout</B>. If it is positive, it will
override the <B>targetden</B> value read above.
</UL>
<U>Data</U> for Key1=<B>PICL</B>:
<B>targetden</B>, <B>integraltime</B>, <B>gain</B>, <B>inorout</B>
<UL>
<LI><B>targetden</B> {0.997} - The targeted density in the bulk
region as defined by the <B><A HREF="#MOND">MOND</A></B> key.
<LI><B>integraltime</B> {13000.0} - integraltime
<LI><B>gain</B> {-113.0} - gain
<LI><B>inorout</B> {2} - see above.
<LI><B>targetn</B> {-50.0} - see above.
</UL>
</UL>
The program will periodically change the GCE <A HREF="#GCENBA">B
parameter</A>
to achieve the targeted density in the 'bulk' region - defined as
the outside region by the <B><A HREF="#MOND">MOND</A></B> key.
<p>
Note that the keys <B>NONE</B> and <B>AVRG</B> are only meaningful
after having restored a checkpoint file
(key <B><A HREF="#RCKP">RCKP</A></B>).
<p>
<A NAME="LIMG"></A>
III.02.08. <em u>Limit the cavity grid</em>
<p>
<B>LIMG</B> [Key1,Data]
<P>
Prerequisite:
<B><A HREF="#PBCN">PBCN</A></B><BR>
Potential successor:
<B><A HREF="#GCEN">GCEN</A></B><BR>
<P>
<u>Key1</u> (used only for GCE simulations to specify treatment of
boundary crossings):
<UL>
<LI><B>NOWA</B> - Free movement of solvents in and out of the grid
region
<LI><B>WARC</B> - Free movement of solvents in and out of the grid
region but counts of border crossings are maintained
<LI><B>WALL</B> - Movement of solvents in and out of the grid
region
is prevented, counts of crossing attempts are maintained<BR>
<U>Data</U>:
<B>glimminx</B>, <B>glimmaxx</B>, <B>glimminy</B>, <B>glimmaxy</B>,
<B>glimminz</B>, <B>glimmaxz</B>
<UL>
<LI>The limited grid will extend from
<B>glimminx</B> to <B>glimmaxx</B>,
<B>glimminy</B> to <B>glimmaxy</B>,
<B>glimminz</B> to <B>glimmaxz</B> in the x, y, and z directions,
respectively.
For any <B>glim*</B>value that reaches or exceeds the limits of the
simulation
cell (centered at (0,0,0), the periodic boundary conditions will
also be invoked.
</UL>
</UL>
<A NAME="IBEN"></A>
III.02.09. <em u>Isothermal isobaric ensemble simulation (TPN)</em>
<p>
<B>IBEN</B> Key1, Key2, Data
<P>
Can not be followed by:
<B><A HREF="#VCHA">VCHA</A></B><BR>
Potential successors:
<B><A HREF="#CNFG">CNFG</A></B>, <B><A HREF="#VCHA">VCHA</A></B><BR>
Prerequisite of:
<B><A HREF="#LCMP">LCMP</A></B><BR>
Related key:
<B><A HREF="#LCMP">LCMP</A></B>.
<p>
Variable to set to 'T' in the <A HREF="#PREP">preprocessor</A>: <B>IB</B>
<P>
<u>Key1</u> (volume sampling technique):
<UL>
<LI><B>NONE</B> - Turn off (TPN) ensemble sampling
<LI><B>UNIF</B> - (TPN) ensemble sampling with uniformly
distributed steps
<LI><B>VB3D</B> - (TPN) ensemble sampling with
<A HREF = "#Mezei_83">virial-biased</A> sampling of Mezei.
<LI><B>VB1D</B> - (TPN) ensemble sampling with refinement of the
<A HREF = "#Mezei_83">virial-biased</A> sampling of Mezei,
using the components of the virial sum to generate cell size
changes
in the X, Y, and Z directions, respectively.<BR>
<u>Key2</u> (volume change (un)isotropy):
<UL>
<LI><B>ISOT</B> - Volume change involves all three axes
<LI><B>IXYZ</B> - Volume change alternates between changes in the
X, Y or Z directions.
<LI><B>ISYZ</B> - Volume change alternates between changes in the
X dimensions and in the Y-Z dimensions.
directions<BR>
<U>Data</U>:
<UL>
<LI><B>press</B> {1.0} - The external pressure specified (in
atm)
<LI><B>vrange</B> {100.0} - The volume change range (in
&#197;<sup>3</sup>)
<LI><B>nvchfreq</B> {nmolec} -
The frequency of volume change attmpts (in MC steps)
<LI><B>nvchrep</B> {nmolec} - The frequency of reports on
volume change
attmpts (in number of volume change attempts)
<LI><B>vlam</B> {0.5} - The virial biasing lambda factor.<BR>
For Key2=<B>IXYZ</B>:
<LI><B>ixyzfreeze(1)</B>, <B>ixyzfreeze(1)</B>, <B>ixyzfreeze(1)</B>
{0,0,0} - If <B>ixyzfreeze(k)</B> is &gt; 0 then no change will be done along
the <b>k</b>-th axis.
</UL>
</UL>
</UL>
Note, that putting the key <B>IBEN</B> after <B>CNFG</B>
will result in ignoring the cell size on the <b>.crd</b> file
and the vaulues inputted with the <B>PBCN</B> key will be used
instead.
<p>
<A NAME="LCMP"></A>
III.02.10. <em u>Virial-biasig factor</em>
<p>
<B>LCMP</B> Data
<p>
Prerequisite:
<B><A HREF="#IBEN">IBEN</A></B><BR>
<p>
<U>Data</U>:
<UL>
<LI><b>complfac</b> {0.5} - The scaling factor to use for virial
biasing (corresponding to the lambda factor for force biasing).
</UL>
<p>
<A NAME="CNFG"></A>
III.02.11. <em u>Initial configuration</em>
<p>
<B>CNFG</B> Key1 [ Key2 Key3 Data ] [Fdata]
<p>
Prerequisites:
<B><A HREF="#SLTA">SLTA</A></B>, <B><A HREF="#FILE">FILE</A></B><BR>
Can not be followed by:
<B><A HREF="#FREE">FREE</A></B>, <B><A HREF="#TORD">TORD</A></B>,
<B><A HREF="#CMPC">CMPC</A></B>, <B><A HREF="#NSLV">NSLV</A></B>,
<B><A HREF="#IBEN">IBEN</A></B>, <B><A HREF="#VCHA">VCHA</A></B>,
<B><A HREF="#PXYZ">PXYZ</A></B><BR>
Potential successors:
<B><A HREF="#RCKP">RCKP</A></B>, <B><A HREF="#SPST">SPST</A></B><BR>
Prerequisite of:
<B><A HREF="#RUNS">RUNS</A></B>, <B><A HREF="#SCAL">SCAL</A></B>,
<B><A HREF="#PRTG">PRTG</A></B>, <B><A HREF="#REGE">REGE</A></B>,
<B><A HREF="#SPRD">SPRD</A></B>, <B><A HREF="#VORO">VORO</A></B>,
<B><A HREF="#STIR">STIR</A></B>, <B><A HREF="#SCRM">SCRM</A></B>,
<B><A HREF="#TORT">TORT</A></B>, <B><A HREF="#LPST">LPST</A></B>,
<B><A HREF="#GENT">GENT</A></B><BR>
Related keys:
<B><A HREF="#GENT">GENT</A></B>,
<B><A HREF="#SPRD">SPRD</A></B>,
<B><A HREF="#SCRM">SCRM</A></B>,
<B><A HREF="#STIR">STIR</A></B>.
<p>
<u>Key1</u> (data source):
<UL>
<LI><B>READ</B> : Initial configuration is read from user supplied
<B>.crd</B> file. For Key2=<B>PDB </B> or Key2=<B>CHRM</B>,
the extensions <B>.pdb</B> and <B>.CRD</B>, resp., will be
assumed first.<BR>
<LI><B>RDEX</B> : When part of the system is only used for grid
generation (see key <B><A HREF="#FLEX">FLEX</A></B>) the full
configuration is read from a PDB or Charmm CRD file and the
moving atoms will be extracted and saved in ASCI format on 
a <B>.crd</B>file.<BR>
<u>Key2</u> (file format):
<UL>
<LI><B>ASCI</B> - Coordinate file will be read/written in
ASCII.<BR>
<LI><B>BNRY</B> - Coordinate file will be read/written in
binary.<BR>
<LI><B>ASAN</B> - Coordinate file is an annotated file written
with <B><A HREF="#WCNF">WCNF</A> SVAN</B>.<BR>
<LI><B>PDB </B> - Coordinate file is imported from
<A HREF="https://www.rcsb.org/pdb/">PDB</A> format<BR>
If the program is dimensioned for generic sites
(<b>nmolec</b> &lt; #MH) then the solvent fractional occupancies and
RMS's are also read from the last two data columns.
<LI><B>CHRM</B> - Coordinate file is imported from CHARMM CRD format<BR>
<u>Key3</u> (configuration check procedure):
<UL>
<LI><B>NOFX</B> - Abort if solute on the <B>.crd</B> file
differs form the input definition<BR>
<LI><B>FXCR</B> - If solute on the <B>.crd</B> file differs
form the input definition by more than 0.01 &#197;,
use the input definition for both<BR>
<LI><B>FXCC</B> - If solute on the <B>.crd</B> file differs
form the input definition, by more than 0.001,
use the input definition for both<BR>
<LI><B>FXIN</B> - If solute on the <B>.crd</B> file differs
form the input definition, use one on the <B>.crd</B> file for
both<BR>
<LI><B>IGND</B> - Ignore if solute on the <B>.crd</B> file
differs form the input definitition<BR>
<U>Data</U>: <B>numrunr</B>, <B>irunvers</B>, <B>jobnamer</B>
<UL>
<LI><B>numrunr</B> {<B>numrun</B>},
<B>irunvers</B> {1},
<B>jobnamer</B> {<B>jobname</B>} - The program will look for the file
<B>&lt;jobnamer&gt;.&lt;numrunr&gt;.crd</B>
(<B>&lt;jobnamer&gt;.crd</B> when <b>numrunr</b>=1) to read the
coordinates from. If the filename is different from the
default coordinate file name, the program will open a new
coordinate file name using <b>numrun</b> as well, and copy the
coordinates there.
</UL>
</UL>
</UL>
<LI><B>RANC</B> - The starting configuration is obtained by
<UL>
<LI>Placing the solute
(as defined by the <B><A HREF="#SLTA">SLTA</A></B> key)
at the cell center,
<i>with its COM shifted to the cell center</i>.
This shift is added to the information that was read by the
<B><A HREF="#SLTA">SLTA</A></B> key.
<LI>Inserting the number of solvents
requested by the <B><A HREF="#NSLV">NSLV</A></B> key at random
location and orientation. Overlap with the solute van der Waals
envelop is prevented.
<LI>When solute molecules are cloned, the cloned copies will be
placed at randomly selected points with randomly selected
orientations in the cell,
with their COM outside the rest of the solute.
No tests for overlap will be performed.
<LI>Force bias and the calculation of
distribution functions will be turned off.
</UL>
See also the key <B><A HREF="#STIR">STIR</A></B>
for the elimination of persistent contacts.<BR>
<u>Key2</u> (file format): see <B><A HREF="#CNFG">CNFG</A> READ</B>
<UL>
<u>Key3</u> (determination of cell size and number of solvents):
<UL>
<LI><B>SIZE</B> - The cell dimensions given
by the <B><A HREF="#PBCN">PBCN</A></B> key
and the number of solvents given by the
<B><A HREF="#NSLV">NSLV</A></B>
key will be used<BR>
<LI><B>NTOS</B> - Cell dimensions determined from externally
provided partial molar volumes and number of solvent molecules.
given by the
<B><A HREF="#NSLV">NSLV</A></B> key.
For
<B><A HREF="#PBCN">PBCN</A> RECT</B>: all edges are the same;
for <B><A HREF="#PBCN">PBCN</A> HEXG</B>: provide the
prism length as plen with <B><A HREF="#PBCN">PBCN</A> HEXG</B><BR>
<LI><B>STOS</B> - Cell dimensions are determined from
externally provided solute and solvent partial molar volumes and
number of required shells. Data inputted with the keys
<B><A HREF="#NSLV">NSLV</A></B> and <B><A HREF="#PBCN">PBCN</A></B>
will be disregarded.<BR>
<LI><B>STON</B> - Number of solvents is determined from the
externally provided partial molar volumes and cell dimensions
given by the <B><A HREF="#PBCN">PBCN</A></B> key.<BR>
<U>Data</U>: <B>pmvslt</B>, <B>pmvslv</B>, <B>rnshll</B>,
<B>cplpar</B>
<UL>
<LI><B>pmvslt</B> - Partial molar volume of the solute in
ml/mole.<BR>
<LI><B>pmvslv</B> {18.12004 (water at 25 deg C)} - Partial
molar volume of the solvent in ml/mole.<BR>
<LI><B>rnshll</B> {2} - The number of solvent shells required
around the solute in the simulation cell (only used for
Key3=<B>STOS</B>)<BR>
<LI><B>cplpar</B> {0.0} The coupling parameter value to use
for the creation of the third copy (for
<B><A HREF="#FREE5">FREE&nbsp;PMF1</A></B> or
<B><A HREF="#FREE6">FREE&nbsp;PMNL</A></B>runs)<BR>
</UL>
</UL>
</UL>
<LI><B>RANI</B> - Same as <B>RANC</B>, except that the solute will
be placed into the simulation cell unchanged (i.e., as defined by the
<B><A HREF="#SLTA">SLTA</A></B> key).<BR>
<LI><B>TRAJ</B> : The initial configuration is gathered from the
trajectory file opened previously.<BR>
<u>Key2</u>(file format): see <B><A HREF="#CNFG">CNFG</A> READ</B><BR>
<LI><B>RPSU</B> : A built-in procedure is used. The first molecule
of an old system is replaced by the solute. The COM of the old and
new solute will coincide. Solvents overlapping with the Van der
Waals envelope of the new solute will be discarded. The
resulting configuration will be written on the <B>.crd</B> file.
Also, <B><A HREF="#DSTC">DSTC</A> NONE</B> and
<B><A HREF="#SAMP">SAMP</A> METC</B> will be set.<BR>
<u>Key2</u>(file format): see <B><A HREF="#CNFG">CNFG</A> READ</B>
<UL>
<U>Data</U>: <B>numrunr</B>, <B>jobname</B>, <B>noslt</B>
<UL>
<LI><B>jobname</B>, <B>numrunr</B> - Old configuration is to be
read from file <B>&lt;jobname&gt;.&lt;numrunr&gt;.crd</B><BR>
<LI><B>noslt</B> - Number of solute atoms in the old
configuration.<BR>
<U>Fdata:</U>
<UL>
<LI>RECTYPE 23: <B>iclslt</B>(1),...,<B>iclslt</B>(<B>noslt</B>)
(14I5)
<LI><B>iclslt</B>(i) - Atomic number (for EPEN potential) or
atom type (for Clementi potential) of the I-th atom in the old
solute.
</UL>
</UL>
</UL>
<LI><B>RPUV</B> : The starting configuration is obtained from an
old system by replacing both the solute and solvent molecules in
such a way that the orientation matrices are preserved. Both
the number of solute and solvent atoms can change.<BR>
<u>Key2</u>: see <B><A HREF="#CNFG">CNFG</A> READ</B>
<UL>
<U>Data</U>: <B>numrunr</B>, <B>jobname</B>, <B>noslt</B>,
<B>noslv</B>, <B>iop37o</B>
<UL>
<LI><B>jobname</B>, <B>numrunr</B> - Old configuration is to be
read from file <B>&lt;jobname&gt;.&lt;numrunr&gt;.crd</B><BR>
<LI><B>noslt</B> - Number of solute atoms in the old
configuration.<BR>
<LI><B>noslv</B> - Number of solvent atoms in the old
configuration.<BR>
<LI><B>iop37o</B> - Potential identifier of the old solvent
(0-1-2 &lt;=&gt; MCY-TIP3P-TIP4P).<BR>
<U>Fdata:</U>
<UL>
<LI>RECTYPE 25's: (not needed for
<B><A HREF="#SLTA">SLTA</A> SLVT</B>).<BR>
<LI><B>noslt</B> records - one for every atom of the old
solute:<BR>
<LI><B>ioslt</B>(i),(<B>coslt</B>(j,i),j=1,3) (I5,3F15.10)<BR>
<LI><B>ioslt</B> - Atomic number for atom i (6 for C, 8 for
O, etc.) for EPEN potential and atom type for all the other
potentials.<BR>
<LI><B>coslt</B> - Coordinates of the old solute.<BR>
<LI>RECTYPE 26's: (read only when <B>iop37o</B> .ne. 0)<BR>
<LI><B>noslv</B> records - one for every atoms of the old
solvent:<BR>
<LI><B>ioslv</B>(i),(<B>coslv</B>(j,i),j=1,3) (I5,3F15.10)<BR>
<LI><B>ioslv</B> - atomic number for solvent atom i.<BR>
<LI><B>coslv</B> - X,Y,Z coordinates of atom i in the old<BR>
solvent.
</UL>
</UL>
</UL>
<LI><B>PMFN</B> - Create an input file for conformational
transition from a file with a fixed solute conformation. It takes
the conformation given after <B><A HREF="#SLTA">SLTA</A></B>
(RECTYPEs 14) as R<sub>o</sub> reads in the R<sub>1</sub>
conformation
(together with new charges) and prepares the linear combination of
R<sub>o</sub> and R<sub>1</sub>.
All three solute conformations
will be stored as a single solute on the <B>.crd</B> file, along
with the value of the parameter CP (read from input). Note, that it
assumes that the solute in the input configuration is in the
R<sub>o</sub> conformation.<BR>
<u>Key2</u>: see <B><A HREF="#CNFG">CNFG</A> READ</B><BR>
<UL>
<U>Data</U>: <B>numrunr</B>, <B>jobname</B>, <B>nstfa0new</B>,
<B>cplpar</B>
<UL>
<LI><B>jobname</B>, <B>numrunr</B> - The input file containing
a full configuration with the solute in its initial conformation is
<B>&lt;jobname&gt;.&lt;numrunr&gt;.crd</B> . The solute
conformation has to be the one described by the data after
<B><A HREF="#SLTA">SLTA</A></B>.<BR>
<LI><B>nstfa0new</B> - the number of FE solute atoms in the
file <B>&lt;jobname&gt;.&lt;numrunr&gt;.crd</B><BR> .
<LI><B>cplpar</B> - The starting value of the coupling
parameter CP.<BR>
<U>Fdata:</U>
<UL>
<LI>RECTYPE 31: <B>iclslt</B>(i),(<B>cslt</B>(j,i),j=1,3),
<B>qslt</B>(i) (I5,4F15.5)
<LI><B>iclslt</B>, <B>cslt</B>, <B>qslt</B> - Description of
the second solute conformation. Identical meaning as for the data
after <B><A HREF="#SLTA">SLTA</A></B>. The number of real solute
atoms is <B>nslt</B>
in the above description. This same number should appear also
with the keyword <B><A HREF="#SLTA">SLTA</A></B>. However, the
program will
replace it with 3*<B>nslt</B>, corresponding to the fact that
it stores three copies of the free-energy solute, one each of the
it stores three copies of the free-energy solute, one each of the
two extremal conformations and the combined
conformation.
</UL>
</UL>
</UL>
<LI><B>RPUU</B> - Same as <B>RPSU</B>, but no solvent molecule is
discarded.<BR>
</UL>
<A NAME="SCRM"></A>
III.02.12. <em u>Scramble torsion angles</em>
<p>
<B>SCRM</B>
<P>
Prerequisite: <B><A HREF="#CNFG">CNFG</A></B><BR>
<P>
When this key is present, all torsion angles are given a random value and
the solute is regenerated in this random conformation.
<P>
<A NAME="SCAL"></A>
III.02.12. <em u>Simulation cell rescaling</em>
<p>
<B>SCAL</B> Data
<P>
Prerequisite:
<B><A HREF="#CNFG">CNFG</A></B><BR>
<P>
<U>Data</U>: <B>e1</B>, <B>e2</B>, <B>e3</B>
<UL>
<LI><B>e1</B>, <B>e2</B>, <B>e3</B> - Same meaning as <B>edgex</B>,
<B>edgey</B>, <B>edgez</B> for the <B><A HREF="#PBCN">PBCN</A></B>
key, defining the new cell. The COM's of the solvents will be
scaled
to fill in the new box evenly.
</UL>
<A NAME="OVST"></A>
III.02.13. <em u>Overlap of the second solute copy with the
first</em>
<p>
<B>OVST</B> Data
<P>
<U>Data</U>: <B>i1</B>, <B>i2</B>, <B>i3</B>
<UL>
<LI><B>i1</B>, <B>i2</B>, <B>i3</B> - The second copy of the solute
will be shifted and rotated to best overlap with the first. The
overlap is based on the coordinates of atoms <B>i1</B>, <B>i2</B>,
<B>i3</B>. If any of them are zero, the first three atoms will be
chosen. This is useful for free-energy calculations to reduce the
change in the solute that the calculation has to model.
</UL>
<A NAME="SPRD"></A>
III.02.14. <em u>Spread out the solute molecules for display</em>
<p>
<B>SPRD</B> [Data]
<P>
Prerequisite: <B><A HREF="#CNFG">CNFG</A></B><BR>
<P>
<U>Data</U>:
<UL>
<LI><B>numrunw</B> {<B>numrun</B>} - the runnumber of the PDB file
to be written.
</UL>
This key generates a PDB file containing the solute molecules
spread out
in the x-y plane so that they don't overlap.
<P>
<A NAME="PBGR"></A>
III.02.15. <em u>Force solute reset with grids</em>
<p>
<B>PBGR</B>
<p>
When this key is present, the grid calculations will reset
each solute atom to the periodic cell before calculating
the cover list. This is a safety option, since it is automatically
set when the inital structure contains solute atoms outside the
simulation cell.
<p>
<A NAME="LIGA"></A>
III.02.16. <em u>Multi-copy ligand</em>
<p>
<B>LIGA</B> Data
<p>
<U>Data</U>:
<UL>
<LI><B>ligands</B> -
The number of solute molecules that are different copies
of the same ligand (they will not see each other).
</UL>
<p>
<A NAME="STUN"></A>
III.02.17. <em u>Stepsize tuning</em>
<p>
<B>STUN</B> Key1, Key2, Key3, Data
<p>
Prerequisite:
<B><A HREF="#MVRT">MVRT</A></B><BR>
Can not be followed by:
<B><A HREF="#MVRT">MVRT</A></B><BR>
Prerequisite of:
<B><A HREF="#TAC0">TAC0</A></B><BR>
<P>
Related key: <B><A HREF="#TAC0">TAC0</A></B>.
<p>
<u>Key1</u> (move type):
<UL>
<LI><B>TRAN</B> - Tune solute molecule translation stepsize
<LI><B>ROTA</B> - Tune solute molecule rotation stepsize
<LI><B>TORS</B> - Tune solute molecule torsion stepsize
<LI><B>LOOP</B> - Tune solute molecule loop move stepsize<BR>
<u>Key2</u> (tune type):
<UL>
<LI><B>PICL</B> - <A HREF="#STUNEref">Process control algorithm for tuning</a>.
<LI><B>PCLG</B> - <A HREF="#STUNEref">Process control algorithm for tuning</a>;
tuning is done on the logarithm of the acceptance rate.
<LI><B>OFFW</B> - Turn of tuning, keep current average stepsize(s),calculated
with the weight <b>wstepsum</b>
<LI><B>OFFC</B> - Turn of tuning, keep current cumulative average stepsize(s)
<LI><B>OFF </B> - Turn of tuning, keep current (tuned) stepsize(s)
</UL>
<U>Data</U>: <b>targetacc</b>, <b>wsum</b>, <b>timeint</b>, <b>gain</b>,
<b>wstepsum</b>, <b>tunedstepmax</b>
<UL>
<LI><b>targetacc</b> {0.3} -  target acceptance rate (&lt; 1.0)
<LI><b>wsum</b> {0.999} - current acceptance rate Pacc<sub>i</sub>
for tuning purposes is <BR>
Pacc<sub>i</sub> - wsum*Pacc<sub>i-1</sub> + (1-wsum)*iacc<sub>i</sub><BR>
where iacc<sub>i</sub> is 1 for accepted and 0 for rejected moves.
<LI><b>timeint</b> {2000.0} - integraltime of tuning
<LI><b>gain</b> {-50.0} -  gain of tuning
<LI><b>wstepsum</b> {wsum} - weight factor used to calculate the
average tuned stepsizes, analogously to the calculation of the
acceptance rates.
<LI><b>tunedstepmax</b> - the maximum value the tuned stepsize(s) can
take. Default value is 2  for Key1=<B>TRAN</B>, 180<sup>o</sup> otherwise.
</UL>
</UL>
</UL>
Note that the solute sampling should be set to alternating
translation and rotation with the
<B><A HREF="#MVRT">MVRT</A></B>
key. For cloned solute molecules (see the key
<B><A HREF="#CLON">CLON</A></B>) the tuning will generate identical
stepsizes for each instance of a torsion angle.
<p>
<A NAME="TAC0"></A>
III.02.19. <em u>Stepsize accumulator reset</em>
<p>
<B>TAC0</B> Key1, Data
<p>
Prerequisite: <B><A HREF="#STUN">STUN</A></B>
<p>
<u>Key1</u> (move type): same as Key1 of
<B><A HREF="#STUN">STUN</A></B>
<p>
<U>Data</U>: <b>nmc_zeroacc</b>
<UL>
<LI><b>nmc_zeroacc</b> - the stepsize accumulator for the step type
specified by Key1 will be reset at the <b>nmc_zeroacc</b>-th MC step.
<b>nmc_zeroacc</b>
has to be a multiple of the frequency of printing full results
(<B>nplt</B> read by the <B><A HREF="#RUNS">RUNS</A></B> key).
This option is designed to work with
<B><A HREF="#STUN">STUN</A> **** OFFC</B>.
</UL>
<p>
<A NAME="SANN"></A>
III.02.19. <em u>Simulated annealing</em>
<p>
<B>SANN</B> Key1, Data
<p>
Prerequisite: <B><A HREF="#TEMP">TEMP</A></B>
<p>
<u>Key1</u> (annealing schedule):
<UL>
<LI><B>NONE</B> - turn off simulated annealing
<LI><B>LINE</B> - temperature is decreased by constant increments
<LI><B>EXPO</B> - temperature at the nstep-th step is set as
T=T<sub>0</sub>*exp(-nstep*expfac)
<LI><B>LOG</B> - temperature at the nstep-th step is set as
T=coef_log/ln(nstep+2)
<LI><B>LIST</B> - temperature values at each step are specified from input<BR>
<U>Data</U> for Key1=<B>LINE</B>: <b>nmctempstep</b>, <b>tempinc</b>
<UL>
<LI> <b>nmctempstep</b> - temperature is changed at every
<b>nmctempstep</b> MC step
<LI><b>tempinc</b> - temperature decrement at each step
<UL>
<LI>If <b>tempinc</b>=0.0 then <b>tempfin</b>, the final temperature;
MMC will calculate <b>tempinc</b>
</UL>
</UL>
<U>Data</U> for Key1=<B>EXPO</B>: <b>nmctempstep</b>, <b>expfac</b>
<UL>
<LI> <b>nmctempstep</b> - temperature is changed at every
<b>nmctempstep</b> MC step
<LI><b>expfac</b> - temperature decrement exponent at each step
<UL>
<LI>If <b>expfac</b>=0.0 then <b>tempfin</b>, the final temperature;
MMC will calculate <b>expfac</b>
</UL>
</UL>
<U>Data</U> for Key1=<B>LOG</B>: <b>nmctempstep</b>, <b>coef_log</b>,
<UL>
<LI> <b>nmctempstep</b> - temperature is changed at every
<b>nmctempstep</b> MC step
<LI><b>coef_log</b> - prefactor of the logarithmic formula
<UL>
<LI>If <b>coef_log</b>=0.0 then <b>tempfin</b>, the final temperature;
MMC will calculate <b>coef_log</b>
</UL>
</UL>
<U>Data</U> for Key1=<B>LIST</B>: <b>nmctempstep</b>, <b>ntemplist</b>,
<b>templist(1)</b>,...,<b>templist(ntemplist)</b>
<UL>
<LI> <b>nmctempstep</b> - temperature is changed at every
<b>nmctempstep</b> MC step
<LI><b>ntemplist</b> - number of temperature values to read (&lt; 101)
<LI><b>templist(i)</b> - the temperature at the i-th temperture step.
After 100 steps, the temperature stays at the last list value.
The run stops if a negative value is reached.
</UL>
</UL>
<p>
<A NAME="SACP"></A>
III.02.20. <em u>Simulated annealing of the chemical potential</em>
<p>
<B>SACP</B> Key1, Data
<p>
Prerequisite: <B><A HREF="#GCEN">GCEN</A></B>
<p>
<u>Key1</u> (iteration length)
<UL>
<LI><B>FIXS</B> - Iteration lengths are fixed, independent of the number of molecules
<LI><B>SCLS</B> - Iteration length is proportional to the number of molecules
</UL>
<p>
<p>
<U>Data</U>: <b>bincr</b>, <b>nmcsacpstep</b>, <b>fract_lim</b>
<UL>
<LI><b>bincr</b> - B parameter increment
<LI><b>nmcsacpstep</b> - for Key1=<B>FIXS</B>
the B parameter will decreased by <b>bincr</b> at
every <b>nmcsacpstep</b> MC step;
for Key1=<B>SCLS</B> the B parameter will decreased at every 
<b>nmolec</b>*<b>nmcsacpstep</b> MC step.
<LI><b>fract_lim</b> - Cavity biasing will be turned off if the fraction of
the number of free grids exceeds <b>fract_lim</b>
</UL>
When this option is used and the trajectory writing 
(key <B><A HREF="#TRAJ">TRAJ</A></B>)
is not used, at each B parameter change the solvent coordinates
will be saved on the history file as PDB MODELs. 
<p>
<A NAME="Input_FREE"></A>
III.3. <B>Free-energy calculation keywords</B>
<OL>
<LI><TT><B><A HREF="#FREE">FREE</A></B>: </TT>Free energy
options<BR>
<LI><TT><B><A HREF="#RAUS">RAUS</A></B>: </TT>AUS parameter
change<BR>
<LI><TT><B><A HREF="#WMAT">WMAT</A></B>: </TT>PMF matching<BR>
<LI><TT><B><A HREF="#WPLT">WPLT</A></B>: </TT>AUS iteration plot<BR>
<LI><TT><B><A HREF="#TIQU">TIQU</A></B>: </TT>TI quadrature<BR>
<LI><TT><B><A HREF="#OVRA">OVRA</A></B>: </TT>Overlap ratio
<LI><TT><B><A HREF="#FETK">FETK</A></B>: </TT>Temperature scaling for TI
evaluation<BR>
</OL>
<A NAME="FREE"></A>
III.03.01. <em u>Free energy options</em>
<p>
<B>FREE</B> Key1, [Key2-4 Data Fdata]
<p>
Prerequisite: <B><A HREF="#PBCN">PBCN</A></B><BR>
Potential successors:
<B><A HREF="#CNFG">CNFG</A></B>, <B><A HREF="#SLTA">SLTA</A></B>,
<B><A HREF="#TORD">TORD</A></B>, <B><A HREF="#RCKP">RCKP</A></B>,
<B><A HREF="#PARD">PARD</A></B><BR>
Related keys:
<B><A HREF="#TIQU">TIQU</A></B>,
<B><A HREF="#WMAT">WMAT</A></B>,
<B><A HREF="#WPLT">WPLT</A></B>.
<p>
<u>Key1</u> (free-energy method choice):
<UL>
<A NAME="FREE1">
<LI><B>WIDO</B> - Excess chemical potential is calculated using the
Widom insertion method with cavity modification.
This option requires either a subsequent <B><A
HREF="#SCAN">SCAN</A></B> key
or the <B><A HREF="#PXAN">PXAN</A></B> key before the
<B><A HREF="#RUNS">RUNS</A></B> key<BR>
<U>Data</U>: <b>rspha</b>, <b>ngridx</b>, <b>ngridy</b>,
<b>ngridz</b>,
<b>ngtry</b>, <b>nitry</b>, <b>ishftx</b>, <b>ishfty</b>,
<b>ishftz</b>, <b>nmolwid</b>; <b>nmolwid</b> times
(<b>ifdummy</b>, <b>ew0</b>)
<UL>
<LI><b>rspha</b> - Cavity radius (solute atom sizes are defined
by the vdw radii see key <B><A HREF="#PRAT">PRAT</A></B>)<BR>
<LI><b>ngridx</b>, <b>ngridy</b>, <b>ngridz</b> -
Number of grid points in the x, y, z directions, respectively.
By default, the whole simulation cell is covered by the grid,
but a smaller portion can also be specified with the
<B><A HREF="#LIMG">LIMG</A></B> key (preceding the
<B><A HREF="#FREE">FREE</A></B> key)<BR>
<LI><b>ngtry</b> {1} - Number of different grids (different random shifts)
to use for each configuration<BR>
<LI><b>nitry</b> {1} - Number of insertion trys per grid<BR>
<LI><b>ishftx</b> {0}, <b>ishfty</b> {0}, <b>ishftz</b> {0} -
When isftx=1, no grid shift is applied in the x direction;
same for y and z.
When isftx=2, succesive grids will be centered at i/ngtry.
<BR>
<b>nmolwid</b> {1} - Number of different solutes to calculate free energy for.<BR>
For each solute:
<UL>
<LI><b>ifdummy</b> {0} - When ifdummy=1, the first atom of the
inserted molecule is assumed to be non-interacting (used only to
ensure the best centering into the cavity).<BR>
<LI><b>ew0</b> {0.0} - expected energy value (will be subtracted
from all calculated energies before forming exponentials, added
back at the end)
</UL>
</UL>
<LI><B>ADDW</B> - An excess chemical potential-like quantity
is calculated for the mutation of a solute atom into
a new group, using the Widom insertion method formalism.
In the figure below, atom <b>X</B> will be replaced
by different groups. These groups have to be defined as
separate molecules, counted as free-energy molecules and
they have to include at least the atoms that are
in 1-4 position to the atom <b>X</B>. In the figure
below they are labeled as <b>A</B>.
<pre><TT>
             L    A             A
   L-L       |    |             |
     |  L    L----A      D      A
     |  |   /      \    /        \
   L-L--L--L        A--R          A--R
     |  |   \      /    \        /    \
     |  L    L----A      X      A      G(X)--G
   L-L       |    |             |      |
     |       L    A             A      G
 
</TT></pre>
This option requires either a subsequent <B><A
HREF="#SCAN">SCAN</A></B> key
or the <B><A HREF="#PXAN">PXAN</A></B> key before the
<B><A HREF="#RUNS">RUNS</A></B> key<BR>
<u>Key2</u> (ligand flexibility):
<UL>
<LI><B>FIXL</B> - the conformation of the ligand (atoms L, A, R, D and X)
is unchanged during the simulation
<LI><B>FLEX</B> -
 the conformation of the ligand varies during the simulation<BR>
<U>Data</U>:
<b>nwgtry</b>, <b>nwitry</b>, <b>nmolwid</b>, <b>nmcransh</b>,
<b>nghanchor</b>;
<b>nghanchor</b>+1 times <b>ighanchor</b>;
<b>nmolwid</b> times <b>ew0</b>
<UL>
<LI><b>nwgtry</b> {1} - Number of different orientations of the
replacement group to try (evenly spaced)<BR>
<LI><b>nwitry</b> {1} - Number of different orientations of the atom
to be replaced<BR>
<LI><b>nmolwid</b> {1} - Number of different groups to use as a replacement.<BR>
<LI><b>nmcransh</b> {0} - At each <b>nmcransh</b>-th simulation step
a new random increment is generated for the orientational sampling.<BR>
<LI><b>nghanchor</b> {1} - Number of anchor atoms
<LI><b>ighanchor</b> - the atomindex of the anchor atoms - negative index
means to drop that atom from the ligand (atom D above);
the (<b>nghanchor</b>)th is the atom <b>R</b> bonded to the atom
<B>X</B> to be replaced and
the (<b>nghanchor</b>+1)th is the atom <b>X</b> to be replaced.
Thus the <b>nganchor</b>-1 atoms of type A and D should be followed by
the atom R, the atom G(X) and the rest of the G atoms (if any)<BR>
For each of the <b>nmolwid</b> group:
<UL>
<LI><b>ew0</b> {0.0} - expected energy value (will be subtracted
from all calculated energies before forming exponentials, added
back at the end)
</UL>
</UL>
</UL>
<LI><B>CHIM</B> - the excess Helmholtz free energy
of changing a solvent molecule into a new substance
is calculated using the
perturbation formula without a coupling parameter.
Each solvent molecule is changed to its new form and the energy
of the new substance is evaluated and fed into the perturbation
method formula. Clearly, it is a low-accuracy technique.
This option requires either a subsequent <B><A
HREF="#SCAN">SCAN</A></B> key
or the <B><A HREF="#PXAN">PXAN</A></B> key before the
<B><A HREF="#RUNS">RUNS</A></B> key<BR>
<u>Key2</u> (Grid direction):
<UL>
<LI><B>XGRD</B> - free energy profile aling the X axis will be calculated
<LI><B>YGRD</B> - free energy profile along the Y axis will be calculated
<LI><B>ZGRD</B> - free energy profile along the Z axis will be calculated
</UL>
<U>Data</U>: <b>ngrid</b>, <b>nwitry</b>, <b>nmolchim</b>,
<b>nmolchim</b> times (<b>ifdummy</b>, <b>ew0</b>)
<UL>
<LI><b>ngrid</b> -
Number of grid points in the direction chosen by Key2.<BR>
<LI><b>nwitry</b> - Number of insertion trys in the place of each deleted
molecule
<b>nmolchim</b> {10} - Number of different solutes to calculate free
energy for.<BR>
For each solute:
<UL>
<LI><b>ifdummy</b> {0} - When ifdummy=1, the first atom of the
inserted molecule is assumed to be non-interacting (used only to
ensure the best centering into the cavity).<BR>
<LI><b>ew0</b> {0.0} - expected energy value (will be subtracted
from all calculated energies before forming exponentials, added
back at the end)
</UL>
</UL>
<A NAME="FREE2">
<LI><B>TICA</B> - Energy coupling is used to change from one solute
to an other:
<p>
E(CP) = CP<sup><B>tiexp</B></sup>*E<sub>1</sub> +
(1-CP)<sup><B>tiexp</B></sup>*E<sub>o</sub> <p>
The control function of the TI integrand will be plotted at the
end of the run and error estimates will be provided for it.
<A NAME="FREE3">
<LI><B>PMLI</B> - Same as <B>TICA</B> except that
the program will calculate the free energy difference between the
states precombined with the <B>CIMX</B> or <B>CCMX</B> key
using the perturbation method.
The control function of
&lt;exp(-(E(uscp1)-E(cplpar))/kT)&gt;<sub>cplpar</sub>
will be plotted at the end of the run and error estimates
will be provided for it.<BR>
<A NAME="FREE6">
<LI><B>PMNL</B></A> - The calculation is performed at a fixed
coupling
parameter value CP (atomic coupling) to get free energy dfferences
between states CP and CP<sub>o</sub> and states CP and
CP<sub>1</sub> with both the perturbation method and overlap ratio
method.<BR>
<LI><u>Key2</u> for Key1=<B>TICA</B>, <B>PMLI</B>, or <B>PMNL</B>
(pre-combining):
<UL>
<LI><B>NOMX</B> - No combining of states. The (excess) free
energy difference will be calculated between the two states
represented by the two free energy solutes defined with the
<B><A HREF="#SLTA">SLTA</A></B> key.<BR>
<LI><B>CIMX</B> - Prepare the initial and final states as
the linear combinations of the inputted ones (after the
<B><A HREF="#SLTA">SLTA</A></B>
key) with coupling parameter values <B>uspar0</B>, <B>uspar1</B>.
The potential coefficients will be similarly combined.
During the run, these states
will then be considered to correspond to coupling parameters 0 and
1, respectively. The coupling parameter <B>cplpar</B> will
transformed accordingly to maintain the ratio of the forward
and backward coupling parameter change.<BR>
<LI><B>CCMX</B> - The same linear combinations will be done as
for <B>CIMX</B> but both for the conformation pair given after
<B><A HREF="#SLTA">SLTA</A></B> and for the conformation on the
.<B>crd</B> file. In
addition to that, the two solute conformations on the <B>.crd</B>
file are assumed to have been be obtained previusly as linear
combinations of the two inputted ones with coupling parameter
values <B>uscpo0, uscpo1</B> (read with formatted input,
see below). This allows the preparation of a new
'window' from a previously used (and equilibrated) 'window'.<BR>
<U>Data</U> (for Key1=<B>TICA</B> only):
<B>ngquadp</B>, <B>igquadp</B>,
 <B>epstol</B>, <B>sigtol</B>, <B>qtol</B>, <B>xyz2tol</B>
<UL>
<LI><B>ngquadp</B> {0}, <B>igquadp</B> {0} -
if not zero, the coupling parameter
will be the <B>igquadp</B>-th point of the <B>ngquadp</B>-point
Gaussian quadrature
<LI><B>epstol</B> {0.0} - When the LJ sigma values of all atoms
in a group differ by less than <B>epstol</B> then that group will
be coupled linearly, i.e., <B>tiexp</B> = 1 will be used.<BR>
<LI><B>sigtol</B> {0.0}, <B>qtol</B> {0.0}, <B>xyz2tol</B>
{0.0} - similar tolerance values for the LJ epsilon, atomic charge
and distance square.
</UL>
<U>Data</U> for key1=<B>PMNL</B>:
<B>gmor0k</B>, <B>gmor1k</B>, <B>gdvork</B> (X20,5F10.0)
<UL>
<LI><B>gmor0k</B>, <B>gmor1k</B> - the energy difference
distributions will be accumulated
for the overlap ratio method calculation
in a grid starting at <B>gmor0k</B>, <B>gmor1k</B>
for the initial and final states, respectively.
Each distribution contains 100 grids with a grid width of
<B>gdvork</B>.<BR>
<U>Fdata</U> (for Key1=<B>TICA</B>, <B>PMLI</B>, or <B>PMNL</B>):
<UL>
<LI>RECTYPE 5: <B>tiexp</B>, <B>cplpar</B>, <B>uspar0</B>,
<B>uspar1</B>, <B>uscpo0</B>, <B>uscpo1</B> (3F5.2,5X,5F10.0)
<LI><B>tiexp</B> - Coupling parameter exponent array. Its
elements give the coupling parameter exponents for the
1/r<sup>12</sup>, 1/r<sup>6</sup> and 1/r potential energy terms
(<A HREF="#Mezei_92">Polynomial TI</A>).
When <B>tiexp</B>(i)&gt;0, in using TI
(<B><A HREF="#FREE2">FREE&nbsp;TICA</A></B>), the CP and
(1-CP) factors for the corresponding potential energy
terms are raised to the <B>tiexpi</B>(i)-th power).<BR>
<LI><B>cplpar</B> - Coupling parameter value to be used in
the energy expression. Ignored when <B>ngquadp</B> &gt; 0 !!<BR>
<LI><B>uspar0</B>, <B>uspar1</B> -
when Key2 = <B>CIMX</B> or <B>CCMX</B>
the free energy difference will be determined
between states represented by these two
coupling parameter values (instead of between the
two states defined by the key
<B><A HREF="#SLTA">SLTA</A></B>),
using the perturbation method (Key1 = <B>PMLI</B> or <B>PMNL</B>)
and the overlap ratio method (Key1 = <B>PMNL</B>).<BR>
<LI><B>uscpo0</B> {0}, <B>uscpo1</B> {1} - The coupling
parameter values that had been used to prepare the solute
previously on the .<B>crd</B> file. It is used when a starting
configuration is generated for an intermediate coupling parameter
value from a configuration that itself was already an
intermediate one (Key2 = <B>CCMX</B>).
</UL>
</UL>
</UL>
<A NAME="FREE5">
<LI><B>PMF1</B></A> - The conformational coupling parameter CP will
be sampled with umbrella sampling using either harmonic or adaptive
weighing.<BR>
<u>Key2</u> for Key1=<B>PMF1</B> (coupling type):
<UL>
<LI><B>GENL</B> - Linear coupling in Cartesian space<BR>
<LI><B>WRMM</B> - Linear coupling in Cartesian space, but the
solute has two rigid molecules that move together as the coupling
parameter changes.<BR>
<LI><B>WRFM</B> - Like <B>WRMM</B>, but the second molecule
solute is kept fixed. The second molecule is not considered part of
the free energy solute and should be specified among the non-free
energy solutes.<BR>
<LI><B>WRFR</B> - Like <B>WRFM</B> but the first molecule is not
only displaced but also rotated.<BR>
<LI><B>TORS</B> - Coupling parameter is defined along torsional
coordinate(s). Requires also the <B><A HREF="#TORD">TORD</A></B>
key.
<LI><B>WRTR</B> - Linear coupling of the center of mass of two
positions of a single molecule, followed by translation and
rotation
in certain directions.<BR>
<u>Key3</u> (solute change type):
<UL>
<LI><B>GEOC</B> - Only the solute geometry changes<BR>
<LI><B>GEQC</B> - Only the solute geometry and partial charges
change<BR>
<LI><B>GDQC</B> - The solute geometry, partial charges and atom
types all change<BR>
<u>Key4</u> (umbrella sampling type):
<UL>
<LI><B>HUSS</B> - Umbrella sampling is based on a harmonic
potential.<BR>
<LI><B>AUSL</B> - Adaptive umbrella sampling is used with linear
interpolation for the AUS weighths<BR>
<LI><B>AUSE</B> - Adaptive umbrella sampling is used with exponential
interpolation for the AUS weighths<BR>
<U>Data</U>: (for <B><A HREF="#FREE5">FREE&nbsp;PMF1</A> WRF* ****
****</B> <B>nmolcnt2</B>
<UL>
<LI><B>nmolcnt2</B> - The second molecule of the PMF calculation
that is kept fixed. This information is used
to generate the common center of the two PMF molecules to make
sure that
all molecules interact with the same images of both PMF molecules
</UL>
<U>Data</U>: (for <B><A HREF="#FREE5">FREE&nbsp;PMF1</A> WRTR ****
****</B> <b>cdpmfx</b>, <b>crpmfx</b>, <b>cdpmfy</b>,
<b>crpmfy</b>, <b>cdpmfz</b>, <b>crpmfz</b>
<UL>
<LI><b>cdpmfx</b>, <b>cdpmfy</b>, <b>cdpmfz</b> - Center of mass
displacement step size limits
(see key <B><A HREF="#STEP">STEP</A></B>) in
the X, Y, and Z directions, respectively. Zero value in any
direction
will restrict the motion to the other, nonzero directions.
<LI><b>crpmfx</b>, <b>crpmfy</b>, <b>crpmfz</b> - Rotation
step size limits (see key <B><A HREF="#STEP">STEP</A></B>) around
the X, Y, and Z axes, respectively.<BR>
<U>Fdata</U> for
<B><A HREF="#FREE5">FREE PMF1</A> **** **** **SS</B>):
<UL>
<LI>RECTYPE 6:
<B>nsweep</B>, <B>cplmin</B>, <B>cplmax</B>, <B>delcpl</B>,
<B>c0cplh</B>,
<B>p0cplh</B>, <B>wcplcha</B> (15x,i5,6F10.0)
<LI><B>nsweep</B> {0}. If &gt; 0, an adaptive umbrella
sampling iteration will continue if the interval was traveled at
least nsweep times.
<LI><B>cplmin</B>, <B>cplmax</B> - minimum and maximum of
the coupling parameter CP to be sampled in this run.
<LI><B>delcpl</B> - Stepsize parameter for CP.
<LI><B>c0cplh</B>, <B>p0cplh</B> - With <B>HUSS</B>, the
umbrella sampling weights are obtained as
exp(<B>c0cplh</B>*(CP-<B>p0cplh</B>)<sup>2</sup>).
<LI><B>wcplcha</B> {1.0} - Relative weight of coupling
parameter moves (compared to other types of solute moves).
</UL>
<U>Fdata</U> (additional) for
<B><A HREF="#FREE5">FREE PMF1</A> **** **** AUS*</B>):
<UL>
<LI>RECTYPE 7: <B>iopnrm</B>, <B>iopeql</B>, <B>iopenc</B>,
<B>nitssk</B>, <B>faclim</B>, <B>fcenc1</B>,
<LI><B>smplmx</B>, <B>ratmax</B>, <B>rldvmx</B>, <B>diffmx</B>
(4I5,6F10.0)
<LI>RECTYPE 8: <B>ngovmn</B>, <B>nsubmn</B>, <B>ngrcor</B>,
<B>nwtst</B>, <B>fcenc2</B>, <B>encexp</B>, <B>tolera</B>
(4I5,6F10.0)
<LI><B>iopnrm</B> - Option to select matching algorithm in
adaptive US.
<UL>
<LI><B>iopnrm</B> &lt; 0: linear n-step optimized:
<LI><B>iopnrm</B> = -1: matching is done on the
probabilities
(not recommended).
<LI><B>iopnrm</B> = -2: matching is done on the
potential of mean force.
<LI><B>iopnrm</B> = -3: matching is done on the potential of
mean force, with each increment matched separately (Dutch method).
<LI><B>iopnrm</B> = 0: nonlinear n-step optimized.
<LI><B>iopnrm</B> &gt; 0: nonlinear n-step optimization,
with
regenerating initial guess from successive 1-step
optimization at every <B>iopnrm</B>-th iteration (after
resorting the iterations based on the range covered).
</UL>
<LI><B>iopeql</B> - When &gt;0, self test for equilibration in
the adaptive umbrella sampling is performed by screening previous
iterations at every iteration and dropping
iterations 'contradicted' by subsequent ones.
<LI><B>iopenc</B> - Controls the strategy to force the
adaptive umbrella sampling to extend the sampled range of the
coupling parameter by temporarily modifying the weights.
<UL>
<LI><b>iopenc</b> &lt; 0 : Instead of Boltzmann probabilities, the US
weights will be calculated from their sampling frequencies
<LI><B>iopenc</B> = 0 : No weight modification.
<LI><B>iopenc</B> = 1 : Weights are modified gradually as
the number of iterations during which a grid is unsampled
grows. ("global encouraging")
<LI><B>iopenc</B> = 2 : Suddenly abandoned regions
will get higher weight:
<LI><B>iopenc</B> = 3 : 1 and 2 combined.
</UL>
<LI><B>nitssk</B> - Number of iterations to be skipped when
equilibration is found to be necessary during adaptive US.
<LI><B>faclim</B> {0.5} - K factor used to limit the
sampling to the requested interval.
<LI><B>fcenc1</B> {2.0} - C factor used to encourage
sampling of new, undersampled region (<b>iopenc</b> = 1 or 3 or &lt; 0),
<LI><B>smplmx</B>, <B>ratmax</B> {2.0} - The allowed ratio
between W(CP) over neighboring grid points is limited to the range
[1/<B>ratmax</B>, <B>ratmax</B>] when the frequency of sampling of
that grid point is under <B>smplmx</B>.
<LI><B>rldvmx</B> {0.0001} - Maximum relative deviation
allowed in nonlinear optimization.
<LI><B>diffmx</B> {0.9} - Threshold value for the iteration
difference indicator.
<LI><B>ngovmn</B> {4} - Minimum number of grids to overlap
for two iterations to be considered in the screening.
<LI><B>nsubmn</B> {5} - Iteration is rejected if one out of
every <B>nsubmn</B> subsequent overlapping iterations give
difference indicator &gt; <B>diffmx</B>.
<LI><B>ngrcor</B> {5} - Number of gridpoints over which the
weighting is to be modified when sampling
encouragement is used.
<LI><B>nwtst</B> - Partial results of adaptive step is
printed for iterations until <B>nwtst</B>. If <B>nwtst</B>=#WI
(see Section <A HREF="#Compilation">VI.</A>), stop the run after
saving the data on the iteration.
<LI><B>fcenc2</B> {1.1} - For global encouraging, unsampled
grid weight will be multiplied by <B>fcenc2</B> in each
iteration.
<UL>
<LI><B>encexp</B> {0.75} - for global encouraging (<b>iopenc</b> = 1 or 3),
sampled grids modifying weights will be replaced by their <B>encexp</B>-th
power.
<LI><B>encexp</B> {1.00} - for sampling-frequency based US weights
</UL>
(<b>iopenc</b> &lt; 0), the sampling frequencies will be raised to their
<B>encexp</B>-th power.
<LI><B>tolera</B> {0.06} - The maximum distance of the
coupling parameter from the interval defined by <B>cplmin</B> and
<B>cplmax</B>.
</UL>
</UL>
</UL>
</UL>
</UL>
</UL>
<A NAME="RAUS"></A>
III.03.02. <em u>Adaptive Umbrella Sampling parameter change</em>
<p>
<B>RAUS</B> Key1, Fdata
<p>
<u>Key1</u> (initialization option):
<UL>
<LI><B>NOIN</B> - No accumulator/counter is changed
<LI><B>CTIN</B> - Initialize the sampling frequency accumulator
<LI><B>ALIN</B> - Initialize all AUS-related accumulators, counters<BR>
<U>Fdata:</U>
<UL>
<LI>RECTYPE's 6-8 as described with the
<B><A HREF="#FREE">FREE&nbsp;PMF1</A> **** **** AUS*</B>
keywords followed by
<LI>RECTYPE 9: <B>iterw0</B>, <B>newlim</B> (2i5)
<LI><B>iterw0</B> - When <B>iterw0</B> &gt; 0, use only iterations
from <B>iterw0</B> for the matching.
When <B>iterw0</B> &lt; 0, restore dropped iterations.
<LI><B>newlim</B> - If <B>iterw0</B> = 1, iterations are discarded
until the newly inputted region is reached and further
<B>nitskp</B> iterations are performed (as equilibration).
</UL>
</UL>
Note that <B>cplmin</B> and <B>cplmax</B> have to stay at their original value.
<p>
<A NAME="WMAT"></A>
III.03.03. <em u>PMF matching</em>
<p>
<B>WMAT</B> Key1, Key2, Key3, Key4, Data, Fdata
<p>
This function joins the adaptive umbrella sampling estimates
of the potential of mean force obtained from the checkpoint files
of calculations on overlapping coupling parameter ranges (windows).
<P>
<u>Key1</u> (checkpoint list specification):
<UL>
<LI><B>REGL</B> - checkpoint file runnumbers are evenly spaced
<LI><B>LIST</B> - checkpoint file runnumbers are irregularly spaced<BR>
<u>Key2</u> (plot option)
<UL>
<LI><B>NONE</B> - matched distributions will not be put on .wmp files
<LI><B>PLPS</B> - matched distributions and sampling frequency values
will be written on the file <B>&lt;jobname&gt;.&lt;numrun&gt;.wmp</B>
(for further plotting)
and a Postscript plot
will be written on the file <B>&lt;jobname&gt;.&lt;numrun&gt;.wmp.ps</B>.
<LI><B>NOPS</B> - only the .wmp file will be written
<LI><B>NOPL</B> - only the Postscript plot will be written<BR>
<u>Key3</u> (plot cropping option)
<UL>
<LI><B>NOCR</B> - distribution beyond match points will be printed/plotted
on the .wmp file(s)
<LI><B>CROP</B> - distribution beyond match points will be croppped
(to produce a contiguous plot)<BR>
<u>Key4</u> (matching algorithm)
<UL>
<LI><B>MATW</B> - Match at the best sampled gridpoint in the overlap range
<LI><B>MATD</B> - Find the matching factor by minimizing the
weighted deviation square sum in the overlap range
<LI><B>MATC</B> - Find the matching factor by first minimizing the
weighted deviation square sum in the overlap range and then select the gridpoint
where the matched curves are the closest and match at that point
</UL>
<U>Data</U> for Key1=<B>REGL</B>:
<B>nruns</B>, <B>numrunr</B>, <B>incrnumrun</B>
<UL>
<LI><B>nruns</B> - Number of runs to match
<LI><B>numrunr</B> {<B>numrun</B>} - Run number for the checkpoint
file containing the first segment (window)
<LI><B>incrnumrun</B> {1} - Run number increment, i.e., the i-th
window's checkpoint file was generated by run number
<B>numrunr</B>+<B>incrnumrun</B>*(i-1).<BR>
</UL>
<U>Data</U> for Key1=<B>LIST</B>: <B>nruns</B>, <B>numrun1</B>,
<B>numrun2</B>
...
<UL>
<LI><B>nruns</B> - Number of runs to match
<LI><B>numrun*</B> -
<A HREF="#File_org">Run numbers</A>
of the checkpoint files to match (<B>nruns</B> number)<BR>
<U>Fdata:</U>
<UL>
<LI>RECTYPE 48: <B>delpl1</B>,
<B>delpl2</B>, <B>r0</B>, <B>cx</B>, <B>pmn</B>, <B>imatread</B>,
<B>cplzero</B>, <B>xlab</B>
(5F10.0,I5,A)
<UL>
<LI><B>delpl1</B>, <B>delpl2</B> - To be added to the values of
w(CP) and p(CP) (to shift plots).
<LI><B>r0</B>, <B>cx</B> - The CP values on the <B>.crd</B> file
will be transformed into <B>r0</B>+<B>cx</B>*CP.
<LI><B>pmn</B> - Minimum sampling frequency for a grid point to
be written on the <B>.wmp</B> file (to keep noise out of the plots).
<LI><B>imatread</B> - Number of match points to read (if any)
<LI><B>cplzero</B> - when not zero, the match PMF will be shifted in
such a way that the PMF at that coupling parameter value will be zero.
<LI><B>xlab</B> - The X axis label when <B>r0</B>&gt;0 or <B>cx</B> is not one.
</UL>
<LI>RECTYPE 49: (read if <B>imatread</B> &gt; 0):
(<B>match</B>(i),i=1,<B>nruns</B>+1) (11I5)
<UL>
<LI><B>match</B> - matching point specification. <B>match</B>(1)
and <B>match</B>(<B>nruns</B>+1) specify the beginning and the end
point of the complete coupling parameter interval. <B>match</B>(2)
is the gridpoint used to match runs 1 and 2. In general,
<B>match</B>(i+1) is the matching point between runs i and i+1.
Zero at any place results in default values. The default matching
points are determined based on the combined sampling frequencies
and the interval defaults to the [0.0,1.0] coupling parameter
interval. The whole record can be omitted.
</UL>
</UL>
</UL>
</UL>
</UL>
</UL>
<A NAME="WPLT"></A>
III.03.04. <em u>AUS iteration plot</em>
<p>
<B>WPLT</B> Key1, Data, [Data]
<p>
<u>Key1</u> (scale for data points):
<UL>
<LI><B>LIN&nbsp;</B> - circles with radius proportional to the extent of
sampling will be drawn at each data point
<LI><B>SQRT</B> - radius will be proportional to the square root of
the sampling extent
<LI><B>LOG&nbsp;</B> - radius will be proportional to the logarithm of
the sampling extent
<BR>
<U>Data</U>: <B>PSfile</B>, <B>nfstplt</B>, <B>maxit</B>,
<B>nfrplt</B>, <B>nrad</B>
<UL>
<LI><B>PSfile</B> - name of the file the plot will be written
in Postscript format
<LI><B>nfstplt</B> {1}, <B>maxit</B> {number of iterations},
<B>nfrplt</B> {1} - plot data for every <B>nfrplt</B> -th iteration
from iteration <B>nfstplt</B> to iteration <B>maxit</B>
<LI><B>nrad</B> {5} - the radius of the largest circle (in pixels)
</UL>
</UL>
This key will produce a family of curves showing the PMF calculated from
each iteration as lines, with circles representing the extent of sampling
at each coupling parameter grid. The matched PMF is shown with
large filled circles.
<p>
<A NAME="TIQU"></A>
III.03.05. <em u>TI quadrature</em>
<p>
<B>TIQU</B> Key1,[Key2] Data, [Fdata]
<P>
<u>Key1</u> (data source specification):
<UL>
<LI><B>DATA</B> - Just read the integrand values
<LI><B>REGL</B> - checkpoint file runnumbers are evenly spaced
<LI><B>LIST</B> - checkpoint file runnumbers are irregularly spaced<BR>
<u>Key2</u> (contribution type, only for Key1=<B>REGL</B> or Key1=<B>LIST</B>)
<UL>
<LI><B>ALL&nbsp;</B> - use both solute-solvent and solute solute (if any) contributions
<LI><B>STSV</B> - use only the solute-solvent contribution
<LI><B>STST</B> - use only the solute-solute contribution
</UL>
<U>Data</U> for Key1=<B>DATA</B>: <B>nquad</B>, <B>integrand values</B>
<UL>
<LI><B>nquad</B> - Number of quadrature points. Currently allowed
values: 3, 5 and 8.
<LI><B>integrand values</B> - <B>nquad</B> numbers
</UL>
<U>Data</U> for Key1=<B>REGL</B>:
<B>nquad</B>, <B>numruns</B>, <B>incrnumrun</B>
<UL>
<LI><B>nquad</B> - Number of quadrature points.
<LI><B>numruns</B> {<B>numrun</B>} -
<A HREF="#File_org">Run number</A> for the checkpoint
file containing the result of the first quadrature point
<LI><B>incrnumrun</B> {1} - Run number increment, i.e., the i-th
quadrature point checkpoint file was generated by run number
<B>numrunr</B>+<B>incrnumrun</B>*(i-1).
</UL>
<U>Data</U> for Key1=<B>LIST</B>: <B>nquad</B>,
<B>numrun1</B>, <B>numrun2</B> ...
<UL>
<LI><B>nquad</B> - Number of quadrature points.
<LI><B>numrun*</B> - <A HREF="#File_org">Run numbers</A>
of the quadrature point checkpoint files (<B>nquad</B> number)<BR>
<U>Fdata:</U> (only for Key1=<B>REGL</B> or Key1=<B>LIST</B>)
<UL>
<LI>RECTYPE 41: <B>ncread</B>, <B>nfac</B>,
<B>iskipread</B>, <B>itrapread</B>, <B>nendread</B> (6I5)
<UL>
<LI><B>ncread</B> - If &gt; 0, the cumulative averages of the TI
integrands will be read from STDIN (unit 5) (RECTYPE's 43-45), <b>ncread</b>
numbers per line. Maximum value: 8.
<LI><B>nfac</B> - The TI integrands read from STDIN (unit 5) will be divided by
<B>nfac</B> if <B>nfac</B> &gt; 0 (ignored if <B>ncread</B>=0).
<LI><B>iskipread</B>{0} - if &gt; 0, allow for a-posterior equilibration
(read RECTYPE 42 below).
<B>ncntin</B> long segments) to skip from the beginning of the run
(a posteriori equilibration).
<LI><B>itrapread</B> {0} - If &gt; 0, perform trapezoid rule integration
(read RECTYPE 46 below).
<LI><B>nendread</B> {0} - If &gt; 0, calculate the TI integral over
different intervals (read RECTYPE 46 below).
</UL>
<LI>RECTYPE 42: (read if <B>iskipread</B> &gt; 0)
<B>(nskp</B>(iq),iq=1,<B>nquad</B>) (8I5)
<UL>
<LI><B>nskp</B>(iq) - Number of control function blocks (i.e.,
<B>ncntin</B> long) to skip from the beginning for quadrature point
iq.
</UL>
<LI>RECTYPE 43: (read if <B>ncread</B> &gt; 0) <B>ident</B>
(a80,/,a80)
<UL>
<LI><B>ident</B> - 2-line description of the data.
<LI>If <B>iread</B> &gt; 0 then <B>nquad</B> times (one data set
for each quadrature point):
</UL>
<LI>RECTYPE 44: <B>nbl</B>, <B>ncntin</B>, <B>cplpr</B>
(2I10,F10.0)
<UL>
<LI><B>nbl</B> - The number of control function blocks to read
in.
<LI><B>ncntin</B> - The length of a control function block.
<LI><B>cplpar</B> - The coupling parameter value.
</UL>
<LI>RECTYPES 45: <B>ticums</B>(ib,iq),ib=1,<B>nbl</B>,
<B>ncread</B> numbers per line (6E12.6)
<UL>
<LI><B>ticums</B> - The cumulative average of the integrand at
each control function block.
</UL>
<LI>RECTYPE 46: (read if <B>itrapread</B> &gt; 0 ):
<b>idef</b>, <B>ymin</B>, <B>ymax</B> (I5,2F10.0)
<UL>
<LI><b>idef</b>,-
If <b>idef</b>=0, the fitted polynomial will be used to calculate
the integrand at the endpoints.
<LI><B>ymin</B>, <B>ymax</B> -
The value of the TI integrand at
coupling parameters 0 and 1 (if available) for trapezoid rule
integration (to be used when <b>idef</b> &gt; 0).
</UL>
<LI>RECTYPE 47: (<B>nendread</B> diffferent lines): <B>xmin</B>,
<B>xmax</B> (6F12.6)
<UL>
<LI><B>xmin</B>, <B>xmax</B> - New coupling parameter limits. The
program will calculate the integral of the Gaussian approximating
polynomial between these limits. This line is only read if
RECTYPE 43 was present.
</UL>
</UL>
</UL>
</UL>
This function performs the numerical (Gaussian) quadrature on
thermodynamic integration runs. It prepares a plot of the
polynomial fitting the quadrature ponts and, optionally, performs
other numerical integrations.
<p>
<A NAME="OVRA"></A>
III.03.06. <em u>Overlap ratio evaluation</em>
<p>
<B>OVRA</B> Data
<p>
This function obtains the free energy from two overlap ratio method
runs.
<P>
<U>Data</U>: <B>numrun1</B>, <B>numrun2</B>
<UL>
<A HREF="#File_org">Run number</A>
<LI><B>numrun1</B>, <B>numrun2</B> - The
<A HREF="#File_org">run numbers</A>
that generated the two checkpoint files containing the
energy distributions the method requires.
</UL>
<A NAME="FETK"></A>
III.03.07. <em u>Temperature scaling for TI</em>
<P>
Prerequisite: <B><A HREF="#TEMP">TEMP</A></B>
<P>
<B>FETK</B> Data
<P>
This option scales the temperature by dividing it with
<b>cplpar<sup>kexp</sup></b> to represent a run where the total system
is made to turn into ideal gas
<p>
<U>Data</U>: <B>kexp</B>, <B>nquad</B>, <B>iquad</B>
<UL>
<LI><B>kexp</B> {4} - the coupling parameter exponent
<LI><B>nquad</B> - the number of Gaussian quadrature points (3, or 5 or 8)
<LI><B>iquad</B> - the quadrature index (&le; <b>nquad</b>)
</UL>
<A NAME="Input_MOLEC"></A>
III.4. <B>Molecule descriptor keywords</B>
<OL>
<LI><TT><B><A HREF="#SLTA">SLTA</A></B>: </TT>Solute
description<BR>
<LI><TT><B><A HREF="#SLVA">SLVA</A></B>: </TT>Solvent
description<BR>
<LI><TT><B><A HREF="#CLON">CLON</A></B>: </TT>Solute cloning<BR>
<LI><TT><B><A HREF="#MAKB">MAKB</A></B>: </TT>Make bonds<BR>
<LI><TT><B><A HREF="#BRKB">BRKB</A></B>: </TT>Break bonds<BR>
<LI><TT><B><A HREF="#BDHH">BDHH</A></B>: </TT>Forcing H-H bonds<BR>
<LI><TT><B><A HREF="#BDRH">BDRH</A></B>: </TT>Forcing bonds of
an H to a different group<BR>
<LI><TT><B><A HREF="#MOLD">MOLD</A></B>: </TT>Define solute
molecule limits<BR>
<LI><TT><B><A HREF="#RDBD">RDBD</A></B>: </TT>Bond list read<BR>
<LI><TT><B><A HREF="#TORD">TORD</A></B>: </TT>Torsion angle
definitions<BR>
<LI><TT><B><A HREF="#CSEG">CSEG</A></B>: </TT>CHARMM segment id
handling<BR>
</OL>
<A NAME="SLTA"></A>
III.04.01. <em u>Solute description</em>
<p>
<B>SLTA</B> Key1 Key2 Key3, Data, [Data, Fdata]
<p>
Prerequisites:
<B><A HREF="#FILE">FILE</A></B>, <B><A HREF="#PBCN">PBCN</A></B>,
<B><A HREF="#SVPT">SVPT</A></B>, <B><A HREF="#SUPT">SUPT</A></B><BR>
Can not be followed by:
<B><A HREF="#SLVA">SLVA</A></B>, <B><A HREF="#PMOD">PMOD</A></B>,
<B><A HREF="#SLVA">SLVA</A></B>, <B><A HREF="#PFRD">PFRD</A></B>,
<B><A HREF="#FREE">FREE</A></B>, <B><A HREF="#PROT">PROT</A></B>,
p
<B><A HREF="#FLXR">FLXR</A></B>, <B><A HREF="#CNST">CNST</A></B>,
<B><A HREF="#CHRG">CHRG</A></B>, <B><A HREF="#CLON">CLON</A></B>,
<B><A HREF="#MAKB">MAKB</A></B>, <B><A HREF="#BRKB">BRKB</A></B>,
<B><A HREF="#RFSL">RFSL</A></B>, <B><A HREF="#MIXR">MIXR</A></B>,
<B><A HREF="#SETC">SETC</A></B>, <B><A HREF="#MODA">MODA</A></B>,
<B><A HREF="#MOLD">MOLD</A></B>, <B><A HREF="#STEP">STEP</A></B>,
<B><A HREF="#RDBD">RDBD</A></B><BR>
Potential successors:
<B><A HREF="#RCKP">RCKP</A></B>, <B><A HREF="#PRMF">PRMF</A></B><BR>
Prerequisite of:
<B><A HREF="#CNFG">CNFG</A></B>, <B><A HREF="#RUNS">RUNS</A></B>,
<B><A HREF="#TORD">TORD</A></B>, <B><A HREF="#WCNF">WCNF</A></B>,
<B><A HREF="#FCGA">FCGA</A></B>, <B><A HREF="#PARD">PARD</A></B>,
<B><A HREF="#FCGD">FCGD</A></B>, <B><A HREF="#PMFM">PMFM</A></B>,
<B><A HREF="#AROM">AROM</A></B>, <B><A HREF="#ENGL">ENGL</A></B>,
<B><A HREF="#STVG">STVG</A></B>, <B><A HREF="#CMPC">CMPC</A></B>,
<B><A HREF="#BNDL">BNDL</A></B>, <B><A HREF="#HBMO">HBMO</A></B>,
<B><A HREF="#SVDP">SVDP</A></B>, <B><A HREF="#GENV">GENV</A></B>,
<B><A HREF="#PXCR">PXCR</A></B>, <B><A HREF="#SCAN">SCAN</A></B>,
<B><A HREF="#HBBR">HBBR</A></B><BR>
Related keys:
<B><A HREF="#CLON">CLON</A></B>,
<B><A HREF="#MAKB">MAKB</A></B>,
<B><A HREF="#BRKB">BRKB</A></B>,
<B><A HREF="#MOLD">MOLD</A></B>,
<B><A HREF="#RDBD">RDBD</A></B>.
<p>
<u>Key1</u> (possible solute periodicity):
<UL>
<LI><B>SMPL</B> - Simple solute.
<LI><B>POLY</B> - Periodic solute along the X axis. Displacement
not made along the X axis.
<LI><B>XTAL</B> - Solute is periodic along all 3 axes (crystal).
Global solute moves are forbidden.<BR>
<u>Key2</u> (input format):
<UL>
<LI><B>MMC</B> - MMC format - specific to this program
(the program
<A HREF="https://mezeim01.u.hpc.mssm.edu/simulaid">Simulaid</A>
can help convert structures in different file formats
(e.g., PDB, CHARMM .CRD) to this format).
<LI><B>MMC4</B> - Older version of MMC format (4-character PF labels)
<LI><B>PDB</B> - <A HREF="https://www.rcsb.org/pdb/">PDB</A>
format input - atomnames will
be used to deduce atomic numbers only<BR>
<LI><B>CRD</B> - CHARMM CRD format input -
atomnames will be used to deduce atomic numbers only<BR>
<u>Key3</u> (data source):
<UL>
<LI><B>READ</B> - Read the solute description data
(the formatted input below describing each solute atom)
from the input file, starting from the next line
<LI><B>FILE</B> - Read the solute description data
from a file with extension <B>.slt</B> .
For Key2=<B>PDB</B>, <B>.pdb</B> is also allowed,
For Key2=<B>CRD</B>, <B>.CRD</B> is also allowed.<BR>
<U>Data</U>: <B>nslt</B>, <B>nsltfe</B>, <B>nsltfe0</B>,
<B>nsltpr</B>, <B>numrunr</B>, <B>jobnamer</B>
<UL>
<LI><B>nslt</B> - Number of solute atoms. Note that free energy
calculations require multiple copies and <B>nslt</B> includes the
atoms in all the copies.
<LI><B>nsltfe</B> {0} - The number of solute atoms in all copies
of the free-energy molecule(s).
<LI><B>nsltfe0</B> {nsltfe/&lt;number of free energy copies
(1,2 or 3)&gt;}
The number of solute atoms in the first
copy of the free-energy molecule(s).
<UL>
<LI>For the Widom ((<B><A HREF="#FREE1">FREE WIDO</A></B>) or the
chimera (<B><A HREF="#FREE1">FREE&nbsp;CHIM</A></B>) there is one copy
of the free energy solute, the atoms in the <b>nmolwid</b> solutes
for which the free energy is calculated. Each molecule has to be a
single (and different) residue.
<LI>For creation/annihilation path calculations
(<B><A HREF="#FREE2">FREE&nbsp;TICA</A></B>) or
(<B><A HREF="#FREE3">FREE&nbsp;PMLI</A></B>) the free energy solute
consists of two copies of the same molecule in its two different
states.
<LI>For calculations over a path of continuous deformation,
(<B><A HREF="#FREE5">FREE&nbsp;PMF1</A></B> or
<B><A HREF="#FREE6">FREE&nbsp;PMNL</A> </B>) the program requires three
copies of the solute: the initial and final states and an
intermediate state. This third copy is essentially a placeholder
for the actula, continually changing intermediate copy used in the
calculation of energies, etc.
</UL>
<LI><B>nsltpr</B> {<B>nslt</B>} - Number of solute atoms to echo
<LI><B>numrunr</B> {<B>numrun</B>} -
<A HREF="#File_org">Run number</A>
of the solute description file (only read when Key3=FILE).
<LI><B>jobnamer</B> {jobname} - File name root of the solute
description file (only read when Key3=FILE).<BR>
<U>Fdata:</U> Description of the solute atoms<BR>
When the solute description is read from a file then the title/comment
lines can contain information about how to translate/rotate
(or vice versa) to a previous position and orientation
(to be used by, e.g., the
<B><A HREF="#WCNF">WCNF</A> **** PRET</B> key). This information
is provided automatically by Simulaid's orientational optimization
features.
The program will scan the first 100 lines for labels 'TRRT' or 'RTTR',
indicating translation+rotation or rotation+translation.
A letter 't' after the label indicates that the next three
(free format) numbers are the
components of the translation vector and a letter 'r' followed by
1 or 2 or 3 indicates that the next three numbers are the corresponding
rows of the rotation matrix. For exemple the header of an <b>.slt</b> file
may contain the lines
<PRE>
! Translation first  TRRTt  1.0  2.0  3.0
! Rotation next      TRRTr1 0.0 1.0 0.0
!                    TRRTr2 1.0 0.0 0.0
!                    TRRTr3 0.0 0.0 1.0
</PRE>
indicates that translation by (1,2,3) should be followed by a multiplication
(from the left) by the matrix whose rows are (0,1,0), (1,0,0,) an (0,0,1).
<UL>
<LI>RECTYPE 14's: <B> nslt</B> records - one for every atoms of the
solute for <B>MMC</B> format:
<B>tslt</B>(i),(<B>cslt</B>(j,i),j=1,3),<B>qslt</B>(i),
<B>igrslt</B>(i), <B>gcent</B>, <B>mcent</B>, <B>labslt1</B>(i),
<B>labslt2</B>(i), <B>indxrdf</B>(i), <B>pf</B>(i),
<B>spgroup</B>(i)<BR>
Format for Key2=<B>MMC</B>: (1x,A6,4F10.0,I5,2a1,2A4,I5,1x,A4,A8)<BR>
Format for Key2=<B>MMC4</B>: (1x,A4,4F10.0,I5,2a1,2A4,I5,1x,A4,A10)
<LI><B>tslt</B>(i) - Atom type specification for atom i (not
EPEN) for solute-solvent potential - either its code number or its
4-character label (AMBER or CHARMM only) - or atomic number for
atom i for EPEN solute-solvent potential.
<LI><B>cslt</B>(k,i) - X,Y,Z coordinates in  of the atom i.
<LI><B>qslt</B>(i) - charge on atom i. For Jorgensen OPLS it is
disregarded and internally stored values will be used, unless this
feature is overridden by <B><A HREF="#CHRG">CHRG</A> INPT</B>.
<LI><B>igrslt</B> - group (residue) index of the solute atoms.
Used only with <B><A HREF="#SUVC">SUVC</A> **GC</B>. Important: The
atoms have to be listed by increasing group index.
<LI><B>gcent</B> - if <B>gcent</B>='G' then this atom will be
the group center, overriding the built in topological algorithm
<LI><B>mcent</B> - if <B>mcent</B>='M' then this atom will be
the molecular center, overriding the built in topological algorithm
<LI><B>labslt1</B>(i) {&nbsp;&nbsp;&nbsp;&nbsp;},
<B>labslt2</B>(i) {&nbsp;&nbsp;&nbsp;&nbsp;}
- Optional atom labels label used in proximity analysis output.
Whenever applicable, they are interpreted
 as residue name and atom names (four characters each,
residue name left adjusted).
<LI><B>indxrdf</B>(i) {i} - optional index pointing to the
proximity distributions this solute atom contributes to.
If all indxrdf(i)'s are zero then individual proximity
distributions
will be calculated for each solute atom. If at least one indxrdf(i)
is nonzero then the atoms with indxrdf(i)=0 will not contribute to
any distribution.
<LI><B>pf</B>(i) - For Key2=<B>MMC</B>: if not blank the
potential function type for the <em u>whole</em> residue containig this atom
(overriding the type specified by the <B><A HREF="#SUPT">SUPT</A></B> key);
<BR> For Key2=<B>MMC4</B>: two possible uses: If <B>pf</B>(i) is a valid
potential function type then the same as for Key2=<B>MMC</B> otherwise
if <B>pf</B>(i) is not blank then it will be added to the potential label
<B>tslt</B>(i) allowing the use of more than four characters (maximum six).
<LI><B>spgroup</B>(i)
{&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;}
If not blank, atoms with the same <B>spgroup</B> may be considered
a single functional group for the proximity analysis (see key
<B><A HREF="#FCGD">FCGD</A></B>).
For the Widom insertion (<B><A HREF="#FREE1">FREE&nbsp;WIDO</A></B>) or
the chimera method (<B><A HREF="#FREE1">FREE&nbsp;CHIM</A></B>)
<B>spgroup</B> in the record of the atom belonging to the
molecule's center will be used as the name of the solute.
</UL>
<LI>RECTYPE 51's: <B> nslt</B> records - one for every atoms of the
solute for <A HREF="https://www.rcsb.org/pdb/">PDB</A> format:
<UL>
<LI><B>mcent,atnam,resnam,(cslt(k,i),k=1,3)</B>
(11x,a1,a4,1x,a4,10x,3f8.0)
<LI><B>resnam,atnam</B> - the analysis label <B>labslt</B> will
be a concatenation of <B>resnam</B> and <B>atnam</B>.<BR>
<LI><B>mcent</B> - if <B>mcent</B>='M' then this atom will be
the molecular center, overriding the built in topological algorithm
</UL>
<LI>RECTYPE 50's: <B> nslt</B> records - one for every atoms of the
solute for <B>CRD</B> format:
<UL>
<LI><B>resnam,mcent,atnam,(cslt(k,i),k=1,3)</B>
(11x,a4,a1,a4,1x,3f10.0)
<LI><B>resnam,atnam</B> - the analysis label <B>labslt</B> will
be a concatenation of <B>resnam</B> and <B>atnam</B>.<BR>
<LI><B>mcent</B> - if <B>mcent</B>='M' then this atom will be
the molecular center, overriding the built in topological algorithm
<U>Fdata:</U> RECTYPEs 19-21 are only needed for
<B><A HREF="#SUPT">SUPT</A> EPEN</B>
(EPEN potentials). The location and the type of each center have
already been
defined above. The program finds their breakdown into
electrons and nuclei automatically thus here only the
repulsion/dispersion term parameters are to be inputted.
<UL>
<LI>RECTYPE 19's: <B>nslte</B> records - for every solute
electronic center, in the same order as the centers were defined:
<B>as</B>(i), <B>bs</B>(i), <B>ds</B>(i) (3F20.10)
<LI><B>as</B> - A parameter for solute electron i
<LI><B>bs</B> - B parameter for solute electron i
<LI><B>ds</B> - C parameter for solute electron i
<LI>RECTYPE 20's: <B>nslve</B> records - for every solvent
electronic center:
<LI><B>asv</B>(i), <B>bsv</B>(i), <B>dsv</B>(i) (3F20.10)
<LI><B>asv</B> - A parameter for solvent electron i
<LI><B>bsv</B> - B parameter for solvent electron i
<LI><B>dsv</B> - C parameter for solvent electron i
</UL>
The units of the various parameters are discussed in
<A HREF="#Marchese">Marchese et al.</A>
<LI>RECTYPE 21: <B>cutin</B>, <B>escale</B>, <B>iopict</B>
(2F20.10,I5)
<LI><B>cutin,</B><B>iopict</B> - The radius of the hard sphere
placed at the COM of the solute when <B>iopict</B> = 1 or 2. (Use
it with some caution! There might be some discontinuity problems.)
<LI><B>escale</B> {1.0} - Energy scaling parameter for the
solute-solvent potential function. All the solute-solvent
pair energies are multiplied by this constant scale factor.
The majority of the potential functions, including EPEN,
break down at very close distances leading to very high
negative energies in certain orientations. Fortunately,
these spurious minima are followed by large barriers as the
molecule moves outward. If the starting configuration has
such close contacts and regular potential is used, then the
total energy of this configuration will be very negative
and it is highly unlikely that the system will ever get out
of this bottleneck. This problem can be circumvented by
redefining the potential function as repulsive in the inner
Van der Waals region.
<UL>
<LI><B>escale</B>=0: Regular EPEN potential without any
inner cutoff.
<LI><B>escale</B>=1: A hard sphere of radius <B>cutin</B> is
placed
at the COM of the solute. This feature is included to expel any
solvent molecule which are in close contact with the
solute molecule. Beyond the <B>cutin </B>distance (see RECTYPE 21),
regular EPEN potential is used.
<LI><B>escale</B>=2: Van der Waals envelopes are created
around the
solute and solvent molecules to check whether any solvent
molecule is in close contact with the solute.
</UL>
</UL>
</UL>
</UL>
</UL>
</UL>
<p>
The program is currently prepared to handle atoms up to atomic
number 86. The atomic number for lone-pair is 89 (in case
of <B>EPEN</B>, it is disregarded) and atomic number 90 is an
electron.
Atomic numbers 91-93 have been reserved for the united atoms
CH<sub>1</sub>-CH<sub>3</sub>, respectively. The various
atomtypes for the different
potential libraries implemented and the actual functional form of
the potentials are specified in a separate document.
<p>
Important: For <B>EPEN</B>, the different types of centers have to
be grouped together. Lone-pairs should follow nuclei and
electrons should follow nuclei and lone-pairs.
<p>
When the coupling parameter is used to determine
conformational transition (<B><A HREF="#FREE5">FREE&nbsp;PMF1</A></B>),
three sets of solute
coordinates are used: the two conformations R<sub>o</sub> and
R<sub>1</sub> and their
combination CP*R<sub>1</sub>+(1-CP)*R<sub>o</sub>. The solute
description thus has to
contain the coordinates, charges of the two conformations as well
as room for the mixture. In this third set, only the atomtypes
have to be given. When <B><A HREF="#CNFG">CNFG</A> PMFN</B>,
however, only the R<sub>o</sub> conformation is required here since
the R<sub>1</sub> conformation will be
given at a different part of the input. It is important that the
different descriptions of the conformation give the solute atoms
in identical order. Solute atom type change during the transition
can not involve EPEN.
<P>
<A NAME="SLVA"></A>
III.04.02. <em u>Solvent description</em>
<p>
<B>SLVA</B> [Data Fdata]
<p>
Prerequisite: <B><A HREF="#SVPT">SVPT</A></B><BR>
Potential successors:
<B><A HREF="#SLTA">SLTA</A></B>, <B><A HREF="#RCKP">RCKP</A></B><BR>
Prerequisite of:
<B><A HREF="#SVDP">SVDP</A></B><BR>
<p>
<U>Data</U>: <B>nslv</B>, <B>islvrep</B>, <B>label</B>, <B>islv4</B>, <B>filename</b>
<UL>
<LI><B>nslv</B> - Number of solvent atoms per solvent molecule.
<LI><B>islvrep</B> {1} - The central atom to be used for
representing the solvent molecule when only one center is needed,
e.g., for proximity analysis, solute-solvent distances will be
calculated from the <B>islvrep</B>-th solvent atom
<LI><B>label</B> {HOH } - The residue label to use with this
molecule.
<LI><B>islv4</B> - {0} When <B>islv4</B>=1 the atom records use only
four characters for the potential label
<LI><b>filename</B> - when present, the solvent desription (see Fdata below)
will be read from a file <b>filename.slv</b>, in the same format as
the Fdata for the solute atoms (key <B><A HREF="#SLTA">SLTA</A></B>)<BR>
<U>Fdata</U>:
<UL>
<LI>RECTYPE 15's: <B>nslv</B> records - one for every atoms of the
solvent:
<B>itypslv</B>(i), (<B>cslv</B>(j,i),j=1,3), <B>qslv</B>(i),
<B>labslv</B>(i)<BR>
<UL>
<LI>For <b>nslv</b> &gt; 0:<BR>
For <B>islv4</B>=0:(1x,a6,4F10.0,2x,a4) or (1x,i6,4F10.0,2x,a4)<BR>
For <B>islv4</B>=1:(1x,a4,4F10.0,2x,a4) or (1x,i4,4F10.0,2x,a4)<BR>
<LI>For <b>nslv</b> = 0: (1x,a6,4f10.0,11x,a4)<BR>
Also, a character <B>G</B> in colum4 53 can override <B>islvrep</B> read above.
</UL>
<LI><B>itypslv</B> - for general solvent
(<B><A HREF="#SVPT">SVPT</A> GENL</B>) it
is the atom type (either a potential label, left-adjusted,
or number, right adjusted - see key
<B><A HREF="#SLTA">SLTA</A></B>), otherwise the atomic number of atom i.
<LI><B>cslv</B> - X,Y,Z coordinates in  of the atom i.
<LI><B>qslv</B> - charge on atom i to be used in the SLT-SLV
interactions.
<LI><B>labslv</B> - label of atom i
</UL>
</UL>
Pseudo-atoms are to be placed after the regular atoms.
<p>
Note: This key is ignored if the solvent is not
<B><A HREF="#SVPT">GENL</A></B> type.
<p>
Note: the dipole of the solvent molecule should be parallel
to the X axis for the purpose of the near-neighbor solvent dipole
correlation function computation
(<B><A HREF="#DSTC">DSTC</A> ALL</B>).
<p>
<A NAME="CLON"></A>
III.04.03. <em u>Solute molecule cloning</em>
<p>
<B>CLON</B> Data
<p>
Potential successor:
<B><A HREF="#SLTA">SLTA</A></B><BR>
<p>
<U>Data</U>: <B>nclone</B>, <B>iclonef(1)</B>, <B>iclonel(1)</B>,
<B>ncopy(1)</B>, ...
<B>iclonef(nclone)</B>, <B>iclonel(nclone)</B>,
<B>ncopy(nclone)</B>
<UL>
<LI><B>nclone</B> - Number of solute molecules to clone
<LI><B>iclonef(1)</B>, <B>iclonel(1)</B> - First and last atoms of
the first molecule to be cloned (atom numbers refer to the input list
read after the key <B><A HREF="#SLTA">SLTA</A></B>)
<LI><B>ncopy(1)</B> - Number of copies required (&gt; 1).
<p>
This operation takes parts of the input solute description read
with the
<B><A HREF="#SLTA">SLTA</A></B> key, makes duplicate descriptions.
This way a
'solute' that consists of several identical molecules can be
specified without needless repetition.
Note that free-energy molecules can not be cloned.
</UL>
<A NAME="MAKB"></A>
III.04.04. <em u>Making additional bonds</em>
<p>
<B>MAKB</B> Data
<p>
Potential successor:
<B><A HREF="#SLTA">SLTA</A></B><BR>
<p>
<U>Data</U>: <B>nmake</B>, <B>imake1(1)</B>, <B>imake2(1)</B>,
...,
<B>imake1(nmake)</B>, <B>imake2(nmake)</B>
<UL>
<LI><B>nmake</B> - Number of bonds to make
<LI><B>imake1(i)</B>, <B>imake2(i)</B> - The <B>i</B>-th new bond
connects atoms <B>imake1(i)</B> and <B>imake2(i)</B>
</UL>
Note that this key is not implemented for the case when cloning
(<B><A HREF="#CLON">CLON</A></B>) is in effect.
<p>
<A NAME="BRKB"></A>
III.04.05. <em u>Breaking bonds</em>
<p>
<B>BRKB</B> Data
<p>
Potential successor:
<B><A HREF="#SLTA">SLTA</A></B><BR>
<p>
<U>Data</U>: <B>nbreak</B>, <B>ibreak1(1)</B>, <B>ibreak2(1)</B>,
...
<B>ibreak1(nbreak)</B>, <B>ibreak2(nbreak)</B><UL>
<LI><B>nbreak</B> - Number of bonds to break
<LI><B>ibreak1(i)</B>, <B>ibreak2(i)</B> - The <B>i</B>-th bond to
be broken
connects atoms <B>ibreak1(i)</B> and <B>ibreak2(i)</B>
</UL>
Note that this key is not implemented for the case when cloning
(<B><A HREF="#CLON">CLON</A></B>) is in effect.
<p>
<A NAME="BDHH"></A>
III.04.06. <em u>Forcing H-H bonds</em>
<p>
<B>BDHH</B>
<p>
By default, the topology search excludes bonds between hydrogens.
If this key is present, these bonds will be kept.
<p>
<A NAME="BDRH"></A>
III.04.07. <em u>Forcing bonds of an H to a different group </em>
<p>
<B>BDRH</B>
<p>
By default, the topology search excludes bonds of hydrogens to heavy
atoms of a different group/residue.
If this key is present, these bonds will be kept.
<p>
<A NAME="MOLD"></A>
III.04.08. <em u>Defining solute molecule limits</em>
<p>
<B>MOLD</B> Data
<p>
Potential successor:
<B><A HREF="#SLTA">SLTA</A></B><BR>
<p>
<U>Data</U>: <B>molslt</B>, <B>ilastm(1)</B>, <B>ilastm(2)</B>,
...
<UL>
<LI><B>molslt</B> - Number of molecules in the solute
<LI><B>ilastm(i)</B> - Last atom of molecule i.
</UL>
This key overrides the molecular topology for defining the
molecules within the solute (or just a part of it).
This is useful when the
molecule is distored and too many incorrect bonds are
found or too many bonds are missing.
When applied to cloned (<B><A HREF="#CLON">CLON</A></B>) solute
atoms, the definitions have to be
repeated for all clones.
The solute atoms not covered by this operation
have to be placed contiguously at the end of the
solute definition read with
the (<B><A HREF="#SLTA">SLTA</A></B>) key.
<p>
<A NAME="RDBD"></A>
III.04.09. <em u>Bond list input instead of calculating</em>
<p>
<B>RDBD</B> Key1 [Data]
<p>
When this key is present, the solute bond list is read from a file.
<p>
<u>Key1</u> (bond list type):
<UL>
<LI><B>MAKB</B> - Bond list is read from a file whose syntax
is the same as the input for the
<B><A HREF="#MAKB">MAKB</A></B> key (including the key itself).
<A HREF="https://mezeim01.u.hpc.mssm.edu/simulaid">Simulaid</a> can prepare this
file along with the <b>.slt</b> file
<LI><B>CPSF</B> - Bond list is read from a file
that is a Charmm topology (<b>.psf</b>) file.
<LI><B>ATOP</B> - Bond list is read from a file
that is an Amber topology file.<BR>
</UL>
<U>Data</U> <b>file name</b>
<UL>
<LI><b> file name</b> - bond list is a file called <b>file name</b>.mkb, 
<b>file name</b>.psf, and <b>file name</b>.top for Key1 =
<B>MAKB</B>, <B>CPSF</B>, and <B>ATOP</B>, resp.
Default <b>file name</b>: <b>jobname</b>.
</UL>
<p>
<A NAME="TORD"></A>
III.04.10. <em u>Torsion angle definitions</em>
<p>
<B>TORD</B> Key1 Key2 Data Fdata
<P>
Prerequisite:
<B><A HREF="#SLTA">SLTA</A></B><BR>
Can not be followed by:
<B><A HREF="#PRMF">PRMF</A></B>, <B><A HREF="#SETC">SETC</A></B>,
<B><A HREF="#LOOP">LOOP</A></B>, <B><A HREF="#FREE">FREE</A></B>,
<B><A HREF="#PARD">PARD</A></B>, <B><A HREF="#SKWT">SKWT</A></B><BR>
Potential successors:
<B><A HREF="#CNFG">CNFG</A></B>, <B><A HREF="#SPST">SPST</A></B><BR>
Prerequisite of:
<B><A HREF="#REGE">REGE</A></B>, <B><A HREF="#TORT">TORT</A></B>,
<B><A HREF="#LPST">LPST</A></B>, <B><A HREF="#GENT">GENT</A></B>,
<B><A HREF="#TAND">TAND</A></B><BR>
Related keys:
<B><A HREF="#LOOP">LOOP</A></B>, <B><A HREF="#PART">PART</A></B>,
<B><A HREF="#TORT">TORT</A></B>, <B><A HREF="#LPST">LPST</A></B>.
<p>
<u>Key1</u> (torsion definiton source):
<UL>
<LI><B>INPT</B><BR>
<U>Data</U>: <B>ntorsinp</B>, <B>nsltcop</B>, <B>rngfac</B>,
<B>rngfacl</B>, <B>inrcat</B>
<UL>
<LI><B>ntorsinp</B> - Number of torsion angles on the solute.
<LI><B>nsltcop</B> {1} - number of solute copies to apply the
inputted torsions. Applicable for free energy simulations where two
or three solute copies are needed (see the key
<B><A HREF="#FREE">FREE</A></B> for details.
<LI><B>rngfac</B> {1.0}
(<B>nsltcop</B> numbers, one for each solute copy) -
The <b>torstep</b> values read in RECTYPE 24's will
be all multiplied with <B>rngfac</B>.
<LI><B>rngfacl</B> {1.0}
(<B>nsltcop</B> numbers, one for each solute copy) -
The <b>steploop</b> values read in RECTYPE 24's will
be all multiplied with <B>rngfacl</B>.
The different solute copies may have different factors.<BR>
<LI><B>incrat</B> {0} - the atom indices read in below will be
incremented
by <B>incrat</B> (to allow the insertion of new molecules into the
solute).<BR>
<U>Fdata</U> (when <B>ntorsinp</B> &gt; 0):
<UL>
<LI>RECTYPE 24's (for each torsion) <B>ik</B>, <B>il</B>,
<B>steploop</B>, <B>torstep</B>, <B>itorgrp</B>,
<B>isense</B>, <B>wtors</B>, <B>deldih</B>
<B>targetang</B>, <B>iskew</B> (2i5,2f10.0,2i5,3f10.0,i5)
<LI><B>ib</B>, <B>ik</B> - The two (bonded) atoms defining the
bond of torsion. The program will assign automatically one
neighbour
each of <B>ik</B> and <B>il</B> to define the actual torsion angle.
For solute molecules that were sepcified with the help of the
<B><A HREF="#CLON">CLON</A></B> key, <B>ik</B> and <B>il</B>
refer to the atoms in the original (uncloned list). Torsions on
cloned molecules will also be cloned automatically.
For free-energy calculations using three solute copies
(<B><A HREF="#FREE5">FREE&nbsp;PMF1</A> </B> or
<B><A HREF="#FREE6">FREE&nbsp;PMNL</A> </B>)
the atom numbers should relate to the <em>third</em> copy.
<LI><B>steploop</B> - The torsional displacement stepsize range
for loop moves.
It is analogous to <b>wpslt*</b> (given with the
<B><A HREF="#PARD">PARD</A></B> key).
Extension biasing is not applied to loop moves.
<LI><B>torstep</B> - The torsional displacement stepsize range.
It is analogous to <b>wpslt*</b> (given with the
<B><A HREF="#PARD">PARD</A></B> key) and its meaning also varies
depending on Key 1 of <B><A HREF="#PART">PART</A></B>.
<LI><B>itorgrp</B> - The torsion group number. For
<B><A HREF="#FREE">FREE&nbsp;PMF1</A> TORS</B> the first group
is the group of torsions moved by the coupling parameter.
Torsions belonging to the same torsion group
will be changed simultaneously.
All torsions of a group must refer to the same solute molecule.
<LI><B>isense</B> - The direction of rotation for
<B>deldih</B>.
+1: clockwise, -1: counterclockwise.
<LI><B>wtors</B> {1.0} Weight for selecting this torsion.
<LI><B>deldih</B> - For torsions involving the coupling parameter
the program will generate the second conformation by a
rotation of <B>deldih</B> degrees (in the direction specified by
<B>isense</B>) whenever it is not zero.
This saves the user the need to actually generate
the coordinates of the transformed solute conformation.<BR>
For torsions not involved in the coupling parameter, the program will
keep this torsion within <B>deldih</B> degrees of its initial value
(as defined by the coordinates read by the
<B><A HREF="#SLTA">SLTA</A></B> key)
whenever it is not zero.<BR>
<B>deldih</B> can not be given as negative.<BR>
<B>targetang</B> and <B>iskew</B> are read only when the
<B><A HREF="#SKWT">SKWT</A></B> key is present:
<UL>
<LI><B>targetang</B> - angle toward which the sampling may be skewed
<LI><B>iskew</B> - when it is not zero, the sampling of the
corresponding torsion will be skewed
</UL>
</UL>
</UL>
<LI><B>ALL </B><BR>
<u>Key2</u> (torsion group definition):
<UL>
<LI><B>MOLG</B> - Each solute molecule will be a torsion group
<LI><B>SING</B> - Each torsion will be a torsion group
<LI><B>BRAG</B> - Each branch from the root atom will be a
torsion group<BR>
<U>Data</U>: <B>defrot</B>, <B>deflooprot</B>, <B>maxtorchain</B>,
<B>nfixbondtyp</B>,
<B>atypx1</B>, <B>atypy1</B>, <B>atypx2</B>, <B>atypy2</B>, ...
<UL>
<LI><B>defrot</B> {10.0 deg or 0.5 for EXB*} - The <b>torstep</b> value
to be used for all torsions
<LI><B>deflooprot</B> {10.0} - The <b>steploop</b> value to be used for all
torsions
<LI><B>maxtorchain</B> {<b>nslt</b>} - Simple torsions moving chains longer
than <B>maxtorchain</B> atoms will be disabled
<LI><B>atypx1</B>, <B>atypy1</B> - list of <B>nfixbondtyp</B> pairs of
atomtypes. Torsion angles around bonds between such pairs of atoms will be
held fixed.
</UL>
</UL>
>LI><B>RDBB</B><BR>
Same input as Key1=<B>INPT</B>, but in addition the program will ensure that
protein backbones (if any) use backbone atoms for the four atom defining
each backbone torsion angle.
>LI><B>ALBB</B><BR>
Same input as Key1=<B>ALL </B>, with the same added function as for 
Key1=<B>RDBB</B>
</UL>
<P>
<A NAME="CSEG"></A>
III.04.11 <em u>CHARMM segment id handling</em>
<p>
<B>CSEG</B>
<P>
When this key is present, the last 4 characters of <b>spgroup</b>
(read with <B><A HREF="#SLTA">SLTA</A> **** MMC</B>) will be
used for the segment id when CHARMM CRD format is written
with the <B><A HREF="#WCNF">WCNF</A> CHRM</B> or
<B><A HREF="#DENF">DENF</A> CHRM</B> keys.
<P>
<A NAME="Input_PF"></A>
III.5. <B>Potential descriptor keywords</B>
<OL>
<LI><TT><B><A HREF="#SUPT">SUPT</A></B>: </TT>Solute potential
type<BR>
<LI><TT><B><A HREF="#SVPT">SVPT</A></B>: </TT>Solvent potential
type<BR>
<LI><TT><B><A HREF="#MIXR">MIXR</A></B>: </TT>LJ parameter mixing
rule<BR>
<LI><TT><B><A HREF="#PRMF">PRMF</A></B>: </TT>Torsion potential
specification<BR>
<LI><TT><B><A HREF="#PMOD">PMOD</A></B>: </TT>Potential library
modifications<BR>
<LI><TT><B><A HREF="#MODA">MODA</A></B>: </TT>Element
modification<BR>
<LI><TT><B><A HREF="#PFRD">PFRD</A></B>: </TT>Potential library input
<LI><TT><B><A HREF="#SUVC">SUVC</A></B>: </TT>Solute-solvent
cutoff<BR>
<LI><TT><B><A HREF="#SUUC">SUUC</A></B>: </TT>Solute-solute
cutoff<BR>
<LI><TT><B><A HREF="#SVVC">SVVC</A></B>: </TT>Solvent-solvent
cutoff<BR>
<LI><TT><B><A HREF="#INCT">INCT</A></B>: </TT>Solvent inner
cutoff<BR>
<LI><TT><B><A HREF="#PROT">PROT</A></B>: </TT>Minimum repulsion for
general solvent
<LI><TT><B><A HREF="#RFCR">RFCR</A></B>: </TT> Reaction-field correction
<LI><TT><B><A HREF="#SETC">SETC</A></B>: </TT>Set various constants
<LI><TT><B><A HREF="#CHRG">CHRG</A></B>: </TT>Solute charge
manipulations<BR>
<LI><TT><B><A HREF="#SPST">SPST</A></B>: </TT>Field-dependent potentials
<LI><TT><B><A HREF="#SPPS">SPPS</A></B>: </TT>Solute
molecule-dependent potentials<BR>
<LI><TT><B><A HREF="#AROM">AROM</A></B>: </TT>List of aromatic carbons
<LI><TT><B><A HREF="#SVIN">SVIN</A></B>: </TT>Solvation parameter input<BR>
<LI><TT><B><A HREF="#ENHB">ENHB</A></B>: </TT>Hydrogen-bond potential<BR>
<LI><TT><B><A HREF="#HBMO">HBMO</A></B>: </TT>Non-default hydrogen-bond<BR>
<LI><TT><B><A HREF="#FLXR">FLXR</A></B>: </TT>Residues to sample<BR>
<LI><TT><B><A HREF="#CNST">CNST</A></B>: </TT>Constraint potential<BR>
<LI><TT><B><A HREF="#VVNE">VVNE</A></B>: </TT>No solvent-solvent elecrostatic in
Solvation parameter input</BR>
<LI><TT><B><A HREF="#IGTT">IGTT</A></B>: </TT>Solute-solute potential is set to zero<BR>
</OL>
<A NAME="SUPT"></A>
III.05.01. <em u>Solute-solvent potential</em>
<p>
<B>SUPT</B> : Key1 [Key2]
<p>
Potential successor:
<B><A HREF="#RCKP">RCKP</A></B><BR>
Prerequisite of:
<B><A HREF="#SLTA">SLTA</A></B>, <B><A HREF="#PROT">PROT</A></B>,
<B><A HREF="#RFSL">RFSL</A></B><BR>
<p>
Related keys:
<B><A HREF="#PMOD">PMOD</A></B>,
<B><A HREF="#MODA">MODA</A></B>,
<B><A HREF="#PFRD">PFRD</A></B>,
<B><A HREF="#MIXR">MIXR</A></B>.
<p>
<u>Key1</u> (solute PF library source):
<UL>
<LI><B>CLMG</B> - Clementi et al.'s library.<BR>
<LI><B>EPEN</B> - EPEN/QPEN potential (parameters to be read
in)<BR>
<LI><B>AM02</B> - Kollman et al.'s new library (AMBER).<BR>
<LI><B>AM94</B> - Kollman et al.'s DB94 library (AMBER).<BR>
<LI><B>CHRM</B> - CHARMM (Parm 22)<BR>
<LI><B>GR87</B> - Berendsen/Van Gunsteren library (GROMOS87).<BR>
<LI><B>GR96</b> - Berendsen/Van Gunsteren library (GROMOS96).<BR>
<LI><B>OPLS</B> - Jorgensen et al.'s library (OPLS). See also
key <B><A HREF="#CHRG">CHRG</A></B><BR>
<LI><B>HSPH</B> - Hard spheres - atom types (i.e., HS diameters)
have to be defined by the <B><A HREF="#PMOD">PMOD</A></B> key
<LI><B>ATNO</B> - Atomtype is just atomic number (only used for
proximity analysis)<BR>
<LI><B>DUMM</B> - Dummy atom (no interactions)
<p>
<u>Key2</u> (solute PF library source):
if present it will apply to the second solute copy
for <B><A HREF="#FREE2">FREE&nbsp;TICA</A></B> and
<B><A HREF="#FREE6">FREE</A> PMFL</B> calculations.
</UL>
<p>
<A NAME="SVPT"></A>
III.05.02. <em u>Solvent-solvent potential</em>
<p>
<B>SVPT</B> Key1 Key2 [Data]
<p>
Potential successor:
<B><A HREF="#RCKP">RCKP</A></B><BR>
Prerequisite of:
<B><A HREF="#SLTA">SLTA</A></B>, <B><A HREF="#SLVA">SLVA</A></B><BR>
<p>
<u>Key1</u> (solvent PF type):
<UL>
<LI><B>MCY </B> - Solvent is MCY-CI water<BR>
<u>Key2</u> (solvent PF parameters):
<UL>
<LI><B>MCY </B> - Use the stored MCY-CI-II parameters.<BR>
<LI><B>YMD </B> - Use the stored Yoon-Morokuma-Davidson
parameters (YMD).
</UL>
<LI><B>TIP3</B> - Solvent is TIPS/SPC/TIP3P (type) water.<BR>
<u>Key2</u> (solvent PF parameter source):
<UL>
<LI><B>TIPS</B> - Use the stored parameters for Jorgensen TIPS<BR>
<LI><B>SPC </B> - Use the stored parameters for SPC.<BR>
<LI><B>TIP3</B> - Use the stored parameters for Jorgensen
TIP3P.<BR>
<LI><B>SPCE</B> - Use the stored parameters for SPC/E.<BR>
<LI><B>READ</B> - Input water-water interaction parameters.<BR>
<U>Data</U>: <B>c6</B>, <B>c12</B>, <B>qh</B> (in kcal,  and
electron)
<UL>
<LI><B>c6</B> - Coefficient of the -1/r<sup>6</sup> term
in kcal <sup>6</sup>.
<LI><B>c12</B> - Coefficient of the 1/r<sup>12</sup> term
in kcal <sup>12</sup>.<BR>
<LI><B>qh</B> - Charge on the hydrogen<BR>
</UL>
</UL>
<LI><B>TIP4</B> - Solvent is TIPS2/TIP4P/B-F (type) water.<BR>
<u>Key2</u> (solvent PF parameter source):
<UL>
<LI><B>BRFW</B> - Use the stored parameters for Bernal-Fowler<BR>
<LI><B>TIP2</B> - Use the stored parameters for Jorgensen TIPS2<BR>
<LI><B>TIP4</B> - Use the stored parameters for Jorgensen
TIP4P.<BR>
<LI><B>READ</B> - See above
</UL>
<LI><B>TIP5</B> - Solvent is TIP5P (type) water. Currently implemented only for
<B><A HREF="#HRDW">HRDW</A> SCLR</B>
<BR>
<u>Key2</u> (solvent PF parameter source):
<UL>
<LI><B>TIP5</B> - use the stored parameters for TIP5P water.<BR>
<LI><B>READ</B> - See above
</UL>
<LI><B>GENL</B> - Solvent is general 1-6-12 molecule.<BR>
<u>Key2</u>: The parameter library used (see
<B><A HREF="#SUPT">SUPT</A></B> for the list).
</UL>
<u>Key2</u>=<B>GENL</B> also requires the key
<B><A HREF="#SLVA">SLVA</A> READ</B>
<P>
<A NAME="MIXR"></A>
III.05.03. <em u>Lennard-Jones parameter mixing rules</em>
<p>
<B>MIXR</B> Key1
<p>
Potential successors:
<B><A HREF="#SLTA">SLTA</A></B>, <B><A HREF="#RCKP">RCKP</A></B><BR>
<p>
<u>Key1</u>:
<UL>
<LI><B>GEGE</B> - Geometric mean for both the SIG's and the
EPS's<BR>
<LI><B>ARGE</B> - Arithmetic mean for the SIG's and geometric mean
for the EPS's.
</UL>
E<sub>LJ</sub>(r)=4*EPS[(SIG/r)<sup>12</sup>-(SIG/r)<sup>6</sup>]
= c12/r<sup>12</sup>-c6/r<sup>6</sup>.
Relevant only for AMBER, CHARMM, GROMOS and OPLS potentials.
<p>
<A NAME="PRMF"></A>
III.05.04. <em u>Torsion potential parameter input</em>
<p>
<B>PRMF</B> Key1 Data
<P>
Can not be followed by: <B><A HREF="#SLTA">SLTA</A></B><BR>
Potential successor: <B><A HREF="#TORD">TORD</A></B><BR>
Related key: <B><A HREF="#PFRD">PFRD</A></B><BR>
<P>
<u>Key1</u>: Same as Key1 of <B><A HREF="#SUPT">SUPT</A></B>
- the name of the potential library.<BR>
<U>Data</U>: Name of the file containing the parameters.
<p>
The input syntax for the parameter files is defined by the
respective libraries.
At present, only AMBER, CHARMM, OPLS (in CHARMM format)
and GROMOS libraries can be read.
Furthermore, for GROMOS input the file name has to be of the
form &lt;name&gt;bon.itp and the file &lt;name&gt;.rpt also
has to be present.
<p>
<A NAME="PMOD"></A>
III.05.05. <em u>Potential (re)definition</em>
<p>
<B>PMOD</B> Key1 Data Fdata
<p>
Potential successor:
<B><A HREF="#SLTA">SLTA</A></B><BR>
<P>
<u>Key1</u>:
<UL>
<LI>Key of the potential function library to be modified (see
key <B><A HREF="#SUPT">SUPT</A></B>). All parameters
will be echoed on the output listing.<BR>
Data: <B>ntyps</B>
<UL>
<LI><B>ntyps</B> - Number of new types<BR>
<U>Fdata</U>
<UL>
<LI>RECTYPE 17's: (ntyp times) <B>ityp</B>, <B>ifcg</B>,
<B>rfshl</B>, <b>pflab</b>
(2I5,F10.0,1X,A4) or (1X,A4,F10.0,1X,A4)
<LI><B>ityp</B> - Atom type number
(see key <B><A HREF="#SLTA">SLTA</A></B>).
For new types, it is preferable to give here an unused type number
(the <B><A HREF="#PLBP">PLBP</A></B> key can give a list of
the type numbers used).
For modifying existing types it can be given as zero
- in this case the type number will be determined
from the potential label <b>pflab</b> (see below).
<LI><B>ifcg</B> - Functional group to which this atom type belongs
to.
It is used only for display and data checking. The following
functional groups are coded:
<pre>
 1:&gt;C&lt;    11:&gt;N-     21:-OH    31:PO4-   41:  Li   51:  Ca
 2:&gt;CH-   12:&gt;NH     22:&gt;PO2   32:HPO3   42: Li+   52:Ca2+
 3:&gt;CH2   13:-NH2    23:&gt;N+&lt;   33:       43:  Be   66:  F-
 4:-CH3   14:=N-     24:N+H1   34:       44:Be2+   67: -Cl
 5:=C&lt;    15:=NH     25:N+H2   35:       45:  Na   69: Cl-
 6:=CH-   16:-=N     26:N+H3   36:       46: Na+   73: -Br
 7:=CH2   17:C=O     27:COO-   37:       47:  Mg   76: Br-
 8:=C=    18:H/CO    28:-SH    38:       48:Mg2+   77:  -I
 9:-=CH   19:-O-     29:-S-    39:       49:   K   78:  I-
10:==C    20:P-O-    30:MNH+   40:       50:  K+
</pre>
<LI><B>rfshl</B> - First shell radius for proximity analysis
(used only with <B><A HREF="#RFSL">RFSL</A></B> <B>STOR</B>).
<LI><B>pflab</b> - Potential label corresponding to potential type
<B>ityp</B> (see key <B><A HREF="#SLTA">SLTA</A></B>).
If left blank and <b>ityp</b> refers to an existing type,
the label corresponding to this existing type will be used.
<LI>Non-bonded parameters. If the first number, <B>iatnos</B>(<B>ityp</B>),
is zero then no more data has to be provided for this atomtype
(and the non-bonded parameters are left unchanged).
<UL>
<LI>RECTYPE 18A's for CLEMENTI potentials: (ntyp times)
<B>iatnos</B>(<B>ityp</B>), <B>a</B>(<B>ityp</B>,o),
<B>a</B>(<B>ityp</B>,h), <B>b</B>(<B>ityp</B>,o),
<B>b</B>(<B>ityp</B>,h), <B>c</B>(<B>ityp</B>,o),
<B>c</B>(<B>ityp</B>,h) (I5,4E15.6,/,2F10.0)
<UL>
<LI><B>iatnos</B>(<B>ityp</B>) - Atomic number of type <B>ityp</B>.
<LI><B>a</B>, <B>b</B>, <B>c</B> - The A,B,C parameters of the
</UL>
Clementi potential in kcal/mol and  units for <B>a</B> and
<B>b</B> and in a.u. and  units for <B>c</B>.
<LI>RECTYPE 18B's for Kollman et al.(AMBER) potentials (both old
and new) and CHARMM: <B>iatnos</B>(<B>ityp</B>),
<B>rmin</B>(<B>ityp</B>), <B>emn</B>(<B>ityp</B>),
<B>rmn14</B>(<B>ityp</B>), <B>emn14</B>(<B>ityp</B>) (I5,4E15.6)
<UL>
<LI><B>iatnos</B>(<B>ityp</B>) - Atomic number of type <B>ityp</B>.
<LI><B>rmin</B>, <B>emin</B> - c6=2.0*<B>emn</B>*(<B>rmin</B>*2.0)<sup>6</sup>
and c12=<B>emn</B>*(<B>rmin</B>*2.0)<sup>12</sup>
in  and kcal/mol units.
Note, that <B>rmin</B>=(2*c12/c6)<sup>(1/6)</sup>/2 and
<B>emin</B>=(c6<sup>2</sup>/c12)/4.
<LI><B>rmin14</B>{<B>rmin</B>}, <B>emin14</B>{<B>emin</B>} -
non bonded parameters for 1-4 interactions.
</UL>
<LI>RECTYPE 18C's for Berendsen-Van Gunsteren (GROMOS 87 or 96) potentials:
<B>iatnos</B>(<B>ityp</B>), <B>a</B>(<B>ityp</B>),
<B>b</B>(<B>ityp</B>), <B>a14</B>(<B>ityp</B>),
<B>b14</B>(<B>ityp</B>) (I5,4E15.6)
<UL>
<LI><B>iatnos</B>(<B>ityp</B>) - Atomic number of type <B>ITYP</B>.
<LI><B>a</B>, <B>b</B>: The A,B parameters (c6 and c12) in
kcal/mol and  units.
<LI><B>a14</B>{<B>a</B>}, <B>b14</B>{<B>b</B>} - The A,B parameters
(c6 and c12) for the 1-4 interactions.
</UL>
<LI>RECTYPE 18D's for Jorgensen (OPLS) potentials:
<B>iatnos</B>(<B>ityp</B>), <B>sig</B>(<B>ityp</B>),
<B>eps</B>(<B>ityp</B>), <B>q</B>(<B>ityp</B>) (I5,3E15.6)
<UL>
<LI><B>iatnos</B>(<B>ityp</B>) - Atomic number of type <B>ityp</B>.
<LI><B>sig</B>, <B>eps</B> - The LJ SIG and EPS parameters in  and
kcal/mol units.
<LI><B>q</B> - The charge associated with atomtype <B>ityp</B>.
</UL>
<LI>RECTYPE 18E's for Hard Spehere potentials:
<B>iatnos</B>(<B>ityp</B>), <B>sig</B>(<B>ityp</B>) (I5,E15.6)
<UL>
<LI><B>iatnos</B>(<B>ityp</B>) - Atomic number of type <B>ityp</B>.
<LI><B>sig</B> - The hard sphere diameter in .
</UL>
</UL>
</UL>
</UL>
</UL>
<A NAME="MODA"></A>
III.05.06. <em u>Atom (element) descriptor modifications</em>
<p>
<B>MODA</B> Data Fdata
<p>
Potential successor:
<B><A HREF="#SLTA">SLTA</A></B><BR>
<p>
Data: <B>nmod</B>
<UL>
<LI><B>nmod</B> - Number of element descriptors to modify<BR>
Fdata:
<UL>
<LI>RECTYPE 10 (<B>nmod</B> records):
<B>ian</B>, </B>ianpr</B>, </B>ifgr</B>, </B>namr</B>,
</B>awr</B>, </B>vdwr</B>, </B>ramaxr</B>, </B>nv</B>
(3i5,1x,a4,3f10.0,i5)
<UL>
<LI><B>ian</B> - atomic number to modify
<LI><B>ianpr</B> - new atomic number for print
(to be used with united atom types)
<LI><B>ifgr</B> - new functional group index
(see key <B><A HREF="#PMOD">PMOD</A></B>)
<LI><B>namr</B> - new name
<LI><B>awr</B> - new atomic weight
<LI><B>vdwr</B> - new Van der Waals radius
<LI><B>ramaxr</B> - new covalent radius
<LI><B>nv</B> - new valence
</UL>
</UL>
</UL>
Using this key the built in constants for each chemical
element can be modified or new ones can be
defined.
Fields left blank or given as zero will retain their original value
<P>
<A NAME="PFRD"></A>
III.05.07. <em u>Potential library input</em>
<p>
<B>PFRD</B> Key1 Data
<p>
Potential successor: <B><A HREF="#SLTA">SLTA</A></B><BR>
Related key: <B><A HREF="#PRMF">PRMF</A></B><BR>
<p>
<u>Key1</u> (library type):
<UL>
<LI><B>CHRM</B> - Read the NB parameters from a Charmm parameter file
<LI><B>OPLS</B> - Read the NB parameters from a Charmm-formatted
parameter file for OPLS types<BR>
<U>Data</U>: <b>paramfile</b> <b>iswitchcase</b>
<UL>
<LI><b>paramfile</b> - the name of the parameter file
<LI><b>iswitchcase</b> {0} - when =1 the potential labels read
will be converted from lower case to upper case
</UL>
</UL>
<P>
<A NAME="SUVC"></A>
III.05.08. <em u>Solute-solvent cutoff</em>
<p>
<B>SUVC</B> Key1 [Data]
<p>
Potential successor:
<B><A HREF="#RCKP">RCKP</A></B><BR>
Prerequisite of:
<B><A HREF="#PBCN">PBCN</A></B><BR>
<p>
<u>Key1</u> (cutoff type):
<UL>
<LI><B>SPCC</B> - Spherical cutoff based on the solute COM
<LI><B>MICC</B> - Minimum image convention based on the solute COM
<LI><B>ISOE</B> - Isoenergy cutoff
<LI><B>SPGC</B> - Spherical cutoff based on solute group COM's
<LI><B>MIGC</B> - Minimum image convention, based on solute group
COM's<BR>
<U>Data</U>: Cutoff radius {R<sub>insc</sub> - see
<B><A HREF="#PBCN">PBCN</A></B>}
</UL>
For the particular case of a single solute molecule in nmolec-1 solvent it
is advisable to use the minimum-image cutoff for solute-solvent
interaction to avoid the single discontinuity at the boundary of
the cutoff sphere.
Multimolecular solutes or for solutes with torsional degree of freedom
require group-based cutoffs.
<p>
In the minimum-image cutoff for each pair of molecules only
the interaction between the nearest images are taken into
account, irrespective of the intermolecular separation. This is
equivalent to the use of a cutoff cube, prism, etc., identical in
size, shape and orientation with the periodic cell.
<p>
The cutoff radius may be based on the centers of solute
groups (see <B>SPGC</B> or <B>MIGC</B> above) - useful for large
solutes. Also,
interionic PMF calculations can use the
<A HREF="#Mezei_94">iso-energy cutoff</A> introduced by Mezei.
<p>
<A NAME="SUUC"></A>
III.05.09. <em u>Solute-solute cutoff</em>
<p>
<B>SUUC</B> Key1
<p>
Potential successor:
<B><A HREF="#RCKP">RCKP</A></B><BR>
Prerequisite of:
<B><A HREF="#PARD">PARD</A></B>, <B><A HREF="#PART">PART</A></B>,
<B><A HREF="#ENGL">ENGL</A></B><BR>
<p>
<u>Key1</u> (cutoff type):
<UL>
<LI><B>NONE</B> - No cutoff, no periodic images considered
<LI><B>MIMC</B> - Minimum image convention based on solute molecule
centers
<LI><B>SPMC</B> - Spherical cutoff based on solute molecule centers
<LI><B>MIGC</B> - Minimum image convention based on solute molecule
group (residues) centers
<LI><B>SPGC</B> - Spherical cutoff based on solute molecule
group (residue) centers<BR>
<U>Data</U>: <B>rc</b>
<UL>
<LI><B>rc</b> {R<sub>insc</sub>} - Cutoff radius
(see <B><A HREF="#PBCN">PBCN</A></B>)
</UL>
</UL>
Note, that the <B>**GC</B> options will update the energies of all
atoms that are in a residue when a torsion move moves
the group center atom.
To reduce the extra calculations, a) you should not use too large
groups when this option is active and b) whenever possible, group
torsions
so that they don't leave too many group members unchanged.
<p>
<A NAME="SVVC"></A>
III.05.10. <em u>Solvent-solvent cutoff</em>
<p>
<B>SVVC</B> Key1 [Data]:
<p>
Potential successors:
<B><A HREF="#RCKP">RCKP</A></B>, <B><A HREF="#PART">PART</A></B><BR>
Prerequisite of:
<B><A HREF="#PBCN">PBCN</A></B><BR>
<p>
<u>Key1</u> (cutoff type):
<UL>
<LI><B>SPCC</B> - Spherical cutoff
<LI><B>MINI</B> - Minimum image convention<BR>
<U>Data</U>: <B>rc, exprc, maprc</B>
<UL>
<LI><B>rc</B> {R<sub>insc</sub> - see
<B><A HREF="#PBCN">PBCN</A></B>} - Cutoff radius
<LI><B>exprc</B> {5.5 } - Special cutoff for the
exponential term in the MCY potential.
<LI><B>maprc</B> {<B>rc</B>+2} - Cutoff for inclusion to the
solvent-solvent <A HREF="#Mezei_88_BM">near-neighbor bitmap</a>
</UL>
</UL>
<A NAME="INCT"></A>
III.05.11. <em u>Inner cutoff on the solvent</em>
<p>
<B>INCT</B> Key1 [Data]
<p>
Potential successor:
<B><A HREF="#RCKP">RCKP</A></B><BR>
<p>
<u>Key1</u>:
<UL>
<LI><B>NONE</B> - None used
<LI><B>EXPL</B> - Exponential repulsion at short distances<BR>
<U>Data</U>: <B>rwwina</B>
<UL>
<LI><B>rwwina</B> {2.0 } - For water-water distance rww &lt;
<B>rwwina</B>, replace the energy by
10.0*exp(<B>rwwina</B><sup>2</sup>-rww<sup>2</sup>) + 1000.0
kcal/mol.
</UL>
</UL>
<A NAME="PROT"></A>
III.05.12. <em u>Minimum repulsion</em>.
<p>
<B>PROT</B> Data
<P>
Prerequisite:
<B><A HREF="#SUPT">SUPT</A></B><BR>
Potential successor:
<B><A HREF="#SLTA">SLTA</A></B><BR>
<p>
<U>Data</U>: <B>c12prot</B>
<UL>
<LI><B>c12prot</B> - For general solvent only, solute-solvent
interactions where no LJ interaction is specified by the potential
library are modified ('protected') with a repulsion term of
<b>c13prot</b>/r12<sup>12</sup> kcal/<sup>12</sup> .
This can be used to avoid numerical instabilities during
creation-annihilation TI (<B><A HREF="#FREE2">FREE&nbsp;TICA</A></B>)
that can result from unlike charges getting too close to each
other.
</UL>
<A NAME="RFCR"></A>
III.05.13. <em u>Reaction-field correction</em>
<p>
<B>RFCR</B> [Data]
<P>
Variable to set to 'T' in the <A HREF="#PREP">preprocessor</A>: <B>RF</B>
<p>
<U>Data</U>: <B>epsrf</B>
<UL>
<LI><B>epsrf</B> {80.0} - The dielectric constant in the
reaction field correction formula.
<B>epsrf</B>=1.0 represents vacuum (i.e., no correction);
<B>epsrf</B>=80.0 represents water.
</UL>
The use of the reaction field correction requires the use of
identical cutoffs for the
solvent-solvent (<B><A HREF="#SVVC">SVVC</A></B>),
solute-solvent (<B><A HREF="#SUVC">SVVC</A></B>) and
solute-solute (<B><A HREF="#SUUC">SUUC</A></B>) interactions.
Also, the cutoffs are to be group based and each group has to be
electrically neutral.
<p>
<A NAME="SETC"></A>
III.05.14. <em u>Set various constants</em>
<p>
<B>SETC</B> Key1 Data
<p>
Potential successors:
<B><A HREF="#SLTA">SLTA</A></B>, <B><A HREF="#TORD">TORD</A></B><BR>
<p>
<UL>
<LI>Key1=<B>DIEL</B>:
<UL><U>Data</U>: <B>dielc</B>
<LI><B>dielc</B> - An overall dielectric constant,
applied (as a divisor) to
<em u>all</em> electrostatic interactions.
If a distance-dependent dielectric function code
is activated it multiplies whichever (Mehler-Solmayer or epr(r)=r)
distance-dependent dielectric function is calculated
(see the <A HREF="#PREP">preprocessor</A>).
</UL>
<LI>Key1=<B>DDIE</B>:
Constants for the Mehler-Solmayer dielectric function<BR>
eps=A +(eps0-A)/[1+  k * exp(-lambda*(eps0-A)*r)]
<UL> <U>Data</U>: <B>eps0_ddd</B>,
<B>a_ddd</B>,  <B>rlambda_ddd</B>, <B>rk_ddd</B>
<LI><B>eps0_ddd</B> {78.4} - eps0
<LI><B>a_ddd</B> {6.0294} - A
<LI><B>rlambda_ddd</B> {0.18733345} - lambda
<LI><B>rk_ddd</B> {213.5782} - k
<LI><B>eps1_dd_ddd</B> {8.0} - when <B>eps1_dd_ddd</B> &gt; 0 then the dielectri
constant will be set to <B>eps1_dd_ddd</B> for distances that are less
than the distance at which the Mehler-Solmayer function reaches
<B>eps1_dd_ddd</B>
</UL>
NOTE: This dielectric constant is not applied to interactions involving any 
of the explicit water models, i.e., only to interactions involving the
solute and a <i>general</i> solvent.<BR>
<LI>Key1=<B>FC14</B>:
<U>Data</U>: <B>esf14</B>, <B>vdwf14</B>
<UL>
<LI><B>esf14</B> {0.833333} Factor to apply to the 1-4
electrostatic interactions
<LI><B>vdwf14</B> {0.5} Factor to apply to the 1-4 Lennard-Jones
(i.e., Van der Waals) interactions.<BR>
If <B>FC14</B> is not used then the program sets the default values
that are potential dependent: (1.0, 1.0) for CHARMM,
(0.5, 0.5) for OPLS, (0.833333, 0.5) for AMBER and all the others as well.<BR>
</UL>
<LI>Key1=<B>FCIN</B>: <U>Data</U>: <B>fcintra</B>
<UL>
<LI><B>fcintra</B> Factor to apply to the solute intramolecular
energies when summing them into the total energy
</UL>
<LI>Key1=<B>BLKS</B>: <U>Data</U>: <B>nblskip</B>, <B>nbllim</B>
<UL>
<LI><B>nblskip</B> {0} - Number of control function blocks to skip when
calculating error estimates (to drop equilibration part).
<LI><B>nbllim</B> {#MI} - Number of the last control function blocks to use
calculating eror estimates.
</UL>
</UL>
<A NAME="CHRG"></A>
III.05.15. <em u>Solute charge manipulations</em>
<p>
<B>CHRG</B> Key1
<p>
Potential successor:
<B><A HREF="#SLTA">SLTA</A></B><BR>
<p>
<u>Key1</u>:
<UL>
<LI><B>ZERO</B> - Set the solute and solvent charges to zero.
This is advisable when a solvent configuration is generated
randomly since it will avoid solvents getting stuck to the solute due
to unphysically close distances involving atoms that have no
repulsive core assigned to them.
<LI><B>INPT</B> - Use inputted charges (after
<B><A HREF="#SLTA">SLTA</A></B>) for OPLS
</UL>
<A NAME="SPST"></A>
III.05.16. <em u>Field-dependent solute potential</em>
<p>
<B>SPST</B> Key1 Data
<p>
Prerequisite: <B><A HREF="#FLXR">FLXR</A></B><BR>
<p>
Can not be followed by:
<B><A HREF="#AROM">AROM</A></B>, <B><A HREF="#SVIN">SVIN</A></B>,
<B><A HREF="#ENHB">ENHB</A></B>, <B><A HREF="#PARD">PARD</A></B>,
<B><A HREF="#TORD">TORD</A></B>, <B><A HREF="#CNFG">CNFG</A></B>,
<B><A HREF="#HBMO">HBMO</A></B><BR>
Related keys: <B><A HREF="#AROM">AROM</A></B>,
<B><A HREF="#ENHB">ENHB</A></B><BR>
Variable to set to 'T' in the <A HREF="#PREP">preprocessor</A>: <B>GM</B><BR>
<p>
<u>Key1</u>:
<UL>
<LI><B>NONE</B> - No field-dependent potential
<LI><B>EMAP</B> - Additional solute potential depending on energy grid maps
are read from files (see below).<BR>
<LI><B>CMAP</B> - Energy grid maps are calculated.
<LI><B>CMPW</B> - Energy grid maps are calculated and written to the
map files.<BR>
<U>Data</U>: <B>ningridmap</B>, <B>ewvdmax</B>
<UL>
<LI><B>ningridmap</B> {131} - Number of gridpoints in the energy maps - has
to be an odd number and has to be the same as the
<A HREF="#Compilation">dimension symbol</A> #GM
<LI><B>ewvdmax</B> {1000.0} - VdW energies above <B>ewvdmax</B> will be
truncated to <B>ewvdmax</B>
<LI><B>scalemap</B> {1.0} - All maps read will be multiplied by <B>scalemap</B>
</UL>
Additional data for Key1=<B>CMAP</B> or <B>CMPW</B>:
<b>edgemap</b>, <b>gridcent_x</b>, <b>gridcent_y</b>, <b>gridcent_z</b>,
<b>gridbuffer</b>
<UL>
<LI><b>edgemap</b> - the edge of the (cubic) grid map.
<LI><b>gridcent_x</b>, <b>gridcent_y</b>, <b>gridcent_z</b> -
the coordinates of the center of the grid map.
<LI><b>gridbuffer</b> - atoms in a box extended by <b>gridbuffer</b>
will contribute to the energy maps.
</UL>
</UL>
For Key1=<B>EMAP</B>,
after establishing the grid constants the program reads the energy maps from
the files
<b>prot.pi</b>,
<b>lp.map</b>,
<b>pa.map</b>,
<b>pc.map</b>,
<b>pd.map</b>,
<b>ph.map</b>,
<b>pn.map</b>,
<b>po.map</b>,
<b>pp.map</b>,
<b>ps.map</b>.
for the electrostatic energy, van der Waals energy for
C, A, N, O, P, S and H atoms, respectively (A is aromatic C) and,
whenever needed, the file <b>lp.map</b> containing the contribution of the
(Autodock 4) solvation term and the file <b>ov.map</b>
containing the O-H VdW term.
These are binary files containig the 3-dimensional array
<b>e(ningridmap,ningridmap,ningridmap)</b> in a single record.
The electrostatic map (read first) also contains a second record containig
title, nx, ny, nz, ep, es, micro, stern, radd, ho, hi2, akp, aks
where
<UL>
<LI>title: 40-character description
<LI>nx,ny,nz: number of grids in the x, y,and z directions
(unless nx=ny=nz=ningridmap the program aborts)
<LI>ep,es: dielectric constant inside and outside the solute
<LI>hi2: grid spacing
<LI>akp,akps: Debey screening constant inside and outside the solute
</UL>
map(hi2*[(ningridmap-1)/2+1],
hi2*[(ningridmap-1)/2+1], hi2*[(ningridmap-1)/2+1]) is
the energy at <0.0,0.0,0.0>;
map(1,1,1) is the energy at
<-hi2*(ningridmap-1)/2, -hi2*(ningridmap-1)/2, -hi2*(ningridmap-1)/2>
<p>
The solvation term E<sub>solv</sub>(i,j) contribution from atoms i and j
is of the form<BR>
E<sub>solv</sub>(i,j) = (V<sub>i</sub>S<sub>j</sub> +
V<sub>j</sub>S<sub>i</sub>)*exp{r(i,j)<sup2</sup>/gaussdist2}<BR>
S<sub>i</sub>=A<sub>i</sub> + qk_par*abs(q(i))<BR>
where gaussdist2, and qk_par are constants
V<sub>i</sub>, and A<sub>i</sub> are atom type-dependent constants
and r(i,j) is the distance between atoms i and j and
q(i) is the partial charge on atom i.
All constants have built-in values that can be changed
with the <B><A HREF="#SVIN">SVIN</A></B> key
(the built-in values serve as defaults).
<p>
Furthermore, if the map file <b>hb.map</b> is present
then the program will treat all fixed oxygens as hydrogen bond donors
and include a 12-10 hydrogen-bond potential.
Hydrogen-bond acceptors can not be included in such a map since the H-X bond's
orientation is conformation-dependent. It can be calculated explicitly
during the simulation
with the <B><A HREF="#ENHB">ENHB</A></B> key
<p>
Note that on restart the energy maps have to be read again the same way
as for startup.
<p>
For Key1=<B>CMAP</B> or Key1=<B>CMPW</B>, the program needs the description of
the atoms contributing to the grid potentials.
These are assumed to be in
a file <b>&lt;jobname&gt;_2.slt</b> that is in the same format
as the records read
by the key (<B><A HREF="#SLTA">SLTA</A></B>, but only the fields
<B>tslt</B>(i), (<B>cslt</B>(j,i),j=1,3), <B>qslt</B>(i),
<B>labslt2</B>(i), <b>indxrdf(i)</b> are read.
There should be a record
for each atom that is not explicitly represented.
<p>
<A NAME="SPPS"></A>
III.05.17. <em u>Molecule-dependent solute potential</em>
<p>
<B>SPPS</B> Key1 Data
<p>
<u>Key1</u>:
<UL>
<LI><B>NONE</B> - No Molecule-dependent potential
<LI><B>FIEL</B> - An additional potential depending on the solute
atoms' positions only is used.<BR>
<U>Data</U>: <B>nint</B>, <B>int1</B>, ..., <B>int(nint)</B>,
The <B>nint</B> integer values and the <B>nreal</B> real values
will be stored in a common block and made available to the
subroutine <B>esltfield</B>.
<LI><B>POSI</B> - An additional potential depending on the solute
molecules' position is used.<BR>
<U>Data</U>: <B>nint</B>, <B>int1</B>, ..., <B>int(nint)</B>,
<B>nreal</B>, <B>real1</B>, ..., <B>real(nreal)</B><p>
The <B>nint</B> integer values and the <B>nreal</B> real values
will be stored
in a common block and made available to the subroutine
<B>esltmolec</B>.
</UL>
This key allowed the use of potential terms depending on the field or on the
molecule's center, orientation, etc. To use this option, the
<U>user also have to provide the corresponding code to the body of the
subroutine <B>esltmolec</B></U>.
<p>
<A NAME="AROM"></A>
III.05.18. <em u>List of aromatic carbons</em>
<p>
<B>AROM</B> Key1 Data
<p>
Prerequisite: <B><A HREF="#SLTA">SLTA</A></B><BR>
Potential successor:
<B><A HREF="#SPST">SPST</A></B><BR>
Prerequisite of:
<B><A HREF="#SVIN">SVIN</A></B><BR>
Related key: <B><A HREF="#SPST">SPST</A></B>.
<P>
<u>Key1</u>:
Solute potential type see key <B><A HREF="#SAMP">SUPT</A></B>.<BR>
<UL>
<u>Data</u>: <b>narom</b>, <b>pflab(1)</b>,...,<b>pflab(narom)</b>
<UL>
<LI><b>narom</b> - number of labels to read
<LI><b>pflab</b> - potential label of aromatic carbon (<b>narom</b> labels)
</UL>
</UL>
This information is needed for deciding what type of energy maps
to use with these atoms - see, e.g.,  <B><A HREF="#SPST">SPST</A> EMAP</B>.
<P>
<A NAME="SVIN"></A>
III.05.19. <em u>Desolvation parameter input</em>
<p>
<B>SVIN</B> Data
<p>
Prerequisite:
<B><A HREF="#AROM">AROM</A></B><BR>
Potential successor:
<B><A HREF="#SPST">SPST</A></B><BR>
Related keys: <B><A HREF="#SPST">SPST</A></B>,
<B><A HREF="#AROM">AROM</A></B>.
<p>
<u>Key1</u>:
<UL>
<LI><B>ALLA</B> -
Adds a desolvation term (Autodock 4 parametrization)
acting between all solute atoms.
Aliphatic carbons will be treated as united atoms
(the charges of hydrogens on carbons are added to the carbon charge).
<LI><B>CARB</B> - same as <B>ALLA</B> but the desolvation term is only applied
between carbon atoms.
<LI><B>CARA</B> - same as <B>CARB</B> but the desolvation term is only applied
between non-polar carbon atoms.
A carbon is considered polar if its charge is &gt; <b>polarcarblim</b><BR>
<LI><B>NOHP</B> - same as <B>ALLA</B> but the desolvation term is not applied
to pairs involving a hydrogen atom.
<LI><B>NOPP</B> - same as <B>ALLA</B> but the desolvation term is not applied
between polar atoms: N,O,S, polar H and polar C.
<u>Data</u>: <b>gaussdist2</b>, <b>gausslimfac</b>, <b>qp_par</b>,
6*[<b>A_solv(i)</b>, <b>V_solv(i)</b>],
<b>polarcarblim</b>
<UL>
<LI><b>gaussdist2</b> {24.50}
<LI><B>gausslimfac</B> {4.0} - desolvation terms will not be calculated for
distances that are larger than <B>gausslimfac</B>*<B>gaussdist2</B>.
<LI><b>qk_par</b> {0.01097}
<LI><b>A_solv(1)</b> {-0.00143} Carbon
<LI><b>V_solv(1)</b> {33.5103 } Carbon
<LI><b>A_solv(2)</b> {-0.00052} Aromatic carbon
<LI><b>V_solv(2)</b> {33.5103 } Aromatic carbon
<LI><b>A_solv(3)</b> {-0.00162} Nitrogen
<LI><b>V_solv(3)</b> {22.4493 } Nitrogen
<LI><b>A_solv(4)</b> {-0.00251} Oxygen
<LI><b>V_solv(4)</b> {17.1573 } Oxygen
<LI><b>A_solv(5)</b> {-0.00214} Phosphorus
<LI><b>V_solv(5)</b> {33.5103} Phosphorus
<LI><b>A_solv(6)</b> {-0.00214} Sulphur
<LI><b>V_solv(6)</b> {33.5103} Sulphur
<LI><b>A_solv(7)</b> {-0.00051} Hydrogen
<LI><b>V_solv(7)</b> { 0.0   } Hydrogen
<LI><b>polarcarblim</b> {0.2} - the carbon polarity threshold
</UL>
</UL>
<p>
<A NAME="ENHB"></A>
III.05.20. <em u>Hydrogen bond potential</em>
<p>
<B>ENHB</B> Key1 Key2 Data
<p>
<p>
Prerequisite: <B><A HREF="#SLTA">SLTA</A></B><BR>
Potential successor:
<B><A HREF="#SPST">SPST</A></B><BR>
Prerequisite of:
<B><A HREF="#HBMO">HBMO</A></B><BR>
Related keys: <B><A HREF="#SPST">SPST</A></B>,
<B><A HREF="#HBMO">HBMO</A></B>,
<B><A HREF="#SVIN">SSVIN</A></B>.
<p>
<u>Key1</u> (solute-solute HB potential type):
<UL>
<LI><B>NONE</B> - Hydrogen-bond terms will be not be used for the
solute-solute interactions.
<LI><B>SMPL</B> - Hydrogen-bond putential is  of the form<BR>
(c12/rHB<sup>12</sup>-c10/rHB<sup>10</sup>)*cos(A...H-X)
<LI><B>2ANG</B> - Hydrogen-bond putential is  of the form<BR>
(c12/rHB<sup>12</sup>-c10/rHB<sup>10</sup>) *
cos<sup><b>iexptheta</b></sup>(A...H-X) *
cos<sup><b>iexpomega</b></sup>(LP-A...H)<BR>
<u>Key2</u> (solute-map HB potential type):
<UL>
Same options as for Key1<BR>
<u>Data</u>: <b>rmaxhb</b>, <b>rminhb</b>, <b>rminhbnc</b>,
<b>qmaxhbdon</b>, <b>qminhbacc</b>, <b>iexptheta</b>, <b>iexpomega</b>,
<b>colpangle</b>
<UL>
<LI><b>rmaxhb</b> {3.0} - upper threshold of the HB potential
<LI><b>rminhb</b> {1.65} - lower threshold of the HB potential
<LI><b>qmaxhbdon</b> {0.0} - when not zero, hydrogens with partial charge &gt;
<b>qmaxhbdon</b> will not act as hydrogen-bond donors
<LI><b>qminhbacc</b> {0.0} - when not zero, oxygens with partial charge &lt;
<b>qminhbacc</b> (i.e.,more negative) will not act as hydrogen-bond acceptors
<LI><b>iexptheta</b> {2} - For Key*=<B>2ANG</B>,
the exponent on the theta factor
<LI><b>iexpomega</b> {4} - for Key*=<B>2ANG</B>,
the exponent on the omega factor
<b>colpangle</b> {120} - The angle between the two lone-pair directions,
used for the calculation of the omega angle
</UL>
</UL>
</UL>
When this key is used
a hydrogen-bond term will be used between polar hydrogens and oxygens
as specified by Key1 and Key2, subject to the filters specified by
<b>rmaxhb</b>, <b>rminhb</b>, <b>qmaxhbdon</b>, <b>qminhbacc</b>.
This term may be used between non-bonded solute atoms and
- if used - the grid-based energy term
(<B><A HREF="#SPST">SPST</A></B> <B>EMAP</B>).
<P>
When the grid energy is used, and there are hydrogen bonds between the
explicitly represented atoms and the atoms contributing to the grid
then the atoms (that are otherwise fixed) involved in hydrogen bonds
have to be specified.
The hydrogen-bond donors and
acceptors will be extracted from the non-felxible part of the solute
as specified by  the <A HREF="#FLXR">FLXR</A></B> key.
</UL>
<p>
<A NAME="HBMO"></A>
III.05.21. <em u>Non-default hydrogen bond</em>
<p>
<B>HBMO</B> Key1 Data
<p>
Prerequisites:
<B><A HREF="#ENHB">ENHB</A></B>, <B><A HREF="#SLTA">SLTA</A></B><BR>
Potential successor:
<B><A HREF="#SPST">SPST</A></B><BR>
Related key: <B><A HREF="#SPST">SPST</A></B>,
<B><A HREF="#ENHB">ENHB</A></B>.<BR>
<p>
<u>Key1</u> (type of action):
<UL>
<LI><B>ADDT</B> - Add new hydrogen bonding type(s). Default types are:
<UL>
<LI>"D&gt;NH": H on a nitrogen
<LI>"D-OH": H on an oxygen
<LI>"DNH+": H on a protonated nitrogen
<LI>"AC=O": Carbonyl oxygen
<LI>"AH-O": Non-carbonyl oxygen
<LI>"AC-O": Ester or ether oxygen
</UL>
New types are defined by their potential label:<BR>
<u>Data</u> for Key1=<B>ADDT</B>:
<b>pflib</b>, <b>ntype</b>, <b>pflab(1)</b>, ..., <b>pflab(ntype)</b>
<UL>
<LI><B>pflib</B> - Potential library of the labels
(see key <B><A HREF="#SUPT">SUPT</A></B>)
<LI><B>ntype</B> - Number of new types
<LI><B>pflab(i)</B> - Potential type label of the ith new type.
</UL>
<LI><B>ADDC</B> - Add new hydrogen bonding coefficients:<BR>
<u>Data</u> for Key1=<B>ADDC</B>:
<b>hbtyp1</b>, <b>hbtyp2</b>, <b>c12</b>, <b>c10</b>, <b>rminhbheavy</b>
<UL>
<LI><b>hbtyp1</b>, <b>hbtyp2</b> - hydrogen bond type labels for the
donor and acceptor if this type of bydrogen bond (either the
default labels listed above or the potential labels read with the key
<B><A HREF="#HBMO">HBMO</A></B> <B>ADDT</B>)
<LI><b>c12</b>, <b>c10</b> the c12 and c10 coefficients of
this type of bydrogen bond.
If both are given as zero then this type of bydrogen bond will not be used.
<LI><b>rminhbheavy</b> {0.0} - For Key*=<B>2ANG</B>,
hydrrogen bond between types <b>hbtyp1</b>, <b>hbtyp2</b>,
oxygen acceptor with one bond only,
the lower threshold of distance between the donor hydrogen's
heavy-atom and the atom the oxygen is bonded.
</UL>
<LI><B>EXCL</B> - Exclude atom types from hydrogen bonding<BR>
<u>Data</u> for Key1=<B>EXCL</B>:
<b>pflib</b>, <b>nhbexclude</b>, <b>pflab(1)</b>,
...,<b>pflab(nhbexclude)</b>
<UL>
<LI><B>pflib</B> - Potential library of the labels
(see key <B><A HREF="#SUPT">SUPT</A></B>)
<LI><b>nhbexclude</b> - number of labels to read
<LI><b>pflab</b> - potential label to exclude (<b>nhbexclude</b> labels)
</UL>
</UL>
<p>
<A NAME="FLXR"></A>
III.05.22. <em u>Residues to sample</em>
<p>
Potential successor:
<B><A HREF="#SLTA">SLTA</A></B><BR>
<p>
<B>FLXR</B> Key1 Data
<p>
Related key: <B><A HREF="#TORD">TORD</A></B>
<p>
Prerequisite of: <B><A HREF="#SPST">SPST</A></B>
<p>
<u>Key1</u> (input syntax):
<UL>
<LI><B>INDX</B> - Ranges of flexible residues are listed as pairs of integers
<LI><B>PDB</B> - PDB-formatted ATOM records, one for each flexible residue<BR>
<p>
<u>Data for Key1=<B>INDX</B></u>: <b>n_residue_range</b>,
<b>nresidue_range</b> times: <b>irf</b>, <b>i2f</b><BR>
<u>Data for Key1=<B>PDB</B></u>: <b>n_residue</b>,
<b>nresidue</b> times: PDB ATOM records specifying the residue numbers of the
flexible residues
</UL>
<p>
For calculations using the energy map
this key defines the residues to be sampled. All other residues will
contribute to the energy maps.
<p>
<A NAME="CNST"></A>
III.05.23. <em u>Constraint potential</em>
<p>
Potential successor:
<B><A HREF="#SLTA">SLTA</A></B><BR>
<p>
<B>CNST</B> Key1 Key2 Data
<p>
<u>Key1</u> (new or modify)
<UL>
<LI><B>NONE</B> - No constraints
<LI><B>INIT</B> - New constraint list
<LI><B>UPDT</B> - Modifying constraint list<BR>
<u>Key2</u> (constraint bond creation)
<UL>
<LI><B>NOCT</B> - No CONECT records will be written on PDB files
<LI><B>WCNT</B> - CONECT records will be written on PDB files representing the
constraints as bonds<BR>
<u>Data for Key1=<B>INIT</B></u>: <b>nconstr_pair</b>,
nconstr_pair times: <b>iatom1</b>, <b>iatom2</b>, <b>d_target</b>,
<b>force_target</b>, <b>tol_target</b><BR>
<u>Data for Key1=<B>UPDT</B></u>: <b>nconstr_pair</b>,
nconstr_pair times: <b>d_target</b>, <b>force_target</b>, <b>tol_target</b><BR>
<UL>
<LI><b>nconstr_pair</b>: The number of solute atom pairs whose distance is
to be constrained
<LI><b>iatom1</b>, <b>iatom2</b>: The two atoms whose distance is
constrained with this constraint.
<LI><b>d_target</b>: The target distance for this constraint
<LI><b>force_target</b>: The constraint force constant. The constraint energy
will be <BR>
<b>force_target</b>*(<b>d_target</b>-actual distance)<sup>2</sup>-
<b>force_target</b>*<b>tol_target</b><sup>2</sup>
<LI><b>tol_target</b>: Distance tolerance
- the constraint energy is set to zero if the actual distance is within
<b>tol_target</b> of <b>d_target</b>
</UL>
</UL>
</UL>
<A NAME="VVNE"></A>
III.05.24. <em u>No solvent-solvent electrostatic interaction</em>
<p>
Potential successor:
<B><A HREF="#SLTA">SLTA</A></B><BR>
<p>
<B>VVNE</B>
<p>
When this key is present solvent-solvent interactions will mot include the
electrostatic terms.
<p>
<A NAME="IGTT"></A>
III.05.25. <em u>Ignore solute-solute energy</em>
<p>
<B>IGTT</B>
<p>
When this key is present all solute-solute interactions are set to zero.
<p>
<A NAME="Input_SAMP"></A>
III.6. <B>Sampling-related keywords</B>
<OL>
<LI><TT><B><A HREF="#MOVE">MOVE</A></B>: </TT>Move selection
strategy<BR>
<LI><TT><B><A HREF="#SAMP">SAMP</A></B>: </TT>Sampling
technique<BR>
<LI><TT><B><A HREF="#NFBU">NFBU</A></B>: </TT>Solute force bias<BR>
<LI><TT><B><A HREF="#STEP">STEP</A></B>: </TT>Stepsizes<BR>
<LI><TT><B><A HREF="#STPS">STPS</A></B>: </TT>Stepsize scaling<BR>
<LI><TT><B><A HREF="#PARD">PARD</A></B>: </TT>Solute molecule
displacement<BR>
<LI><TT><B><A HREF="#SWAP">SWAP</A></B>: </TT>Solute molecule
swap<BR>
<LI><TT><B><A HREF="#MV2S">MV2S</A></B>: </TT>Correlated 2-solute move<BR>
<LI><TT><B><A HREF="#PART">PART</A></B>: </TT>Solute torsion<BR>
<LI><TT><B><A HREF="#MVRT">MVRT</A></B>: </TT>Alternating
translation and rotation<BR>
<LI><TT><B><A HREF="#PRFI">PRFI</A></B>: </TT>Insertion/deletion
pref. sampl.<BR>
<LI><TT><B><A HREF="#SEED">SEED</A></B>: </TT>Random number seed
input<BR>
<LI><TT><B><A HREF="#NONB">NONB</A></B>: </TT>Non-Boltzmann
solute samplings<BR>
<LI><TT><B><A HREF="#STSC">STSC</A></B>: </TT>Temporary stepsize scaling
<LI><TT><B><A HREF="#LOOP">LOOP</A></B>: </TT>Loop torsion moves<BR>
<LI><TT><B><A HREF="#IGJA">IGJA</A></B>: </TT>Ignore Jacobian<BR>
<LI><TT><B><A HREF="#SKWT">SKWT</A></B>: </TT>Skewed torsion sampling<BR>
<LI><TT><B><A HREF="#IGSV">IGSV</A></B>: </TT>Non-interactive solvent<BR>
<LI><TT><B><A HREF="#RNDG">RNDG</A></B>: </TT>Random number generator type<BR>
</OL>
<A NAME="MOVE"></A>
III.6.01. <em u>Molecule selection strategy specification</em>
<p>
<B>MOVE</B> Key1 [Data] [Fdata]
<p>
Potential successor:
<B><A HREF="#RCKP">RCKP</A></B><BR>
Prerequisite of:
<B><A HREF="#PRFI">PRFI</A></B><BR>
<p>
<u>Key1</u> (molecule selection strategy):
<UL>
<LI><B>RAND</B> - Molecules are selected randomly (uniformly) for
move<BR>
<LI><B>CYCL</B> - Cyclic selection - increasing order<BR>
<LI><B>CYCI</B> - Cyclic selection - decreasing order<BR>
<LI><B>SHCY</B> - <A HREF="#Mezei_81">Shuffled cyclic procedure</A>
of Mezei is used.<BR>
<LI><B>PRSL</B> - Preferential sampling with piecewise linear
weight function. Preferential sampling (PS) systematically selects
solvents near the solute more frequently to be moved to improve the
satatistics of the solute-related properties.<BR>
<LI><B>PRSP</B> - Preferential sampling with polynomial weigth
function<BR>
<LI><B>PRSE</B> - Preferential sampling with exponential weigth
function<BR>
<u>Key2</u> (distance choice for PS):
<UL>
<LI><B>COM </B> - PS is based on (solvent's COM's) distance from
solute COM<BR>
<LI><B>NRST</B> - PS is based on (solvent's first atom's)
distance from the nearest solute heavy atom. Not valid for EPEN
solute.<BR>
<LI><B>CXYZ</B> - PS is based on distance from inputted center<BR>
center<BR>
For Key2=<B>CXYZ</B> <U>Data</U>:
<UL>
<LI><B>CXYZ</B> - The coordinates of the center from where the
PS distance is calculated.
</UL>
<U>Fdata</U> for Key1=<B>PRSL</B>:
<UL>
<LI>RECTYPE 1: <B>npf</B> (I5)
<LI>RECTYPE 2: (<B>rpf</B>(i),<B>apf</B>(i),i=1,-<B>npf</B>)
(10F7.0)
<LI><B>rpf, apf</B> - The weight function value w(r) at each
<B>rpf</B>(i) is <B>apf</B>(i). For r &lt; <B>rpf</B>(1)
w(r)=<B>apf</B>(1) while, for r &gt; <B>rpf</B>(<B>npf</B>)
w(r)=<B>apf</B>(<B>npf</B>). For intermediate r's (<B>rpf</B>(1)
&lt; r &lt; <B>rpf</B>(<B>npf</B>)) w(r) obtained by linear
interpolation betwen the two nearest <B>rpf</B>'s.
</UL>
<U>Fdata</U> for Key1=<B>PRSP</B>:
<UL>
<LI>RECTYPE 1: <B>npf</B> (I5)
<LI><B>npf</B> = 1+ the smallest exponent having non-zero
coefficient.
<LI>RECTYPE 3: (<B>apf</B>(i),i=1,<B>npf</B>) (10F7.0)
<LI><B>apf</B> - The coefficients in the weighing polynomial.
Units are in . The weighing function is constant for r &lt;
<B>apf</B>(1) and
 If r &gt; <B>apf</B>(1) then w(r)=<B>apf</B>(2)+<B>apf</B>(3)/r
+<B>apf</B>(4)/r<sup>2</sup>+ ... . w(<B>apf</B>(1))
gives the value of the constant.
</UL>
<U>Fdata</U> for Key1=<B>PRSE</B>:
<UL>
<LI>RECTYPE 4: <B>stiff</B>, <B>stvar</B>(2F7.0)
<LI><B>stiff,stvar</B> - The weighting function is
exp(-(<B>stiff</B>*/<B>stvar</B><sup>2</sup>)*r<sup>2</sup>) where
r is the solute-solvent distance as specified by Key2.
</UL>
</UL>
</UL>
<A NAME="SAMP"></A>
III.6.02. <em u>Sampling technique selection</em>
<p>
<B>SAMP</B> Key1 [Data Fdata]
<p>
Variable to set to 'T' in the <A HREF="#PREP">preprocessor</A>
for Key1=<B>FBPR</B> or <B>FBSC</B>: <B>FR</B>
<p>
<u>Key1</u> (possible displacement biasing strategy):
<UL>
<LI><B>FBPR</B> - Force biased.<BR>
<U>Data</U>: <B>fblam</B> {0.5} .
<UL>
<LI><B>fblam</B> - the lambda factor for force biasing (the
strength of the bias)
</UL>
<LI><B>METC</B> - Standard Metropolis, moves in a cube.<BR>
<LI><B>METS</B> - Standard Metropolis, moves in a sphere.<BR>
<LI><B>METX</B> - Standard Metropolis, fixed length moves (half of
<b>cedslt</b> or <b>cedslv</b> as read by the key
<B><A HREF="#STEP">STEP</A></B><BR>
<LI><B>METR</B> - Standard Metropolis, length of the moves
should fall within a range
<LI><B>FBSC</B> - Force biased, scaled depending on the slt-slv
distance.<BR>
For Key1=<B>METR</B>: <U>Data</U>: <B>rngminfac</B>
<UL>
<LI><B>rngminfac</B> {0.8} - the step magnitude will be between
<B>rngminfac</B>*<b>ceds**</B> and <b>ceds**</B>
</UL>
For Key1=<B>FB**</B>:
<U>Data</U>: <B>fblam</B>
<UL>
<LI><B>fblam</B> {0.5} - the lambda factor for force biasing (the
strength of the bias)
</UL>
For Key1=<B>FBPR</B>:
<U>Fdata:</U>
<UL>
<LI>RECTYPE 27: <B>nfblam</B> (I5)<BR>
<LI><B>nfblam</B> - Number of data points in the table specifying
the force-bias scaling.<BR>
<LI>RECTYPE 28:
(<B>rfblam</B>(i),<B>afblam</B>(i),i=1,<B>nfblam</B>)
(10F7.0)
<LI><B>rfblam</B>(i), <B>afblam</B>(i) - The force-bias lambda
parameter will be multiplied by a scale factor that is interpolated
between the values <B>afblam</B>(i) at solute-solvent distance
<B>rfblam</B>(i).
</UL>
</UL>
<A NAME="NFBU"></A>
III.6.03. <em u>Solute sampling technique selection</em>
<p>
<B>NFBU</B>
<p>
If present, the Metropolis method will be used for
for moves translating/rotating the whole solute
(se key <B><A HREF="#STEP">STEP</A></B>),
even when the key <B><A HREF="#SAMP">SAMP</A></B> specifies
force-biased sampling.
In this case leaving the C@TS lines
comments saves some computer time.
<p>
<A NAME="STEP"></A>
III.6.04. <em u>Solute and solvent stepsizes</em>
<p>
<B>STEP</B> Data
<p>
Potential successor:
<B><A HREF="#SLTA">SLTA</A></B><BR>
Prerequisite of:
<B><A HREF="#RUNS">RUNS</A></B><BR>
<p>
<U>Data</U>: <B>cedslt</B>, <B>rtxslt</B>, <B>cedslv</B>,
<B>rtxslv</B>, <B>nsltfreq</B>
<UL>
<LI><B>cedslt</B> - Shift interval in  for the COM of the solute:
The displacement of the solute COM will be limited into the
interval [-<B>cedslv</B>/2,<B>cedslv</B>/2].<BR>
<LI><B>rtxslt</B> - Rotation interval in degrees in which the
solute is to be rotated.<BR>
<LI><B>cedslv</B> - Shift interval in  for the COM of the
solvent.<BR>
<LI><B>rtxslv</B> - Rotation interval in degrees by which the
solvent is to be rotated.<BR>
<LI><B>nsltfreq</B> - The frequency of all solute moves. Solute
move may be any of the following type: displacement-rotation
(<B>cedslt</B> &gt; 0.0 and <B>nmolfx</B>=0 ); coupling parameter
change (<B><A HREF="#FREE5">FREE&nbsp;PMF1</A></B>); partial solute move
(<B><A HREF="#PARD">PARD</A></B>);
partial solute torsion (<B><A HREF="#PART">PART</A></B>) or
solute group swap (<B><A HREF="#SWAP">SWAP</A></B>).
</UL>
The above variables are used to perturb a chosen molecule,
solute or any of the solvent during the Metropolis walk. As a
simple rule of thumb, these variables are chosen by trial and
error to obtain 30-40% acceptance rate.
For water the recommended values are:
0.30  for the COM of the solvent and 30 degrees for the rotation
angle for standard Metropolis and 0.55  and 40 degrees for Force
Bias. The stepsize of the solute should be adjusted to
provide similar acceptance rate as the solvent. In general, the
larger the solute the smaller stepsizes are to be used.
<p>
<A NAME="STPS"></A>
III.6.05. <em u>Stepsize scaling</em>
<p>
<B>STPS</B> Data
<P>
<U>Data</U>: <B>notmovcyc</B>
<UL>
<LI><B>notmovcyc</B> {1} - Solvent stepsizes are scaled down by 10%
when a solvent is found that was not moved during the last
10*<B>notmovcyc</B> cycles. One cycle is <B>nmolec</B> MC steps.
</UL>
<A NAME="PARD"></A>
III.6.06. <em u>Partial displacement of the solute</em>
<p>
<B>PARD</B> Key1, Key2, Data
<P>
Prerequisites: <B><A HREF="#SLTA">SLTA</A></B>,
<B><A HREF="#SUUC">SUUC</A></B><BR>
Can not be followed by: <B><A HREF="#FREE">FREE</A></B><BR>
Potential successors:
<B><A HREF="#TORD">TORD</A></B>, <B><A HREF="#SPST">SPST</A></B><BR>
Prerequisite of:
<B><A HREF="#SWAP">SWAP</A></B>, <B><A HREF="#MV2S">MV2S</A></B><BR>
Related key:
<B><A HREF="#SWAP">SWAP</A></B>.
<P>
<u>Key1</u> (rotation stepsize strategy):
<UL>
<LI><B>UNIF</B> - Rotation stepsize is kept constant<BR>
<LI><B>EXBI</B> - Rotation stepsize is scaled inversely with the
maximum extension of the molecule
in the plane perpendicular to the rotation axis.<BR>
<LI><B>EXBQ</B> - Rotation stepsize is scaled inversely with the
square root of the maximum extension<BR>
<LI><B>EXBA</B> - Rotation stepsize is scaled inversely with the
square root of the root mean square extension<BR>
<u>Key2</u> (molecule selection strategy):
<UL>
<LI><B>RAND</B> - Molecules are selected randomly (uniformly) for
move<BR>
<LI><B>CYCL</B> - Cyclic selection - increasing order<BR>
<LI><B>SHCY</B> - <A HREF="#Mezei_81">Shuffled cyclic procedure</A>
of Mezei is used.<BR>
<LI><B>CYCI</B> - Cyclic selection - decreasing order<BR>
<U>Data</U>: <B>wsltpdis</B>, <B>cedpsltx</B>, <B>rtxpsltx</B>,
<B>wrotx</B>, <B>cedpslty</B>, <B>rtxpslty</B>, <B>wroty</B>,
<B>cedpsltz</B>, <B>rtxpsltz</B>, <B>wrotz</B>, <B>nsltpardis</B>
<UL>
<LI><B>wsltpdis</B> - Relative frequency of partial solute
displacements<BR>
<LI><B>cedpslt*</B> - Displacement size parameters
corresponding to the x,y, and z directions, resp.<BR>
<LI><B>rtxpslt*</B> - For Key1=<B>UNIF</B>: Rotation size
parameters
corresponding to the x,y, and z directions, resp.<BR>
<LI><B>rtxpslt*</B> - For Key1=<B>EXBI</B>: Rotation size
parameter will be calculated as
rotmax=rtxpslt* / extmax where extmax is the maximum extension of
the molecule to be rotated in the plane perpendicular to the
rotation axis.<BR>
<LI><B>rtxpslt*</B> - For Key1=<B>EXBQ</B>: Rotation size
parameter will
be calculated as rotmax=rtxpslt* / (2.0 * sqrt(extmax))<BR>
<LI><B>wrot*</B> - Weight for selecting an axis for rotation.<BR>
<LI><B>nsltpardis</B> - When &gt;0, the molecule(s) containing the
first <B>nsltpardis</B> atoms after the free energy molecule (if any)
will be subjected to independent moves (translation and
rotation).
When negative, the molecules containing the last -<B>nsltpardis</B>
atoms will be moved (This allows their treatment in subsequent
analysis as solvents).
Note, that <B>nsltpardis</B> refers to the explicite solute atom
list,
and <u>not</u> the full list of solute atoms after cloning (if
any).
</UL>
</UL>
</UL>
<A NAME="SWAP"></A>
III.6.07. <em u>Solute molecule swap</em>
<p>
<B>SWAP</B> Data
<P>
Prerequisite: <B><A HREF="#PARD">PARD</A></B>
<P>
<U>Data</U>: <B>wswap</B>
<p>
<UL>
<LI>The relative frequency of attempting to swap
two randomly selected solute molecules
</UL>
<A NAME="MV2S"></A>
III.6.08. <em u>Correlated 2-solute molecule move</em>
<p>
<B>MV2S</B> Data
<P>
Prerequisite: <B><A HREF="#PARD">PARD</A></B>
<P>
<U>Data</U>: <B>wmv2s</B>, <b>r2scut</b>, <b>rtxcslt</b>, <b>iaxis2s</b>
<p>
<u>Key1</u> (stepsize selection strategy): See Key1 of
key <B><A HREF="#PART">PART</A></B>
<UL>
<LI><B>wmv2s</B> - The relative frequency of attempting to make
a correlated move of two solute molecules
<LI><B>r2scut</B> - The distance threshold for selection of the
solute molecule pairs
<LI><b>rtxcslt</b> - The solute rotation stepsize (see
key <B><A HREF="#PARD">PARD</A></B>)
<LI><b>iaxis2s</b> - The axis of correlated rotation
</UL>
<A NAME="PART"></A>
III.6.09. <em u>Partial torsion of the solute</em>
<p>
<B>PART</B> Key1, Key2, Data
<P>
Prerequisites: <B><A HREF="#SLTA">SLTA</A></B>,
<B><A HREF="#SUUC">SUUC</A></B><BR>
Can not be followed by: <B><A HREF="#SVVC">SVVC</A></B><BR>
Prerequisite of:
<B><A HREF="#TAUC">TAUC</A></B><BR>
Related keys:
<B><A HREF="#LOOP">LOOP</A></B>,
<B><A HREF="#TORD">TORD</A></B>.
<P>
<u>Key1</u> (stepsize selection strategy):
<UL>
<LI><B>UNIF</B> - Rotation stepsize is kept constant
<LI><B>EXBI</B> - Rotation stepsize is scaled inversely with the
maximum extension of the part of the molecule
that is affected by this torsion,
calculated from the line of the torsion bond
<LI><B>EXBQ</B> - Rotation stepsize is scaled inversely with the
square root of the maximum extension
<LI><B>EXBA</B> - Rotation stepsize is scaled inversely with the
square root of the root mean square extension<BR>
<u>Key2</u> (strategy for selecting the solute molecule for
torsion):
<UL>
<LI><B>RAND</B> - Molecules
are selected randomly (uniformly) for move
<LI><B>CYCL</B> - Cyclic selection - increasing order.
<LI><B>SHCY</B> - <A HREF="#Mezei_81">Shuffled cyclic procedure</A>
of Mezei is used for selecting the
molecule for torsion change.<BR>
<LI><B>CYCI</B> - Cyclic selection - decreasing order.<BR>
<u>Key3</u> (strategy for selecting the torsion on a solute
molecule):
<UL>
<LI><B>RAND</B> - Torison groups are selected randomly
(uniformly) for change
<LI><B>CYCL</B> - Cyclic selection - increasing order.
<LI><B>CYCI</B> - Cyclic selection - decreasing order.<BR>
<U>Data</U>: <B>wsltptor</B>
<UL>
<LI><B>wsltptor</B> - Relative frequency of partial solute
torsion
</UL>
</UL>
</UL>
</UL>
<A NAME="MVRT"></A>
III.6.10. <em u>Alternating translation and rotation</em>
<p>
<B>MVRT</B> Key1
<P>
Potential successor:
<B><A HREF="#STUN">STUN</A></B><BR>
Prerequisite of:
<B><A HREF="#STUN">STUN</A></B><BR>
<P>
<u>Key1</u> (solute and/or solvent moves)
<UL>
<LI><B>BOTH</B> - Both the solute and the sovent will be
translated and rotated simulatneously
<LI><B>SVAL</B> - Translations and rotations will alternate for
the solvent
<LI><B>STAL</B> - Translations and rotations will alternate for
the solute
<LI><B>TVAL</B> - Translations and rotations will alternate for
both the solute and the solvent<BR>
<U>Data</U>: <B>wsltdisp</B>, <B>wsltrot</B>, <B>wslvdisp</B>, <B>wslvrot</B>
<UL>
<LI><B>wsltdisp</B>, <B>wsltrot</B> - Relative frequences of
solute translatons and rotations
<LI><B>wslvdisp</B>, <B>wslvrot</B> - Relative frequences of
solvent translatons and rotations
</UL>
</UL>
<A NAME="PRFI"></A>
III.6.11. <em u>Insertion/deletion preferential sampling</em>
<p>
<B>PRFI</B> Key1 Key2 [Data] [Fdata]
<p>
Prerequisites: <B><A HREF="#GCEN">GCEN</A></B>,
<B><A HREF="#MOVE">MOVE</A></B>
<p>
The syntax is the same as for the preferential sampling for the
solvent moves. Thus, except for not allowing <B>RAND</B>,
<B>CYCL</B>, <B>SHCY</B>, <B>CYCI</B> for
Key1, the input is the same as for the key
<B><A HREF="#MOVE">MOVE</A></B>.
<p>
<A NAME="SEED"></A>
III.6.12. <em u>Random number seed initialization</em>
<p>
<B>SEED</B> Data
<p>
<U>Data</U>: <b>ix1</b>, <b>ix2</b>, <b>ix3</b>, <b>ix4</b>, <b>ixscr</b>
<UL>
<LI><b>ix1</b>, <b>ix2</b>, <b>ix3</b>, <b>ix4</b> - the new random
number seed will be
<b>ix1</b>*2^48+<b>ix2</b>*2^32+<b>ix3</b>*2^16+<b>ix4</b>
<LI><b>ixscr</b> - the new scrambler seed will be <b>ixscr</b>
</UL>
The random number generator will be reinitialized with these seeds.
<p>
<A NAME="NONB"></A>
III.6.13. <em u>Non-Boltzmann solute samplings</em>
<p>
<B>NONB</B> Key1 Data
<p>
<u>Key1</u> (sampling type):
<UL>
<LI><B>SCTR</B> - sampling will use reduced solute torsion terms
<LI><B>SCRP</B> - sampling will use reduced solute-solvent repulsion
term
<LI><B>SCNB</B> - sampling will use reduced intra-solute non-bonded terms
<LI><B>SCTN</B> - sampling will use reduced torsion and
intra-solute non-bonded terms
<LI><B>TMOD</B> - solute energy will be scaled with a different
temperature
<LI><B>TSAL</B> - sampling follows Tsallis statistics
(<A HREF="#TSALLIS">REF.</A>) for the solute energy
<BR>
<u>Data</u>: <b>param(s)</b>, <b>ew0</b>, <b>signincfac</b>
<UL>
<LI><b>param(s)</b> - the extent of non-Boltzmann sampling:
<UL>
<LI>For Key1=<B>SCTR</B>,
the scale factor (&lt; 1) to apply on the
torsion energy.
<LI>For Key1=<B>SCRP</B>, the scale factor (&lt; 1) to apply on the
solute-solvent repulsion.
<LI>For Key1=<B>SCNB</B>, the scale factor (&lt; 1) to apply on the
intra solute non-bonded term.
<LI>For Key1=<B>SCTN</B>, the scale factor (&lt; 1) to apply on the
torsion energy, followed by the scale factor (&lt; 1) to apply on the
intra solute non-bonded term.
<LI>For Key1=<B>TMOD</B>, the temperature for the solute energy
(&gt; T (see key <B><A HREF="#TEMP">TEMP</A></B>))
<LI>For Key1=<B>TSAL</B>, the q parameter
</UL>
<LI><b>ew0</b> {value at start} - the value to
subtract from energy factor of the weight factor's exponent
<LI><b>signincfac</b> {1.1} - for Key1=<B>TSAL</B> only:
Add to the exponential argument
-<b>signincfac</b> times its initial value if
it turns out to be negative (to avoid taking
the log of a negative number)
</UL>
</UL>
<A NAME="STSC"></A>
III.6.14. <em u>Temporary stepsize scaling</em>
<p>
<B>STSC</B> Key1 Data
<p>
<u>Key1</u> (move type to scale):
<UL>
<LI><B>SVTR</B> - Scale solvent translation stepsize
<LI><B>SVRO</B> - Scale solvent rotation stepsize
<LI><B>STTR</B> - Scale whole solute translation stepsize
<LI><B>STRO</B> - Scale whole solute rotation stepsize
<LI><B>SMTR</B> - Scale solute molecule translation stepsizes
<LI><B>SMRO</B> - Scale solute molecule rotation stepsizes
<LI><B>NTOR</B> - Scale normal torsion stepsizes
<LI><B>LTOR</B> - Scale loop torsion stepsizes<BR>
<u>Data</u>: <b>scalefac</b>
<UL>
<LI><b>scalefac</b> {1.0} - the factor to use for this type move
</UL>
</UL>
The scaled stepsizes will only be used for this run - the next
calculation (initiated by a new
<B><A HREF="#RUNS">RUNS</A></B> key) will use the originally inputted
values.
<P>
<A NAME="LOOP"></A>
III.06.15. <em u>Loop torsion moves</em>
<p>
<B>LOOP</B> Key1, Key2, [ Data]
<P>
Potential successor:
<B><A HREF="#TORD">TORD</A></B><BR>
Prerequisite of:
<B><A HREF="#LPST">LPST</A></B><BR>
<P>
Variable to set to 'T' in the <A HREF="#PREP">preprocessor</A>: <B>LO</B>
<p>
Related key:
<B><A HREF="#IGJA">IGJA</A></B>
<p>
When this key is present the program will analyze the torsion angle
list and determine which torsions can be the first torsion of a loop
move. Loop moves change six consecutive torsions in such a way that
the bond of the first and last torsion remain fixed,
thereby limiting the change in the molecular geometry
due to the change in the torsionangle(s).
<P>
The formulae giving the loop-closing torsion angles
were obtained based on a distance geometry approach.
The current version is limited to loops where the torsions are
on consecutive bonds (-------),
or on consecutive bonds broken by a rigid bond (-=-----, --=----,
---=---, ----=--, -----=-)
or on a peptide backbone with rigid peptide bonds
(-[=--]<sub>3</sub>, -[-=-]<sub>3</sub>).<BR>
<u>Key1</u> (solution selection algorithm):
<UL>
<LI><B>PROX</B> - Search only in the vicinity of the reference
conformation for the nearest solution.
<LI><B>SCAN</B> - Scan the whole solution space to obtain all solutions
and select the nearest. Currently it is implemented only for
consecutive torsions.
<LI><B>JACW</B> - Scan the whole solution space to obtain all solutions
and select one with Jacobian weight. Currently it is implemented only for
consecutive torsions.  <BR>
<u>Key2</u> (iteration direction(s))
<UL>
<LI><B>FORW</B> - Loop-closing iteration is in the 'forward' direction only
<LI><B>BCKW</B> - Loop-closing iteration is in the 'backward' direction only
<LI><B>TRNG</B> - Loop-closing iteration direction is chosen based on the
closeness of the triangle inequlities
<LI><B>BOTH</B> - Loop closing iterations are run in both direction and the
solution with better precision is chosen<BR>
<U>Data</U>: <B>wloop</B>, <B>d3grid</B>, <B>d3tol</B>, <B>dijtol</B>,
<B>d24tol</B>, <B>drevtol</B>, <B>dextol</B>, <B>dihpmtol</B>
<UL>
<LI><B>wloop</B> {1.0} - Probability of selecting a loop torsion move
whenever possible (instead of a simple rotation)
<LI><B>d3grid</B> {0.1} - r3 range will be searched
by <B>d3grid</B> increments (or smaller when a root appears near).
<LI><B>d3tol</B> {0.000001} - Convergence tolerance of the distance
between putative r3 positions in the loop-closing algorithm
<LI><B>dijtol</B> {0.01} - Tolerance of the bond length squares
for the no-torsion debug option
<LI><B>d24tol</B> {0.1} - Tolerance of (r2-r4)^2
in the loop-closing algorithm
<LI><B>drevtol</B> {0.2} - Distance square tolerance for the
check of the reverse proximity criterion - only used for
Key2=<B>PROX</B>
<LI><B>dextol</B> {0.2} - Tolerance of the filtering distance squares
<LI><B>dihpmtol</B> {5.0} - Frozen bond torsion angle signs will be
conserved for angles that deviate from planarity by
less than <B>dihpmtol</B> degrees.
</UL>
</UL>
</UL>
<P>
<A NAME="IGJA"></A>
III.06.16. <em u>Ignore Jacobian</em>
<p>
<B>IGJA</B>
<p>
Related key:
<B><A HREF="#LOOP">LOOP</A></B>,
<P>
When this key is present, the acceptance probability of a local move
(see key <B><A HREF="#LOOP">LOOP</A></B>) will ignore the Jacobian.
This will allow skipping the calculation of the loop-closure problem
for the reverse move and thereby speeding up the calculation.
The speedup, however, comes at the expense of the loss of Boltzmann
sampling and thus this option is to be used for search or minimization runs.
<p>
<A NAME="SKWT"></A>
III.06.17. <em u>Skewed Torsion sampling</em>
<p>
<B>SKWT</B> Key1 Data
<p>
Related key: <B><A HREF="#TORD">TORD</A></B>,
<P>
<u>Key1</u> (type of skewing)
<UL>
<LI><B>DRIV</B> - for selected bonds,
torsion attempts will be made <b>wskew</b> times
more frequently in the target direction than away from it.
<LI><B>LOOP</B> - for loops involving selected bonds,
the loop solution closest to the target angles will be chosen
<b>wskew</b> times more likely than a randomly chosen solution.
<LI><B>DRLP</B> - a combination of both
options described by the keys <B>DRIV</B> and <B>LOOP</B>
<LI><B>PROX</B> - for loops involving selected bonds,
the loop move will be rejected with 1-1/<b>wskew</b> probability whenever
the Rloop torsions moved away from their target (in the least-square sense).
<LI><B>DRPR</B> - a combination of both options described
by the keys <B>DRIV</B> and <B>PROX</B>
<LI><B>DLPR</B> - a combination of the options described
by the keys <B>DRIV</B>,  <B>LOOP</B>, and <B>PROX</B>
.<BR>
Data: <B>wskew</B>, <b>wskewstepmin</b>, <b>resumefac</b>
<UL>
<LI><B>wskew</B> - the weight used for skewing the sampling for the
torsions selected for skewed sampling.
<LI><b>wskewstepmin</b> {1.0) - for stepsize tuning runs, the skewed
sampling is suspended whenever the stepsize falls below
<b>wskewstepmin</b> degrees.
<LI><b>resumefac</b> {1.2} - for stepsize tuning runs, the skewed
sampling is resumed whenever the stepsize is raised above
<b>wskewstepmin</b>*<b>resumefac</b> degrees.
</UL>
</UL>
When this key is present then
the formatted data after the key <B><A HREF="#TORD">TORD</A></B>
will specify a target angle for some of the torsions.
<p>
<A NAME="IGSV"></A>
III.06.18. <em u>Non-intractive solvent</em>
<p>
<B>IGSV</B>
<p>
When this key is present then solvent-solvent interactions will be ignored.
<P>
<A NAME="RNDG"></A>
III.06.19. <em u>Random number generator information</em>
<p>
<B>RNDG</B> Key1 Data
<p>
<u>Key1</u> (generator type)
<UL>
<LI><B>MMC </B> - MMC RNG: 64-bit congruantial RNG; scrambled in 64 number segme
<LI><B>LCG </B> - Linear congruential generator
<LI><B>MRTW</B> - Mersenne twister generator
<LI><B>INPT</B> - A list of random numbers is read from a file; before reusing
they are scrambled
<LI><B>INNS</B> - A list of random numbers is read from a file; they are reused
unchanged when the list is exhausted
<LI><B>OUTP</B> - A list of random numbers generated by MMC RNG is written to a
the run will stop after writing the file
</UL>
<U>Data for Key1=<B>LCG </B></U>: <b>lcg_fac</b>, <b>lcg_add</b>,
<b>lcg_exp</b>, <b>seed</b>
<UL>
<LI><b>lcg_fac</b> {314159269} - factor C (see below)
<LI><b>lcg_add</b> {453806245} - additive constant A (see below);
default values from Forsythe.
<LI><b>seed</b> {1357} - First random number
<LI><b>lcg_exp</b> {31} - modulus exponent E (see below)
<UL>
ir(next)=mod(ir*C+A,2<sup>E</sup>)
</UL>
</UL>
<p>
<U>Data for Key1=<B>MRTW</B></U>: <b>seed</b>
<UL>
<LI><b>seed</b> {4357} - random number seed
</UL>
<p>
<U>Data for Key1=<B>INPT</B> or Key1=<B>OUTP</B></U>:
<B>rndgfile</B>, <b>nrand</b>
<UL>
<LI><B>rndgfile</B> - name of the file the random numbers
are read from or written to (Key1 = <B>INPT</B> or <B>OUTP</B>)
<LI><b>nrand</b> - number of random numbers to write (Key1 = <B>OUTP</B>)
</UL>
<p>
<A NAME="Input_FLOW"></A>
III.7. <B>Calculation flow keywords</B>
<OL>
<LI><TT><B><A HREF="#RUNS">RUNS</A></B>: </TT>Run MC
simulation<BR>
<LI><TT><B><A HREF="#SCAN">SCAN</A></B>: </TT>Scan trajectory for
proximity analysis<BR>
<LI><TT><B><A HREF="#PXAN">PXAN</A></B>: </TT>Proximity analysis
frequencies<BR>
<LI><TT><B><A HREF="#RCKP">RCKP</A></B>: </TT>Checkpoint file read
<BR>
<LI><TT><B><A HREF="#STOP">STOP</A></B>: </TT>Stop the run<BR>
</OL>
<P>
<A NAME="RUNS"></A>
III.07.01. <em u>Execution of a MC simulation</em>
<p>
<B>RUNS</B> Data
<p>
Prerequisites:
<B><A HREF="#SLTA">SLTA</A></B>, <B><A HREF="#CNFG">CNFG</A></B>,
<B><A HREF="#TEMP">TEMP</A></B>, <B><A HREF="#FILE">FILE</A></B>,
<B><A HREF="#STEP">STEP</A></B><BR>
<p>
<U>Data</U>: <B>nmcmax</B>, <B>nmcrep</B>, <B>nplt</B>,
<B>nmcrec</B>, <B>ncntin</B>, <B>nmcadp</B>
<UL>
<LI><B>nmcmax</B> - Number of MC steps to run.
<LI><B>nmcrep</B> {<B>nmcmax</B>/100} -
After every <B>nmcrep</B> steps certain useful
intermediate output, e.g., energy, energy-minimum, molecule moved,
fraction of accepted steps, solute binding energy is printed.
</UL>
After reading a checkpoint file (see key <B><A HREF="#RCKP">RCKP</A></B>)
the rest of the data is ignored.
<UL>
<LI><B>nplt</B> - After every <B>nplt</B> steps, various statistics
and distribution functions, such as the radial distribution
functions g(R), binding energy distributions QCBE, coordination
number distributions QCCN, pair-energy distributions QCPE, etc.
are printed and are optionally stacked on the <B>.dst</B> plot file
see <B><A HREF="#DSTC">DSTC</A> **** FILE</B>)
(for later analysis). <B>nplt</B> must
be an integer multiple of <B>nmcrep</B>. If it is not, the program
will automatically change it to the nearest multiple.
<LI><B>nmcrec</B> {2*nmolec} - Frequency of saving a full
configuration on
the <B>.hst</B>file (see key <B><A HREF="#TRAJ">TRAJ</A></B>)
<LI><B>ncntin</B> {greater of 10000 and <B>nmcmax</B>/100} -
Length of the interval for
control-function (block average) calculations. Some of the control
functions thus obtained will be plotted on the printer at the end
of the output. (Recommended value: 100000.)
<LI><B>nmcadp</B> {100000} - Frequency of adaptive umbrella
sampling weight updates. <B>nmcadp</B> must be an integer multiple
of <B><A HREF="#File_org">nrecd</a></B>
(see also key <B><A HREF="#CHKP">CHKP</A></B>.
If it is not, the program will automatically
change it to the nearest multiple.
</UL>
<A NAME="SCAN"></A>
III.07.02. <em u>Proximity analysis from history</em>
<p>
<B>SCAN</B> [Key1] Data
<p>
Prerequisite:
<B><A HREF="#SLTA">SLTA</A></B><BR>
<p>
<u>Key1</u> (source of data):
<UL>
<LI><B>TRAJ</B> - analyze a trajectory
<LI><B>TRNC</B> - analyze a trajectory that is the continuation of
the previous trajectory (but with different runnumber),
i.e., don't initalize the analysis accumulators.
This key assumes that a previous <B>SCAN TRAJ</B> was
already executed in the <i>same run</i>.
<B><A HREF="#TRAJ">TRAJ</A> ALLV</B>.
<LI><B>CONF</B> - analyze the current configuration
</UL>
For Key1=<B>TRAJ</B> or <B>TRNC</B>:<BR>
<U>Data</U>: <B>nmcpxmax</B>, <B>navgpx</B>, <B>nranpx</B>,
<B>nmcpxdsc</B>, <B>npxres</B>, <B>npxcntin</B>,
<B>lumppr</B>, <B>nsymdim</B>, <B>ifrsdim</B>
<UL>
<LI><B>nmcpxmax</B> - The program will scan the history file until
the <B>nmcpxmax</B>-th configuration and perform proximity
analysis.
If no
<B><A HREF="#CNFG">CNFG</A></B> command reading in a configuration
preceded <B><A HREF="#SCAN">SCAN</A></B>, the
initial coordinates will be gathered from the history file.
<LI><B>navgpx</B> - frequency of selecting a structure for
the proximity analysis contributions<BR>
<LI><B>nranpx</B> {0} - number of random points to generate (for
volume-element estimation) after the analysis of each
configuration<BR>
<LI><B>nmcpxdsc</B> {0} - Analysis to be started after the first
<B>nmcpxdsc</B>-th configuration<BR>
<LI><B>npxres</B> {npxmax} - Proximity analysis results printed
after each <B>npxres</B>-th configuration<BR>
<LI><B>npxcntin</B> {<B>ncntin</B>} - Block size
(number of configurations to average) for proximity
analysis error estimates<BR>
<LI><B>lumppr</B> {2} - Printed error estimates are based on
2<sup><B>lumppr</B></sup>*<B>npxctn</B> long blocks<BR>
<LI><B>nsymdim</B> {3} - For each random point symmetry-related
points will be generated in <B>nsymdim</B> dimension<BR>
<LI><B>ifrsdim</B> {1} - Start symmetry generation at the
<B>ifrsdim</B>-th dimension
</UL>
For Key1=<B>CONF</B>:<BR>
<U>Data</U>: <B>nranpx</B>, <B>lumppr</B>, <B>nsymdim</B>,
<B>ifrsdim</B> - see above.
<p>
Structures in history files in CHARMM
(<B><A HREF="#TRAJ">TRAJ</A> CHRM</B>) or AMBER
(<B><A HREF="#TRAJ">TRAJ</A> AMBR</B>) trajectory syntax
are labeled 1,2,... while structures in all other syntaxes
are labeled by the MC stepnumber they were saved at.
<P>
<A NAME="PXAN"></A>
III.07.03. <em u>Proximity analysis frequency parameters
(concurrent to MC run)</em>
<p>
<B>PXAN</B> Data
<P>
Can not be followed by: <B><A HREF="#PXGR">PXGR</A></B>,
<B><A HREF="#PXDP">PXDP</A></B>, <B><A HREF="#PXBE">PXBE</A></B>,
<B><A HREF="#PXWW">PXWW</A></B>, <B><A HREF="#TAND">TAND</A></B>,
<B><A HREF="#ATFR">ATFR</A></B>, <B><A HREF="#FLDG">FLDG</A></B>,
<B><A HREF="#RTIM">RTIM</A></B>, <B><A HREF="#DIFC">DIFC</A></B>,
<B><A HREF="#VORO">VORO</A></B>, <B><A HREF="#CHKP">CHKP</A></B><BR>
Related keys:
<B><A HREF="#PXLM">PXLM</A></B>,
<B><A HREF="#PCPA">PCPA</A></B>,
<B><A HREF="#PXYZ">PXYZ</A></B>.
<P>
<U>Data</U>: <B>navgpx</B>, <B>nranpx</B>, <B>nmcpxdsc</B>,
<B>npxres</B>,
<B>npxcntin</B>, <B>lumppr</B>, <B>nsymdim</B>,
<B>ifrsdim</B>
<p>
These variables are the same as defined for the keyword
<B><A HREF="#SCAN">SCAN</A></B>.
<P>
Using this keyword the proximity analysis will be performed
concurrently to the MC simulation (eliminating the need of saving
the history file).
<p>
<A NAME="RCKP"></A>
III.07.04. <em u>Continue from a checkpoint file</em>
<p>
<B>RCKP</B> [Key1 Key2 Data]
<p>
Prerequisite:
<B><A HREF="#FILE">FILE</A></B><BR>
Can not be followed by:
<B><A HREF="#TITL">TITL</A></B>, <B><A HREF="#PBCN">PBCN</A></B>,
<B><A HREF="#SUPT">SUPT</A></B>, <B><A HREF="#SVPT">SVPT</A></B>,
<B><A HREF="#TRAJ">TRAJ</A></B>, <B><A HREF="#INCT">INCT</A></B>,
<B><A HREF="#MIXR">MIXR</A></B>, <B><A HREF="#SUVC">SUVC</A></B>,
<B><A HREF="#SVVC">SVVC</A></B>, <B><A HREF="#SUUC">SUUC</A></B>,
<B><A HREF="#NSLV">NSLV</A></B>, <B><A HREF="#MOVE">MOVE</A></B>,
<B><A HREF="#FREE">FREE</A></B>, <B><A HREF="#SLTA">SLTA</A></B>,
<B><A HREF="#SLVA">SLVA</A></B>, <B><A HREF="#CNFG">CNFG</A></B><BR>
Potential successor:
<B><A HREF="#VOLE">VOLE</A></B><BR>
Related keys:
<B><A HREF="#CHKP">CHKP</A></B>,
<B><A HREF="#WCKP">WCKP</A></B>,
<B><A HREF="#RMCK">RMCK</A></B>,
<B><A HREF="#SCKP">SCKP</A></B>.
<p>
The program will read the content of the current checkpoint file,
<B>.ckp</B> and restore from there
the state of the variables, including
the file unit numbers and the run number <B>numrun</B> (!),
allowing the continuation of an earlier run as if it never stopped.
If you want to reinput some of the data used by the run the run may abort
as certain inputs are forbidden in different context.
In this case either use the
<B><A HREF="#RUNS">RUNS</A></B> key with zero steps to run or the
<B><A HREF="#FILE">FILE</A></B> key.
Use with caution, though!
<p>
<u>Key1</u> (checkpoint file checking procedure):
<UL>
<LI><B>STPD</B> - Stop whenever the self test discovers a
discrepancy.<BR>
<LI><B>IGND</B> - Ignore whenever the self test discovers a
discrepancy.<BR>
<LI><B>FIXD</B> - Attempt to fix discrepancies, but stop if fails
to.<BR>
<LI><B>FIXD</B> - Attempt to fix discrepancies, but don't stop if fails
to.<BR>
<u>Key2</u> (trajectory checking procedure):
<UL><LI><B>NOFX</B> - Abort if trajectory file is damaged
<LI><B>FXTR</B> - Run with damaged trajectory file.<BR>
<U>Data</U>: <B>numrunr</B>, <B>jobnamer</B>
<UL>
<LI><B>numrunr</B> {<B>numrun</B>}, <B>jobnamer</B>
{<B>jobname</B>} - The program will look for the file
<B>&lt;jobnamer&gt;.&lt;numrunr&gt;.ckp</B> to restore the data
from. If the filename is different from the default checkpoint file
name, the program will open a new checkpoint file name with the
default name, i.e., <B>&lt;jobname&gt;.&lt;numrun&gt;.ckp</B> and
save subsequent checkpoint data there.
</UL>
</UL>
</UL>
If the checkpoint file was found too short,
the program will assume that the last configuration was still properly
read (since that is the first item in the file) and write it on a new file
<B>&lt;jobname&gt;.99.crd</B> in binary form (to be read with
<B><A HREF="#CNFG">CNFG</A> READ BNRY NOFX 99</B>).
<P>
<A NAME="STOP"></A>
III.07.05. <em u>Termination of the calculation</em>
<p>
<B>STOP</B> [Key1, Key2, Data]
<p>
<u>Key1</u> (possible exit self test request):
<UL>
<LI><B>SLFT</B> - Perform a <A HREF="#SLFT">self-test</A> before
stopping.<BR>
<u>Key2</u> (Same as Key1 for <B>SLFT</B> see details there):
<UL>
<LI><B>BASC</B> - Basic test<BR>
<LI><B>FULL</B> - Full test (This is the default here)<BR>
<LI><B>ALL </B> - FULL test plus <B>BASC</B> test on selected
molecules:<BR>
<U>Data</U> for Key2=<B>ALL </B>: <B>nmolck</B>
<UL>
<LI><B>nmolck</B> {<B>nmolec</B>} - <B>BASC</B> tests are
performed on molecules 1 - <B>nmolck</B>
</UL>
</UL>
<LI><B>NOSF</B> - Don't perform a <A HREF="#SLFT">self-test</A>
before stopping.
</UL>
<A NAME="Input_OUTPUT"></A>
III.8. <B>Output related keywords</B>
<OL>
<LI><TT><B><A HREF="#OUTP">OUTP</A></B>: </TT>Print various
system charateristics<BR>
<LI><TT><B><A HREF="#PRNT">PRNT</A></B>: </TT>Input echo level<BR>
<LI><TT><B><A HREF="#DSTP">DSTP</A></B>: </TT>Bulk solvent
distribution
print<BR>
<LI><TT><B><A HREF="#PRAC">PRAC</A></B>: </TT>Acceptance rate print
detail<BR>
<LI><TT><B><A HREF="#BNDL">BNDL</A></B>: </TT>Listing of bonds
found<BR>
<LI><TT><B><A HREF="#PRCO">PRCO</A></B>: </TT>Compilation option
listing<BR>
<LI><TT><B><A HREF="#PLBP">PLBP</A></B>: </TT>Potential parameter
echo<BR>
<LI><TT><B><A HREF="#KMNP">KMNP</A></B>: </TT>PDB file property
threshold<BR>
<LI><TT><B><A HREF="#EPLT">EPLT</A></B>: </TT>Switch between
C<sub>v</sub> and U<sub>tot</sub> in convergence plot<BR>
<LI><TT><B><A HREF="#PRPL">PRPL</A></B>: </TT>Print plotted data<BR>
<LI><TT><B><A HREF="#NMVP">NMVP</A></B>: </TT>'Not-moved'
threshold to print solvents<BR>
<LI><TT><B><A HREF="#PXPA">PXPA</A></B>: </TT>Empty proximity
region print<BR>
<LI><TT><B><A HREF="#PXPR">PXPR</A></B>: </TT>Proximity analysis
printing options<BR>
<LI><TT><B><A HREF="#PDBT">PDBT</A></B>: </TT>PDB atomname convention
<LI><TT><B><A HREF="#PLCV">PLCV</A></B>: </TT>Postscript convergence plots
</OL>
<A NAME="OUTP"></A>
III.08.01. <em u>Print various system characteristics</em>
<p>
<B>OUTP</B> Key1
<p>
<u>Key1</u> (type of output):
<UL>
<A NAME="REPT"></A>
<LI><B>REPT</B> -
The program prints a full description of the system (normally
printed before each simulation run).
<LI><B>PRAT</B> -
The program prints a list of atomic constants stored (atomic
weights, radii, valences). The vdw radii stored can be used
for cavity search (see keys <B><A HREF="#GCEN">GCEN</A></B> and
<B><A HREF="#FREE1">FREE&nbsp;WIDO</A></B>) or solute size definition
(see keys <B><A HREF="#FILT">FILT</A></B> and
<B><A HREF="#PHS ">PBCN&nbsp;PHS </A></B>).
<LI><B>PRPB</B> - The program prints the centers of the periodic cells.
<LI><B>PRFC</B> - The program prints the list of functional groups.
<LI><B>PROP</B> - The program prints the internal option and
debug option arrays.
If <B><A HREF="#PRNT">PRNT</A> ECHO</B> was present,
for each option number that is defined by a main key MMC lists
these keys with their option number.
<p>
</UL>
<p>
<A NAME="PRNT"></A>
III.08.02. <em u>Input echo level selection</em>
<p>
<B>PRNT</B> Key1
<p>
<u>Key1</u> (level of echoing input):
<UL>
<LI><B>NECH</B> - The lines with unformatted input will only be
echoed<BR>
<LI><B>ECHO</B> - Each individual unformatted datum will be echoed,
use of default values will be stated
(note, that in some instances these inital defaults may be
overridden),
opening and closing of files will be announced,
extra information about the cavity grid will be printed.
<LI><B>DETL</B> - Besides the output described with
Key1=<B>ECHO</B>,
the formatted input will be also echoed labeled with the
corresponding
record type number.<BR>
</UL>
<A NAME="DSTP"></A>
III.08.03. <em u>Bulk solvent distribution function print level
specification</em>
<p>
<B>DSTP</B> Key1
<P>
Related key: <B><A HREF="#DSTC">DSTC</A></B>.
<p>
<u>Key1</u>:
<UL>
<LI><B>NONE</B> - Do not print distribution functions at all.<BR>
<LI><B>GRKC</B> - Print only the overall solvent-solvent and
solute-solvent g(R), coordination number distributions.
When torsions are active and cloned, print the
individual torsion angle distributions.
The functions are printed in a compact form.<BR>
<LI><B>ALL </B> - Print output for all the properties, including
g(R) and coordination number distribution.
The functions are printed in a compact form.<BR>
<LI><B>GKCC</B> - same as <B>GRKC</B> but the
functions are printed in a longer format to allow easy plotting<BR>
<LI><B>ALLC</B> - same as <B>ALL </B> but the
functions are printed in a longer format to allow easy plotting<BR>
</UL>
<A NAME="PRAC"></A>
III.08.04. <em u>Acceptance rate listing detail</em>
<p>
<B>PRAC</B> Key1[,Key2]
<p>
<u>Key1</u>:
<UL>
<LI><B>SUMR</B> - Only the mean,minimum and maximum acceptance
rates are printed.<BR>
<LI><B>INDI</B> - Individual acceptance rates are also printed
(wherever collected)<BR>
<LI><B>NTRY</B> - Individual number of trials (or rate of trials)
are also printed.<BR>
<u>Key2</u>:
<UL>
<LI><B>USED</B> - The periodic output prints the acceptance rate
for the last steps movetype<BR>
<LI><B>ALLA</B> - The periodic output prints the acceptance rates
for all types of moves
</UL>
</UL>
<A NAME="BNDL"></A>
III.08.05. <em u>Bond list print</em>
<p>
<B>BNDL</B> [Key1 Data]
<p>
Prerequisite: <B><A HREF="#SLTA">SLTA</A></B>
<p>
<u>Key1</u>:
<UL>
<LI><B>DIST</B> - Print only the neighbor list and bond lengths
<LI><B>ANGL</B> - Print also the bond angles
<LI><B>TORS</B> - Print also the torsion angles
<LI><B>ANRS</B> - Print also both the bond and the torsion angles
<LI><B>CNCT</B> - Print PDB-style CONECT records for the bonds found<BR>
Data: <b>nf</b>, <b>nl</b>
<UL>
<LI><b>nf</b> {1} - Start neighbour list at atom <b>nf</b><BR>
<LI><b>nl</b> {nslt}- End neighbour list at atom <b>nl</b>
</UL>
</UL>
Note: atoms with one neighbor will not be listed separately, only
as neighbors.
<p>
<A NAME="PRCO"></A>
III.08.06. <em u>Compilation option print</em>
<p>
<B>PRCO</B> [Key1 Data]
<p>
<u>Key1</u>:
<UL>
<LI><B>NOSA</B> - don't create a preprocessor
<LI><B>SAVE</B> - create a preprocessor reproducing the compilation<BR>
<LI><B>SVMN</B> - create a preprocessor with minimal size, tailored
to the currently running system<BR>
<U>Data for Key1=<B>SAVE</B></U>: <B>numrunr</B>
<UL>
<A HREF="#File_org">Run number</A>
<LI><B>numrunr</B> {<B>numrun</B>} - the
<A HREF="#File_org">run number</A>
of the checkpoint file to use. If <B>numrunr</B>=0, then the
currently running program options will be printed.
</UL>
</UL>
The program prints all compilation options and array sizes as
well as version dates used either by the currently running
ptogram or the program that created the checkpoint file
with <B>jobname.numrunr.ckp</B>.
If requested by Key1, a file <B>pre_jobname.f</B> will be
generated that can be inserted into the preprocessor <b>pre.f</b>
to compile the program the same way.
Furthermore, if the file <b>pre.f</b> is present in the current
directory then the file <B>pre_jobname.f</B> will contain
the full source code of the modified preprocessor.
<p>
<A NAME="PLBP"></A>
III.08.07. <em u>Potential parameter listing</em>
<p>
<B>PLBP</B> Key1 [Key2]
<p>
<u>Key1</u>:
<UL>
<LI><B>USED</B> - Print the parameters in the potential libraries used.
If the solute was not read yet, the list will be printed with the
run description<BR>
<LI><B>ALL</B> - Print the parameters of all potential libraries <BR>
<LI><B>COMU</B> - Like Key1=<B>USED</B>, but for general solvent,
the combined solute-solvent parameters will also be listed.<BR>
<LI><B>COMU</B> - Like Key1=<B>ALL </B>, but for general solvent,
the combined solute-solvent parameters will also be listed.<BR>
<LI><B>SNGL</B> - Print the parameters of a selected single library<BR>
<UL>
<u>Key2</u> (for Key1=<B>SNGL</B> only)
<LI><B>PPPP</B> - the potential code that can be specified with the
<A HREF="#SUPT"><B>SUPT</B></A> key
</UL>
</UL>
<p>
<A NAME="KMNP"></A>
III.08.8. <em u>Coordination number threshold for property
output</em>
<p>
<B>KMNP</B> [Data]
<p>
<U>Data</U>: <b>rkpdbmin</b>
<UL>
<LI><b>rkpdbmin</b> {0.0} - When writing a PDB file with the
<B><A HREF="#WCNF">WCNF</A> PDB*</B> command, properties
(other then first and second shell coordination numbers and
first shell volume) belonging to
solute atoms whose primary coordination number is below
<b>rkpdbmin</b> are set to zero.
</UL>
<A NAME="EPLT"></A>
III.08.9. <em u>Switch between C<sub>v</sub> and U<sub>tot</sub>
in the convergence plot</em>
<p>
<B>EPLT</B> Key1
<p>
<u>Key1</u>:
<UL>
<LI><B>UICV</B> - Energy block averages and C<sub>v</sub> are
plotted<BR>
<LI><B>UIUT</B> - Energy block averages and cumulative total energy
are plotted.
</UL>
<A NAME="PRPL"></A>
III.08.10. <em u>Listing of function values for plots</em>
<p>
<B>PRPL</B> Key1
<p>
<u>Key1</u>:
<UL>
<LI><B>NONE</B> - Draw the plots only on the output file<BR>
<LI><B>CPLD</B> - List the plotted Umbrella Sampling distribution
values.<BR>
<LI><B>ENRG</B> - List the plotted energy and heat-capacity values.
<LI><B>CONV</B> - List the data plotted with the
<B><A HREF="#PLCV">PLCV</A></B> key
<LI><B>ALL </B> - List all plotted data
</UL>
<A NAME="NMVP"></A>
III.08.11. <em u>'Not-moved' threshold</em>
<p>
<B>NMVP</B> Data
<P>
<U>Data</U>: <B>notmoved</B>
<UL>
<LI><B>notmoved</B> {2} - Solvents with acceptance rate &lt;
<B>notmoved</B> percent and their distance from the nearest solute
atom and their energy with the solute will be listed when a) the
run is longer than 200*<B>nmolec</B> steps and b) more than 10
attempts were made to move that solvent. However, for
<B><A HREF="#PRNT">PRNT</A> DETL</B> the list will be produced
anyway. The list is sorted by the acceptance rate.
</UL>
<A NAME="PXPA"></A>
III.08.12. <em u>Empty proximity region print</em>
<p>
<B>PXPA</B> Key1
<p>
<u>Key1</u>:
<UL>
<LI><B>ALL </B> - All proximity regions will be printed<BR>
<LI><B>NOEM</B> - Proximity regions not containing solvent will not
be printed (neither to the standard output nor to the <B>.pxp</B>
file).
</UL>
<A NAME="PXPR"></A>
III.08.13. <em u>Proximity analysis print levels</em>
<p>
<B>PXPR</B> [Key1 Key2 Key3 Key4]
<p>
<u>Key1</u> (distribution function print level):
<UL>
<LI><B>DSTR</B> - Print all calculated distribution functions<BR>
<LI><B>NDST</B> - Don't print the calculated distribution
functions<BR>
<LI><B>NDSF</B> - Don't even print the calculated forces on the
solute atoms<BR>
<u>Key2</u> (volume element function print level):
<UL>
<LI><B>NVLE</B> - Don't print the radial volume element
estimates<BR>
<LI><B>VOLE</B> - Print the radial volume element
estimates<BR>
<u>Key3</u> (functional group analysis):
<UL>
<LI><B>FULT</B> - Print the full functional group analysis
table before the condensed list<BR>
<LI><B>NFLT</B> - Don't print the full functional group
analysis table before the condensed list<BR>
<LI><B>NFGT</B> - Don't print functional group analysis table
at all<BR>
<LI><B>FFGT</B> - Print (only) the full functional group
analysis table<BR>
<u>Key4</u> (parameter echo):
<UL>
<LI><B>NECH</B> - Don't echo the run parameters before each
proximity analysis table<BR>
<LI><B>FECH</B> - Echo the run parameters before each
proximity analysis table
</UL>
</UL>
</UL>
</UL>
The proximity analysis results consist of two parts: distribution
functions
(and their characteristics) and tables of calculated avareges.
The distribution printing is governed by Key1 and Key 2.
The program can print the the averages arranged in three different
tables:
<UL>
<LI>In the original order, further averaged by residues
<LI>Ordered by functional groups (chemical or user defined - see
the key
<B><A HREF="#FCGD">FCGD</A></B>)
<LI>Condensed in the same way as the RDF's are (as governed by
<B>indxrdf</B> - see key <B><A HREF="#SLTA">SLTA</A></B>). Key3
governs which of these tables will be printed.
</UL>
<A NAME="PDBT"></A>
III.08.14. <em u>PDB atomname convention</em>
<p>
<B>PDBT</B> Key1
<p>
<u>Key1</u> (name conversion):
<UL>
<LI><B>INPT</B> - no conversion
<LI><B>REGU</B> - convert to PDB convention
<LI><B>LEFT</B> - left adjust atomnames
</UL>
When printing a structure in PDB format (e.g., keys
<B><A HREF="#WCNF">WCNF</A></B>,
<B><A HREF="#GENS">GENS</A></B>,
<B><A HREF="#FILT">FILT</A></B>)
this option allows the choice of PDB conventions.
<p>
<A NAME="PLCV"></A>
III.08.15. <em u>Postscript convergence plots</em>
<p>
<B>PLCV</B> Key1, Data
<p>
<u>Key1</u> (Property):
<UL>
<LI><B>ENRG</B> - energy
<LI><B>VOLU</B> - volume (after a run with <A HREF="#IBEN">IBEN</A></B>)
<LI><B>NUMB</B> - Number of molecules
(after a run with <A HREF="#GCEN">GCEN</A></B>)
<LI><B>BPAR</B> - GCE B parameter
(after a run with <A HREF="#BTUN">BTUN</A> PICL</B>)
<LI><B>FETI</B> - Free energy (after <A HREF="#TIQU">TIQU</A></B>)
and the polynomial fitting to the TI integrand<BR>
Data: <b>xmn</b>, <b>xdv</b>, <b>nx</b>,
<b>ymn</b>, <b>ydv</b>, <b>ny</b>,
<b>linwid</b>
<UL>
<LI><b>xmn</b> {0} - the beginning of the X-axis
<LI><b>nx</b> {10} - the number of ticks on the X axis
<LI><b>xdv</b> {Nmc(max)/10} - the distance between ticks on the X axis<BR>
Note: Default values will be used for all when <b>nx</b>=0
<LI><b>ymn</b> {Y(min)} - the beginning of the Y-axis
<LI><b>ny</b> {10} - the number of ticks on the Y axis
<LI><b>ydv</b> {(Y(max)-Y(min))/10} -
the distance between ticks on the Y axis<BR>
Note: Default values will be used for all when <b>ny</b>=0
<LI><b>linwid</b> {1} - the plot line width
</UL>
</UL>
<p>
This key can be used to generate a Postscript plot of the block averages
and cumulative averages of the energy, volume, number of molecules or the
free energy calculated with thermodynamic integration.
The name of the plot file will be of the form
<B>&lt;jobname&gt;.&lt;ext&gt;.&lt;runnumber&lt;.ps</B>
where <b>&lt;ext&gt;</b> is
<b>eng</b>, <b>vol</b>, <b>nml</b>, <b>bpr</b>, <b>fen</b> for
Key1= <B>ENGR</B>, <B>VOLU</B>, <B>NUMB</B>, <B>BPAR</B>, <B>FETI</B>,
resp.
If the .ps file already is present, a new file will be opened vith
an added (or incremented) version number.
<p>
<A NAME="Input_ANAL"></A>
III.9. <B>General analysis keywords</B>
<OL>
<LI><TT><B><A HREF="#DSTC">DSTC</A></B>: </TT>Bulk solvent
distribution calculation<BR>
<LI><TT><B><A HREF="#FCGA">FCGA</A></B>: </TT>Functional group
analysis<BR>
<LI><TT><B><A HREF="#ATFR">ATFR</A></B>: </TT>Calculation of mean
force on solute atom<BR>
<LI><TT><B><A HREF="#FLDG">FLDG</A></B>: </TT>Field gradient
calculation on solvents<BR>
<LI><TT><B><A HREF="#SENS">SENS</A></B>: </TT>Sensitivity
analysis<BR>
<LI><TT><B><A HREF="#ENGL">ENGL</A></B>: </TT>Energy contribution
calculation<BR>
<LI><TT><B><A HREF="#FILT">FILT</A></B>: </TT>Solvent filtering<BR>
<LI><TT><B><A HREF="#HBBR">HBBR</A></B>: </TT>Calculate
hydrogen-bond bridges<BR>
<LI><TT><B><A HREF="#CMPC">CMPC</A></B>: </TT>Configuration
comparison<BR>
<LI><TT><B><A HREF="#DENF">DENF</A></B>: </TT>Combined solvent
conformation<BR>
<LI><TT><B><A HREF="#PRTG">PRTG</A></B>: </TT>Print cavity
grids<BR>
<LI><TT><B><A HREF="#STVG">STVG</A></B>: </TT>Estimate solute and
cavity volumes with multiple grids<BR>
<LI><TT><B><A HREF="#CPOC">CPOC</A></B>: </TT>Calculate cavity/pocket occupancie
<LI><TT><B><A HREF="#GENS">GENS</A></B>: </TT>Calculate generic
solvent sites
<LI><TT><B><A HREF="#IWSL">IWSL</A></B>: </TT>Prepare site assignment file,
entropy input
<LI><TT><B><A HREF="#GSAN">GSAN</A></B>: </TT>Generic site analysis choice
<LI><TT><B><A HREF="#GSAO">GSAO</A></B>: </TT>Generic site analysis options
<LI><TT><B><A HREF="#GABF">GABF</A></B>: </TT>Generic site graph analysis
by frame
<LI><TT><B><A HREF="#GENT">GENT</A></B>: </TT>Configuration with
input torsion angles
<LI><TT><B><A HREF="#TAND">TAND</A></B>: </TT>Calculate torsion
angle distribution<BR>
<LI><TT><B><A HREF="#TORT">TORT</A></B>: </TT>Energy analysis
along a torsion<BR>
<LI><TT><B><A HREF="#LPST">LPST</A></B>: </TT>Energy analysis
of backbone loop solutions<BR>
<LI><TT><B><A HREF="#IDAG">IDAG</A></B>: </TT>Aggregate insertion
and deletion sites
<LI><TT><B><A HREF="#SUPI">SUPI</A></B>: </TT>Trajectory superimposition
<LI><TT><B><A HREF="#SVDP">SVDP</A></B>: </TT>Solvent dipole distribution
<LI><TT><B><A HREF="#MOND">MOND</A></B>: </TT>Bulk density
monitoring<BR>
<LI><TT><B><A HREF="#BLKW">BLKW</A></B>: </TT>Bulk density
monitoring<BR>
</OL>
<A NAME="DSTC"></A>
III.09.01. <em u>Bulk solvent
distribution function calculation level specification</em>
<p>
<B>DSTC</B> Key1 [Key2]
<P>
Related key: <B><A HREF="#DSTP">DSTP</A></B>.
<P>
<u>Key1</u> (level of distribution function calculations):
<UL>
<LI><B>NONE</B> - Do not calculate any bulk solvent distribution
function
(to save time, particularly in vector mode).<BR>
<LI><B>SOME</B> - The pair properties, such as the pair energy
distribution and dipole-dipole correlation functions are not
calculated.<BR>
<LI><B>ALL </B> - All distribution function calculations are
done.<BR>
<u>Key2</u> (creation of plot file):
<UL>
<LI><B>NOFI</B> - Nothing is written on the <B>.dst</B> file.<BR>
<LI><B>FILE</B> - g(r)s, QCDFs are written on
the <B>.dst</B> file for plot.<BR>
<U>Fdata:</U>
<UL>
<LI>RECTYPEs 32-33 are read only for
<B><A HREF="#DSTC">DSTC</A> SOME</B> or
<B><A HREF="#DSTC">DSTC</A> ALL</B>.<BR>
<LI>RECTYPE 32: <B>rdmx</B>, <B>ri</B>, <B>rfsslv</B>,
<B>empslv</B>, <B>egrslv</B> (5F15.5)<BR>
<UL>
<LI><B>rdmx</B> {<B>ct</B>, the solvent cutoff} - Maximum range
in  for the COM g(R) for the solute an the solvent. Its default
(and maximum) value is the same as that of CUTOF. Beyond this
distance the g(R) computation would have to include more
than one image of a molecule, leading to both the inclusion
of unwanted correlations and additional computational
complications.<BR>
<LI><B>ri</B> {0.05} - Grid interval in  for g(R) computation.
In the program, g(R) is computed in tabular form at each grid
interval starting from 0 to <B>rdmx</B>.<BR>
<LI><B>rfsslv</B> {3.3} - First solvation shell radius (in ) of the
solvent molecule. This quantity is used in conjunction with coord.
no. calculations. All molecules which lie inside a sphere
of radius <B>rfsslv</B> (centered at the COM of a given solvent
molecule) are said to be coordinated to that solvent
molecule. It is usually set at the location of the first
minimum of the g(R).<BR>
<LI><B>empslv</B> - The midpoint of the solvent binding energy
distribution (in kcal/mol) for the quasi-component of the
binding energy calculation (QCBE).<BR>
<LI><B>egrslv</B> {0.25} - Grid interval in kcal/mol for the
above QCBE calculation. The QCBE is again tabulated for 100
gridpoints starting from (-50*<B>egrslv</B>+<B>empslv</B>) to
(50*<B>egrslv</B>+<B>empslv</B>).<BR>
</UL>
<LI>RECTYPE 33: <B>rssmin</B>, <B>rssmax</B>, <B>rfsslt</B>,
<B>empslt</B>, <B>egrslt</B>, <B>ksltmn</B> (5F15.10,I5)<BR>
<UL>
<LI><B>rssmin</B> {<B>rfsslt</B>}, <B>rssmax</B> {<B>rdmx</B>}
- The solvent-solvent distributions (g(R), coordination number
distributions, near-neighbor pair energy and orientational correlation
distributions;
solute-solvent pair energy distributions and
dipole correlation functions are computed
only for solvent molecules
in the concentric shell around the solute between <B>rssmin</B> and
<B>rssmax</B>.<BR>
<LI><B>rfsslt</B> {5.0} - First solvation shell radius in  for the
solute. Used in conjunction with the coordination number
calculation.<BR>
<LI><B>empslt</B> - Midpoint (in kcal/mol) for the QCBE of
solute.<BR>
<LI><B>egrslt</B> {0.5} - Grid interval (in kcal/mol) for the
QCBE of solute.<BR>
<LI><B>ksltmn</B> {0} - Solute coordination number will be
collected from k=<B>ksltmn</B>.<BR>
</UL>
<LI>RECTYPE 34: (only for <B><A HREF="#DSTC">DSTC</A> ALL</B>)
dmnslt, dmxslt, dmnslv, dmxslv (4F15.10)<BR>
<UL>
<LI><B>dmnslt</B> {0}, <B>dmxslt</B> {<B>rfsslt</B>} -
solute-solvent pair energy distributions and
solute-solvent dipole correlation functions are computed
only for solvent molecules
in the concentric shell around the solute between <B>dmnslt</B> and
<B>dmxslt</B>.<BR>
<LI><B>dmnslv</B> {0}, <B>dmxslv</B> {<B>rfsslv</B>} -
near-neighbour pair energy and
solvent-solvent dipole correlation functions are computed
only for solvent-solvent distances
between <B>dmnslv</B> and <B>dmxslv</B>.<BR>
</UL>
<LI>RECTYPE 35: (only for <B><A HREF="#DSTC">DSTC</A> ALL</B>)
emnslt,egpslt,emnslv,egpslv (4F15.10)<BR>
<UL>
<LI><B>emnslt</B> - Minimum of the solute-solvent pair-energy
distribution scale (in kcal/mol).<BR>
<LI><B>egpslt</B> {0.5} - Grid interval (in kcal/mol) for the
above pair-energy distribution.<BR>
<LI><B>emnslv</B> - Minimum of the solvent-solvent pair-energy
distribution scale (in kcal/mol).<BR>
<LI><B>egpslv</B> {0.25} - Grid interval (in kcal/mol) for the
above pair-energy distribution. The pair-energy distributions are
very helpful to locate any surprise (overlooked) minimum in
the potential function.<BR>
</UL>
</UL>
</UL>
</UL>
<A NAME="FCGA"></A>
III.09.02. <em u>Functional-group analysis of the solute</em>.
<p>
<B>FCGA</B> [Key1]
<p>
Prerequisite:
<B><A HREF="#SLTA">SLTA</A></B><BR>
Potential successor:
<B><A HREF="#FCGD">FCGD</A></B><BR>
<p>
<u>Key1</u> (specifies groups to be printed):
<LI><B>ALL </B> - All functional groups will be listed
<LI><B>OERR</B> - Only the ???? and VERR functional groups will be
listed. Atoms in these groups are likely to have
missing or superfluous bonds.
<P>
The solute is broken up into functional groups and a list of
bonded atoms is given (based on atom-atom distances).
It is also useful for the detection of errors in the
connectivity generated by the program.
The errors found can be corrected with the keys
<B><A HREF="#MAKB">MAKB</A></B> or <B><A HREF="#BRKB">BRKB</A></B>
<p>
<A NAME="ATFR"></A>
III.09.03. <em u>Calculation of average force on the solute
atoms</em>
<p>
<B>ATFR</B><p>
<p>
Potential successor:
<B><A HREF="#PXAN">PXAN</A></B>
<p>
Also requires the
<B><A HREF="#PXAN">PXAN</A></B> or the
<B><A HREF="#SCAN">SCAN</A></B> key.
<p>
<A NAME="FLDG"></A>
III.09.04. <em u>Calculation of average field gradient at the
solvents</em>
<p>
<B>FLDG</B> Key1, Key2, Data
<p>
Prerequisite:
<B><A HREF="#PXCR">PXCR</A></B><BR>
Potential successor:
<B><A HREF="#PXAN">PXAN</A></B><BR>
<p>
Variable to set to 'T' in the <A HREF="#PREP">preprocessor</A>: <B>FG</B>
<p>
<u>Key1</u> (specifies the type of file to write the gradients):
<UL>
<LI><B>NONE</B> - Turn of field gradient calculation
<LI><B>BNRY</B> - Binary file, with all contributions
<LI><B>ASCI</B> - Ascii file, with all contributions
<LI><B>BNTO</B> - Binary file, only the total field gradients
<LI><B>ASTO</B> - Ascii file, only the total field gradients<BR>
<u>Key2</u> (do or don't do solvent sites)
<UL>
<LI><B>ALLG</B> - Calculate field gradients at solute and solvent sites
<LI><B>NOSV</B> - Don't calculate field gradients at the solvent sites<BR>
Data: <B>fgruucut</B>, <B>fgruvcut</B>, <B>fgrvvcut</B>,
<B>rproxmax</B>
<UL>
<LI><B>fgruucut</B> - solute-solute cutoff
<LI><B>fgruvcut</B> - solute-solvetn cutoff
<LI><B>fgrvvcut</B> - solvent-solvent cutoff
<LI><B>rproxmax</B> - field gradients will be calculated only at
solvents nearer than <B>rproxmax</B> to their proximal solute atom
</UL>
</UL>
</UL>
<p>
Only works for general solvent.
Also requires the
<B><A HREF="#PXAN">PXAN</A></B> or the
<B><A HREF="#SCAN">SCAN</A></B> key.
<p>
<A NAME="SENS"></A>
III.09.05. <em u>Sensitivity coefficent calculation</em>
<p>
<B>SENS</B> Data
<p>
<U>Data</U>: <B>logfreq</B>
<UL>
<LI><B>logfreq</B> - The sensitivity coefficient contributions
w.r.t. the LJ potential parameters and the solute site charges will
be calculated at every 10<sup><B>logfreq</B></sup> step. Valid only
for three-site water solvents.
</UL>
<A NAME="ENGL"></A>
III.09.06. <em u>Calculate energy contributions</em>
<p>
<B>ENGL</B> Key1 Key2 [Data]
<P>
Prerequisites:
<B><A HREF="#SLTA">SLTA</A></B>, <B><A HREF="#SUUC">SUUC</A></B><BR>
<P>
<u>Key1</u> (source):
<UL>
<LI><B>AVRG</B> - Print only the averages over a trajectory<BR>
<LI><B>INDI</B> - Print results for each frame of a trajectory<BR>
<u>Key2</u> (sum exclusion):
<UL>
<LI><B>ALLI</B> - Include all interactions<BR>
<LI><B>NOGI</B> - Exclude interactions within each solute group/residue<BR>
<LI><B>NOMI</B> - Exclude interactions within each solute molecule<BR>
<u>Key3</u> (limit type):
<UL>
<LI><B>ARNG</B> - Interaction range specification refers to atom numbers<BR>
<LI><B>GRNG</B> - Interaction range specification refers to
group (residue) numbers.<BR>
Data: <B>iafrst</B>, <B>ialast</B>,
<B>iafrst2</B>, <B>ialast2</B>, <B>ctonnb</B>
<UL>
<LI><B>iafrst</B> {1}, <B>ialast</B> {nslt} - Restrict analysis to solute
atoms/groups from iafrst to ialast
<LI><B>iafrst2</B> {1}, <B>ialast2</B> {nslt} - Restrict interactions
of the atoms/groups selected for analysis with
atoms/groups from iafrst2 to ialast2.
<LI><B>ctonnb</B> {<b>cutslt</b>} - switching function cutoff
will be used from ctonnb to cutslt
(read with key <B><A HREF="#SUVC">SUVC</A></B>)
</UL>
</UL>
</UL>
</UL>
For a configuration analyzed, the program prints
<P>
ia, ig,E(es), E(d), E(r), E14(es), E14(d), E14(r)
(2i10,3e12.5,3e12.5)<BR>
ia, Esw(es), Esw(d), Esw(r)(i10,3e12.5)<BR>
ia, ER(es), ER14(d), ERsw(r), sigma, Nnb, N14, Nw
(i10,3e12.5,3i6)<BR>
<P>
where E(*) refers to different terms of energy between solute atom
ia and the rest with (es), (d) and (r) indicating electrostatic
(1/r), dispersion (1/r<sup>6</sup>) and repulsion
(1/r<sup>12</sup>) terms, respectively;
14 refers to 1-4 interactions; w refers to solute-water
interactions;
R refers to the sensitivity coefficient of the energy with respect
to the Lennard-Jones sigma of solute atom ia;
Nnb, N14, Nw are the number of non-bonded, 1-4 and solute-water
terms contributing, respectively.
<P>
The calculated contributions are also calculated for each group (residue)
as defined in the <b>.slt</b> file.
<P>
For trajectory analysis,a the calculated properties are averaged over
the trajectory; the standard deviations are also calculated.
<P>
Also requires the
<B><A HREF="#PXAN">PXAN</A></B> or the
<B><A HREF="#SCAN">SCAN</A></B> key.
<P>
<A NAME="FILT"></A>
III.09.07. <em u> Solute and/or solvent filtering</em>
<p>
<B>FILT</B> Key1 Key2 Key3 Key4 Key5 Data :
<p>
Prerequisite: <B> <A HREF="#CNFG">CNFG</A></B> or
<B><A HREF="#TRAJ">TRAJ</A></B>
<p>
<u>Key1</u> (basis of filtering):
<UL>
<LI><B>SOLV</B> - Filter solvents by solvation state (inside, invagination
or shell)
<LI><B>GEOM</B> - Filter solvents by position, system size and coupling
parameter
<LI><B>NMBR</B> - Filter out solute and/or solvent segments
<LI><B>ENRG</B> - Filter solvents by solute-solvent energy
<LI><B>FRAM</B> - Filter frames from a trajectory<BR>
<u>Key2</u> (target of filtering), skip when Key1=<B>FRAM</B> :
<UL>
<LI><B>CONF</B> - Filter the current configuration
<LI><B>TRAJ</B> - Filter a trajectory
<LI><B>TRAC</B> - Filter a trajectory, center each frame
<LI><B>TRAO</B> - Filter a trajectory, and write the file of original solvent
numbers
<LI><B>TRCO</B> - Filter a trajectory, center each frame and
write the file of original solvent numbers<BR>
Note: there is some (optional)
<A HREF="#add_inp">additional data input</A> described
at the end of the section.<BR>
<u>Key3</u> (format of filetered result):
<UL>
<LI>For Key2=<B>CONF</B>: see Key2 of <B><A HREF="#CNFG">CNFG</A></B>
<LI>For Key2=<B>TRAJ</B> or Key1=<B>FRAM</B>: see Key1 of <B><A
		HREF="#TRAJ">TRAJ</A></B><BR>
Note: each data input can be followed by the file version increment,
<A HREF="#incvers"><b>incvers</b></A>
</UL>
</UL>
</UL>
For Key1=<B>SOLV</B>:
<UL>
<UL>
<UL>
<u>Key4</u> (filtering specifics):
<UL>
<LI><B>INSD</B> - Discard all solvents that are not 'inside'
the solute, based on solvent neighborhood analysis.<BR>
<LI><B>INSV</B> - Discard all solvents that are not 'inside'
the solute or in invaginations of the solute,
based on solvent neighborhood analysis.<BR>
<LI><B>INVG</B> - Discard all solvents that are not in
invaginations of the solute, based on solvent neighborhood
analysis.<BR>
<LI><B>CRCV</B> - Discard solvent based on the
<A HREF ="#CRCV">circular variance</A> of
the vectors from the solvent to its neighboring solute atoms, as
specified by the data items <b>rslv</b> and <b>cvlim</b>.<BR>
<LI><B>CRCH</B> - Same as <B>CRCV</B> above, except the circular
variance is calculated using the solute heavy atoms only.<BR>
<LI><B>CVLS</B> - Similar to <B>CRCV</B> above, but
 a specified number (<b>leavemol</b>) of innermost (<b>cvlim</b> &gt; 0)
or outermost (<b>cvlim</b> &lt; 0) solvents<BR>
<LI><B>SHL1</B> - Discard all solvents that are outside the
first solvent shell.
<LI><B>SHL2</B> - Discard all solvents that are outside the
second solvent shell.
<LI><B>SHL3</B> - Discard all solvents that are outside the
third solvent shell.
<LI><B>NCLS</B> - Keep a specified number (<b>leavemol</b>)
solvents that are nearest to the solute.<BR>
<u>Key5</u>: Like Key2 of <B><A HREF="#GCEN">GCEN</A></B>
(Sec. III.2.5) to specify the solute radius source
(i.e., <B>RVDW</B>, <B>RSIG</B>, <B>RSGH</B>)
to be used for determining solvent <A
HREF="#bulk">connectivity</A>.<BR>
For Key4 = <B>INSD</B>, <B>INSV</B>, or <B>INVG</B>:
<UL>
<U>Data</U>: <B>rslv</B>, <B>nnoffset</B>, <B>minmem</B>
<LI><B>rslv</B> {1.4} - Solvent radius (in ) to be used for
determining solvent <A HREF="#bulk">connectivity</A>.
<LI><B>nnoffset</B> {0} - when filtering for solvents in
invaginations, solvent will be considered to be in an invagination
if the number of its solvent neighbors is not less than the number
of
its solute atom neighbors minus <B>nnoffset</B> and is part of an
<A HREF="#bulk">'outside'</a> cluster. Thus increasing
<B>nnoffset</B> will retain solvents that are deeper in
invaginations.
<LI><B>minmem</B> {100} - When &gt; 0, all solvent clusters
having at least <B>minmem</B> members will be considered 'outside'
(instead of the largest one).
</UL>
<A NAME="bulk"></a>
When Key1=<B>SOLV</B>, the program determines connected
solvent clusters and considers either the largest or all clusters
having at least <B>minmem</B> members to be 'outside' solvent and
the rest to be 'inside' (unless the nearest solute atom to the
inside
candidate solvent is outside the first two solvent layer). Two
solvents are connected if their COM's are within 2*<B>rslv</B>. A
solute atom is neighbor to a solvent if the solvent's COM is within
<B>R<sub>VdW</sub></B>+<B>rslv</B> to that solute
atom.<BR>
For Key4 = <B>NCLS</B>:
<UL>
<U>Data</U>: <B>leavemol</B>,
<B>nafltrange</b>, <B>iafltrange1</b>, <B>iafltrange2</b>, ...
<LI><B>leavemol</B> - The number of solvent molecules
nearest to the solute to be kept
<LI><B>nafltrange</b> {0} - the solute atoms to solvate will
 be restricted to <B>nafltrange</b> ranges
<LI><B>iafltrange1</b>, <B>iafltrange2</b> (<B>nafltrange</b> times)
- the first and last solute atoms of each range to solvate
</UL>
For Key4 = <B>SHL1</B>, <B>SHL2</B>, or <B>SHL3</B>:
<UL>
<U>Data</U>: <B>rslv</B>,
<B>nafltrange</b>, <B>iafltrange1</b>, <B>iafltrange2</b>, ...
<LI><B>rslv</B> {1.4} - Solvent radius (in ) to define a shell
(1st shell width: rslt+<B>rslv</B>; 2nd shell width: 2*<B>rslv</B>)
<LI><B>nafltrange</b> {0} - the solute atoms to solvate will
 be restricted to <B>nafltrange</b> ranges
<LI><B>iafltrange1</b>, <B>iafltrange2</b> (<B>nafltrange</b> times)
- the first and last solute atoms of each range to solvate
</UL>
For Key4=<B>CRCV</B> or <B>CRCH</B>
<UL>
<U>Data</U>: <B>rcrcv</B>, <b>cvlim</b>, <b>nsltcvfst</b>,
<b>nsltcv</b>
<LI><B>rcrcv</B> {10.0} - Circular variance is calculated with
solute atoms within <B>rcrcv</B> + R(solute atom).
It can be considered as the radius of the test sphere rolled over
the surface
used to calculate solvent accessibility.
<LI><b>cvlim</b> {0.5} - When <b>cvlim</b>&gt;0 then solvents
with circular variance &gt; <b>cvlim</b> are kept (inside the
solute);
when <b>cvlim</b>&lt;0 then solvents with circular variance &lt;
-<b>cvlim</b>
are kept (outside the solute).
<LI><B>nsltcvfst</B> {1} - the first solute atom to use for the
circular variance calculation
<LI><B>nsltcv</B> {<b>nslt</b>} - the last solute atom to use for the
circular variance calculation
</UL>
For Key4 = <B>CVLS</B>:
<UL>
<U>Data</U>: <B>rcrcv</B>, <b>cvlim</b>, <B>nsltcvfst</B>, <B>nsltcv</B>, <B>leavemol</B>
<LI><B>rcrcv</B>, <b>cvlim</b>, <B>nsltcvfst</B>, <B>nsltcv</B> - See above
<LI><B>leavemol</B> - The number of innermost/outermost
solvent molecules to be kept
</UL>
</UL>
</UL>
</UL>
</UL>
For Key1=<B>GEOM</B>:
<UL>
<UL>
<U>Data</U>: <B>rminx</B>, <B>rmaxx</B>, <B>rminy</B>,
<B>rmaxy</B>, <B>rminz</B>, <B>rmaxz</B>, <B>centx</B>,
<B>centy</B>, <B>centz</B>, <B>rmin</B>, <B>rmax</B>,
<B>cplmin</B>, <B>cplmax</B>, <B>nmolmin</B>, <B>nmolmax</B>,
<B>isltfilt</B>.
<UL>
<LI><B>rminx</B>, <B>rmaxx</B>, <B>rminy</B>, <B>rmaxy</B>,
<B>rminz</B>, <B>rmaxz</B> {box limits}-
Solvents whose COM falls between
(<B>rminx</B>,<B>rminy</B>,<B>rminz</B>)
and (<B>rmaxx</B>,<B>rmaxy</B>,<B>rmaxz</B>)
will be kept (default leaves all solvents)
<LI><B>rmin</B> {0}, <B>rmax</B> {outer box radius} -
solvents whose COM's distance from
<B>cent</B> is betwen <B>rmin</B> and <B>rmax</B> will be kept
(default leaves all solvents).
<LI><B>cplmin</B> {0.0}, <B>cplmax</B> {1.0} - Whenever applicable,
configurations where the coupling parameter is outside the range
[<B>cplmin</B>,<B>cplmax</B>] will be dropped.
<LI><B>nmolmin</B> {1}, <B>nmolmax</B>
{<A HREF="#Compilation">maxmol</A>} - Whenever applicable,
configurations where the number of solvent
molecules is outside the range
[<B>nmolmin</B>,<B>nmolmax</B>] will be dropped.
<LI><B>isltfilt</B> [0] - If =1 then the solute atoms will also be filtered.
</UL>
</UL>
</UL>
For Key1=<B>NMBR</B>:
<UL>
<UL>
<U>Data</U>:
<B>nsltseg</B>, <B>ifsltseg1</B>, <B>ilsltseg1</B>,
<B>ifsltseg2</B>, <B>ilsltseg2</B>, ..., <B>nslvseg</B>,
<B>ifslvseg1</B>, <B>ilslvseg1</B>, <B>ifslvseg2</B>,
<B>ilslvseg2</B>
<UL>
<LI><B>nsltseg</B>, <B>ifsltseg1</B>, <B>ilsltseg1</B>, ... -
There will be <B>nsltseg</B> solute atom segments to be
deleted, and their limits (inclusive) are
<B>ifsltseg1</B>, <B>ilsltseg1</B>, etc. (up to a maximum of 250
segments).
<B>i*sltseg*</B> values refer to atom numbers.
<LI><B>nslvseg</B>, <B>ifslvseg1</B>, <B>ilslvseg1</B>, ... -
There will be <B>nslvseg</B> solvent molecule segments to be
deleted, and their limits (inclusive) are
<B>ifslvseg1</B>, <B>ilslvseg1</B>, etc. (up to a maximum of 250
segments).
The solvent number does NOT include the solute; i.e., solvent No 1 is
molecule No 2.
</UL>
</UL>
</UL>
For Key1=<B>ENRG</B>:
<UL>
<UL>
<U>Data</U>: <B>esusvmin</B>, <B>esusvmax</B>
<UL>
<LI><B>esusvmin</B>, <B>esusvmax</B> -
Drop solvents with solute-solvent energy outside the range
[ <B>esusvmin</B>, <B>esusvmax</B>] kcal/mol.
</UL>
</UL>
</UL>
For Key1=<B>FRAM</B>:
<UL>
<UL>
<u>Key4</u> (energy recalculation option) :
<UL>
<LI><B>ETRA</B> - Use the total energy on the trajectory file<BR>
<LI><B>ECAL</B> - Calculate the total energy for each  frame on
the trajectory file<BR>
<UL>
<U>Data</U>: <B>etotmin</B>, <B>etotmax</B>,
<B>nslvmin</B>, <B>nslvmax</B>
<UL>
<LI><B>etotmin</B>, <B>etotmax</B> - minimum and maximum total energy
of frames (configurations) kept
<LI><B>nslvmin</B>, <B>nslvmax</B> - minimum and maximum number of solvents
of frames (configurations) kept
</UL>
</UL>
</UL>
</UL>
</UL>
<u>ADDITIONAL INPUT</u>:<BR>
<A NAME="incvers">
<U>Data: </U><B>incvers</B></A>
<UL>
<LI><B>incvers</B> {10} - The filtered trajectory will
have a version number that is incremented from the initial
trajectory file version number (<B>runvers</B>, see key
<B><A HREF="#FILE">FILE</A></B>) by <B>incvers</B>.
</UL>
<A NAME="add_inp">
<u>ADDITIONAL INPUT</u></a> for Key2=<B>TR**</B>:<BR>
<U>Data:</U> <B>nfrmmax</B>, <B>nfrmfreq</B>, <B>nfrmskip</B>
<UL>
<LI><B>nfrmmax</B> {100000000} - gather configurations until the
<B>nfrmmax</B>-th frame (snapshot).
<LI><B>nfrmfreq</B> {1} - use every <B>nfrmfreq</B>-th frame only
<LI><B>nfrmskip</B> {0} - skip the first <B>nfrmskip</B> frames
</UL>
<A NAME="HBBR"></A>
III.09.08. <em u>Calculate hydrogen-bond bridges</em>
<p>
<B>HBBR</B> Key1 Data
<P>
Prerequisite:
<B><A HREF="#SLTA">SLTA</A></B><BR>
<P>
<u>Key1</u> (anchor atoms specification)
<UL>
<LI><B>ALST</B> - Solute atoms that can anchor a hydrogen-bond bridge
are listed
<LI><B>BBON</B> - Hydrogen-bond bridges are anchored by
protein backbone atoms only
<LI><B>TLST</B> - Solute atom names that can anchor a
hydrogen-bond bridge are listed
<LI><B>CHRG</B> - Solute atoms with partial charge above a threshold
can anchor a hydrogen-bond bridge<BR>
<u>Data</u> <b>rxhmax</b>, <b>hbanglemin</b>
<UL>
<b>rxhmax</b> {3.0} - maximum X...H distance for a hydrogen bond
<b>hbanglemin</b> {90.0} - minimum X...H-Y angle for a hydrogen bond<BR>
<u>Additional data for Key1=<B>ALST</B></u>: <b>nanchor</b>,
<b>ianchor(1)</b>, <b>ianchor(2)</b>, ...
<UL>
<LI><b>nanchor</b> - number of solute anchor atoms
<LI><b>ianchor(i)</b> - ith anchor atoms (<b>nanchor</b> numbers)<BR>
</UL>
<u>Additional data for Key1=<B>TLST</B></u>: <b>ntype</b>,
<b>typename(1)</b>, <b>typename(2)</b>, ...
<UL>
<LI><b>ntype</b> - number of solute anchor atoms
<LI><b>typename(i)</b> - th anchor atom type (<b>ntype</b> numbers)
Replace leading blanks by "."<BR>
</UL>
<u>Additional data for Key1=<B>CHRG</B></u>: <b>qbmin</b>
<UL>
<b>qbmin</b> {0.3} - solute atoms with partial charge less than
-<b>qbmin</b> or solute hydrogens with partial charge
greater than <b>qbmin</b> can anchor hydrogen-bond bridges
</UL>
</UL>
</UL>
<A NAME="CMPC"></A>
III.09.09. <em u>Configuration comparison</em>
<p>
<B>CMPC</B> Key1, Key2, Key3, Data
<p>
Prerequisite: <B><A HREF="#SLTA">SLTA</A></B><BR>
Potential successor:
<B><A HREF="#CNFG">CNFG</A></B><BR>
<p>
<u>Key1</u> (input file format):
<UL>
<LI><B>ASCI</B>, <B>BNRY</B>, <B>ASAN</B>, <B>PDB </B>, <B>CHRM</B>
-
File format. See Key2 of Key <B><A HREF="#CNFG">CNFG</A></B><BR>
<u>Key2</u> (comparison operation):
<UL>
<LI><B>SHRT</B> - Shift and overlay the second conformation's
molecules on the first (valid only when the
<B><A HREF="#PART">PART</A></B> key is active<BR>
<LI><B>SHON</B> - Shift the second conformation's molecules on the
first<BR>
<LI><B>ALLD</B> - Don't shift or rotate the molecules relative to
each other<BR>
<u>Key3</u> (possible special treatment for lipids):
<UL>
<LI><B>GENL</B> - General solute molecules<BR>
<LI><B>LIPI</B> - Lipid bilayer: 2nd half is
in mirror image orientation to the first.<BR>
<U>Data</U>: <b>numrun0</b>, <b>numrun1</b>, <b>numrund</b>,
<B>ix1</b>, <B>dx1</b>, <B>ix2</b>, <B>dx2</b>,
<B>numshow</B>
<UL>
<LI><b>numrun0</b> - first conformation's
<A HREF="#File_org">run number</A><BR>
<LI><b>numrun1</b> - second conformation's run number<BR>
<LI><b>numrund</b> - output files' run numbers will be incremented
by <b>numrund</b><BR>
<LI><B>ix1</b>, <B>dx1</b> - molecules will be <B>dx1</b> &#197;
apart
along the <B>ix1</b> axis<BR>
<LI><B>ix2</b>, <B>dx2</b> - molecules will be <B>dx2</b> &#197;
apart
along the <B>ix2</b> axis<BR>
<LI><B>numshow</B> {all} - show only the top (most different)
<B>numshow</B> molecules<BR>
If Key3=<B>LIPI</B>:
<B>ixflip</B>, <B>lipfst</B>, <B>liplst</B>
<LI><B>ixflip</B> - invert the sign of the <B>ixflip</B>-th
coordinate
for the second half of the lipid molecules<BR>
<LI><B>lipfst</B> {1} - the first lipid is the
<B>lipfst</B>-th solute molecule<BR>
<LI><B>liplst</B> {all} - the last lipid is the
<B>liplst</B>-th solute molecule
</UL>
</UL>
</UL>
</UL>
This operation reads in two configurations, calculates the average,
minimum and maximum RMS between the solute molecules
(possibly subject to the operation specified by Key2)
and spreads out the compared molecules of the solute
in the <B>ix1</b>-<B>ix2</b> plane for comparison.
The centers of the molecules will be shifted
to this plane in a rectangular arrangement.
<p>
<A NAME="DENF"></A>
III.09.10. <em u>Combining solvents from a trajectory</em>
<p>
<B>DENF</B> Key1 [Data]
<P>
Prerequisite: <B><A HREF="#TRAJ">TRAJ</A></B>
<p>
<u>Key1</u> (output format):
<UL>
<LI><B>PDB </B> - Combined file will be in
<A HREF="https://www.rcsb.org/pdb/">PDB</A> format
<LI><B>CHRM</B> - Combined file will be in CHARMM CRD format
<LI><B>INSG</B> - Combined file will be in Insight free (sxyzrq)
format
</UL>
<U>Data</U>: <B>nmcmax, nmcfreq, nmcskip</B>
<UL>
<LI><B>nmcmax</B> {100000000} - gather configurations until the
<B>nmcmax</B>-th step.
<LI><B>nmcfreq</B> {1} - use configuration saved at every
<B>nmcfreq</B>-th simulation step
<LI><B>nmcskip</B> {0} - skip the first <B>nmcskip</B> simulation steps
</UL>
This operation scans the trajectory specified and puts the
first atom of all the solvent molecules into a single
configuration onto a file
<B>&lt;jobname&gt;_den.dat</B> (with the run and version number
taken from the trajectory file's name)
in InsightII free format) as He atoms
If the output file already exists, a file with incremented version
number
will be created.
If parts of the solute were allowed to move,
the affected solute atoms are
also collected. The resulting configuration is a simple
approximation to the three-dimensional density.
<p>
<A NAME="PRTG"></A>
III.09.11. <em u>Cavity grid listing</em>
<p>
<B>PRTG</B> Key1 Key2 Key3 [Data]
<p>
Prerequisites: <B><A HREF="#CNFG">CNFG</A>,
<A HREF="#GCEN">GCEN</A></B><BR>
Potential successor:
<B><A HREF="#MOND">MOND</A></B> or 
<B><A HREF="#BLKW">BLKW</A></B><BR>
Variable to set to 'T' in the <A HREF="#PREP">preprocessor</A>: <B>PG</B>
<P>
Grid points representing cavities around the solute will be
enumerated and separated into mutually disconnected clusters.
The first cluster is the set of gridpoints surrounding the
solute and the remaining ones (if any) represent cavities
inside the solute. The program writes out the solute and the
internal cavities to the file
<B>&lt;jobname&gt;.&lt;numrunw&gt;.grd</B>
and, if requested, the solute and external gridpoints (in the same format) to
<B>&lt;jobname&gt;.&lt;numrunw&gt;_2.&lt;ext&gt;</B> iwhere
&lt;ext&gt; is <b>pdb</B>, <b>CRD</b> or <b>grd</b> for PDB, CHARMM, Insight
free format, resp..
The gridpoints are written as helium atoms with zero charge
and the group (residue) number is the group number of the last
solute atom plus the cluster number.
Note, that only a layer of gridpoints near the solute atoms are
written
providing a bounding shell for each cavity.
<P>
<u>Key1</u> (grid file format):
<UL>
<LI><B>NOLP</B> - No grid file is written
<LI><B>PDB</B> - Grid files are in PDB format
<LI><B>CHRM</B> - Grid files are in CHARMM CRD format
<LI><B>ASAN</B> - Grid files are in annotated ASCII format
(see <B><A HREF="#WCNF">WCNF</A> SVAN</B>)<BR>
<u>Key2</u> (need for grids in the interior of a cavity)
<UL>
<LI><B>ALLG</B> - All gridpoints in a cavity will be written
<LI><B>SRFG</B> - Only gridpoints on the surface of a cavity
will be written<BR>
<u>Key3</u> (controls printing of cavity/pocket environment)
<UL>
<LI><B>NOLS</B> - No atom or residue list lining a cavity/pocket will be given
<LI><B>GRLS</B> - List of residues lining a cavity/pocket will be printed
<LI><B>ATLS</B> - List of heavy atoms
lining a cavity/pocket will be printed
<LI><B>AGLS</B> - Both list of heavy atoms and of residues
lining a cavity/pocket will be printed <BR>
<U>Data</U>: <B>nnexmin</B>, <B>nnexmax</B>, <B>cvlim</B>, <B>rcutcv</B>,
<B>volcavmin</B>, <B>volpckmin</B>, <B>rnearlim</B>, <B>rsltfac</B>,
<B>cvlimcav</B>, <B>randsh</B>, <B>diamslvnew</B>,
<B>nlshave</B>, <B>maxnnsurf</B>, <B>noputbacksfg</B>, <B>numrunw</B>
<UL>
<LI><B>nnexmin</B>, {1} <B>nnexmax</B> {0} - external gridpoints will be
written only if the number of their grid neighbor is between
<B>nnexmin</B>, <B>nnexmax</B> (inclusive). Can be a  huge file!
<LI><B>cvlim</B> {0.0}
- When <B>cvlim</B> &gt; 0.0 external grid points will be dropped
when their circular variance
is &lt; <B>cvlim</B>.
The remaining points will be clustered and printed, filtered by
<B>volcavmin</B> or <B>volpckmin</B>.
<LI><B>volcavmin</B> - {0.0} minimum volume estimate of a cavity
to be printed
<LI><B>volpckmin</B> - {0.0} minimum volume estimate of a pocket
to be printed
<LI><B>rcutcv</B> - {10.0} the cutoff for all XV calculations
<LI><B>rnearlim</B> - {10<sup>1/2</sup>} atoms lininig a cavity/pocket
have to be closer then <B>rnearlim</B> &#197; to a cavity grid.
<LI><B>rsltfac</B> - {0.9} Solute atom radii will be multiplied by
<B>rsltfac</B> for the calculation of grid coverings.
<LI><B>randsh</B> {0.5} - grid will be shifted by <B>randsh</B>*gridsize
<LI><B>cvlimcav</B> - {0.0} circular variance threshold for the cavity centers:
cavites whose centers circular variance is below <B>cvlimcav</B> will be
dropped from consideration
<LI><B>diamslvnew</B> {0.0} - When <B>diamslvnew</B> &gt; 0.0 the grids
are also generated with cavity radius <B>diamslvnew</B>. If it is
significantly smaller than the original <B>rsph</B> value inputted
after <B><A HREF="#GCEN">GCEN</A></B> then the second set might
partition into more
discrete clusters, indicating the presence of invaginations, in
addition to real cavities. These new gridpoints will be written to
files <B>&lt;jobname&gt;.&lt;numrunw&gt;_3.&lt;ext&gt;</B>,
<B>&lt;jobname&gt;.&lt;numrunw&gt;_4.&lt;ext&gt;</B>.
<LI><B>nlshave</B> {0} - When <B>nlshave</B> &gt; 0 both the cavities and
the pockets will be subjected by a sub-cluster search by removing
<b>nlshave</b> layers from each cluster and check if the resulting set of
gridpoints fall apart. If yes, these sub-clusters will be also written
to the corresponding PDB files with chain id 'S'.
<LI><B>maxnnsurf</B> {3} - Grids with <B>maxnnsurf</B> or less
neighbors will be considered surface grids and will be shaved off.
<LI><B>noputbacksfg</B> {0} - When <B>noputbacksfg</B>=1 the shaved
gridpoints will be left out from the subclusters
<LI><B>numrunw</B> {<B>numrun</B>} - the runnumber of the PDB files
to be written.
</UL>
</UL>
</UL>
</UL>
<A NAME="STVG"></A>
III.09.12. <em u>Estimate solute and cavity
volumes with multiple grids</em>
<p>
<B>STVG</B> [Data]
<p>
Prerequisites: <B><A HREF="#CNFG">CNFG</A>,
<B><A HREF="#GCEN">GCEN</A></B>,
<A HREF="#SLTA">SLTA</A></B><BR>
Variable to set to 'T' in the <A HREF="#PREP">preprocessor</A>: <B>PG</B>
<P>
<U>Data</U>: <B>diamslvnew</B>, <B>ngridrep</B>
<UL>
<LI><B>diamslvnew</B> {diamslv} - Calculate the number of grids covered by
the solute using <B>diamslvnew</B> as the solvent radius and
estimate the solute volume from it.
If <B>diamslvnew</B> is different from <B>diamslv</B> (read by the
<B><A HREF="#GCEN">GCEN</A></B> key) then the solute radii will be also
scaled by <B>diamslvnew</B>/<B>diamslv</B>.
<LI><B>ngridrep</B> {10} - repeat the grid cover calcculation with
<B>ngridrep</B>
randomly shifted grid and print the average volumes.
</UL>
The progam estimates the volume of the solute and the volume of
each cavity
based on the number of gridpoints covered/left uncovered, resp.
The first 'cavity' is the volume outside the solute.
The averaging assumes that the cavities occur in the same order in
each sample.
<p>
<A NAME="CPOC"></A>
III.09.13. <em u>Calculate cavity occupancies</em>
<p>
<B>CPOC</B> Key1 Key2 Key3
<P>
Prerequisite:
<B><A HREF="#CNFG">CNFG</A></B>,<B><A HREF="#PRTG">PRTG</A></B>,
<B><A HREF="#GCEN">GCEN</A></B><BR>
<P>
This key calculates the (average) number of solvents in each cavity
as determined by the key <B><A HREF="#PRTG">PRTG</A></B><BR>
<u>Key1</u> (select cavity/pocket)
<UL>
<LI><B>CVTY</B> - calculate the occupancies of internal cavities
<LI><B>PCKT</B> - calculate the occupancies of surface pockets<BR>
<u>Key2</u> (input type specification)
<UL>
<LI><B>CONF</B> - calculate occupancies for a single structure (read by the
<B><A HREF="#CNFG">CNFG</A></B> key)
<LI><B>TRAJ</B> - calculate average occupancies for a trajectory (specified by the
<B><A HREF="#TRAJ">TRAJ</A></B> key). Note that the pockets/cavities are defined
by the conformation read at the start of the run thus it is assumed that
the solute is rigid<BR>
<u>Key3</u> (saving cavity solvents)
<UL>
<LI><B>NOSA</B> - cavity/pocket solvents will not be saved
<LI><B>SAVE</B> - When Key2=<B>CONF</B>, a PDB file containing the solute
and solvents found to be in a cavity/pocket will be saved
</UL>
</UL>
</UL>
<P>
<A NAME="GENS"></A>
III.09.14. <em u>Calculate generic solvent sites</em>
<p>
<B>GENS</B> Key1 Key2 Key3 Key4 Key5 Data
<p>
Prerequisite: <B><A HREF="#PBCN">PBCN</A></B><BR>
Can not be followed by: <B><A HREF="#IWSL">IWSL</A></B>
Related keys: <B><A HREF="#GABF">GABF</A></B>,
<B><A HREF="#GSAN">GSAN</A></B>,
<B><A HREF="#GSAO">GSAO</A></B>, <B><A HREF="#IWSL">IWSL</A></B>
Variable to set to 'T' in the <A HREF="#PREP">preprocessor</A>: <B>PG</B>
<p>
<u>Key1</u> (type of calculation)
<UL>
<LI><B>CALC</B> - Calculate generic solvation sites from a
trajectory
with the algorithm of <A HREF="#Gensite">Mezei and Beveridge</a>
<LI><B>CCAL</B> - Continue the generic solvation site calculation.
This option requires a  preceding <B><A HREF="#RCKP">RCKP</A></B> key.
<LI><B>COMP</B> - Compare two site estimates, read from two files,
by performing a Hungarian method matching. If the solute is also present,
write a PDB file containing the sites, sorted by their circular variance
(inside sites first).
<LI><B>CLST</B> - Cluster sites that are within a threshold distance
in a a single site and combine their occupancies.
</UL>
For Key1=<B>CALC</B> or Key1=<B>CCAL</B>:
<UL>
<u>Key2</u>: (updating option)
<UL>
<LI><B>FIXI</B> - Except for the first scan (iteration)
of the history file, keep
the site estimate fixed during an iteration (recommended).
<LI><B>CONT</B> - Continually update the site estimate during an
iteration.<BR>
<u>Key3</u>: (inital site estimate source)
<UL>
<LI><B>GETC</B> - Obtain initial sites from a system read
previously
by the <B><A HREF="#CNFG">CNFG</A></B> key.
<LI><B>GETH</B> - Obtain initial sites from the trajectory file
<LI><B>RDAS</B> - Read initial sites from a file specified here in
<B>ASCI</B> format (see key <B><A HREF="#CNFG">CNFG</A></B>)
<LI><B>RDBI</B>- Read initial sites from a file specified here in
<B>BNRY</B> form (see key <B><A HREF="#CNFG">CNFG</A></B>)
<LI><B>RDAN</B> - Read initial sites from a file specified here
in <B>ASAN</B> format (see key <B><A HREF="#CNFG">CNFG</A></B>)<BR>
<LI><B>RDPD</B> - Read initial sites from a file specified here
in <B>PDB</B> format (see key <B><A HREF="#CNFG">CNFG</A></B>)
<LI><B>RDCH</B> - Read initial sites from a file specified here
in <B>CHRM</B> format (see key <B><A HREF="#CNFG">CNFG</A></B>)<BR>
<u>Key4</u>: (output file format - see key <B><A HREF="#CNFG">CNFG</A></B>)
CNFG
<UL>
<LI><B>WPDB</B> - Output file format is <B>PDB</B>, the file
<B>&lt;jobname&gt;_gs.</b> will contain the sites only
<LI><B>WASC</B> - Output file format is <B>ASCI</B>, the file
<B>&lt;jobname&gt;_gs.</b> will contain the sites only
<LI><B>WANN</B> - Output file format is <B>ASAN</B>, the file
<B>&lt;jobname&gt;_gs.</b> will contain the sites only
<LI><B>SPDB</B> - Output file format is <B>PDB</B>, the file
<B>&lt;jobname&gt;_gs.</b> will include the solute of the last frame
<LI><B>SASC</B> - Output file format is <B>ASCI</B>, the file
<B>&lt;jobname&gt;_gs.</b> will include the solute of the last frame
<LI><B>SANN</B> - Output file format is <B>ASAN</B>, the file
<B>&lt;jobname&gt;_gs.</b> will include the solute of the last frame
<LI><B>RPDB</B> - Output file format is <B>PDB</B>, the file
<B>&lt;jobname&gt;_gs.</b> will include the solute read at the start
<LI><B>RASC</B> - Output file format is <B>ASCI</B>, the file
<B>&lt;jobname&gt;_gs.</b> will include the solute read at the start
<LI><B>RANN</B> - Output file format is <B>ASAN</B>, the file
<B>&lt;jobname&gt;_gs.</b> will include the solute read at the start<BR>
<u>Key5</u>: (order of generated sites)
<UL>
<LI><B>FSRT</B> - Generated sites will be sorted in the order of
decreasing occupancy, fully occuped sites also sorted in the order of
increasing site RMS
<LI><B>DSRT</B> - Generated sites will be sorted in the order of
increasing site RMS<BR>
<LI><B>CSRT</B> - Generated sites will be sorted in the order of
increasing site convergence<BR>
<LI><B>PSRT</B> - Generated sites will be sorted in the order of
increasing distance from the solute<BR>
<LI><B>CVSR</B> - Generated sites will be sorted in the order of
decreasing site CV wrt the solute (going from inside to outside)<BR>
<LI><B>PXSR</B> - Generated sites will be sorted in the order of
increasing proximity solute index<BR>
<u>Data</u>: <b>ns</b>, <b>nmcmax</b>, <b>nmcfreq</b>,
<b>nmcdisc</b>, <b>rcutxst</b>, <b>rmateps</b>, <b>epsgsite</b>,
<b>rmaxmatch</b>,
<b>rmix</b>, <b>maxiter</b>,
<b>nsmin</b>, <b>nsmax</b>, <b>cplpmin</b>, <b>cplpmax</b>,
<b>cvfilt_traj</b>, <b>cvfilt_trajcut</b>,
<b>froccmin</b>, <b>cvfilt</b>, <b>distsltmax</b>,
<b>rmsdlim</b>, <b>froccminrep</b>,
<b>clstmerge_dmax</b>, <b>froccpairmax</b>, <b>rmaxrematch</b>
<UL>
<LI><b>ns</b> - Generate <B>ns</B> sites to start with. If <b>ns</b>=0,
the initial number of sites will be the number of solvents in the input
structure.
When a structure assigns a solvent to each site and there are solvents
left over those solvents will added as new sites.
<LI><b>nmcmax</b> - trajectory file will be scanned until simulation step
number <b>nmcmax</b>
<LI><b>nmcfreq</b> - Use every <b>nmcfreq</b>-th configuration
from the trajectory file
<LI><b>nmcdisc</b> - Discard the first <b>nmcdisc</b>
configurations.
<u>Note</u>: for MC history files,
<b>nmcmax</b>, <b>nmcfreq</b>, <b>nmcdisc</b>
should refer to MC stepnumbers, not structure numbers.
<LI><b>rcutxst</b> {8.0} - For the matching, consider only sites
within <b>rcutxst</b> &#197; from each solvent (just to speed up the
matching procedure)
<LI><b>rmateps</b> {0.01} - Convergence threshold for optimizing
the Hungarian method matching
<LI><b>epsgsite</b> {0.1} - Convergence threshold for the RMSD
between subsequent site estimates
<LI><b>rmaxmatch</b> {1000.0} - If a solvent is farther than
<b>rmaxmatch</b> &#197; from its assigned site, it will be added as
a new site to the site list and <b>ns</b> will be increased accordingly.
<LI><b>rmix</b> {0.0} - When <b>rmix</b> &gt; 0.0 the previous
site estimate will be mixed with the new site estimate:
new=<b>rmix</b>*old+(1-<b>rmix</b>)*new . This might help the
iteration to converge.
<LI><b>maxiter</b> {100} - maximum number of iterations over the
trajectory. If it is less than the iterations already performed,
only the results will be printed.
<LI><b>nsmin</b> {0}, <b>nsmax</b> {#MH} - Use only configurations
where the number of solvents is in the (inclusive) range
[<b>nsmin</b>,<b>nsmax</b>] (meaningful only for analyzing
grand-canonical ensemble simulations - see the
<B><A HREF="#GCEN">GCEN</A></B> key).
<LI><b>cplpmin</b> {-0.1}, <b>cplpmax</b> {2.0} - Use only
configurations where the coupling parameter is in the range
(meaningful for analyzing PMF calculations, i.e.,
<B><A HREF="#FREE5">FREE&nbsp;PMF1</A></B>)
[<b>cplpmin</b>,<b>cplpmax</b>]
<LI><b>cvfilt_traj</b> - When &gt;0.0 the solvents read from the trajectory
with circular variance (w.r.t the solute) &lt; <b>cvfilt_traj</b>
will be dropped.
<LI><b>cvfilt_trajcut</b> - The CV will be calculated
using a distance cutoff of <b>cvfilt_trajcut</b> &#197;.
<LI><b>froccmin</b> {0.0} - Sites with fractional occupancy &lt;
<b>froccmin</b> will be eliminated.
<LI><b>cvfilt</b> - When &gt;0.0 sites whose CV (w.r.t. the solute) is
&lt; <b>cvfilt</b> will be eliminated.
<LI> <b>distsltmax</b> {0.0} - When &gt;0.0 sites farther than
<b>distsltmax</b> from the nearest solute atom will be eliminated.
<LI><b>rmsdlim</b> {0.0} - When &gt;0.0 sites with site rmsd
&gt; <b>rmsdlim</b> will be dropped.
<LI><b>froccminrep</b> {0.0} - When &gt;0.0 a representative
configuration where all sites with occupancy &gt; <b>froccminrep</b>
(must be &gt; <b>froccmin</b>) will also be generated.
<LI><b>clstmerge_dmax</b> {0.0} - When &gt;0.0 sites that are closer to each
other than <b>clstmerge_dmax</b> will be combined into a single site.
<LI><b>froccpairmax</b> {2.0} - Site pairs whose sum of occupancies
exceed <b>froccpairmax</b> will not be paired for clustering
<LI><b>rmaxrematch</b> {0.0} - When &gt; 0.0 the assigments of solvents
to sites will be repeated using the filtered sites only (see <b>froccmin</b>,
<b>cvfilt</b>, <b>distsltmax</b>, above). For this match only solvents whose
distance from the site is &lt; <b>rmaxrematch</b> &#197; will be accepted.
</UL>
When Key3=<B>RD**</B> additional Data:
<UL>
<LI><b>file</b>, <b>numrun</b>, <b>irunvers</b>, <b>nslt</b>, <b>nslv</b>,
<b>islvrep</b> - information specifying the content of the input
file as described above with Key1=<B>COMP</B>
</UL>
</UL>
</UL>
</UL>
</UL>
For Key1=<B>COMP</B>:
<UL>
<u>Key2</u>, <u>Key3</u> (input file syntaxes)<UL>
<LI><B>ASCI</B> or <B>BNRY</B> or <B>ASAN</B> or <B>PDB </B> or
<B>CHRM</B> - the formats of the two input files as described with
the <B><A HREF="#CNFG">CNFG</A></B> key.</BR>
<u>Key4</u> (site distance measure M)
<UL>
<LI><B>DIST</B> - M = site-site distance
<LI><B>ENVN</B> - M = <# of neigbours> - # of identical neighbors
(both within a user defined threshold).<BR>
<LI><B>ENVI</B> - M = (<# of neighbors>+1)/(# of identical neighbors+1)<BR>
<LI><B>ENVM</B> - M = (site-site distance) *
(<# of neighbors>+1)/(# of identical neighbors+1)<BR>
<LI><B>TANI</B> - M = (1 - Tanimoto coef of the neighbor lists)<BR>
If <b>nexp</b> &gt; 1 then M =&gt; M^N<BR>
If <b>nexp</b> &lt; -1 then M =&gt; M^(1/N)<BR>
<u>Data</u>: <b>ns1</b>, <b>ns2</b>,
<b>file1</b>, <b>numrun1</b>, <b>numvers1</B>, <b>nslt1</b>, <b>nslv1</b>,
<b>islvrep1</b>,
<b>file2</b>, <b>numrun2</b>, <b>numvers2</B>, <b>nslt2</b>, <b>nslv2</b>,
<b>islvrep2</b>, <b>rcutxst</b>, <b>rmateps</b>, <b>rcutcrcv</b>,
<b>rmsdgrid</b>, <b>rneigsite</b>, <b>nexp</b>
<UL>
<LI><b>ns1</b>, <b>ns2</b> - Number of sites in systems 1 and 2,
resp.  If either of them is zero, the number of sites will be
the number of solvents in the corresponding structure.
<LI><b>file1</b>,<b>file2</b> - jobnames of the files
for systems 1 and 2, resp.
<LI><b>numrun1</b>,<b>numrun2</b> - runnumbers of the files
for systems 1 and 2, resp.
<LI><b>numvers1</b>,<b>numvers2</b> - version numbers of the files
for systems 1 and 2, resp.
<LI><b>nslt1</b>,<b>nslt2</b> - number of solute atoms in the files
<b>&lt;jobname&gt;.&lt;runnumber&gt;.crd</b>
for systems 1 and 2, resp.
<LI><b>nslv1</b>,<b>nslv2</b> - number of solvent atoms (per
molecule) on the files for systems 1 and 2, resp.
<LI><b>islvrep1</b>,<b>islvrep2</b> - index of the solvent atom
representing the site for systems 1 and 2, resp.
<LI><b>rcutxst</b> {3.0} - For the matching, consider only sites
within <b>rcutxst</b> &#197; from each solvent
<LI><b>rmateps</b> {0.01} - Convergence threshold for optimizing
the Hungarian method matching
<LI><b>rcutcrcv</b> {8.0} - Cutoff for the circular variance calculation
<LI><b>rmsdgrid</b> {0.15} - The grid size for the matching distance
distribution
<LI><b>rneigsite</b> {5.0} - The neighbor distance threshold when
Key4 is not <B>DIST</B>
<LI><b>nexp</b> {1} - the exponent to use with Key4=<B>TANI</B>
</UL>
</UL></UL>
For Key1=<B>CLST</B>:
<UL>
<u>Key2</u> (input file syntax)<UL>
<LI><B>ASCI</B> or <B>BNRY</B> or <B>ASAN</B> or <B>PDB </B> or
<B>CHRM</B> - the format of the input file as described with
the <B><A HREF="#CNFG">CNFG</A></B> key.</BR>
<u>Data</u>: <b>ns</b>,
<b>file</b>, <b>numrun</b>, <b>nslt</b>, <b>nslv</b>,
<b>islvrep</b>, <b>clstmerge_dmax</b>
<UL>
<LI><b>ns</b> - Number of sites in the system
If <b>ns</b>=0, the number of sites will be
the number of solvents in the input structure.
<LI><b>file</b> - jobname of the files
containing the generic sites to be clustered
<LI><b>numrun</b> - runnumber of the file
<LI><b>numvers</b> - version number of the file
<LI><b>nslt</b>- number of solute atoms in the file
<LI><b>nslv</b> - number of solvent atoms (per
molecule) in the file
<LI><b>islvrep</b> - index of the solvent atom
representing the site
<LI><b>clstmerge_dmax</b> {2.0} - Sites that are closer to each
other than <b>clstmerge_dmax</b> will be combined into a single site.
<LI><b>froccpairmax</b> {2.0} - Site pairs whose sum of occupancies
exceed <b>froccpairmax</b> will not be paired for clustering
</UL>
</UL>
</UL>
</UL>
</UL>
<P>
Both <B>GENS CALC</B> and <B>GENS COMP</B>
are based on using the so-called Hungarian method
to obtain an optimal matching of solvent coordinates to (generic)
site coordinates.
With <B>GENS CALC</B> the calculation scans a simulation trajectory
and dedudeces iteratively a set of generic sites. For each
configuration, the solvent coordinates are matched to the site
estimates, allowing for the same solvent to contribute to different
sites during the simulation.
<P>
When <B>GENS CALC</B> is used,
the program will write several files:
<UL>
<LI><B>&lt;jobname&gt;_gs.</b>: the generated sites.
The PDB data columns give the fractional occupancy and site RMS.
<LI><B>&lt;jobname&gt;.gs_rep.pdb</b>:
The configuration found to be closest to the configuration
represented by the sites.
The PDB data columns give the fractional occupancy and distance between
the representative solvent and the site CV w.r.t the solute.
<LI><B>&lt;jobname&gt;.pdb_gs_rep_all.pdb</b>: An other representative
configuration selected with the additional criterion that all sites
with occupancy &gt; <b>froccminrep</b> are present
<LI><B>&lt;jobname&gt;.pdb_gs_cmp.pdb</b>: a composite configuration
where for each site (with occupancy &gt; <b>froccmin</b>)
the solvent closest to it is selected.
The PDB data columns give the fractional occupancy and site RMS.
Note that these solvents may
be in incompatible conformations, especially when low occupancy
sites are included.
<LI><B>&lt;jobname&gt;._gs_avg.pdb</b>: a composite configuration
where the waters are in their average orientation (prepared
for water solvent only).
The PDB data columns give the fractional occupancy and
the site CV w.r.t the solute.
<LI><B>&lt;jobname&gt;._gs_cls.pdb</b>: the sites after clustering
those that are closer than <b>clstmerge_dmax</b> to each other.
The PDB data columns give the fractional occupancy and
the site RMSD.
<LI><B>&lt;jobname&gt;._gs_cla.pdb</b>: only for water solvent - waters in
their average orientation at the clustered sites.
The PDB data columns give the fractional occupancy and
the site CV w.r.t the solute.
<LI><B>&lt;jobname&gt;._gs_fcl.pdb</b>: the sites after filtering by
CV and clustering by distance
The PDB data columns give the fractional occupancy and
the site CV w.r.t the solute.
<LI><B>&lt;jobname&gt;._gs_fcl.inf</b>: Detailed information
(coordinates, occupancy, CV, RMSD, energy, proximal solute atom)
on the sites in each cluster and their averages over the cluster members
</UL>
<P>
Depending on Key4, the sites or solvents may be preceded by
the full solute.
Representative and the composite configurations list
the solvents in the order of the sites they match to,
but may be offset due to missing matches.
Also, the standard output has a combined list of the sites
and their representatives with their properties.
<P>
The run and version number for these files are taken from
the trajectory file and extension defined by Key3.
If the output file already exists, a file with incremented version
number will be created.
<p>
<A NAME="IWSL"></A>
III.09.15. <em u>Prepare site assignment file, entropy calculation,
clustering sites</em>
<p>
<B>IWSL</B> Key1, Data
<p>
Potential successor: <B><A HREF="#GENS">GENS</A></B>
<p>
<u>Key1</u> (ask for entropy input)
<UL>
<LI><B>WSTO</B> - Write solvent site assignment file (.wsl) and
an input file (.sto) for the program STOW to calculate entropy.
<LI><B>NOST</B> - Write  only solvent site assignment file (.wsl).<BR>
<u>Data</u>
<b>clstfin_dmax</b>, <b>nclust_clstfin</b>, <b>cvfilt_stow</b>,
<b>froccmin_stow</b>, <b>rmsdlim_stow</b>, <b>maxclstmem_stow</b>,
<b>subreground</b>
<UL>
<LI><b>clstfin_dmax</b> {0.0} - When &gt; 0.0
the filtered sites will be clustered
(using single-link clustering) with a distance cutoff of <b>clstfin_dmax</b>.
If Key1=<B>WSTO</B> than the default value is 3.5 A.
<LI><b>nclust_clstfin</b> {0} - When &gt; 0
the filtered sites will be clustered (using single-link clustering)
by varying the distance threshold to yield <b>nclust_clstfin</b>
clusters.
<LI><b>cvfilt_stow</b> {0.0} - When &gt;0.0 the clusters where the
lowest CV value is less then <b>cvfilt_stow</b> will be skipped
from the STOW input.
<LI><b>occfilt_stow</b> {0.0} - When &gt;0.0 the clusters where the
lowest occupancy value is less then <b>occfilt_stow</b> will be skipped
from the STOW input.
<LI><b>rmsdlim_stow</b> {0.0} - When &gt;0.0 the clusters where the
highest site rmsd exceeds <b>rmsdlim_stow</b> will be skipped
from the STOW input.
<LI><b>maxclstmem_stow</b> {10} - Clusters with more than <b>maxclstmem_stow</b>
members will be skipped from the STOW input.
<LI><b>subreground</b> {0.7} - When calculating the number of neighbour sites,
the mean number of waters in the neighborhood of a site
will be rounded up if the fractional part is &gt; <b>subreground</b>
</UL>
</UL>
<P>
<A NAME="GSAN"></A>
III.09.16. <em u>Generic site analyses</em>
<p>
<B>GSAN</B> Key1 [Key2 [Data]]
<p>
Prerequisite: <B><A HREF="#IWSL">IWSL</A></B><BR>
Can not be followed by: <B><A HREF="#GENS">GENS</A></B><BR>
Related key: <B><A HREF="#GSAO">GSAO</A></B><BR>
<P>
<u>Key1</u> (analysis type)
<UL>
<LI><B>DPCV</B> - the covariance matrix among the water
dipoles corresponding to each site will be
written to a file with extension <b>.cov</b>.
<LI><B>DPCR</B> - the correlation matrix among the water
dipoles corresponding to each site will be
written to a file with extension <b>.cor</b>.
<LI><B>ENRG</B> - the matrix of the average energies between waters belonging
all site pairs will be written to a file with extension <b>.eng</b>.
<LI><B>HBND</B> - the matrix of the fraction of time
waters belonging to each site are hydrogen bonded
will be written to a file with extension <b>.hbd</b> and
site-solute hydrogen bond statistics will be calculated.
<LI><B>HBTT</B> - in addition to the site-site hydrogen-bond occupancy
matrix, the matrix of site-solute and solute solute hydrogen bond
occupancies will be printed.
The solute hydrogen bonds are condensed by residues and residues not
forming hydrogen bond with sites are omitted.
<LI><B>OCCC</B> - the matrix the joint and alternative occupancies
will be written to a file with extension <b>oro</b>.
<LI><B>JMPA</B> - the average number of frames between solvent jumps
will be calculated<BR>
<u>Key2</u> (eigenvalue/vector calculation, normalization, order)<BR>
<UL>
For Key1=<B>DPCV</B>, <B>DPCR</B>, <B>ENRG</B>, or <B>HBND</B> only:
<LI><B>NOEV</B> - Don't calculate the eigenvalues/eigenvectors of this matrix
<LI><B>CAEV</B> - Calculate the eigenvalues/eigenvectors of this matrix
and write it to a file with extension <b>.eiv</b><BR>
For Key1=<B>OCCC</B> only:<BR>
<LI><B>ONRM</B> - Normalize the occupancies calculated with the occupancy
expected from uniform distribution<BR>
For Key1=<B>JMPA</B> only:<BR>
<LI><B>RDOR</B> - Read in the original solvent numbers from a file created
by  <B><A HREF="#FILT">FILT</A> TR*O</B><BR>
<u>Data</u> for Key1=<B>HBND</B>: <b>rxhmax</b>, <b>hbanglemin</b>,
<b>occmin_at</b>, <b>occmin_res</b>
<UL>
<LI><b>rxhmax</b> {3.0} - maximum X...H distance for a hydrogen bond
<LI><b>hbanglemin</b> {90.0} - minimum X...H-Y angle for a hydrogen bond<BR>
<LI><b>occmin_at</b> {0.0} - minimum occupancy to show in the
solute atom - site hydrogen bond list
<LI><b>occmin_res</b> {0.0} - minimum occupancy to show in the
solute residue - site hydrogen bond list
</UL>
<u>Data</u> for Key1=<B>JMPA</B> and Key2=<B>RDOR</B>:
<b>numrunorig</b>, <b>iversorig</b>
<UL>
<LI><b>numrunorig</b> {<b>numrun</b>}, <b>iversorig</b>  {1} -
the run number and version number of the original solvent number file
</UL>
</UL>
</UL>
</UL>
</UL>
</UL>
<p>
This key activates the selected analysis options after a generic
site calculation (with the  key
<B><A HREF="#GENS">GENS</A></B>)
<P>
<A NAME="GSAO"></A>
III.09.17. <em u>Generic site analysis options</em>
<p>
<B>GSAO</B> Key1 Key2 Key3 [Data]
<p>
Prerequisite: <B><A HREF="#IWSL">IWSL</A></B><BR>
Can not be followed by: <B><A HREF="#GENS">GENS</A></B><BR>
Related key: <B><A HREF="#GSAN">GSAN</A></B><BR>
<P>
<u>Key1</u> (matrix sort option)
<UL>
<LI><B>NOSR</B> - print the matrix in the original site order
<LI><B>SXYZ</B> - print the matrix sorted by one of the site coordinates
<LI><B>RXYZ</B> - print the matrix sorted by the site-origin distance
* sign of the one of the coordinates
<LI><B>LEXI</B> - print the matrix sorted by a quasi-lexicographical order
<u>Key2</u> (output format specification)
<UL>
<LI><B>ASCI</B> - formatted output, five numbers per line (text file)
<LI><B>ASCL</B> - formatted output, one line per site (text file)
<LI><B>BNRY</B> - unformatted output (binary file)<BR>
<u>Key3</u> (vector to use for covariance  when
Key1 of <B><A HREF="#GSAN">GSAN</A></B>
=<B>DPCV</B> or <B>DPCR</B>)<BR>
<UL>
<LI><B>DIPL</B> - Covariances are calculated for the HOH bisector
<LI><B>HHVC</B> - Covariances are calculated for the H-H vector
<LI><B>DPHH</B> - Covariances are calculated for the vector
normal to the HOH plane<BR>
<u>Data</u> for Key1=<B>SXYZ</B> or <B>RXYZ</B>: <b>iaxis</b>
<UL>
<LI><b>iaxis</b> sorting wil be based on the x, y, or z coordinates
for <b>iaxis</b> = 1, 2, or,3, resp.
</UL>
<u>Data</u> for Key1=<B>LEXI</B>:
<b>w</b>, <b>iaxis1</b> <b>iaxis2</b> <b>iaxis3</b>
<UL>
<LI><b>w</b>, <b>iaxis1</b> <b>iaxis2</b> <b>iaxis3</b> - Sites will be
sorted based on
<b>w</b>^2*c(<b>iaxis1</b>) +<b>w</b>*c(<b>iaxis2</b>) +c(<b>iaxis2</b>)
</UL>
</UL>
</UL>
</UL>
<p>
<A NAME="GABF"></A>
III.09.18. <em u>Generic site graph analysis by frame</em>
<p>
<B>GABF</B> Key1 [Data]
<p>
<u>Key1</u> (print option)
<UL>
<LI><B>NODP</B> - No frame-by-frame printout
<LI><B>DTPR</B> - Frame-by-frame printout
<LI><B>DLPR</B> - Frame-by-frame printout with list of hydrogen bonds<BR>
<u>Data</u> <b>rxhmax</b>, <b>hbanglemin</b>, <b>hbmatmin</b>
<UL>
<LI><b>rxhmax</b> {3.0} - maximum X...H distance for a hydrogen bond
<LI><b>hbanglemin</b> {90.0} - minimum X...H-Y angle for a hydrogen bond<BR>
<LI><b>hbmatmin</b> {0.0} - the clustering will use an adjacency matrix
whose elements are 1 when the frame-averaged connectivity probability is
&gt; <b>hbmatmin</b>
</UL>
</UL>
Calculate the hydrogen-bond matrix for each frame,
find the connected clusters and accumulate (and average) the matrix where
all cluster members are connected.
The matrix of the connectivity probabilities will be written to a file
with extension <b>.hbf</b>.
<p>
<A NAME="GENT"></A>
III.09.19. <em u>Configuration with input torsion angles</em>
<p>
<B>GENT</B> Data
<p>
Prerequisites: <B><A HREF="#CNFG">CNFG</A></B>,
<B><A HREF="#TORD">TORD</A></B>.
<p>
<u>Data</u> <b>ntorch</b>
<UL>
<LI><b>ntorch</b> - Number of torsion angles to specify
</UL>
<u>Fdata:</u> <b>ntorch</b> records
<UL>
<LI><b>ia1</b>, <b>ia2</b>, <b>ang</b> (2i5,f10.0)
<UL>
<LI><b>ia1</b>, <b>ia2</b> - Atoms of torsion bond to change
<LI><b>ang</b> - New torsion angle
</UL>
</UL>
If the atoms <b>ia1</b> and <b>ia2</b> are listed in the reverse order
on the torsion list, the torsion angle used will be -<b>ang</b>.
<P>
<A NAME="TAND"></A>
III.09.20. <em u>Calculate torsion angle distribution</em>
<p>
<B>TAND</B>
<p>
Prerequisite:
<B><A HREF="#TORD">TORD</A></B><BR>
Potential successor:
<B><A HREF="#PXAN">PXAN</A></B><BR>
<p>
When this key is present, analysis of a configuration or trajectory
(activated by the <B><A HREF="#SCAN">SCAN</A></B> key)
will include the calculation of torsion angle distributions for
torsions defined by the <B><A HREF="#TORD">TORD</A></B> key.
The number of angular grids will be determined by the
value of #TD set by the <A HREF="#Compilation">preprocessor</A>.
Besides the distribution, the program prints the standard deviation
(meaningful for not too wide distribution only), the
<A HREF="#CRCV">circular variance</A>
(on a scale of 0 to 1) and the number of grids sampled.
<p>
<A NAME="TORT"></A>
III.09.21. <em u>Energy analysis along a torsion</em>
<p>
<B>TORT</B> Key1 Data
<p>
Prerequisites: <B><A HREF="#CNFG">CNFG</A></B>,
<B><A HREF="#TORD">TORD</A></B>.
<P>
<u>Key1</u> (reference energy)
<UL>
<LI><B>ABSE</B> - The actual energy terms will be printed
<LI><B>RELE</B> - Each energy term represents the difference
between the current and initial conformation
<u>Data</u> <b>ntsamp</b> times (<B>ia1</B>, <B>ia1</B>),
<B>anginc</B>, <B>PDBfile</B>
<UL>
<LI><B>ia1</B>, <B>ia1</B> - atomindices of the
atoms forming the torsion bond to be sampled
<LI><B>anginc</B> {5.0} - angle increment for the scan (in degrees)
<LI><B>engmax</B> {99999.9} - maximum energy to print
<LI><B>PDBfile</B> - if present, each structure generated will be
written to <B>PDBfile</B>.pdb
</UL>
</UL>
This key calculates the various non-bonded and torsion terms
as the selected torsion angle is driven through the full circle.
The values printed are relative to the initial torsion angle.
<P>
If a conformation was generated previously with the
<B><A HREF="#GENT">GENT</A></B> key then the torsion scan will
start at those angles (otherwise at 0).
<p>
<A NAME="LPST"></A>
III.09.22. <em u>Energy analysis of backbone loop solutions</em>
<p>
<B>LPST</B> Key1 Data
<p>
Prerequisites: <B><A HREF="#CNFG">CNFG</A></B>,
<B><A HREF="#LOOP">LOOP</A></B>, <B><A HREF="#TORD">TORD</A></B><BR>
<P>
<u>Key1</u> (reference energy)
<UL>
<LI><B>ABSE</B> - The actual energy terms will be printed
<LI><B>RELE</B> - Each energy term represents the difference
between the current and initial conformation
<P>
<u>Data</u> <B>ia1</B>, <B>ia1</B>, <B>PDBfile</B>
<UL>
<LI><B>ia1</B>, <B>ia1</B> - atomindices of the
atoms forming the torsion bond
<LI><B>PDBfile</B> -if present, each structure generated will be
written to <B>PDBfile</B>.pdb
</UL>
</UL>
This key calculates all possible backbone solutions for the six
torsions following the selected torsion angle,
and calculates the same energy contributions as the
<B><A HREF="#TORT">TORT</A></B> key.
<p>
<p>
<B>HBBR</B> Key1 Data
<P>
<A NAME="IDAG"></A>
III.09.23. <em u>Aggregate insertion and deletion sites</em>
<p>
<B>IDAG</B> Key1, Key2, Data
<p>
Prerequisite: <B><A HREF="#IDLG">IDLG</A></B><BR>
Related key:
<B><A HREF="#GCEN">GCEN</A></B>
<p>
<u>Key1</u> (select insert/delete/both)
<UL>
<LI><B>BOTH</B> - Both insertion and deletion sites will be extracted
<LI><B>INSR</B> - Only insertion sites will be extracted
<LI><B>DELE</B> - Only deletion sites will be extracted<BR>
<u>Key2</u> (filtering option)
<UL>
<LI><B>NOFT</B> - All sites will be considered
<LI><B>FILT</B> - Sites will be subject to a proximity filter:<BR>
<u>Data</u> <b>rmax</b>, <b>nslt1</b>, <b>nslt2</b>, <b>nmcscan</b>
<UL>
<LI><b>rmax</b>, <b>nslt1</b>, <b>nslt2</b> -
Only present when Key2=<B>FILT</B>.
Sites selected should be closer than <b>rmax</b> to one of the
solute atoms in the range <b>nslt1</b>-<b>nslt2</b>
<LI><b>nmcscan</b> {0} - when not zero, the log file will be scanned
until the <b>nmcscan</b>-th MC step.
</UL>
</UL>
</UL>
This option will prepare a PDB file containing the solute and
the sites of insertions and/or deletions as read from the
insertion deletion log file
(created by the <B><A HREF="#IDLG">IDLG</A></B> key during a
grand-canonical ensemble run (see key <B><A HREF="#GCEN">GCEN</A></B>)).
Insertion sites will be represented as oxygens
while deletion sites as nitrogens.
<P>
For solute atom records,
the occupancy and temperature factor columns will contain the distance of
the closest deletion and/or insertion sites (as specified by Key1)
and the number of such sites, resp.
For the deletion and/or insertion site records,
the occupancy and temperature factor columns will contain the index of
the proximal solute atom and the distance from it, resp.
<P>
<A NAME="SUPI"></A>
III.09.24. <em u>Trajectory superimposition</em>
<p>
<B>SUPI</B> Key1, Key2, Data
<P>
Prerequisite: <B><A HREF="#TRAJ">TRAJ</A></B><BR>
<P>
When filtering a trajectory, the filtered configuration
will be superimposed to  reference configuration.<BR>
<u>Key1</u> (reference structure)
<UL>
<LI><B>CONF</B> - Reference structure is the structure read with
the <B><A HREF="#CNFG">CNFG</A></B> key
<LI><B>TRAJ</B> - Reference structure is the first structure read with
the <B><A HREF="#TRAJ">TRAJ</A></B> key<BR>
<u>Key2</u> (atoms to be used)
<UL>
<LI><B>RNGE</B> - Solute atoms to be used will be defined as index range(s)
<LI><B>PROX</B> - Solute atomd to be used will be defined as the
vicinity of a selected atom in the reference structure<BR>
<u>Data</u> for Key2=RNGE:
<B>nsltseg</B>, <B>ifsltseg1</B>, <B>ilsltseg1</B>,
<B>ifsltseg2</B>, <B>ilsltseg2</B>, ..., <B>nslvseg</B>,
<B>ifslvseg1</B>, <B>ilslvseg1</B>, <B>ifslvseg2</B>,
<B>ilslvseg2</B>
<UL>
There will be nsltseg solute atom segments to be used,
and their limits (inclusive) are ifsltseg1, ilsltseg1, etc.
(up to a maximum of 250 segments).
i*sltseg* values refer to atom numbers.
</UL>
<u>Data</u> for Key2=PROX: <B>iaslt_cent</b>, <b>range</b>
<UL>
Atoms within <b>range</b> &#197; of solute atom
<B>iaslt_cent</b> will be used for superimposition
</UL>
</UL>
</UL>
<P>
<A NAME="SVDP"></A>
III.09.25. <em u>Solvent dipole distribution</em>
<p>
<B>SVDP</B> Data
<p>
Prerequisites:
<B><A HREF="#SLVA">SLVA</A></B>, <B><A HREF="#SLTA">SLTA</A></B><BR>
<p>
<u>Data</u> <b>exa</b>, <b>eyb</b>, <b>ezc</b>
<UL>
<LI><b>exa</b>, <b>eyb</b>, <b>ezc</b> - unit vectors in the direction
of coodinate axes (default: Cartesian axes)
</UL>
The program will calculate the distributions of the angles between the
solvent's dipole vector and the three coordinate axes.
<p>
<A NAME="MOND"></A>
III.09.26. <em u>Monitoring bulk solvent density</em>
<p>
<B>MOND</B> [Data]
<P>
Prerequisite:
<B><A HREF="#PBCN">PBCN</A></B><BR>
Prerequisite of:
<B><A HREF="#BTUN">BTUN</A></B><BR>
Can not be followed by:
<B><A HREF="#PRTG">PRTG</A></B>
<P>
<U>Data</U>:
<B>glimminx</B>, <B>glimmaxx</B>, <B>glimminy</B>, <B>glimmaxy</B>,
<B>glimminz</B>, <B>glimmaxz</B>
<UL>
<LI>The solvent density will be monitored inside and outside the
rectangle defined by the ranges
<B>glimminx</B> to <B>glimmaxx</B>,
<B>glimminy</B> to <B>glimmaxy</B>,
<B>glimminz</B> to <B>glimmaxz</B> in the x, y, and z directions,
respectively.
For non-rectangular simulation cells the center of the rectangle
has to be the origin of the coordinate system.
</UL>
<A NAME="BLKW"></A>
III.09.27. <em u>Monitoring bulk solvent density</em>
<p>
<B>BLKW</B> [Data]
<P>
Prerequisite:
<B><A HREF="#PBCN">PBCN</A></B><BR>
Prerequisite of:
<B><A HREF="#BTUN">BTUN</A></B><BR>
Can not be followed by:
<B><A HREF="#PRTG">PRTG</A></B>
<P>
<U>Data</U>: <b>wlayer</b>
<UL>
<LI>The solvent density will be monitored inside and outside the
rectangle defined by the ranges
<B>glimminx</B> to <B>glimmaxx</B>,
<B>glimminy</B> to <B>glimmaxy</B>,
<B>glimminz</B> to <B>glimmaxz</B> in the x, y, and z directions,
respectively, where these limits of the inside box are determined
to be <b>wlayer</b> away from the edges of the simulation rectangle.
It is only applicable with rectangular or cubic bundary condition and
the center of the rectangle
has to be the origin of the coordinate system.
</UL>
<P>
<A NAME="Input_PROXI"></A>
III.10. <B>Proximity analysis keywords</B>
<OL>
<LI><TT><B><A HREF="#PXYZ">PXYZ</A></B>: </TT>Permute coordinates
read<BR>
<LI><TT><B><A HREF="#PXCR">PXCR</A></B>: </TT>Proximity criterion
choice<BR>
<LI><TT><B><A HREF="#FCGD">FCGD</A></B>: </TT>Functional group
definition<BR>
<LI><TT><B><A HREF="#PXLM">PXLM</A></B>: </TT>Omitting atoms from
proximity considerations<BR>
<LI><TT><B><A HREF="#RFSL">RFSL</A></B>: </TT>Set first shell
radii<BR>
<LI><TT><B><A HREF="#RMOD">RMOD</A></B>: </TT>First shell radius
modifications<BR>
<LI><TT><B><A HREF="#DBLG">DBLG</A></B>: </TT>Double proximity
radial gridsize<BR>
<LI><TT><B><A HREF="#DIPC">DIPC</A></B>: </TT>Dipole
fluctutations<BR>
<LI><TT><B><A HREF="#GENV">GENV</A></B>: </TT>Proximity volume
element generation<BR>
<LI><TT><B><A HREF="#GRAV">GRAV</A></B>: </TT>Proximity RDF
averageing<BR>
<LI><TT><B><A HREF="#PXTD">PXTD</A></B>: </TT>Proximity table
difference calculation<BR>
<LI><TT><B><A HREF="#PXDP">PXDP</A></B>: </TT>Solute-solvent
proximity dipole distr. calculation<BR>
<LI><TT><B><A HREF="#PXGR">PXGR</A></B>: </TT>Solute-solvent
proximity g(r) and coordination number calculation<BR>
<LI><TT><B><A HREF="#PXBE">PXBE</A></B>: </TT>Solute-solvent
proximity binding energy calculation<BR>
<LI><TT><B><A HREF="#PXWW">PXWW</A></B>: </TT>Solvent-solvent
proximity distribution calculation<BR>
<LI><TT><B><A HREF="#VOLE">VOLE</A></B>: </TT>Volume element
regeneration<BR>
<LI><TT><B><A HREF="#VORO">VORO</A></B>: </TT>Voronoi analysis<BR>
<LI><TT><B><A HREF="#RTIM">RTIM</A></B>: </TT>Solvent residence
time<BR>
<LI><TT><B><A HREF="#DIFC">DIFC</A></B>: </TT>Solvent diffusion
constant<BR>
<LI><TT><B><A HREF="#TAUC">TAUC</A></B>: </TT>Torsion angle autocorrelation<BR>
</OL>
<p>
<A NAME="PXYZ"></A>
III.10.01. <em u>Permute coordinates read from history file</em>
<p>
<B>PXYZ</B> Data
<p>
Potential successor:
<B><A HREF="#CNFG">CNFG</A></B><BR>
<p>
Data: <b>ix</b>, <b>iy</b>, <b>iz</b> <BR>
<UL>
<LI><b>ix</b>, <b>iy</b>, <b>iz</b> - the X, Y, and Z coordinates
will be permuted to the <b>ix</b>-th, <b>iy</b>-th and <b>iz</b>-th
place before analysis or before writing out a new trajectory
with the key <B><A HREF="#WTRA">WTRA</A></B>.
For example, to switch to the convention used for CHARMM
(Z axis is the prism axis, hexagon vertex on the Y axis) use
PXYZ 1 3 2 .
</UL>
The coordinates will be permuted before analysis or before writing
the
new trajectory.
This functionality allows the use of a trajectory file where
the hexagonal prism's orientation is different from
the convention MMC is using.
Note, that it is only meaningful to use this key with
hexagonal prism cell.
<p>
Note also that the change is applied only to the configurations
analyzed.
Thus, depending on whether the last configuration read was
analyzed,
the configuration current after a <B><A HREF="#SCAN">SCAN</A></B>
may or may not have the coordinates permuted.
<p>
<A NAME="PXCR"></A>
III.10.02. <em u>Proximity criterion choice</em>
<p>
<B>PXCR</B> Key1 Key2 Key3 [Data]
<p>
Prerequisite:
<B><A HREF="#SLTA">SLTA</A></B><BR>
Prerequisite of:
<B><A HREF="#GENV">GENV</A></B>, <B><A HREF="#PXGR">PXGR</A></B>,
<B><A HREF="#PXWW">PXWW</A></B>, <B><A HREF="#PXBE">PXBE</A></B>,
<B><A HREF="#PXLM">PXLM</A></B>, <B><A HREF="#TIMM">TIMM</A></B>,
<B><A HREF="#FLDG">FLDG</A></B>, <B><A HREF="#RTIM">RTIM</A></B>,
<B><A HREF="#DIFC">DIFC</A></B>, <B><A HREF="#PXDP">PXDP</A></B>,
<B><A HREF="#PXTD">PXTD</A></B>, <B><A HREF="#PXWR">PXWR</A></B><BR>
 
<p>
<u>Key1</u> (solvent density source):
<UL>
<LI><B>TDEN</B> - Solvent density is obtained form the cell volume
and solute p.m.v.<BR>
<LI><B>PDEN</B> - Solvent density is obtained from the considered
primary region volume<BR>
<LI><B>WDEN</B> - Solvent density is obtained from the solvent's
p.m.v.<BR>
<LI><B>MDEN</B> - Solvent density is obtained form the cell volume
and the solute volume (including the layer of solvent radius), as
calculated from onsets of the primary RDF's and the Monte Carlo
estimates of the shell volumes.<BR>
<u>Key2</u> (tesselation strategy):
<UL>
<LI><B>BISE</B> -
<A HREF="#Mehrotra">Original proximity criterion</A> based on
bisectors<BR>
<LI><B>RPVW</B> -
<A HREF="#Mezei_88">Radical plane-based proximity criterion</A>,
using Van der Waals radii<BR>
<LI><B>RPCH</B> -
<A HREF="#Mezei_88">Radical plane-based proximity criterion</A>,
using atomic charges<BR>
<u>Key3</u> (ignoring atoms):
<UL>
<LI><B>ALL </B> - Use all atoms on the solute for proximity
index calculation<BR>
<LI><B>NOHD</B> - Ignore hydrogens on the solute for proximity
index calculation<BR>
<LI><B>URBG</B> - Use only solute atoms in the ranges
(specified below) for proximity index calculation<BR>
<LI><B>URNH</B> - same as Key3=<B>URBG</B> but, in addition,
ignore all hydrogens<BR>
Solute atoms ignored by Key3 will not contribute
to any rdf either.<BR>
<U>Data</U>:
<UL>
<LI>For Key3=RNG* only:
<b>nrange</b>; <b>irangestart</b>, <b>irangestop</b> (<b>nrange</b>
times)
- only atoms in the ranges <b>irangestart</b> - <b>irangestop</b>
will be used for proximity index calculation
<LI>For Key2=RPCH only:
<B>rqwmin</B>, <B>rqwmax</B> - the atomic charges will be
mapped (linearly onto the [<B>rqwmin</B>,<B>rqwmax</B>] interval to
be used as radii in defining the radical planes.
Default values: <B>rqwmin</B> {0.0}, <B>rqwmax</B> {1.5}.<BR>
<LI> <B>pmvslt</B>, <b>pmvslv</b>
<UL>
<LI><B>pmvslt</B> {0.0} - solute partial molar volume in
ml/mol<BR>
<LI><B>pmvslv</B> {18.12004} - solvent partial molar volume
in ml/mol
</UL>
</UL>
</UL>
</UL>
</UL>
<p>
<A NAME="FCGD"></A>
III.10.03. <em u>Functional group definition</em>
<p>
<B>FCGD</B> Key1 Data Fdata
<p>
Prerequisite: <B><A HREF="#SLTA">SLTA</A></B></BR>
Can not be followed by: <B><A HREF="#FCGA">FCGA</A></B><BR>
<p>
<u>Key1</u> (definition type):
<UL>
<LI><B>STND</B> - Standard (chemically defined) functional groups
will be used.<BR>
<LI><B>INTO</B> - Functional groups will be defined by the user.
The information will only be used for tabulating proximity
results.<BR>
<LI><B>INTG</B> - Functional groups will be defined by the user.
The information will be used for
tabulating proximity results and for grouping the
calculated distributions, overriding the
<b>indxrdf</B>
information given after the <B><A HREF="#SLTA">SLTA</A></B>
key.<BR>
<U>Data</U>:
<UL>
<LI><B>ninpgrp</B> - Number of functional groups to be defined.<BR>
<LI><B>ninpspecd</B> - Number of descriptors to be read.<BR>
<U>Fdata:</U>
<UL>
<LI> RECTYPE 11: (<B>ninpgrp</B> times):
<B>short name</B>, <B>long name</B>, <b>scope</b> (a4,1x,a8,i5)
<UL>
<LI><B>short name</B>, <B>long name</B> - The names of the newly
defined functional groups.
<LI><b>scope</b> {0} - The scope of a definition.
<UL>
<LI><b>scope</b>=0: All atoms satisfying the definition(s)
will form a single functional group
<LI><b>scope</b>=1: Atoms satisfying the definition(s)
<i>and</i> belonging to different residues (solute groups)
will form different functional groups
<LI><b>scope</b>=2: Atoms satisfying the definition(s)
<i>and</i> belonging to different solute molecules
will form different functional groups
</UL>
</UL>
<LI> RECTYPE 12: (<B>ninpspecd</B> times):
<B>resnam</B>, <B>atnam</B>, <B>labfg8</B>, <B>ifgrp</B>
(a4,1x,a4,1x,a10,i5)
<UL>
<LI> <B>resnam</B>, <B>atnam</B>, <B>labfg8</B>, <B>ifgrp</B>
- Solute atoms with residue name <B>resnam</B>, atom name
<B>atnam</B>,
(see <B>labslt</B> for key <B><A HREF="#SLTA">SLTA</A></B>)
long atom label (<B>spgroup</B>, see key <B><A
HREF="#SLTA">SLTA</A></B>)
of <B>labfg8</B> are defined to belong to the <B>ifgrp</B>-th
functional group.
Blank field for any of these three variables will match everything.
Asterisk in any of these fields will match any character.
</UL>
</UL>
</UL>
</UL>
Note, that care must be exercised when this option is used after
restoring a checkpoint file.
If the <B>INTG</B> key was used before resulting in the non-default
grouping of rdf's then using it again or using the <B>STND</B> key
(to revert to the default) will abort the run since the original
atom by atom information is lost.
<p>
<A NAME="PXLM"></A>
III.10.04. <em u>Omitting atoms from proximity considerations</em>
<p>
<B>PXLM</B> Data
<p>
Prerequisite: <B><A HREF="#PXCR">PXCR</A></B>
<p>
<U>Data</U>: <B>nfadel</B>, <B>nladel</B>
<UL>
<LI><B>nfadel</B> {0} - The first <B>nfadel</B> solute atoms will
be
omitted from proximity considerations (i.e., when assigning
solvents
to solute atoms, these omitted atoms will be ignored).
<LI><B>nladel</B> {0} - The last <B>nladel</B> solute atoms will be
omitted from proximity considerations.
<p>
</UL>
<A NAME="RFSL"></A>
III.10.05. <em u>First shell radius definition</em>
<p>
<B>RFSL</B> Key1 [Data]
<p>
Prerequisite:
<B><A HREF="#SUPT">SUPT</A></B><BR>
Potential successor:
<B><A HREF="#SLTA">SLTA</A></B><BR>
<p>
Can not be followed by: <B><A HREF="#SLTA">SLTA</A></B><BR>
<u>Key1</u> (source of radii):
<UL>
<LI><B>STOR</B> - Use stored first shell radii (shown on the list
obtained by <B><A HREF="#PLBP">PLBP</A></B>.<BR>
<U>Data</U>: <B>rfsfac</B>, <B>rfsadd</B>
<UL>
<LI> <B>rfsfac</B> {1.0}
- All first shell radii will be multiplied by <B>rfsfac</B>.
<LI> <B>rfsadd</B> {0.0}
- All first shell radii will be incremented by <B>rfsadd</B>
 (after multiplication by <B>rfsfac</B>).
</UL>
<LI><B>SIGM</B> - First shell radii will be obtained as the mean of
the L-J sigma parameters of the solvent and solute atoms.
The solvent's sigma is the sigma of the <B>islvrep</B>-th atom
(see key <B><A HREF="#SLVA">SLVA</A></B>) of a general solvent or
the sigma of the water oxygen.<BR>
<U>Data</U>: <B>rfsfac</B>, <B>rfsadd</B>, <B>sigmah</B>
<UL>
<LI> <B>rfsfac</B> {1.05}
- All first shell radii will be multiplied by <B>rfsfac</B>.
<LI> <B>rfsadd</B> {0.0}
- All first shell radii will be incremented by <B>rfsadd</B>
 (after multiplication by <B>rfsfac</B>).
<LI> <B>sigmah</B> {0} When Key1=<B>SIGM</B> and <B>sigmah</B> is
not zero, <B>sigmah</B> will be the sigma value used for the solute
hydrogen atoms.
</UL>
<LI><B>RDFM</B> - Replace the first shell radii with the calculated
minima
of the primary RDF's.
</UL>
<p>
<A NAME="RMOD"></A>
III.10.06. <em u>First shell radius modification</em>
<p>
<B>RMOD</B> Data
<p>
<U>Data</U>: <B>nmod</B>, nmod times : <B>pfll</B>, <B>pfl</B>,
<B>rfsl</B>
<UL>
<LI><B>nmod</B> - number of potential types to modify
<LI><B>pfll</B> - potential library label
(see key <B><A HREF="#SUPT">SUPT</A></B>)
<LI><B>pfl</B> - atom type label or atom type number to modify
<LI><B>rfsl</B> - new first shell radius for potential type
<B>pfl</B> in the potential library <B>pfll</B>
</UL>
<p>
<A NAME="DBLG"></A>
III.10.07. <em u>Radial gridsize doubling</em>
<p>
<B>DBLG</B>
<p>
This key will double the gridsize for the proximity analysis
radial grid, modify all the accumulators involved
and print the full result pages with the modified gridsize.
If the <B><A HREF="#PXPL">PXPL</A></B> key was used earlier, the
proximity distributions will also be written on the <b>.pxp</b>
file.
Note, that the change is permanent for the remainder
of the calculations.
<p>
<A NAME="DIPC"></A>
III.10.08. <em u>Dipole fluctuation calculation</em>
<p>
<B>DIPC</B>
<p>
The fluctuation of the dipole moments will be calculated
in the proximity regions to give estimates of the dielectric
constant.
<p>
<A NAME="GENV"></A>
III.10.09. <em u>Monte Carlo volume element estimation</em>
<p>
<B>GENV</B> Key1 Data [Data]
<p>
Prerequisites:
<B><A HREF="#SLTA">SLTA</A></B>, <B><A HREF="#PXCR">PXCR</A></B>,
<B><A HREF="#PXGR">PXGR</A></B><BR>
<p>
<u>Key1</u> (possible visualization):
<UL>
<LI><B>NOWR</B> - Don't write <B>.pxv</B> file
<LI><B>WRIT</B> - Write file <B>.pxv</B> to visualize proximity
regions<BR>
<U>Data</U>: <B>nransn</B>, <B>iwritefs</B>, <B>iwritels</B>
<UL>
<LI><B>nransn</B> - The number of (additional) random points to
generate this time. This will increase the precision of volume
estimates, but only when the solute conformation remains
fixed during the simulation.
<LI><B>iwritefs</B> {1}, <B>iwritels</B> {<B>nslt</B>} - points
in the proximity of solute atoms from <B>iwritefs</B> to
<B>iwritels</B> will be written to the
file <B>&lt;jobname&gt;.&lt;numrun&gt;.pxv</B>
<LI><B>maxprint</B> {100} - maximum number of points written per
proximity region
</UL>
</UL>
The calculation of the proximity RDF's require the volume element
of
shells in the various proximity regions. This volume is calculated
by a Monte Carlo method: random points of uniform distributions are
generated and the volume of each shell is assumed to be
proportional to
the number falling into them.
<p>
Generating a certain number of random points
after each snapshot analyzed
(see keys <B><A HREF="#SCAN">SCAN</A></B> or <B><A
HREF="#PXAN">PXAN</A></B>),
instead of generating them in one step at the beginning or end,
allows for an approximate accounting of the variation of these
volume element as a result of conformational changes in the solute.
<p>
<A NAME="GRAV"></A>
III.10.10. <em u>Calculating RDF extrema averages</em>
<p>
<B>GRAV</B> Key1 [Data]
Prerequisite:
<B><A HREF="#PXGR">PXGR</A></B><BR>
<p>
<p>
<u>Key1</u> (averaging strategy):
<UL>
<LI><B>ELEM</B> - Averages calculated over chemical elements
<LI><B>PTYP</B> - Averages calculated over potential types<BR>
<U>Data</U>: <B>rkgrmin</B>, <B>r2kgrmin</B>, <B>rtkgrmin</B>
<UL>
<LI><B>rkgrmin</B> {0.5} - Minimum first shell coordination number
to contribute to averages
<LI><B>rkgrmin</B> {2.5} - Minimum first+second shell coordination
number to contribute to averages
<LI><B>rkgrmin</B> {10.0} - Minimum total coordination number to
contribute to averages
</UL>
</UL>
If primary or total RDF's were calculated, the program averages
the location and values of the first and second maxima, and the
first minima for atoms that are sufficiently solvated
as defined by the threshold values inputted above.
<p>
<A NAME="PXTD"></A>
III.10.11. <em u>Proximity table difference generation</em>
<p>
<B>PXTD</B> Data
Prerequisite:
<B><A HREF="#PXCR">PXCR</A></B><BR>
<p>
<p>
<U>Data</U>: <b>filename1</b>, <b>runnum1</b>, <b>filename2</b>,
<b>runnum2</b>
<UL>
<LI><b>filename1</b> - File name root of the first checkpoint file
<LI><b>runnum1</b> -
<A HREF="#File_org">Run number</A>
of the first checkpoint file
<LI><b>filename2</b> - File name root of the second checkpoint file
<LI><b>runnum2</b> - Run number of the second checkpoint file
</UL>
The program reads both checkpoint files and calculated the
differences of
the table entries (shell volumes, coordination numbers, densities,
energies, etc.). The two checkpoint files should refer to the same
system.
<p>
<A NAME="PXDP"></A>
III.10.12. <em u>Solute-solvent dipole distribution function
calculation</em>
<p>
<B>PXDP</B>
<p>
Prerequisites:
<B><A HREF="#PXCR">PXCR</A></B>, <B><A HREF="#PXGR">PXGR</A></B><BR>
Potential successor:
<B><A HREF="#PXAN">PXAN</A></B><BR>
<p>
This key will result in the calculation of two functions for the
solute atoms,
based on the angle THETA between the line drawn from the solute
atom to the
first solvent atom and the bisector of the angle S2-S1-S3
(i.e., assuming that the solvent is water and the first atom
is the oxigen, the dipole direction):
<UL>
<LI>The distribution of THETA for waters in the first shell around
the solute atom.
<LI>The average of THETA as a function of the waters' distance from
their proximal solute atom.
</UL>
<p>
<A NAME="PXGR"></A>
III.10.13. <em u>Solute-solvent proximity radial distribution
function and coordination number calculation</em>
<p>
<B>PXGR</B> Key1 Key2 Key3 Data
<p>
Prerequisite:
<B><A HREF="#SLTA">SLTA</A></B><BR>
Prerequisite of:
<B><A HREF="#GENV">GENV</A></B>, <B><A HREF="#PXGR">PXGR</A></B>,
<B><A HREF="#PXWW">PXWW</A></B>, <B><A HREF="#PXBE">PXBE</A></B>,
<B><A HREF="#PXLM">PXLM</A></B>, <B><A HREF="#TIMM">TIMM</A></B>,
<B><A HREF="#FLDG">FLDG</A></B>, <B><A HREF="#RTIM">RTIM</A></B>,
<B><A HREF="#DIFC">DIFC</A></B>, <B><A HREF="#PXDP">PXDP</A></B>,
<B><A HREF="#PXTD">PXTD</A></B>, <B><A HREF="#PXWR">PXWR</A></B><BR>
<p>
<u>Key1</u> ('total' solute-solvent g(r) calc.):
<UL>
<LI><B>NOTO</B> - Don't calculate total solute-solvent g(r)'s<BR>
<LI><B>PXTO</B> - Calculate total solute-solvent g(r)'s (centered
at each solute atom)<BR>
<u>Key2</u> (proximity solute-solvent g(r) calc.):
<UL>
<LI><B>NOPR</B> - Don't calculate solute-solvent proximity
g(r)'s or proximity Kc's<BR>
<LI><B>PRKC</B> - Calculate proximity coordination numbers<BR>
<LI><B>PRIM</B> - Calculate solute-solvent proximity g(r)'s,
coordination numbers and their distributions<BR>
<u>Key3</u> (density weighting for proximity regions):
<UL>
<LI><B>NOWT</B> - No density weighting is used for the primary
solute-solvent g(r)'s<BR>
<LI><B>TDWT</B> - Primary solute-solvent g(r)'s are weighted by
the total primary area solvent density<BR>
<LI><B>PDWT</B> - Primary solute-solvent g(r)'s are weighted by
the density of the primary area used for the g(r) calculation<BR>
<U>Data</U>: <B>rgpxmax</B>, <B>rpxdiv</B>, <B>rpxnear</B>
<UL>
<LI><B>rgpxmax</B> {10.0} - Range of the solute-solvent g(r)'s in &#197;<BR>
<LI><B>rpxdiv</B> {0.1} - Grid size to be used with the
solute-solvent g(r)'s in &#197;
<LI><B>rpxnear</B> {1.0} - When a solvent gets closer than
<B>rpxnear</B> &#197; to the solute, a strong warning is issued
and the configuration is saved
</UL>
</UL>
</UL>
</UL>
The total g(r) is calculated for the solute atoms by considering
all solvents.
The proximity g(r) is calculated for solute atoms by considering
only
solvents proximal to it.
Coordination numbers (and their distributions) will be calculated
if either
total g(r) or proximity g(r) calculation was requested.
<p>
<A NAME="PXBE"></A>
III.10.14. <em u>Solute-solvent proximity binding energy
calculation</em>
<p>
<B>PXBE</B> Data
<p>
Prerequisite:
<B><A HREF="#PXCR">PXCR</A></B><BR>
Potential successor:
<B><A HREF="#PXAN">PXAN</A></B><BR>
<P>
<U>Data</U>: <B>epmink</B>, <B>epdivk</B>
<UL>
<LI><B>epmink</B> - Minimum of the solute-solvent pair
energy distributions (in kcal/mol)
<LI><B>epdivk</B> - Grid size of the solute-solvent pair energy
distributions (in kcal/mol), assuming that there are 100 grid points.
</UL>
This calculation will generate the distribution of the solute
solvent
energies for solvents in the first shell, as well as the sum of
solute solvent
energies, summed both for all solvent is the first shell and
for all solvents in the whole proximity region.
<p>
<A NAME="PXWW"></A>
III.10.15. <em u>Solvent-solvent proximity distribution
calculation</em>
<p>
<B>PXWW</B> Key1 Key2 Data
<p>
Prerequisite:
<B><A HREF="#PXCR">PXCR</A></B><BR>
Potential successor:
<B><A HREF="#PXAN">PXAN</A></B><BR>
<p>
<u>Key1</u> (coordination number and pair energy distr. calc.):
<UL>
<LI><B>NOPE</B> - No proximity solvent-solvent coordination number
and pair-energy averages will be calculated<BR>
<LI><B>PEWW</B> - Calculate proximity solvent-solvent
coordination number and pair-energy averages<BR>
<u>Key2</u> (g(r) calculation):
<UL>
<LI><B>NOGR</B> - No proximity solvent-solvent g(r) will be
calculated<BR>
<LI><B>GRWW</B> - Calculate proximity solvent-solvent g(r)<BR>
<U>Data</U>: <B>rsolmin</B>, <B>rsolmax</B>,
[<B>rfsww</B>, <B>rceww</B>,]
[<B>rgpxmax</B>, <B>rpxdiv</B>,
<B>nijgvv</B>, nijgvv times: <B>islvirep(i)</B>,
<B>islvjrep(i)</B> ...]
<UL>
<LI><B>rsolmin</B>, <B>rsolmax</B>: solvent-solvent proximity
analysis will use only solvents whose distance from
their proximal solute atom is in the range
[<B>rsolmin</B>,<B>rsolmax</B>]<BR>
If Key1 = <B>PEWW</B>:
<UL>
<LI><B>rfsww</B>: the solvent first shell
radius to be used in the coordination number and pair energy
calculations.
<LI><B>rceww</B>: the solvent binding energy calculations will
include all solvents within a sphere of radius <B>rceww</B>.
</UL>
If Key2 = <B>GRWW</B>:
<UL>
<LI><B>rgpxmax</B> {10.0} - Range of the solvent-solvent g(r)'s<BR>
<LI><B>rpxdiv</B> {0.1} - g(r) grid size to be used.
If the key <B><A HREF="#PXGR">PXGR</A></B> is also present,
the larger of the <B>rgpxmax</B> values will be used and
the two <B>rpxdiv</B> values have to be the same.
<LI><B>nijgvv</B> {0} - The number of different solvent-solvent
RDF's to calculate.<BR>
<LI><B>islvirep(i)</B>, <B>islvjrep(i)</B>: the i-th
solvent-solvent RDF
will be calculated between solvent centers <B>islvirep(i)</B>
and <B>islvjrep(i)</B>.
</UL>
</UL>
</UL>
</UL>
<p>
The solvent-solvent coordination number is the mean number of
solvents
within a sphere of radius <B>rfsww</B>, averaged over all solvents
considered. The solvent-solvent pair energy is the average of
water-water energies, calculated for waters within a distance of
<B>rfsww</B> and
at least one of the pair being considered for the solvent-solvent
analysis.
The solvent-solvent binding energy is the sum of water-water
energies, averaged fover all solvents considered.
<p>
<A NAME="VOLE"></A>
III.10.16. <em u>Volume element preservation/regeneration</em>
<p>
<B>VOLE</B> Key1
<p>
Can not be followed by: <B><A HREF="#RCKP">RCKP</A></B>
<u>Key1</u>:
<UL>
<LI><B>DROP</B> - Volume element estimates are kept for successive
runs using the same solute
<LI><B>KEEP</B> - Volume element estimates are regenareted for each
run.
</UL>
<A NAME="VORO"></A>
III.10.17. <em u>Voronoi polyhedra analysis</em>
<p>
<B>VORO</B> Key1, Key2, Key3, Data
<p>
Prerequisites:
<B><A HREF="#CNFG">CNFG</A></B>, <B><A HREF="#PBCN">PBCN</A></B><BR>
Potential successor:
<B><A HREF="#PXAN">PXAN</A></B><BR>
<p>
<u>Key1</u> (use of PBC):
<UL>
<LI><B>PBC </B> - Analyze with periodic images included<BR>
<LI><B>NPBC</B> - Analyze with periodic images ignored <BR>
<u>Key2</u>:
<UL>
<LI><B>CONF</B> - Analyze single configuration<BR>
<LI><B>TRAJ</B> - Analyze trajectory <BR>
<u>Key3</u>:
<UL>
<LI><B>ALL </B> - Use all atoms in Voronoi construction
<LI><B>NOHD</B> - Ignore hydrogens in Voronoi construction<BR>
</UL>
</UL>
</UL>
NOT IMPLEMENTED YET
<p>
<A NAME="RTIM"></A>
III.10.18. <em u>Solvent residence time calculation</em>
<p>
<B>RTIM</B> Key1, Data
<p>
Prerequisite:
<B><A HREF="#PXCR">PXCR</A></B><BR>
Potential successor:
<B><A HREF="#PXAN">PXAN</A></B><BR>
Variable to set to 'T' in the <A HREF="#PREP">preprocessor</A>: <B>DR</B>
<P>
<u>Key1</u> (shell boundary)
<UL>
<LI><B>FRST</B> - Residence time within first shell<BR>
<LI><B>SCND</B> - Residence time within first two shells <BR>
<LI><B>READ</B> - Residence time within user-defined shell radius<BR>
<U>Data</U>: <B>timestep</B>, <LI><B>aucmin</B>, <LI><B>nzerosignore</B>,
<B>rtimlim</B>
<UL>
<LI><B>timestep</B> - The simulation timestep used
<LI><B>aucmin</B> - The exponential fit will use the autocorrelation
function until its value drops below <B>aucmin</B>
<LI><B>nzerosignore</B> {0} - In the list of 0's and 1's representing out
and in the solvation shell, resp., set to one zeros when there are at most
<B>nzerosignore</B> zeros are between ones.
<LI><B>rtimlim</B> {10.0} - Residence times will be calculated for
staying within <B>rtimlim</B> &#197; of the solute (only when Key1=<B>READ</B>)
</UL>
</UL>
The program will calculate the autocorrelation function
of the residence function (0: out; 1: in) and save it
on a file &lt;jobname&gt;.rtm
to be processed into actual times by a seperate program.
<p>
<A NAME="DIFC"></A>
III.10.19. <em u>Solvent diffusion constant calculation</em>
<p>
<B>DIFC</B> Data
<p>
Prerequisite:
<B><A HREF="#PXCR">PXCR</A></B><BR>
Variable to set to 'T' in the <A HREF="#PREP">preprocessor</A>: <B>DR</B><BR>
<p>
<U>Data</U>: <B>timestep</B>, <B>dcgrid</B>, <B>rdc1</B>, <B>rdc2</B>
<UL>
<LI><B>timestep</B> - The simulation timestep used
<LI><B>dcgrid</B> - The gridsize (in &#197;) for the diffusion
calculation
<LI><B>rdc1</B> {0.0}, <B>rdc2</B> {10.0}, - Diffusion constants
will be calculated for solvents whose distance from the solute
is in the range of [<B>rdc1</B>,<B>rdc2</B>] &#197;
(in addition to calculations for the first and second solvation shell).
</UL>
<p>
The program will calculate the conformational average of the
solvent displacements in each proximity region; within the first shell,
the first two shells and the range defined by <B>rdc1</B> and <B>rdc2</B>.
<p>
<A NAME="TAUC"></A>
III.10.20. <em u>Torsion angle autocorrelation calculation</em>
<p>
<B>TAUC</B> Data
<p>
Prerequisite: <B><A HREF="#PART">PART</A></B>
<p>
<U>Data</U>: <B>nmctorauc</B>,<b>tauc_timestep</b>, <b>tauc_min</b>,
<b>ntaucprint</b>
<UL>
<LI><B>nmctorauc</B> - The torsion angles are saved at every
<B>nmctorauc</B>th step for autocorrelation calculation.
<LI><b>tauc_timestep</b> {1} - The time elapsed between two samples
<LI><b>tauc_min</b> {0.0} - The minimum of the autocorrelation function
to be used for the fit
<LI><b>ntaucprint</b> {20} - the number of autocorrelation function values
to print (evenly distributed)
</UL>
<p>
<A NAME="Input_NUM"></A>
III.11. <B>Numerical precision related keywords</B>
<OL>
<LI><TT><B><A HREF="#HRDW">HRDW</A></B>: </TT>Computer
architecture<BR>
<LI><TT><B><A HREF="#SLFT">SLFT</A></B>: </TT>Self test<BR>
<LI><TT><B><A HREF="#SLFT">SLFT</A></B>: </TT>Self test tolerances<BR>
<LI><TT><B><A HREF="#FIXD">FIXD</A></B>: </TT>Round-off error
elimination<BR>
<LI><TT><B><A HREF="#REGE">REGE</A></B>: </TT>Regenerate from
torsion angles<BR>
</OL>
<A NAME="HRDW"></A>
III.11.01. <em u>Computer architecture-dependent algorithm
selection</em>
<p>
<B>HRDW</B> Key1
<p>
Potential successor:
<B><A HREF="#NSLV">NSLV</A></B><BR>
<p>
<u>Key1</u> (code-determining hardware selection):
<UL>
<LI><B>SCAL</B> - 'Conventional' scalar machine
(32-bit words)<BR>
<LI><B>VC32</B> - Vector machine, 32 bits word, requiring split
loops to preserve argument order (e.g., SGI, IBM)<BR>
<LI><B>VC64</B> - Vector machine, 64 bits word (e.g., Cray)<BR>
</UL>
<A NAME="SLFT"></A>
III.11.02. <em u>Periodic self test</em>
<p>
<B>SLFT</B> Key1 [Key2] [Data]<BR>
Related key:
<B><A HREF="#PDBT">PDBT</A></B>.
<p>
At each <B>nrep</B>-th (see key <A HREF="#RUNS"><B>RUNS</B></A>) step
the program performs a
<A HREF="#Mezei_86">consistency check</A> as specified by Key1.
<p>
<u>Key1</u> (self test level):
<UL>
<LI><B>DEFL</B> - Only default testing, i.e., tests at the start of
a run and the compulsory test (to turn off previously invoked self
testing or just allow to read in Key2 or Data)<BR>
<LI><B>BASC</B> - Basic test (equivalent to one move's worth of
computations) on moved molecule: consistency check on energy
and forces, center of mass, orientation.<BR>
<LI><B>FULL</B> - Full test (equivalent to initial setup
calculation): check energy,
virial sum and coordination numbers by recomputing them from
scratch; check symmetry of near-neighbor bitmap; check random
number generator; for vector hardware algorithms
(<B><A HREF="#HRDW">HRDW</A>
VC**</B>), check energy with nonvectorized routines.<BR>
<LI><B>ALL </B> - FULL test plus <B>BASC</B> test on all molecules.
<p>
<u>Key2</u> (action when test fails):
<UL>
<LI><B>STPD</B>, <B>IGND</B>, <B>FIXD</B>, or <B>FXDI</B> - described above
for the keyword <B><A HREF="#RCKP">RCKP</A></B>.
<LI><B>STWD</B> - If no inconsistency was found,
save a copy of the checkpoint file to
<b>&lt;jobname&gt;.99.ckp</b>;
if any inconsistency was found stop (like for <B>STPD</B>).<BR>
<U>Data</U>: <B>nmcslftst</B>
<UL>
<LI><B>nmcslftst</B> {5000000} Self tests performed at every
<B>nmcslftst</B>-th MC step.
</UL>
</UL>
</UL>
<A NAME="STOL"></A>
III.11.03. <em u>Input self-test tolerances</em>
<p>
<B>STOL</B> Data
<P>
Prerequisite: <B><A HREF="#CNFG">CNFG</A></B>
<p>
<U>Data</U>: <B>engtol</B>, <B>virtol</B>, <B>tortol</B>,
<B>comtol</B>, <B>zmattol</B>, <B>cslttol</B>, <B>d12tol</B>,
<B>d13tol</B>, <B>wsumtol</B><BR>
Self test tolerances for various quantities:
<UL>
<LI><B>engtol</B> {0.0001} - for relative deviation in energies
<LI><B>virtol</B> {0.001} - for relative deviation in virial sums
<LI><B>tortol</B> {0.1} - for torsion angles
<LI><B>comtol</B> {0.001} - for centers of masses
<LI><B>zmattol</B> {0.0001} - for orientation matrix orthonormality
<LI><B>cslttol</B> {0.001} - for solute atom positions (on distance squares)
<LI><B>d12tol</B> {0.1} - for bond lengths
(see <B><A HREF="#LOOP">LOOP</A> LIMM</B>)
<LI><B>d13tol</B> {0.3} - for 1-3 distances
(see <B><A HREF="#LOOP">LOOP</A> LIMM</B>)
<LI><B>wsumtol</B> {0.0001} - preferential sampling weight sums
</UL>
The default of <B>engtol</B> is relaxed by a factor of 10 for
(TPN) ensemble run (see key <B><A HREF="#IBEN">IBEN</A></B>),
for runs with more than 1000 solute atoms or 1000 solvent
molecules, for systems with active torsions
(see key <B><A HREF="#PART">PART</A></B>), or when
the <B>C@G7</B> lines are active;
the default of <B>tortol</B> is relaxed to 20.0
when loop moves don't even touch the rest of the chain
(see <B><A HREF="#LOOP">LOOP</A> LIMM</B>);
the default of <B>cslttol</B> is relaxed to 0.001
for systems with more than 200 solute atoms.
<P>
<A NAME="FIXD"></A>
III.11.04. <em u>Energy contribution recalculation
to eliminate round-off errors</em>
<p>
<B>FIXD</B> Key 1 Data
<P>
<U>Key1:</U> (action on self-test failure):
<UL>
<LI><B>STPD</B> - Stop run if self-test run before energy recalculation fails
<LI><B>IGND</B> - Ignore if self-test run before energy recalculation fails
<LI><B>FIXD</B> - Attempt to fix if self-test run
before energy recalculation fails, stop if fix is unsuccesfull<BR>
<LI><B>FXDI</B> - Attempt to fix if self-test run
before energy recalculation fails, but run no matter how the fix worked<BR>
<U>Data</U>: <B>nmcacord</B>, <B>nmcgcord</B>
<UL>
<LI><B>nmcacord</B> - Recalculate from scratch all the current
energy contribution values and solute geometries
after every <B>nmcacord</B> MC step.
<LI><B>nmcgcord</B> - Recalculate from scratch only the
solute geometries after every <B>nmcacord</B> MC step.
</UL>
</UL>
There are several quantities that the program calculates by
accumulating the changes to them. Round-off errors may cause
some self tests to fail. This option eliminates periodically these
round-off errors.
<p>
<A NAME="REGE"></A>
III.11.05. <em u>Regenerate solute from torsion angles
to eliminate round-off errors</em>
<p>
<B>REGE</B>
<P>
Prerequisite: <B><A HREF="#TORD">TORD</A></B>,
<B><A HREF="#CNFG">CNFG</A></B><BR>
<P>
When this keyword is invoked, the program takes the torsion angles
read from the .crd file and applies them to the solute molecule
given as
input after the <B><A HREF="#SLTA">SLTA</A></B> key.
The rigid core of the molecule will be taken from the .crd file.
This way the input molecule's bond lengths and bond angles will be
made to agree with the input definition.
<P>
<A NAME="Input_DEVELOP"></A>
III.12. <B>Developpers' keywords</B>
<OL>
<LI><TT><B><A HREF="#TEST">TEST</A></B>: </TT>Special code-testing<BR>
<LI><TT><B><A HREF="#OPTN">OPTN</A></B>: </TT>Direct option
input<BR>
<LI><TT><B><A HREF="#DBUG">DBUG</A></B>: </TT>Input debug keys<BR>
listing<BR>
<LI><TT><B><A HREF="#CHKP">CHKP</A></B>: </TT>Checkpoint file saving
frequency<BR>
</OL>
See also the key <B><A HREF="#OUTP">OUTP</A> PROP</B>.
<P>
<A NAME="TEST"></A>
III.12.01. <em u>Special code testing</em>
<p>
<B>TEST</B> Key1 [Data]
<P>
<U>Key1:</U> (test type):
<UL>
<LI><B>PBCT</B> - Periodic boundary condition test by system
translation
<p>
<U>Data</U>: <B>xcn</b>, <b>ycn</b>, <b>zcn</b>
<UL>
<LI><B>xcn</b> {10.0} - x coordinates of the new center<BR>
<LI><B>ycn</b> {20.0} - y coordinates of the new center<BR>
<LI><B>zcn</b> {30.0} - z coordinates of the new center<BR>
</UL>
The program shifts the system to
(<b>xcn</b>,<b>ycn</b>,<b>zcn</b>) and applies the PBC shifts
to the part of the system that is found to be outside the cell.
This will
result in a system that gives a different representation of the
same periodic system as before.
Comparison of the total energy (and its components) calculated on
the system with and without the shift should not show any
difference
(apart from round-off related deviations). Any difference should be
indicative of incorrect PBC handling.
<LI><B>CPLW</B> - Test coupling parameter weight calculation
<LI><B>FSLT</B> - Test solute force calculation by
finite difference quotients<BR>
<U>Data</U>: <B>nfrq</b>, <b>dx</b>
<UL>
<LI><B>nfrq</b> {1} - every <n>nfrq</b>-th solvent molecule
(starting with the first) will be tested
<LI><b>dx</b> {0.1} - the delta x value for the finite difference calculation
</UL>
<LI><B>FSLV</B> - Test solvent force calculation by
finite difference quotients<BR>
<U>Data</U>: <B>nfrq</b>, <b>dx</b>
<UL>
<LI><B>nfrq</b> {1} - every <n>nfrq</b>-th solute
atom (starting with the first) will be tested
<LI><b>dx</b> {0.1} - the delta x value for the finite difference calculation
</UL>
</UL>
<p>
<A NAME="OPTN"></A>
III.12.02. <em u>Direct option input</em>
<p>
<B>OPTN</B> Data
<P>
<U>Data</U>: <B>nopt</B>, nopt times: <B>index</B>, <B>value</B>
<UL>
<LI><B>index</B>, <B>value</B> - Set the internal option array
<B>iop</B> as <B>iop</B>(<B>index</B>)=<B>value</B>.
</UL>
This key allows to set the internal option arrays without the
keyword
mechanism normally used.
<p>
<A NAME="DBUG"></A>
III.12.03. <em u>Input debug keys</em>
<p>
<B>DBUG</B> Data
<P>
<U>Data</U>: <B>nkeys</B>, nkeys times: <B>index</B>, <B>value</B>
<UL>
<LI><B>index</B>, <B>value</B> - Set the internal debug option
array
<B>(idebug)</B> as <B>idebug</B>(<B>index</B>)=<B>value</B>.
</UL>
numbers and the file unit numbers will be also listed. If
<p>
<A NAME="CHKP"></A>
III.12.04. <em u>Checkpoint file saving frequency</em>
<p>
<B>CHKP</B> Data
<p>
Potential successor:
<B><A HREF="#PXAN">PXAN</A></B>
<p>
<u>Data</u>: <B>nrecd</B>
<UL>
<LI><B>nrecd</B> - After every <B>nrecd</B> steps,
information in all of the common blocks are saved on the <B>.ckp</B> file.
This option overrides the default checkpoint file saving frequency
allowing the localization of a program failure (detected usually by a
failed self test).
See also the key <B><A HREF="#SCKP">SCKP</A></B> for additional
options.
<P>
<A NAME="BUFF"></A>
III.01.15. <em u>History file buffer save</em>
<p>
<B>BUFF</B>
<p>
When this key is present, the buffers of the open history files are
written out whenever a new checkpoint file is saved.
This is currently achieved by rewindig the file and skipping to
the end.
</UL>
<A NAME="Input_SUMMARY"></A>
III.13. <B>Keyword summary</B>
<UL>
<LI><TT><B><A HREF="#AROM">AROM</A></B> 05.18 </TT>Aromatic carbon list
<LI><TT><B><A HREF="#ATFR">ATFR</A></B> 09.03 </TT>Calculation of
mean force on solute atoms
<LI><TT><B><A HREF="#BLKW">BLKW</A></B> 09.27 </TT>Bulk density
monitoring
<LI><TT><B><A HREF="#BNDL">BNDL</A></B> 08.05 </TT>Listing of bonds
found
<LI><TT><B><A HREF="#BRKB">BRKB</A></B> 04.05 </TT>Break bonds
<LI><TT><B><A HREF="#BTUN">BTUN</A></B> 02.07 </TT>Tune GCE B
parameter
<LI><TT><B><A HREF="#BUFF">BUFF</A></B> 01.15 </TT>History file
buffer save
<LI><TT><B><A HREF="#CPOC">CPOC</A></B> 09.13 </TT>Cavity/pocket occupancy
<LI><TT><B><A HREF="#CHKP">CHKP</A></B> 12.04 </TT>Checkpoint file
saving frequency
<LI><TT><B><A HREF="#CHRG">CHRG</A></B> 05.15 </TT>Solute charge
manipulations
<LI><TT><B><A HREF="#CLON">CLON</A></B> 04.03 </TT>Solute cloning
<LI><TT><B><A HREF="#CMPC">CMPC</A></B> 09.09 </TT>Configuration
comparison
<LI><TT><B><A HREF="#CNFG">CNFG</A></B> 02.11 </TT>Initial
configuration
<LI><TT><B><A HREF="#CSEG">CSEG</A></B> 04.11 </TT>CHARMM segment
id handling
<LI><TT><B><A HREF="#CNST">CNST</A></B> 05.23 </TT>Constraint potential
<LI><TT><B><A HREF="#DBLG">DBLG</A></B> 10.07 </TT>Double proximity
radial gridsize
<LI><TT><B><A HREF="#DBUG">DBUG</A></B> 12.03 </TT>Input debug keys
<LI><TT><B><A HREF="#DENF">DENF</A></B> 09.10 </TT>Combined solvent
conformation
<LI><TT><B><A HREF="#DIFC">DIFC</A></B> 10.19 </TT>Solvent diffusion
constant
<LI><TT><B><A HREF="#DIPC">DIPC</A></B> 10.08 </TT>Dipole
fluctutations
<LI><TT><B><A HREF="#DSTC">DSTC</A></B> 09.01 </TT>Distribution
calculation
<LI><TT><B><A HREF="#DSTP">DSTP</A></B> 08.03 </TT>Distribution
print
<LI><TT><B><A HREF="#ENGL">ENGL</A></B> 09.06 </TT>Energy
contribution calculation
<LI><TT><B><A HREF="#ENHB">ENHB</A></B> 05.20 </TT>Hydrogen-bond term
<LI><TT><B><A HREF="#EPLT">EPLT</A></B> 08.09 </TT>Energy
convergence plot form
<LI><TT><B><A HREF="#FCGA">FCGA</A></B> 09.02 </TT>Functional group
analysis
<LI><TT><B><A HREF="#FCGD">FCGD</A></B> 10.03 </TT>Functional group
definition
<LI><TT><B><A HREF="#FILE">FILE</A></B> 01.01 </TT>File name root
<LI><TT><B><A HREF="#FILT">FILT</A></B> 09.07 </TT>Solvent
filtering
<LI><TT><B><A HREF="#FIXD">FIXD</A></B> 11.04 </TT>Round-off error
elimination
<LI><TT><B><A HREF="#FLDG">FLDG</A></B> 09.04 </TT>Field gradient
calculation on solvents
<LI><TT><B><A HREF="#FLXR">FLXR</A></B> 05.22 </TT> Residues to sample
<LI><TT><B><A HREF="#FREE">FREE</A></B> 03.01 </TT>Free energy
options
<LI><TT><B><A HREF="#GABF">GABF</A></B> 09.18 </TT> Generic site
graph analysis by frame
<LI><TT><B><A HREF="#GCEN">GCEN</A></B> 02.05 </TT>Grand-canonical
ensemble
<LI><TT><B><A HREF="#GENS">GENS</A></B> 09.14 </TT>Generic
solvation site calculation
<LI><TT><B><A HREF="#GENT">GENT</A></B> 09.19 </TT>Configuration with
input torsion angles
<LI><TT><B><A HREF="#GENV">GENV</A></B> 10.09 </TT>Proximity volume
element
<LI><TT><B><A HREF="#GRAV">GRAV</A></B> 10.10 </TT>Proximity RDF
averageing
<LI><TT><B><A HREF="#GSAN">GSAN</A></B> 09.17 </TT>Generic site analyses
<LI><TT><B><A HREF="#HBBR">HBBR</A></B> 09.08 </TT>Hydrogen-bond
bridges
<LI><TT><B><A HREF="#HBMO">HBMO</A></B> 06.21 </TT>Non-default hydrogen-bond
<LI><TT><B><A HREF="#HRDW">HRDW</A></B> 11.01 </TT>Computer
architecture
<LI><TT><B><A HREF="#IBEN">IBEN</A></B> 02.09 </TT>(TPN) ensemble
<LI><TT><B><A HREF="#IDAC">IDAC</A></B> 01.08 </TT>Proximity
<LI><TT><B><A HREF="#IDAG">IDAG</A></B> 09.23 </TT>Aggregate insertion and
deletion sites</em>
<LI><TT><B><A HREF="#IDLG">IDLG</A></B> 01.09 </TT>I/D log file
<LI><TT><B><A HREF="#IGSV">IGSV</A></B> 06.18 </TT>Non-interactive solvent
<LI><TT><B><A HREF="#IGJA">IGJA</A></B> 06.16 </TT>Ignore Jacobian
<LI><TT><B><A HREF="#INCT">INCT</A></B> 05.11 </TT>Solvent inner
cutoff
<LI><TT><B><A HREF="#IWSL">IWSL</A></B> 09.15 </TT>Prepare site assignment file,
entropy input
<LI><TT><B><A HREF="#KMNP">KMNP</A></B> 08.08 </TT>PDB file
property threshold
<LI><TT><B><A HREF="#LCMP">LCMP</A></B> 02.10 </TT>Virial biasing factor
<LI><TT><B><A HREF="#LIGA">LIGA</A></B> 02.16 </TT>Multi-copy ligand
<LI><TT><B><A HREF="#LIMG">LIMG</A></B> 02.08 </TT>Limit the cavity
grid
<LI><TT><B><A HREF="#LOOP">LOOP</A></B> 06.15 </TT>Loop torsion move
<LI><TT><B><A HREF="#LPST">LPST</A></B> 09.22 </TT>Loop solution list
<LI><TT><B><A HREF="#MAKB">MAKB</A></B> 04.04 </TT>Make bonds
<LI><TT><B><A HREF="#MINE">MINE</A></B> 01.07 </TT>Minimum energy structure extr
into history file
<LI><TT><B><A HREF="#MIXR">MIXR</A></B> 05.03 </TT>LJ parameter
mixing rule
<LI><TT><B><A HREF="#MODA">MODA</A></B> 05.06 </TT>Element
modification
<LI><TT><B><A HREF="#MOLD">MOLD</A></B> 04.08 </TT>Define solute
molecules
<LI><TT><B><A HREF="#MOND">MOND</A></B> 09.26 </TT>Bulk density
monitoring
<LI><TT><B><A HREF="#MOVE">MOVE</A></B> 06.01 </TT>Move selection
<LI><TT><B><A HREF="#MVRT">MVRT</A></B> 06.10 </TT>Alternate translation
and rotation
<LI><TT><B><A HREF="#MV2S">MV2S</A></B> 06.08 </TT>Correlated
2-solute molecule move
<LI><TT><B><A HREF="#NFBU">NFBU</A></B> 06.03 </TT>Solute force
bias
<LI><TT><B><A HREF="#NMVP">NMVP</A></B> 08.11 </TT>Print solvents
not moved
<LI><TT><B><A HREF="#NONB">NONB</A></B> 06.13 </TT>Non-Boltzmann
samplings
<LI><TT><B><A HREF="#NSLV">NSLV</A></B> 02.01 </TT>Number of solvent
moleculess
<LI><TT><B><A HREF="#OPTN">OPTN</A></B> 12.02 </TT>Direct option
input
<LI><TT><B><A HREF="#OUTP">OUTP</A></B> 08.01 </TT>Print various
system characteristics
<LI><TT><B><A HREF="#OVRA">OVRA</A></B> 03.06 </TT>Overlap ratio
evaluation
<LI><TT><B><A HREF="#OVST">OVST</A></B> 02.14 </TT>Solute copy
overlay
<LI><TT><B><A HREF="#PARD">PARD</A></B> 06.06 </TT>Solute molecule
displacement
<LI><TT><B><A HREF="#PART">PART</A></B> 06.09 </TT>Solute torsion
<LI><TT><B><A HREF="#PBCT">PBCT</A></B> 11.01 </TT>Periodic
boundary condition test by system translation
<LI><TT><B><A HREF="#PBCN">PBCN</A></B> 02.04 </TT>Periodic
boundary conditions
<LI><TT><B><A HREF="#PBGR">PBGR</A></B> 02.16 </TT>Solute reset
with grids
<LI><TT><B><A HREF="#PDBT">PDBT</A></B> 08.14 </TT>PDB atomname convention
<LI><TT><B><A HREF="#PFRD">PFRD</A></B> 05.07 </TT>Potential
library input
<LI><TT><B><A HREF="#PLBP">PLBP</A></B> 08.07 </TT>Potential
parameter echo
<LI><TT><B><A HREF="#PLCV">PLCV</A></B> 08.15 </TT>Postscript
convergence plots
<LI><TT><B><A HREF="#PMOD">PMOD</A></B> 05.05 </TT>Potential
library modifications
<LI><TT><B><A HREF="#PRAC">PRAC</A></B> 08.04 </TT>Acceptance rate
print detail
<LI><TT><B><A HREF="#PRCO">PRCO</A></B> 08.06 </TT>Compilation
option listing
<LI><TT><B><A HREF="#PRFI">PRFI</A></B> 06.11
</TT>Insertion/deletion pref. sampl.
<LI><TT><B><A HREF="#PRMF">PRMF</A></B> 05.04 </TT>Torsion
potential specification
<LI><TT><B><A HREF="#PRNT">PRNT</A></B> 08.02 </TT>Input echo level
<LI><TT><B><A HREF="#PROT">PROT</A></B> 05.12 </TT>Minimum
repulsion for general solvent
<LI><TT><B><A HREF="#PRPL">PRPL</A></B> 08.10 </TT>Plotted data print
<LI><TT><B><A HREF="#PRTG">PRTG</A></B> 09.11 </TT>Print cavity
grids
<LI><TT><B><A HREF="#PXAN">PXAN</A></B> 07.03 </TT>Proximity
analysis frequencies
<LI><TT><B><A HREF="#PXBE">PXBE</A></B> 10.14 </TT>Solute-solvent
proximity
<LI><TT><B><A HREF="#PXCR">PXCR</A></B> 10.02 </TT>Proximity
criterion choice
<LI><TT><B><A HREF="#PXDP">PXDP</A></B> 10.12 </TT>Solute-solvent
proximity dipole distr. calculation
binding energy calculation
<LI><TT><B><A HREF="#PXTD">PXTD</A></B> 10.11 </TT>Proximity table
difference calculation
<LI><TT><B><A HREF="#PXGR">PXGR</A></B> 10.13 </TT>Solute-solvent
proximity g(r) calculation
<LI><TT><B><A HREF="#PXLM">PXLM</A></B> 10.04 </TT>Omitting atoms
from proximity considerations
<LI><TT><B><A HREF="#PXPA">PXPA</A></B> 08.12 </TT>Empty proximity
region print
<LI><TT><B><A HREF="#PXPL">PXPL</A></B> 01.10 </TT>Proximity
analysis distribution plot file writing
<LI><TT><B><A HREF="#PXPR">PXPR</A></B> 08.13 </TT>Proximity
analysis printing options
<LI><TT><B><A HREF="#PXWR">PXWR</A></B> 01.11 </TT>Proximity
information file
<LI><TT><B><A HREF="#PXWW">PXWW</A></B> 10.15 </TT>Solvent-solvent
proximity
distribution calculation
<LI><TT><B><A HREF="#PXYZ">PXYZ</A></B> 10.01 </TT>Permute
coordinates read
<LI><TT><B><A HREF="#RAUS">RAUS</A></B> 03.02 </TT>AUS parameter
change
<LI><TT><B><A HREF="#RCKP">RCKP</A></B> 07.04 </TT>Checkpoint file
read
<LI><TT><B><A HREF="#RDBD">RDBD</A></B> 07.04 </TT>Bond list input
<LI><TT><B><A HREF="#REGE">REGE</A></B> 11.05 </TT>Regenerate from
torsion angles
<LI><TT><B><A HREF="#RFCR">RFCR</A></B> 05.14 </TT>Reaction-field
correction
<LI><TT><B><A HREF="#RFSL">RFSL</A></B> 10.05 </TT>Set first shell
radii
<LI><TT><B><A HREF="#RMCK">RMCK</A></B> 10.05 </TT>Checkpoint file
removal
<LI><TT><B><A HREF="#RMOD">RMOD</A></B> 10.06 </TT>First shell
radius modifications
<LI><TT><B><A HREF="#RNDG">RNDG</A></B> 06.19 </TT>Random number
generator information
<LI><TT><B><A HREF="#RTIM">RTIM</A></B> 10.18 </TT>Solvent
residence time
<LI><TT><B><A HREF="#RUNS">RUNS</A></B> 07.01 </TT>Run MC
simulation
<LI><TT><B><A HREF="#SAMP">SAMP</A></B> 06.02 </TT>Sampling
technique
<LI><TT><B><A HREF="#SANN">SANN</A></B> 02.19 </TT>Simulated annealing
<LI><TT><B><A HREF="#SACP">SACP</A></B> 02.30 </TT>Simulated annealing
of the chemical potential
<LI><TT><B><A HREF="#SCAL">SCAL</A></B> 02.13 </TT>Scale simulation
cell
<LI><TT><B><A HREF="#SCAN">SCAN</A></B> 07.02 </TT>Scan trajectory
for proximity analysis
<LI><TT><B><A HREF="#SCKP">SCKP</A></B> 01.04 </TT>Archive checkpoint
files
<LI><TT><B><A HREF="#SCRM">SCRM</A></B> 02.12 </TT>Scramble torsion angles
<LI><TT><B><A HREF="#SEED">SEED</A></B> 06.11 </TT>Random number
seed input
<LI><TT><B><A HREF="#SENS">SENS</A></B> 09.05 </TT>Sensitivity
analysis
<LI><TT><B><A HREF="#SETC">SETC</A></B> 05.14 </TT>Set various constants
<LI><TT><B><A HREF="#SKWT">SKWT</A></B> 06.17 </TT>Skewed torsion sampling
<LI><TT><B><A HREF="#SLFT">SLFT</A></B> 11.02 </TT>Self test
<LI><TT><B><A HREF="#SLTA">SLTA</A></B> 04.01 </TT>Solute
description
<LI><TT><B><A HREF="#SLVA">SLVA</A></B> 04.02 </TT>Solvent
description
<LI><TT><B><A HREF="#SPPS">SPPS</A></B> 05.17 </TT>Special potential
<LI><TT><B><A HREF="#SPRD">SPRD</A></B> 02.15 </TT>Spread out
solute molecules
<LI><TT><B><A HREF="#SPST">SPST</A></B> 05.16 </TT>Field-dependent
potentials
<LI><TT><B><A HREF="#STEP">STEP</A></B> 06.04 </TT>Stepsizes
<LI><TT><B><A HREF="#STIR">STIR</A></B> 01.13 </TT>Contact elimination
<LI><TT><B><A HREF="#STOL">STOL</A></B> 11.03 </TT>Self test
<LI><TT><B><A HREF="#STOP">STOP</A></B> 07.05 </TT>Stop the run
<LI><TT><B><A HREF="#STPS">STPS</A></B> 06.05 </TT>Stepsize scaling
<LI><TT><B><A HREF="#STPX">STPX</A></B> 02.06 </TT>Stop GCE run at
nx
<LI><TT><B><A HREF="#STSC">STSC</A></B> 06.14 </TT>Temporary stepsize scaling
<LI><TT><B><A HREF="#STUN">STUN</A></B> 02.17 </TT>Stepsize tuning
<LI><TT><B><A HREF="#STVG">STVG</A></B> 09.12 </TT>Estimate solute
volume
<LI><TT><B><A HREF="#SUPI">SUPI</A></B> 09.24 </TT>Trajectory superimposition
<LI><TT><B><A HREF="#SUPT">SUPT</A></B> 05.01 </TT>Solute potential
type
<LI><TT><B><A HREF="#SUVC">SUVC</A></B> 05.08 </TT>Solute-solvent
cutoff
<LI><TT><B><A HREF="#SUUC">SUUC</A></B> 05.09 </TT>Solute-solute
cutoff
<LI><TT><B><A HREF="#SVDP">SVDP</A></B> 09.25 </TT>Solvent dipole distribution
<LI><TT><B><A HREF="#SVIN">SVIN</A></B> 05.19 </TT>Solvation parameter input
<LI><TT><B><A HREF="#SVPT">SVPT</A></B> 05.02 </TT>Solvent
potential type
<LI><TT><B><A HREF="#SVVC">SVVC</A></B> 05.10 </TT>Solvent-solvent
cutoff
<LI><TT><B><A HREF="#SWAP">SWAP</A></B> 06.07 </TT>Solute molecule
swap
<LI><TT><B><A HREF="#TAC0">TAC0</A></B> 02.18 </TT>Stepsize accumulator reset
<LI><TT><B><A HREF="#TAND">TAND</A></B> 09.20 </TT>Torsion angle
distribution
<LI><TT><B><A HREF="#TAUC">TAUC</A></B> 10.20 </TT>Torsion angle autocorrelation
<LI><TT><B><A HREF="#TEMP">TEMP</A></B> 02.02 </TT>Simulation
temperature
<LI><TT><B><A HREF="#TITL">TITL</A></B> 02.03 </TT>Description
<LI><TT><B><A HREF="#TIQU">TIQU</A></B> 03.05 </TT>TI quadrature
<LI><TT><B><A HREF="#TORD">TORD</A></B> 04.10 </TT>Torsion angle
definitions
<LI><TT><B><A HREF="#TORT">TORT</A></B> 09.21 </TT>Energy analysis
along a torsion angle
<LI><TT><B><A HREF="#TRAJ">TRAJ</A></B> 01.04 </TT>History file
specification
<LI><TT><B><A HREF="#VOLE">VOLE</A></B> 10.16 </TT>Volume element
regeneration
<LI><TT><B><A HREF="#VORO">VORO</A></B> 10.17 </TT>Voronoi analysis
<LI><TT><B><A HREF="#VVNE">VVNE</A></B> 05.24 </TT>Solvent-solvent electrostatic
<LI><TT><B><A HREF="#WCKP">WCKP</A></B> 01.03 </TT>Write checkpoint
file
<LI><TT><B><A HREF="#WCNF">WCNF</A></B> 01.02 </TT>Configuration
save
<LI><TT><B><A HREF="#WMAT">WMAT</A></B> 03.03 </TT>PMF matching
<LI><TT><B><A HREF="#WMAT">WPLT</A></B> 03.04 </TT>AUS iteration plot
<LI><TT><B><A HREF="#WTRA">WTRA</A></B> 01.12 </TT>Trajectory
conversions
</UL>
<!-- HardPage --><p>
<A NAME = "Examples">IV.</A> EXAMPLES
<p>
In this section a number of commented input files are presented
illustrating several different options.
<OL>
<LI><A HREF="#Exa_GLY">Glycine in water - minimal input; restart from checkpoint
<LI><A HREF="#Exa_ETH">Ethanol in ethanol (neat liquid)</A><BR>
<LI><A HREF="#Exa_GCE">Grand-canonical ensemble</A><BR>
<LI><A HREF="#Exa_TPN">Constant pressure ensemble</A><BR>
<LI><A HREF="#Exa_DMP">Dimethylphosphate-Sodium complex with
freely moving sodium</A><BR>
<LI><A HREF="#Exa_TOR">Torsion angle sampling</A><BR>
<LI><A HREF="#Exa_PMF">Potential of mean force between dimethyl
phosphate and
sodium ions with adaptive umbrella sampling</A><BR>
<LI><A HREF="#Exa_CAT">Creation/annihilation polynomial path
thermodynamic integration</A><BR>
<LI><A HREF="#Exa_FDT">Finite difference thermodynamic
integration</A><BR>
<LI><A HREF="#Exa_WID">Widom insertion method</A><BR>
<LI><A HREF="#Exa_PXA">Proximity analysis, trajectory
filtering</A><BR>
<LI><A HREF="#Exa_FLT">Conformation filtering</A><BR>
<LI><A HREF="#Exa_GNS">Generic site calculation</A><BR>
<LI><A HREF="#Exa_PHS">Primary hydration shell calculation</A><BR>
<LI><A HREF="#Exa_CAV">Cavity and pocket determination</A><BR>
</OL>
<p>
<A NAME="Exa_GLY"></A>
IV.1. <B>Glycine in water - minimal input; restart from checkpoint file</B>
<P>
This is a simple canonical ensemble MC run of a simple solute in
water.
<pre>
<A HREF="#FILE">FILE</A> glycine ! File names start with glycine.
! Run # = 1
<A HREF="#NSLV">NSLV</A> 215      ! One glycine and 215 waters
<A HREF="#TEMP">TEMP</A> 298.0     ! 298 Kelvin
<A HREF="#SVVC">SVVC</A> SPCC 7.75 ! Solvent-solvent cutoff=7.75 
<A HREF="#SUVC">SUVC</A> MICC 0.0  ! MI on the solute
<A HREF="#PBCN">PBCN</A> FCC  14.81526   ! Face-centered cubic cell
<A HREF="#STEP">STEP</A> 0.15 10.0 0.55 40.0 40
! Solute and solvent stepsizes, slt move frequency
<A HREF="#SUPT">SUPT</A> AM94   ! Solute-solvent potential is AMBER
<A HREF="#SVPT">SVPT</A> TIP4 TIP4!Solvent-solvent potential: TIP4P
<A HREF="#SLTA">SLTA</A> SMPL MMC READ  10       ! 10 solute atoms
   35     2.131    -1.202    -0.377    -.278623
   34     2.096     1.058     0.348    -.260774
   33    -0.549    -1.370     0.0      -.385556
    1     1.526     0.0       0.0       .295148
   18     0.0       0.0       0.0      -.040730
   20    -1.602    -1.374     0.044     .147624
   20    -0.249    -1.903    -0.836     .158476
   22    -0.351     0.514    -0.895     .061766
   22    -0.381     0.536     0.868     .086565
   23     3.068    -0.947    -0.337     .216103
<A HREF="#CNFG">CNFG</A> RANC ASCI SIZE !Random initial configuration
! Run 100000 steps
<A HREF="#RUNS">RUNS</A> 100000 10000 25000 0100000  100000  100000
! Run  an other 100000 steps, starting from the end of the first run.
! Run # = 2
<A HREF="#RUNS">RUNS</A> 100000 10000 25000 0100000  100000  100000
<A HREF="#STOP">STOP</A>
</pre>
Click <A HREF="https://mezeim01.u.hpc.mssm.edu/mmc/examples/exa1.out.html"
TARGET=OUTW>HERE</A> to view the output of this run
<P>
Files needed for run: none<BR>
Files created by the run:
<UL>
<LI>glycine.crd: initial configuration; <BR>
<LI>glycine.2.crd: final configuration after the first run
<LI>glycine.3.crd: final configuration after the last  run
<LI>glycine.ckp: checkpoint file after the first run ;<BR>
<LI>glycine.2.ckp: checkpoint file after the last run ;<BR>
</UL>
Suppose we decide to extend the run for an other 100000 steps.
It can be done by reading in the checkpoint file created after the
run:
<pre>
! Extend the previus run with an other 100000 steps
<A HREF="#FILE">FILE</A> glycine 2! Set Run # to 2
<A HREF="#RCKP">RCKP</A> ! Restore the status from the checkpoint file
! glycine.2.ckp
<A HREF="#RUNS">RUNS</A> 100000 10000 25000 0100000  100000  100000
<A HREF="#STOP">STOP</A>
</pre>
Click <A HREF="https://mezeim01.u.hpc.mssm.edu/mmc/examples/exa0.out.html"
TARGET=OUTW>HERE</A> to view the output of this run
<P>
Files needed by the run:
<UL>
<LI>glycine.2.crd: starting configuration of the second run
<LI>glycine.2.ckp: checkpoint file after the second run<BR>
</UL>
Files modified by the run:
<UL>
<LI>glycine.3.crd: final configuration after the last run
<LI>glycine.2.ckp: checkpoint file after the last run<BR>
</UL>
<!-- HardPage --><p>
<A NAME="Exa_ETH"></A>
IV.2. <B>Ethanol in ethanol (neat liquid)</B>.
<P>
This input runs a MC simulation in liquid ethanol.
One ethanol molecule
is considered the 'solute'
<pre>
<A HREF="#FILE">FILE</A> etnl
<A HREF="#PRNT">PRNT</A> ECHO       ! Echo formatted input
<A HREF="#TITL">TITL</A> Ethanol
<A HREF="#HRDW">HRDW</A> SCLR       ! Scalar machine
<A HREF="#SVVC">SVVC</A> SPCC 10.0  ! Solvent-solvent cutoff
<A HREF="#SUVC">SUVC</A> SPGC 10.0  ! Solute-solvent cutoff
<A HREF="#PBCN">PBCN</A> RECT 26.0
! Unit cell is a cube with 26  egdes
<A HREF="#NSLV">NSLV</A> 181     ! One solute and 181 solvents
<A HREF="#TEMP">TEMP</A> 298.0   ! Simulation temperature: 298 K
<A HREF="#MOVE">MOVE</A> SHCY
! Molecules are selected by the shuffled cyclic scheme
<A HREF="#STEP">STEP</A>  0.50 30.0   0.50  30.0 40
! Solute and solvent stepsize params
<A HREF="#SUPT">SUPT</A> AM94        ! AMBER solute
<A HREF="#SVPT">SVPT</A> GENL AM94 ! General solvent, AMBER library
<A HREF="#SAMP">SAMP</A> METC        ! Metroplis sampling
<A HREF="#SLVA">SLVA</A> 9      ! Read 9 solvent atoms
   18    3.108016  0.653187 -8.526237 -0.198000
   22    2.814817 -0.348721 -8.761003  0.066000
   22    2.517394  1.015322 -7.710808  0.066000
   22    2.956134  1.278341 -9.381230  0.066000
   18    4.596542  0.674197 -8.131964  0.131000
   22    4.748424  0.049042 -7.276970  0.066000
   22    5.187164  0.312062 -8.947392  0.066000
   36    4.988388  2.013194 -7.818209 -0.566000
   23    5.982338  2.026292 -7.548412  0.303000
<A HREF="#SLTA">SLTA</A>  SMPL MMC READ 9   !read 9 solute atoms
   18    3.108016  0.653187 -8.526237 -0.198000    1
   22    2.814817 -0.348721 -8.761003  0.066000    1
   22    2.517394  1.015322 -7.710808  0.066000    1
   22    2.956134  1.278341 -9.381230  0.066000    1
   18    4.596542  0.674197 -8.131964  0.131000    1
   22    4.748424  0.049042 -7.276970  0.066000    1
   22    5.187164  0.312062 -8.947392  0.066000    1
   36    4.988388  2.013194 -7.818209 -0.566000    1
   23    5.982338  2.026292 -7.548412  0.303000    1
<A HREF="#DSTC">DSTC</A> NONE ! No distribution function calc.
<A HREF="#CNFG">CNFG</A> READ ASCI NOFX
! Read initial coordinates in ASCII from etnl.crd
<A HREF="#RUNS">RUNS</A> 1000000 10000 500000 100000 10000
!Run 1M steps
<A HREF="#RUNS">RUNS</A> 1000000 10000 500000 100000 10000
!The second run will start at the end of the first.
<A HREF="#STOP">STOP</A>
</pre>
Click <A HREF="https://mezeim01.u.hpc.mssm.edu/mmc/examples/exa2.out.html"
TARGET=OUTW>HERE</A> to view the output of this run
<P>
Files needed for run: <BR>
etnl.crd: initial configuration.<BR>
Files created by the run:
<UL>
<LI>etnl.ckp: checkpoint file after the first run ;<BR>
<LI>etnl.2.ckp: checkpoint file after the second run;<BR>
<LI>etnl.2.crd: final configuration after the first run;<BR>
<LI>etnl.3.crd: final configuration after the second run.<BR>
</UL>
<!-- HardPage --><p>
<A NAME="Exa_GCE"></A>
IV.3. <B>Grand-canonical ensemble</B>
<P>
This input runs MC simulation in the grand-canonical ensemble.
The solute is the protein BPTI
(from the <A HREF="https://www.rcsb.org/pdb/">PDB</A>).
<pre>
<A HREF="#FILE">FILE</A> bpti
<A HREF="#PRNT">PRNT</A> DETL !Echo formatted data, print extra inf
<A HREF="#HRDW">HRDW</A> VC32          ! 32-bit vector
<A HREF="#SVVC">SVVC</A> SPCC 7.75     ! Solvent cutoff
<A HREF="#SUVC">SUVC</A> MIGC 0.0      ! MI on the solute
<A HREF="#PBCN">PBCN</A> RECT 24.87  40.39 66.2 !Rectangular PBC
<A HREF="#MOVE">MOVE</A> RAND          ! Random selections
<A HREF="#TEMP">TEMP</A> 298
<A HREF="#NSLV">NSLV</A> 10   !10 solvent molecules to start with
<A HREF="#STEP">STEP</A>  0.00   00.0   0.55    40.0  60
! slt, slv stepsize, slt move freq.
<A HREF="#SVPT">SVPT</A> TIP4 TIP4
<A HREF="#SUPT">SUPT</A> AM94
<A HREF="#SAMP">SAMP</A> FBSC 0.5  ! Scaled force-biased sampling
    2
 4.00   0.2   7.0     1.0
<A HREF="#PMOD">PMOD</A> AM94 1
   75   27
    8            1.6     0.2
<A HREF="#SLTA">SLTA</A>  SMPL MMC READ 892
 N3      -2.55400   7.05800  12.80100  -0.37640    1  ARG N
   .....
   .....   891 more lines describing solute atoms ....
   .....
<A HREF="#DSTC">DSTC</A> NONE
<A HREF="#GCEN">GCEN</A> CAVB RSIG ALTI
! Grand-canonical ens., cavity biased,
! LJ sigma-based cavity radii, alternating i/d
    1.0    2.5      1000.0       30   30   30 1000    1    1   0000100     50000
<A HREF="#CNFG">CNFG</A> RANC ASCI SIZE
!Random configuration of 10 (from <A HREF="#NSLV">NSLV</A>) waters
! using the given cell size and writing ASCII file
<A HREF="#RUNS">RUNS</A> 10000  100 25000 100000 100000 100000
<A HREF="#STOP">STOP</A>
</pre>
Click <A HREF="https://mezeim01.u.hpc.mssm.edu/mmc/examples/exa3.out.html"
TARGET=OUTW>HERE</A> to view the output of this run
<P>
Files needed for run: none.<BR>
Files created by the run:
<UL>
<LI>bpti.ckp: checkpoint file after the run;<BR>
<LI>bpti.crd: initial configuration.<BR>
<LI>bpti.2.crd: final configuration after the run.<BR>
</UL>
<!-- HardPage --><p>
<A NAME="Exa_TPN"></A>
IV.4. <B>Constant pressure ensemble run</B>
<p>
This input runs MC simulation in the isothermal-isobaric ensemble.
The solute is the protein a Na+ - DMP- ion pair.
<pre>
<A HREF="#FILE">FILE</A> dmpna
<A HREF="#TITL">TITL</A> run 2 slt molecs in water, only moves
<A HREF="#TITL">TITL</A> NA+DMP-: KOLLMAN-STRAATSMA(NA+)
<A HREF="#HRDW">HRDW</A> VC32          ! 32-bit vector
<A HREF="#SVVC">SVVC</A> SPCC 7.75     ! Solvent cutoff
<A HREF="#SUVC">SUVC</A> MIGC 0.0      ! MI on the solute
<A HREF="#PBCN">PBCN</A> RECT 32.031  21.354 21.354
<A HREF="#TEMP">TEMP</A> 298
<A HREF="#NSLV">NSLV</A> 484
<A HREF="#STEP">STEP</A>  0.10   00.0   0.55    40.0 05
<A HREF="#SVPT">SVPT</A> TIP3 TIP3
<A HREF="#SUPT">SUPT</A> AM94
<A HREF="#SAMP">SAMP</A> FBSC 0.5  ! Scaled force-biased sampling
    2
 4.00   0.2   7.0     1.0
<A HREF="#PMOD">PMOD</A> AM94 1   ! Potential library modification
   49   46
   11  1.59955E+00     0.478833E-01
<A HREF="#SLTA">SLTA</A>  SMPL MMC READ 8
   49     2.70000   0.0       0.00000   1.00000    1
   38     0.0       0.0       0.0       0.91200    4
   35     0.74000   0.0       1.28171  -0.65500    4
   35     0.74000   0.0      -1.28171  -0.65500    4
   37    -0.99164  -1.25560   0.00000  -0.41000    4
   37    -0.99164   1.25560   0.00000  -0.41000    4
    4    -1.92075   1.43300   1.07251   0.10900    4
    4    -1.92075  -1.43300  -1.07251   0.10900    4
<A HREF="#IBEN">IBEN</A> UNIF ISOT 1.0 250.0 500 100 !(TPN) ensemble
<A HREF="#CNFG">CNFG</A> READ ASCI
<A HREF="#RUNS">RUNS</A> 2000000 50000 1000000 0100000 100000
<A HREF="#STOP">STOP</A> SLFT
</pre>
Click <A HREF="https://mezeim01.u.hpc.mssm.edu/mmc/examples/exa4.out.html"
TARGET=OUTW>HERE</A> to view the output of this run
<P>
Files needed for run:
<UL>
<LI>dmpna.crd: initial configuration.<BR>
</UL>
Files created by the run:
<UL>
<LI>dmpna.ckp: checkpoint file after the run;<BR>
<LI>dmpna.crd: initial configuration.<BR>
<LI>dmpna.2.crd: final configuration after the run.<BR>
</UL>
<!-- HardPage --><p>
<A NAME="Exa_DMP"></A>
IV.5. <B>Dimethylphosphate-Sodium complex with freely moving
sodium</B>
<p>
This input show how to allow tranlational/rotational degrees of
freedom for molecules considered 'solute' by the program.
<pre>
<A HREF="#FILE">FILE</A> dmpna
<A HREF="#TITL">TITL</A> Na+DMP-: Kollman-Straatsma(Na+)
TITL SPC water. H-R PBC
<A HREF="#HRDW">HRDW</A> VC32          ! 32-bit vector algorithm
<A HREF="#SVVC">SVVC</A> SPCC 7.75     ! Solvent spherical cutoff
<A HREF="#SUVC">SUVC</A> MIGC 0.0
! MI on the solute based on group center distances
<A HREF="#SUUC">SUUC</A> MIMC          ! Molec-based MI for intrasolute
<A HREF="#PBCN">PBCN</A> RECT 32.031  21.354 21.354
! Rectangular PBC, long box
<A HREF="#TEMP">TEMP</A> 298   |  <A HREF="#NSLV">NSLV</A> 484
!Simulation at 298 Kelvin with 485 solvent molecules
<A HREF="#STEP">STEP</A> 0.1 0.0 0.55 40.0 10
! 0 slt stepsize, 0.55 &#197;, 40 deg solvent steps
<A HREF="#SVPT">SVPT</A> TIP3 TIP3 ! Solvent-solvent pot: TIP3P
<A HREF="#SUPT">SUPT</A> AM94      ! Solute-solvent pot: AMBER
<A HREF="#SAMP">SAMP</A> FBSC 0.5  ! Scaled force-biased sampling
    2
 4.00   0.2   7.0     1.0
<A HREF="#MOVE">MOVE</A> PRSP NRST
! Preferential sampling based on nearest slt atom dist
    3
 3.00   3.5   4.0    03.0    7.0    1.0
<A HREF="#PMOD">PMOD</A> AM94 1    ! Potential library modification
   49   46
   11  1.59955E+00     0.478833E-01
! Solute's "own" solvent will be used for slt-slv
<A HREF="#SLTA">SLTA</A>  SMPL MMC READ 8
! 8 solute atom - regular solute, read from input stream
   38     0.0       0.0       0.0       0.91200    1
   35     0.74000   0.0       1.28171  -0.65500    1
   35     0.74000   0.0      -1.28171  -0.65500    1
   37    -0.99164  -1.25560   0.00000  -0.41000    1
   37    -0.99164   1.25560   0.00000  -0.41000    1
    4    -1.92075   1.43300   1.07251   0.10900    1
    4    -1.92075  -1.43300  -1.07251   0.10900    1
   49     2.70000   0.0      -0.00000   1.00000    2
<A HREF="#PARD">PARD</A> UNIF SHCY 1.0 0.3 0.0 1.0 0.3 0.0 ~
1.0 0.3 0.0 1.0 -1 ! Last slt group is independently moved
<A HREF="#DSTC">DSTC</A> NONE   ! No distribution function calc
<A HREF="#SLFT">SLFT</A> BASC   ! Basic self tests performed
<A HREF="#CNFG">CNFG</A> READ ASCI ! Input from dmpna.crd in ASCII
<A HREF="#RUNS">RUNS</A> 2000000 50000 1000000 0100000 100000
! Run 2M steps
<A HREF="#STOP">STOP</A>
</pre>
Click <A HREF="https://mezeim01.u.hpc.mssm.edu/mmc/examples/exa5.out.html"
TARGET=OUTW>HERE</A> to view the output of this run
<p>
Files needed for run: <BR>
dmpna.crd: initial configuration.<BR>
Files created by the run:
<UL>
<LI>dmpna.ckp: checkpoint file after the first run;<BR>
<LI>dmpna.2.crd: final configuration after the run.<BR>
</UL>
<!-- HardPage --><p>
<A NAME="Exa_TOR"></A>
IV.6. <B>Torsion angle sampling</B>
<p>
This input show how to sample torsional degrees of freedom
for molecules considered 'solute' by the program.
<pre>
<A HREF="#FILE">FILE</A> aa3
<A HREF="#TITL">TITL</A> Three amino acids in vacuum
<A HREF="#TITL">TITL</A> Solute molecules move, swap and torsions
<A HREF="#HRDW">HRDW</A> VC32          ! 32-bit vector
<A HREF="#SVVC">SVVC</A> SPCC 12.00    ! Solvent cutoff
<A HREF="#SUVC">SUVC</A> MIGC 0.0      ! MI on the solute
<A HREF="#SUUC">SUUC</A> MIMC          ! Molec-based MI for intrasolute
<A HREF="#PBCN">PBCN</A> HEXG 99.00  99.00  99.00    !hexagonal PBC
<A HREF="#TEMP">TEMP</A> 998
<A HREF="#NSLV">NSLV</A> 0   ! Only solute
<A HREF="#STEP">STEP</A>  0.00   00.0   0.55    40.0 10
<A HREF="#SVPT">SVPT</A> TIP3 TIP3 ! Solvent-solvent pot: CHARMM
<A HREF="#SUPT">SUPT</A> CHRM
<A HREF="#SLTA">SLTA</A>  SMPL MMC READ 34
 NH1      1.24900   2.31500  -0.79400  -0.47000    1  ILE N
 CT1      1.55500   3.73000  -0.91700   0.07000    1  ILE CA
 C        1.31700   4.34000   0.45300   0.51000    1  ILE C
 O        2.09800   5.17100   0.91200  -0.51000    1  ILE O
 CT1      0.73100   4.43400  -2.05000  -0.09000    1  ILE CB
 H        0.48500   2.01200  -1.35900   0.31000    1  ILE HN
 HB       2.61200   3.83400  -1.12700   0.09000    1  ILE HA
 HA      -0.33600   4.20200  -1.80500   0.09000    1  ILE HB
 NH1      1.49900   2.78700   3.28800  -0.47000    2  LEU N
 CT1      2.55800   2.23900   4.11700   0.07000    2  LEU CA
 C        3.78000   3.10300   3.86100   0.51000    2  LEU C
 O        4.51100   3.44400   4.79000  -0.51000    2  LEU O
 CT2      2.81700   0.74400   3.78200  -0.18000    2  LEU CB
 CT1      3.92200   0.01400   4.58300  -0.09000    2  LEU CG
 CT3      4.04200  -1.44600   4.12400  -0.27000    2  LEU CD1
 CT3      3.68900   0.07700   6.10000  -0.27000    2  LEU CD2
 H        1.16000   2.16100   2.58800   0.31000    2  LEU HN
 HB       2.28700   2.35700   5.15800   0.09000    2  LEU HA
 HA       1.86200   0.19700   3.95500   0.09000    2  LEU HB1
 HA       3.03300   0.64900   2.69400   0.09000    2  LEU HB2
 HA       4.90300   0.49900   4.36300   0.09000    2  LEU HG
 HA       4.86700  -1.95800   4.66500   0.09000    2  LEU HD11
 HA       3.09800  -1.99800   4.32300   0.09000    2  LEU HD12
 HA       4.25600  -1.49900   3.03500   0.09000    2  LEU HD13
 HA       4.49200  -0.47200   6.63700   0.09000    2  LEU HD21
 HA       3.69100   1.12200   6.47100   0.09000    2  LEU HD22
 HA       2.71400  -0.38500   6.36400   0.09000    2  LEU HD23
 NH1      3.65600   6.57500   5.52300  -0.47000    3  GLY N
 CT2      3.90500   6.38700   6.93600  -0.02000    3  GLY CA
 C        5.38300   6.32500   7.16600   0.51000    3  GLY C
 O        5.89600   6.89700   8.12500  -0.51000    3  GLY O
 H        3.16200   5.83500   5.07000   0.31000    3  GLY HN
 HB       3.47900   5.44100   7.23900   0.09000    3  GLY HA1
 HB       3.51900   7.24100   7.47600   0.09000    3  GLY HA2
<A HREF="#PRMF">PRMF</A> CHRM par_all22_prot_na.inp
!Torsion pot param file
<A HREF="#PARD">PARD</A> UNIF RAND 1.0 0.5 30.0 1.0 0.5 30.0 ~
 1.0 0.5 30.0 1.0 34 ! Note the continuation character ~
<A HREF="#PART">PART</A> UNIF RAND RAND 1.0
!Torsion strategy and frequency
<A HREF="#TORD">TORD</A> INPT 7  !Definiton of active torsions
    5    2     180.0      30.0    1
    2    3     180.0      30.0    1
   10    9     180.0      30.0    2
   10   11     180.0      30.0    2
   14   15     180.0      30.0    3
   13   14     180.0      30.0    4
   14   16     180.0      30.0    4
<A HREF="#CNFG">CNFG</A> RANC ASCI
<A HREF="#SWAP">SWAP</A> 0.2 !Specify swapping frequency
<A HREF="#RUNS">RUNS</A> 100000 100  50000 10000 5000
<A HREF="#STOP">STOP</A> SLFT
</pre>
Click <A HREF="https://mezeim01.u.hpc.mssm.edu/mmc/examples/exa6.out.html"
TARGET=OUTW>HERE</A> to view the output of this run
<P>
Files needed for run:
<UL>
<LI>aa3.crd: initial configuration.<BR>
<LI>par_all22_prot_na.inp: torsion potential parameters<BR>
</UL>
Files created by the run:
<UL>
<LI>aa3.ckp: checkpoint file after the run;<BR>
<LI>aa3.crd: initial configuration.<BR>
<LI>aa3.2.crd: final configuration after the run.<BR>
</UL>
<!-- HardPage --><p>
<A NAME="Exa_PMF"></A>
IV.7. <B>Potential of mean force between dimethyl phosphate and
sodium ions with
<A HREF="#Mezei_87">adaptive umbrella sampling</A></B>
<p>
This input performs one type of
<A HREF="#Mezei_86_NYAS">free energy simulation</A>:
it calculates the potential of mean force between two solute
molecules using adaptive umbrella sampling.
<pre>
<A HREF="#FILE">FILE</A> dmpna_pmf     ! SPC water. H-R PBC
<A HREF="#TITL">TITL</A> Na+DMP-: Kollman-Straatsma(Na+),
<A HREF="#TITL">TITL</A> #1   4 &#197; range  MI/RC G/A  Run A
<A HREF="#HRDW">HRDW</A> VC32          ! 32-bit vector hardware
<A HREF="#SVVC">SVVC</A> SPCC 7.75     ! Solvent-solvent cutoff
<A HREF="#SUVC">SUVC</A> MIGC 0.0      ! MI on the solute
<A HREF="#PBCN">PBCN</A> RECT 32.031 21.354 21.354 !Rectangular PBC
<A HREF="#MOVE">MOVE</A> SHCY          ! Shuffled-cyclic selections
<A HREF="#TEMP">TEMP</A> 298   |  <A HREF="#NSLV">NSLV</A> 485
<A HREF="#STEP">STEP</A>  0.10 0.0 0.55  40.0  10
! 0.1 &#197;slt stepsize, no slt rotation, 0.55 &#197;,
!40 deg solvent steps; solute moves at every 10th MC step
<A HREF="#SVPT">SVPT</A> TIP3 TIP3 ! Solvent-solvent pot: TIP3P
<A HREF="#SUPT">SUPT</A> AM94      ! Solute-solvent pot: AMBER
<A HREF="#SAMP">SAMP</A> FBSC 0.5  ! Scaled force-biased sampling
    2
 4.00   0.2   7.0     1.0
<A HREF="#FREE5">FREE PMF1</A> WRMM GEOC AUSE 2
!2-part PMF both parts of the slt moves
                      0.00      0.25      0.050     100.0     0.1
    2    0    3    1   0.01     17.      0.0       0.0      0.00002    0.9
    4    5    5   05    2.43      0.75
<A HREF="#PMOD">PMOD</A> AM94 1    ! Potential library modification
   49   46
   11  1.59955E+00     0.478833E-01
<A HREF="#SLTA">SLTA</A>  SMPL MMC READ 24 16 8
!3*8 solute atom - regular solute, last 1 atom is in the 2nd part
   38     0.0       0.0       0.0       0.91200    1
   35     0.74000   0.0       1.28171  -0.65500    1
   35     0.74000   0.0      -1.28171  -0.65500    1
   37    -0.99164  -1.25560   0.00000  -0.41000    1
   37    -0.99164   1.25560   0.00000  -0.41000    1
    4    -1.92075   1.43300   1.07251   0.10900    1
    4    -1.92075  -1.43300  -1.07251   0.10900    1
   49     2.70000   0.0       0.00000   1.00000    2
   38     0.0       0.0       0.0       0.91200    1
   35     0.74000   0.0       1.28171  -0.65500    1
   35     0.74000   0.0      -1.28171  -0.65500    1
   37    -0.99164  -1.25560   0.00000  -0.41000    1
   37    -0.99164   1.25560   0.00000  -0.41000    1
    4    -1.92075   1.43300   1.07251   0.10900    1
    4    -1.92075  -1.43300  -1.07251   0.10900    1
   49     6.70000   0.0      -0.00000   1.00000    2
   38     0.0       0.0       0.0       0.91200    1
   35     0.74000   0.0       1.28171  -0.65500    1
   35     0.74000   0.0      -1.28171  -0.65500    1
   37    -0.99164  -1.25560   0.00000  -0.41000    1
   37    -0.99164   1.25560   0.00000  -0.41000    1
    4    -1.92075   1.43300   1.07251   0.10900    1
    4    -1.92075  -1.43300  -1.07251   0.10900    1
   49     2.70000   0.0       0.00000   1.00000    2
<A HREF="#DSTC">DSTC</A> NONE
<A HREF="#CNFG">CNFG</A> READ ASCI NOFX
! input from dmpna_pmf.crd in ASCII
<A HREF="#RUNS">RUNS</A> 5000000 20000 2500000 0  100000 200000
<A HREF="#STOP">STOP</A>
</pre>
Click <A HREF="https://mezeim01.u.hpc.mssm.edu/mmc/examples/exa7.out.html"
TARGET=OUTW>HERE</A> to view the output of this run
<P>
Files needed for run: <BR>
dmpna_pmf.crd: initial configuration.<BR>
Files created by the run:
<UL>
<LI>dmpna_pmf.ckp: checkpoint file after the first run;<BR>
<LI>dmpna_pmf.2.crd: final configuration after the run.<BR>
</UL>
<!-- HardPage --><p>
<A NAME="Exa_CAT"></A>
IV.8. <B> Creation/annihilation polynomial path
thermodynamic integration</B>
<P>
This is an example of a different
<A HREF="#Mezei_86_NYAS">free energy simulation</A>
type:
calculation the solvation excess Helmholtz free energy
of NO in water with polynomial thermodynamic integration
over a creation/annihilation path.
<PRE>
<A HREF="#FILE">FILE</A> NO_ti 10
<A HREF="#TITL">TITL</A> Excess Helmholtz free energy of solvation
of NO in water
<A HREF="#TITL">TITL</A> Calculated by three-point Gaussian
quadrature
<A HREF="#ECHO">PRNT</A> ECHO
<A HREF="#HRDW">HRDW</A> VC32          ! 32-bit vector
<A HREF="#SVVC">SVVC</A> MINI         ! Solvent cutoff
<A HREF="#SUVC">SUVC</A> MIGC 0.0      ! MI on the solute
<A HREF="#PBCN">PBCN</A> RECT 14.74    !Rectangular PBC
<A HREF="#MOVE">MOVE</A> RAND          ! Random selections
!Set creation/annihilation TI using the first quadrature point
!Lambda exponents: 4, 3, 2 (for 1/r^12, 1/r^6, 1/r terms, resp.)
<A HREF="#FREE2">FREE TICA</A>  NOMX
 4.0   3.0  2.0     0.112702
<A HREF="#TEMP">TEMP</A> 298
<A HREF="#NSLV">NSLV</A> 108
<A HREF="#STEP">STEP</A>  0.00   00.0   0.40    35.0 50
<A HREF="#SVPT">SVPT</A> TIP3 TIP3 ! Solv-solv potential is TIP3P
<A HREF="#SUPT">SUPT</A> CHRM
!Add NO parameters
<A HREF="#PMOD">PMOD</A> CHRM 2
  150   70    3.3    N/NO
    7 1.824            0.17
  151   70    3.3    O/NO
    8 1.751            0.159
<A HREF="#SAMP">SAMP</A> METC 0.5
!Make sure the dummy atoms in the first copy are treated as one molecule
<A HREF="#MOLD">MOLD</A> 2 2 4
!Solute has 4 atoms, 4 free energy atoms, and the first copy has 2 atoms
<A HREF="#">SLTA</A> SMPL MMC READ 4 4 2    !Dummy solute
 DUM      -.570      000.0    0.00000   0.00000    1
 DUM      0.58       000.0    0.00000   0.00000    1
  150     0.57000    000.0    0.00000   0.02800    2
  151     -.58000    000.0    0.00000  -0.02800    2
<A HREF="#DSTC">DSTC</A> NONE
<A HREF="#CNFG">CNFG</A> READ ASCI NOFX
<A HREF="#FIXD">FIXD</A> 2500000
!Equilibration
<A HREF="#RUNS">RUNS</A> 2000000 100000 100000 500000 100000
<A HREF="#RMCK">RMCK</A> !Remove unneded checkpoint file
!Production run, runnumber=11
<A HREF="#RUNS">RUNS</A> 10000000 100000 100000 500000 100000
<A HREF="#FILE">FILE</A>  NO_ti  20
!Set lambda to the 2nd quadrature points
<A HREF="#FREE2">FREE TICA</A>  NOMX
 4.0   3.0  2.0     0.500000
<A HREF="#CNFG">CNFG</A> READ ASCI NOFX  12
<A HREF="#RUNS">RUNS</A> 2000000 100000 100000 500000 100000
<A HREF="#RMCK">RMCK</A> !Remove unneded checkpoint file
!Production run, runnumber=21
<A HREF="#RUNS">RUNS</A> 10000000 100000 100000 500000 100000
<A HREF="#FILE">FILE</A>  NO_ti  30
!Set lambda to the 3rd quadrature points
<A HREF="#FREE2">FREE TICA</A>  NOMX
 4.0   3.0  2.0     0.887298
<A HREF="#CNFG">CNFG</A> READ ASCI NOFX  22
<A HREF="#RUNS">RUNS</A> 2000000 100000 100000 500000 100000 00000
<A HREF="#RMCK">RMCK</A> !Remove unneded checkpoint file
!Production run, runnumber=31
<A HREF="#RUNS">RUNS</A> 10000000 100000 100000 500000 100000
!Evaluate the quadrature from the data on the three checkpoint files
<A HREF="#TIQU">TIQU</A> REGL ALL  3 11 10
    0  001    0
<A HREF="#STOP">STOP</A> SLFT
</PRE>
Click <A HREF="https://mezeim01.u.hpc.mssm.edu/mmc/examples/exa8.out.html"
TARGET=OUTW>HERE</A> to view the output of this run
<P>
Files needed for run:
<UL>
<LI>NO_ti.10.crd: initial configuration.<BR>
</UL>
Files created by the run:
<UL>
<LI>NO_ti.11.crd: configuration after equilibration<BR>
<LI>NO_ti.11.ckp: checkpoint file with the data for the 1st
quadrature point<BR>
<LI>NO_ti.12.crd: configuration after 1st quadrature run<BR>
<LI>NO_ti.21.crd: configuration after equilibration<BR>
<LI>NO_ti.22.crd: configuration after 2nd quadrature run<BR>
<LI>NO_ti.21.ckp: checkpoint file with the data for the 2nd
quadrature point<BR>
<LI>NO_ti.31.crd: configuration after equilibration<BR>
<LI>NO_ti.32.crd: configuration after 3rd quadrature run<BR>
<LI>NO_ti.31.ckp: checkpoint file with the data for the 3rd
quadrature point<BR>
</UL>
<!-- HardPage --><p>
<A NAME="Exa_FDT"></A>
IV.9. <B>Finite difference thermodynamic integration</B>
<p>
This is a example combines an other two
<A HREF="#Mezei_86_NYAS">free energy simulation</A>
types.
It calculates the Helmholtz free energy difference between ethanol
and
acetone in water, using the
<A HREF="#Mezei_86_FDTI">finite differenece thermdynamic
integration</A>.
The integrand at each quadrature point is
obtained by the popular perturbation method.
<pre>
<A HREF="#FILE">FILE</A> pm 1
<A HREF="#TITL">TITL</A> Perturbation method test ethane -&gt;
acetaldehyde
<A HREF="#TITL">TITL</A> 3-point Gaussian quadrature (probably
inadequate)
<A HREF="#HRDW">HRDW</A> VC32
<A HREF="#SVVC">SVVC</A> SPCC 10.0
<A HREF="#SUVC">SUVC</A> MICC
<A HREF="#PBCN">PBCN</A> FCC    19.57563
<A HREF="#TEMP">TEMP</A> 298 | <A HREF="#NSLV">NSLV</A> 500
<A HREF="#STEP">STEP</A>    0.10   10.0  0.55      30.0  30
<A HREF="#SVPT">SVPT</A> TIP3 TIP3
<A HREF="#SUPT">SUPT</A> CHRM
!Lambda=0.112702 run
<A HREF="#FREE3">FREE PMLI</A> CCMX
                    0.112702  0.102702  0.122702         0.0      1.0
<A HREF="#SLTA">SLTA</A> SMPL MMC FILE 16 16 8 16 1
<A HREF="#CNFG">CNFG</A> READ ASCI NOFX 50
!Equilibrate
<A HREF="#RUNS">RUNS</A> 1000000 100000 10000 1000000 100000
<A HREF="#RMCK">RMCK</A> !Remove equilibration checkpoint file to
save disk space
!Production run
<A HREF="#RUNS">RUNS</A> 5000000 100000 10000 1000000 100000
!New coupling parameter - lambda=0.5
<A HREF="#FREE3">FREE PMLI</A> CCMX
                    0.50      0.49      0.51        0.102702 0.122702
!Restore solute to original initial and final state
<A HREF="#SLTA">SLTA</A> SMPL MMC FILE 16 16 8 16 1
!Read starting config from disk to prepare the new
!initial and final states
<A HREF="#CNFG">CNFG</A> READ ASCI NOFX
!Equilibrate
<A HREF="#RUNS">RUNS</A> 1000000 100000 10000 1000000 100000
<A HREF="#RMCK">RMCK</A>
!Production run
<A HREF="#RUNS">RUNS</A> 5000000 100000 10000 1000000 100000
!New coupling parameter - lambda=0.887298
<A HREF="#FREE3">FREE PMLI</A> CCMX
                    0.887298  0.877298  0.897298    0.49      0.51
!Restore solute to original initial and final state
<A HREF="#SLTA">SLTA</A> SMPL MMC FILE 16 16 8 16 1
<A HREF="#CNFG">CNFG</A> READ ASCI NOFX
!Equilibrate
<A HREF="#RUNS">RUNS</A> 1000000 100000 10000 1000000 100000
<A HREF="#RMCK">RMCK</A>
!Production run
<A HREF="#RUNS">RUNS</A> 5000000 100000 10000 1000000 100000
!Evaluate the TI quadrature
<A HREF="#TIQU">TIQU</A> 2 2
    3    0    1    0    0    0
<A HREF="#STOP">STOP</A> SLFT FULL
</pre>
Click <A HREF="https://mezeim01.u.hpc.mssm.edu/mmc/examples/exa9.out.html"
TARGET=OUTW>HERE</A> to view the output of this run
<P>
Files needed for run:
<UL>
<LI>pm.crd: initial configuration.<BR>
<LI>pm.slt: description of the two solutes.<BR>
</UL>
Files created by the run:
<UL>
<LI>pm.crd: initial configuration.<BR>
<LI>pm.2.ckp, pm.4.ckp, pm.6.ckp: checkpoint files at the
quadrature points.
<LI>pm.2.crd, pm.3.crd, pm.4.crd, pm.5.crd, pm.6.crd, pm.7.crd:
Coordinate files at the end of equilibration and production
runs.
</UL>
<!-- HardPage --><p>
<A NAME="Exa_WID"></A>
IV.10. <B>Widom insertion method</B>
<p>
This input shows how to run a Widom insertion calculation
based on a pure water trajectory run earlier by CHARMM.
<pre>
<A HREF="#FILE">FILE</A> methane
<A HREF="#TITL">TITL</A> Calculate the chemical potential of Methane in Water
<A HREF="#TITL">TITL</A> using Widom cavity insertion method
<A HREF="#HRDW">HRDW</A> VC32          ! 32-bit vector
<A HREF="#SVVC">SVVC</A> MINI 10.0
<A HREF="#SUVC">SUVC</A> MIGC 10.0      ! MI on the solute
<A HREF="#PBCN">PBCN</A> RECT 15.5  15.5 15.5  !Rectangular PBC
<A HREF="#TEMP">TEMP</A> 300
<A HREF="#NSLV">NSLV</A> 124 !no of solvent molecs in the CHARMM run
<A HREF="#SVPT">SVPT</A> TIP4 TIP4 ! Solvent-solvent potential is TIP4P
<A HREF="#SUPT">SUPT</A> CHRM      !Solute-solvent potential is CHARMM
<A HREF="#FREE">FREE</A> WIDO 2.6 64 64 64 2 5
<A HREF="#SLTA">SLTA</A> SMPL MMC READ 5 5
 CT3     -0.13265   0.14073   0.03004  -0.36000    1  CSP3C1
 HA       0.84914   0.15299   0.45527   0.09000    1  CSP3H11
 HA      -0.35123  -0.84076  -0.33574   0.09000    1  CSP3H12
 HA      -0.84914   0.40992   0.77775   0.09000    1  CSP3H13
 HA      -0.18078   0.84077  -0.77776   0.09000    1  CSP3H14
<A HREF="#TRAJ">TRAJ</A> CHRM RGFX  !Open trajectory in CHARMM format
<A HREF="#SCAN">SCAN</A> TRAJ 1000 5 00 0 500 2 1  2 3 1
<A HREF="#STOP">STOP</A>
</pre>
Click <A HREF="https://mezeim01.u.hpc.mssm.edu/mmc/examples/exa10.out.html"
TARGET=OUTW>HERE</A> to view the output of this run
<P>
File needed for run:
<UL>
<LI>methane.hst: the CHARMM trajectory (DCD) file
</UL>
Files created by the run:
<UL>
<LI>methane.ckp: checkpoint files<BR>
<LI>methane.pxc: analysis checkpoint file
</UL>
<!-- HardPage --><p>
<A NAME="Exa_PXA"></A>
IV.11. <B>Proximity analysis, trajectory filtering</B>
<p>
This input show how to set up proximity analysis
of a trajectory given as a series of PDB files and
how to use this analysis to filter this trajectory file.
<pre>
<A HREF="#FILE">FILE</A> 5ht1
<A HREF="#TITL">TITL</A> GCE run with fixed solute
<A HREF="#HRDW">HRDW</A> VC32        ! 32-bit vector
<A HREF="#SVVC">SVVC</A> SPCC 7.75  ! Solvent-solvent cutoff
<A HREF="#SUVC">SUVC</A> MIGC 0.0 ! Group-cent based MI on the slt
<A HREF="#PBCN">PBCN</A> RECT 70.0 75.0 58.0  !Rectangular PBC
<A HREF="#TEMP">TEMP</A> 298
<A HREF="#SVPT">SVPT</A> TIP3 TIP3
<A HREF="#SUPT">SUPT</A> CHRM  ! Solute-solvent potential is CHARMM
<A HREF="#MOLD">MOLD</A> 1 3192 ! Bypass the topology code
<A HREF="#SLTA">SLTA</A> SMPL MMC FILE  3192
!Solute description is read from 5ht1.slt in MMC format
<A HREF="#TRAJ">TRAJ</A> ALLP RGFX ALST !Open trajectory in PDB format
<A HREF="#FILT">FILT</A> SOLV TRAJ ALLP SHL1 RVDW 1.4 0 0 10
!filter out solvents outside the first shell
<A HREF="#TRAJ">TRAJ</A> ALLP RGFX ALST 0 1 11
!Open filtered trajectory
<A HREF="#PXCR">PXCR</A> WDEN BISE
!select proximity criterion type
<A HREF="#PXWR">PXWR</A> ASCI
!Create ASCII proximity information file
<A HREF="#PXBE">PXBE</A> -100.0  2.0
!Calculate solute-solvent energy analysis
<A HREF="#SCAN">SCAN</A> TRAJ 250000 1 5000 0 2000 1000 2  3 1
!Run analysis
<A HREF="#FILE">FILE</A> 5ht1 1    !Reset run number to one
<A HREF="#TRAJ">TRAJ</A> ALLP RGFX ALST 0 1 11 !Open filtered traj
<A HREF="#FILT">FILT</A> ENRG TRAJ ALLP -100.0 -1.0 10
!Filter solvent by the solute-solvent energy
<A HREF="#TRAJ">TRAJ</A> ALLP RGFX ALST 0 1 11
!Open filtered trajectory again
<A HREF="#FILT">FILT</A> ENRG TRAJ ALLP -1.0 100.0 11
!Filter solvent by solute-solvent energy using a different range
<A HREF="#TRAJ">TRAJ</A> ALLP RGFX ALST 0 1 21
!Open trajectory after 1st energy filter
<A HREF="#PXWR">PXWR</A> NONE
<A HREF="#DENF">DENF</A> INSG 0 1 0
!Create single configuration aggregating all solvents
<A HREF="#TRAJ">TRAJ</A> ALLP RGFX ALST 0 1 22
!Open trajectory after 2nd energy filter
<A HREF="#DENF">DENF</A> INSG 0 1 0
!Create single configuration aggregating all solvents
<A HREF="#STOP">STOP</A>
</pre>
Click <A HREF="https://mezeim01.u.hpc.mssm.edu/mmc/examples/exa11.out.html"
TARGET=OUTW>HERE</A> to view the output of this run
<P>
Files needed for run:
<UL>
<LI>5ht1.hst: Trajectory file in PDB format
</UL>
Files created by the run:
<UL>
<LI>5ht1_11.hst: Filtered (1st shell only) trajectory file<BR>
<LI>5ht1.pxi: proximity information file
(see <A HREF="#PXWR">PXWR</A>)<BR>
<LI>5ht1.ckp: checkpoint file<BR>
<LI>5ht1.pxc: proximity analysis checkpoint file<BR>
<LI>5ht1_21.hst: Filtered (1st energy filtering) trajectory
file<BR>
<LI>5ht1_22.hst: Filtered (2nd energy filtering) trajectory
file<BR>
<LI>5ht1_den_21.dat: Aggregated solvents (in Insight free format)
file
from 1st energy filtering<BR>
<LI>5ht1_2.dat: Aggregated solvents (in Insight free format) file
from 2nd energy filtering
</UL>
<!-- HardPage --><p>
<A NAME="Exa_FLT"></A>
IV.12. <B>Conformation filtering</B>
<p>
This example shows the elimination of waters that are not
internal to the solute according to their circular variance.
The filtering is done on a CHARMM CRD file.
<pre>
<A HREF="#FILE">FILE</A> scednas
<A HREF="#PRNT">PRNT</A> ECHO
<A HREF="#SVVC">SVVC</A> SPCC 12.00
<A HREF="#SUVC">SUVC</A> SPGC 20.0
<A HREF="#PBCN">PBCN</A> RECT 99.00  99.00  99.00 ! Large box
<A HREF="#NSLV">NSLV</A> 16426
<A HREF="#SVPT">SVPT</A> TIP3 TIP3 ! Solvent-solvent potential is TIP3
<A HREF="#SUPT">SUPT</A> ATNO ! No potential information
<A HREF="#BRKB">BRKB</A>  32 ~
    92    97    92    99   231   236   231   238   292   297   292   299 ~
   323   328   323   330   404   409   404   411   593  3691   856   861 ~
   856   863  1149  1154  1149  1156  1784  1789  1784  1791  1825  1830 ~
  1825  1832  1882  2138  2030  2035  2030  2037  2090  2095  2090  2097 ~
  2299  2304  2299  2306  2318  2323  2318  2325  2467  2472  2467  2474 ~
  2792  2797  2792  2799
!Break spurious bonds that would confuse the routine establishing
!the molecular connectivities
!Alternative: <A HREF="#MOLD">MOLD</A> 1 3697
<A HREF="#SLTA">SLTA</A> SMPL CRD FILE 3697
<A HREF="#CNFG">CNFG</A> READ CHRM NOFX
<A HREF="#FILT">FILT</A> SOLV CONF CHRM CRCV RSIG 6.0 0.5 0 11
<A HREF="#CNFG">CNFG</A> READ CHRM NOFX
<A HREF="#FILT">FILT</A> SOLV CONF CHRM CRCV RSIG 6.0 0.4 0 12
<A HREF="#STOP">STOP</A>
</pre>
Click <A HREF="https://mezeim01.u.hpc.mssm.edu/mmc/examples/exa12.out.html"
TARGET=OUTW>HERE</A> to view the output of this run
<P>
Files needed for run:
<UL>
<LI>scednas.slt: Solute portion of the CHARMM CRD file
<LI>scednas.CRD: Full configuration (including the solute) in CHARMM CRD format
</UL>
Files created by the run:
<UL>
<LI>scednas_11.CRD: Filtered conformation (keeping only internal waters)
<LI>scednas_12.CRD: Filtered conformation (keeping only internal waters),
weakened filtering (keeping more waters)
</UL>
<!-- HardPage --><p>
<A NAME="Exa_GNS"></A>
IV.13. <B>Generic site calculation</B>
<p>
<pre>
<A HREF="#FILE">FILE</A> GENSfile
<A HREF="#SVVC">SVVC</A> SPCC 10.0  ! Solvent cutoff
<A HREF="#SUVC">SUVC</A> MIGC 0.0   ! MI on the solute
<A HREF="#PBCN">PBCN</A> RECT 56.0011  67.5172  81.0524 !Rectangular PBC
<A HREF="#SVPT">SVPT</A> TIP3 TIP3
<A HREF="#SUPT">SUPT</A> CHRM
<A HREF="#SLTA">SLTA</A> SMPL MMC FILE 5402  0 0 0 1
<A HREF="#TRAJ">TRAJ</A> ALLH FLEX ALST
<A HREF="#GENS">GENS</A> CALC FIXI GETH SPDB FSRT ~
 125 5000000 1 10000 6.5 0.01 0.10 6.0 0.2 100 10
!Number of sites requested (initially): 500
!Number of simulation steps to read: 5000000
!Use every 1-th config (counted in MC stepnumbers)
!Discard the first 10000 MC steps
!For the matching, consider only sites within 6.5 &#197;
!Convergence threshold for the Hungarian method matching: 0.01
!Convergence threshold for the RMSD between iterations: 0.1
!Add a new site if a match is farther than 6.0 &#197;
!Previous site estimate will be mixed in with a factor of 0.2
!Maximum number of iterations over the trajectory: 199
!Minimum number of solvents in a configuration to be of use: 10
<A HREF="#STOP">STOP</A> NOSF
</pre>
Click <A HREF="https://mezeim01.u.hpc.mssm.edu/mmc/examples/exa13.out.html"
TARGET=OUTW>HERE</A> to view the output of this run
<p>
Files needed for run:
<UL>
<LI>GENSfile.slt: Solute portion of the CHARMM CRD file
<LI>GENSfile.CRD: Full configuration (including the solute)
in CHARMM CRD format
<LI>GENSfile.hst: Simulation trajectory in CHARMM CRD format
</UL>
Files created by the run:
<UL>
<LI>GENSfile_gs.pdb: Solute plus the generic sites in PDB format
<LI>GENSfile_gs_cmp.pdb: Solute plus a composite of solvents
that are nearest to sites, in PDB format
<LI>GENSfile_gs_cmp.rep: Solute plus a the most representative configuration
</UL>
from the simulation history, in PDB format
<!-- HardPage --><p>
<A NAME="Exa_PHS"></A>
IV.14. <B>Primary hydration shell calculation</B>
<p>
<pre>
<A HREF="#FILE">FILE</A> alaval
<A HREF="#TITL">TITL</A> compacted alaval
<A HREF="#TITL">TITL</A> PHS, 2 shells
<A HREF="#HRDW">HRDW</A> VC32
<A HREF="#NSLV">NSLV</A> 190       !190 waters
<A HREF="#TEMP">TEMP</A> 298.0
<A HREF="#SUUC">SUUC</A> SPGC 21.00
<A HREF="#SVVC">SVVC</A> SPCC 21.00
<A HREF="#SUVC">SUVC</A> SPGC 21.00
<A HREF="#PBCN">PBCN</A> PHS RSIG UPML 0.15 3.0 8.0 1.0 21.0 0.997 100 1000.0 3.
!PHS boundary conditions
<A HREF="#STEP">STEP</A>    0.00   0.0  0.40      30.0  20
<A HREF="#SUPT">SUPT</A> CHRM
<A HREF="#SVPT">SVPT</A> TIP3 TIP3
<A HREF="#MIXR">MIXR</A> ARGE
<A HREF="#SLTA">SLTA</A> SMPL MMC FILE 56
<A HREF="#SAMP">SAMP</A> METC MOVE SHCY   !Plain Metropolis moves
<A HREF="#PART">PART</A> UNIF SHCY CYCI 0.1          !Solute torsions active
<A HREF="#PRMF">PRMF</A> CHRM  par_all22_prot_na.inp
<A HREF="#TORD">TORD</A> ALL SING 30.0  ! All torsions moved with same stepsize
<A HREF="#CNFG">CNFG</A> READ ASCI
<A HREF="#RUNS">RUNS</A> 200000 5000  100000  90000  9000000  90000000  1
<A HREF="#STOP">STOP</A>
</pre>
Click <A HREF="https://mezeim01.u.hpc.mssm.edu/mmc/examples/exa14.out.html"
TARGET=OUTW>HERE</A> to view the output of this run
<p>
Files needed for run:
<UL>
<LI>alaval.slt: Solute description file
<LI>alaval.crd: Coordinate file of the initial structure
<LI>par_all22_prot_na.inp: Charmm parameter file for torsion potential
description
</UL>
File created by the run:
<UL>
<LI>alaval.ckp: Checkpoint file
</UL>
<!-- HardPage --><p>
<A NAME="Exa_CAV"></A>
IV.15. <B>Cavity and pocket determination</B>
<P>
<pre>
<A HREF="#FILE">FILE</A> f8
<A HREF="#PRNT">PRNT</A> ECHO
<A HREF="#TITL">TITL</A> 1tf_8_13
<A HREF="#NSLV">NSLV</A> 0
<A HREF="#TEMP">TEMP</A> 298.0
<A HREF="#SVVC">SVVC</A> SPCC 9.05     ! Solvent cutoff
<A HREF="#SUUC">SUVC</A> SPCC 12.0
<A HREF="#PBCN">PBCN</A> RECT  70.0 70.0 70.0
<A HREF="#STEP">STEP</A>    0.00   00.0  0.55      40.0  30
<A HREF="#SUPT">SUPT</A> CHRM
<A HREF="#SAMP">SAMP</A> METC 0.5  ! Scaled force-biased sampling
<A HREF="#MOVE">MOVE</A> RAND
<A HREF="#SVPT">SVPT</A> TIP3 TIP3 ! Solvent-solvent potential is TIP4P
<A HREF="#SLTA">SLTA</A> SMPL PDB FILE  3928
<A HREF="#GCEN">GCEN</A> CAVB RSIG ! Grand-canonical ensemble (establishes grid)
    2.0    2.5      1000.0      250  250  250 1000    1    1  010000    20000
<A HREF="#CNFG">CNFG</A> READ PDB
<A HREF="#PRTG">PRTG</A> PDB ALLG AGLS 0 0 0.8 10.0 0.2 1.0 3.3 0.9 0.5 10.0
<A HREF="#STOP">STOP</A> NOSF
</pre>
Click <A HREF="https://mezeim01.u.hpc.mssm.edu/mmc/examples/exa15.out.html"
TARGET=OUTW>HERE</A> to view the output of this run
<p>
Files needed for run:
<UL>
<LI>f8.pdb: Solute description file
</UL>
Files created by the run:
<UL>
<LI>f8_2.pdb: PDB file of solute plus cavity grid coordinates
<LI>f8_3.pdb: PDB file of solute plus pocket grid coordinates
</UL>
<!-- HardPage --><p>
<A NAME = "Brief">V.</A> BRIEF DESCRIPTION OF THE OUTPUT.
<p>
Since the output of the program is largely self-explanatory,
only a relatively brief description is given.
<p>
The output starts with information about the program: self
identification, dimensions used - important for compiling other
programs that read the interrupt file of the program - the dates
the program and the common blocks were modified and the date of
the run.
<p>
By default, the command lines are echoed verbatim.
The command <B><A HREF="#PRNT">PRNT</A> ECHO</B>
allows for direct echoing of the formatted data read as well.
At the start of each run the the options selected are
explained, numbers read are printed with their interpretation and
unit assumed.
These explanations are preceded by the keyword that was the
source of the information, allowing the easy interpretation
of the messages.
<p>
The solute description consists of three parts:
<UL>
<LI>List of atoms (before <A HREF="#CLON">cloning</A>)
and their properties (see below)
<LI>Summary of the properties of residues (charge, radius)
<LI>Summary of the properties of molecules (charge, radius, number of rings)
</UL>
Each atom record consists of the following items
(several of these may be blank):
<UL>
<LI>Sequence number
<LI>Chemical symbol
<LI>Hydrogen-bond donor/acceptor label:
<UL>
<LI>D1: &gt;NH donor
<LI>D2: -OH donor
<LI>A1: &gt;C=O acceptor
<LI>A2: R-O-R acceptor
</UL>
<LI>Potenial code (label and number)
<LI>Potential library (see key <B><A HREF="#SUPT">SUPT</A></B>)
<LI>Atom type label (p: polar; a: aromatic) used for desolvation energy
calculation (see key <B><A HREF="#SVIN">SVIN</A></B>)
<LI>Functional group type
<LI>Marker for group (residue) center: G
<LI>Marker for molecule (segment) center: M
<LI>X, Y, Z coordinates in a representative conformation
<LI>Partial charge
<LI>Lennard-Jones epsilon
<LI>Lennard-Jones sigma
<LI>Molecule number
<LI>Group number
<LI>Mobility marker
<UL>
<LI>FIX: atom is not moved during the simulation
<LI>MOV: atom is moved only by whole solute molecule translation/rotation
(see key <B><A HREF="#PARD">PARD</A></B>)
<LI>TOR: atom is moved by torsion angle change
(see key <B><A HREF="#PART">PART</A></B>)
<LI>INV: Atom stays unchanged between the two topologies
during thermodynamic integration
(see key <B><A HREF="#FREE">FREE</A> TICA</B>)
<LI>SIM: Atom stays nearly unchanged between the two topologies
during thermodynamic integration
(see key <B><A HREF="#FREE">FREE</A> TICA</B>)
</UL>
<LI>Residue name
<LI>Atom name
<LI>First shell radius for proximity analysis
(see key <B><A HREF="#PXCR">PXCR</A></B>) or solute atom volume
(see key <B><A HREF="#SVIN">SVIN</A></B>)
<LI>Proximity group index or
solute atom surface
(see key <B><A HREF="#SVIN">SVIN</A></B>)
<LI>User-defined functional group name
(see key <B><A HREF="#FCGD">FCGD</A></B>)
<LI>User-defined functional group label
(see key <B><A HREF="#FCGD">FCGD</A></B>)
</UL>
<p>
A very important feature of the program is the automatic
self test on continuing a previously interrupted calculation
(<B><A HREF="#RCKP">RCKP</A></B>) as well as at every 2.5 millionth
Monte Carlo step. It can also be called more frequently with the
help of
<B><A HREF="#SLFT">SLFT</A></B>. If the
self test failed a message "Discrepancy found" appears and the
program stops (unless <B><A HREF="#RCKP">RCKP</A> IGND</B> or
<B><A HREF="#RCKP">RCKP</A> FIXD</B> was used).
Discrepancies may arise when the checkpoint file (<B>.ckp</B>)
is corrupted
or in case of a program error.
If the latter is the case,
<A HREF="mailto:Mihaly.Mezei@mssm.edu">
contact the author of the program</A>.
A relatively harmless error can be
an accumulation of round-off errors to unacceptable levels.
Such errors can be prevented using the
<B><A HREF="#FIXD">FIXD</A> FIXD</B> keys.
<p>
The initial total energy and its components, and virial
sum header precedes the periodic output (at each <B>nmcrep</B>-th
step) during the simulation. This periodic output (one to three lines
each) contains the stepnumber (N) the energy of the currently
accepted configuration (E), the cumulative total energy average
(&lt;E&gt;),
the minimum and maximum energy value sampled
(Emn and Emx),
with the configuration number/1000 or /1000000
where it occurred after them
in parenthesis, the solute binding energy (Us) the
acceptance rate (a) of the current move type
(for solute molecule move or torsion the acceptance rate printed
is for the solute molecule or torsion angle sampled in this step)
and the index of the molecule selected for move
(I) and the move type coded as follows.
<UL>
<LI><B>MOV</B>: solvent or whole solute displacement
<LI><B>CPL</B>: coupling parameter change
<LI><B>PSM</B>: solute molecule displacement
<LI><B>TOR</B>: solute torsion change
<LI><B>LP*</B>: solute torsion change with loop move (* is replaced by
the loop type)
<LI><B>SWC</B>: solute molecule swap
<LI><B>2ST</B>: 2-solute molecule correlated rotation
<LI><B>SPC</B>: special move (if implemented)
</UL>
For torsion changes the torsion number is also printed.
For torsion on cloned solute, the torsion number corresponding to the
original torsion list is also given.
For molecule swap, the second molecule's index is also printed.
The line is ended by the symbol <b>A</B> or <b>R</B>
for accepted and rejected moves, resp.
The second line gives the solute intramolecular energy terms
when the solute is not rigid.
An additional line gives grand-canonical ensemble
information: current and average number of solvent molecules, smallest
and largest number of solvent molecules sampled (Nmn, Nmx), the
acceptance rate of the insertion/deletion steps (id acc), the
number of uncovered gridpoints (Ngf) and the probability of
finding a cavity (pcav).
<p>
Periodically (at each <B>nmcplt</B>-th step), and at the end of
each run a complete report of the averages and distributions is
computed and printed. This report starts with the thermodynamic
averages: energy, energy square, standard deviation and range for
energy, total virial sum, solute virial sum and the pressure.This
is followed by the error estimates using the method of batch
means for the total energy or pressure. The error estimates are
given as two standard deviations - corresponding to ca 95%
confidence intervals. Since he method of batch means assumes that
the actual batch means are statistically independent the program
performs a test for independence - the so-called run test. At the
and of this section there are two tables containing the critical
values for the run test: for independence, the number of runs
found should be between the two numbers corresponding to the
number of "ups" (n1) and "downs" (n2) in the two tables. The
program prints CORRELATED or UNCORRELATED when the run test was
failed or passed, respectively and print ??? or &gt;&gt;&gt; when
the number of "ups" or "downs" were either too low or too high for
the table. The test is repeated with double the blocksize until
the actual run lengths prevent further doubling.
<p>
During the run, various informations are gathered relevant
to the convergence of the calculation: Number of times the solute
is attempted to move, accepted to move; average translational and
rotational displacements; the minimum, maximum and average
correlations between the initial and final orientations; the
total displacement of the system (square root of the molecular
displacement square sums), information on the average molecular
displacements, the correlation between two successive moves of a
molecule (averaged over all moves and all molecules) and
convergence indicators involving average molecular displacement
square, acceptance ratio and correlation between successive
moves; if preferential sampling was used
(<B><A HREF="#MOVE">MOVE</A> PRS*</B>) the relative
frequencies the molecules were attempted to move (perturbed),
acceptance rates for individual molecules (useful to detect
"stuck" molecules).
<p>
For the various free energy options the distributions and
free energy values calculated are printed here. Also,
calculations involving the perturbation method print the solute
binding energy estimate on the two limiting state ensembles to
be used for convergence test and the range of energy values
that went into the exponential averages - for reliability these
ranges should not exceed cca 10 kcal/mol.
<p>
For adaptive umbrella sampling, the range covered by each
iteration is plotted with letters U and D indicating that the
probability distribution is increasing or decreasing. In the plot
each line corresponds to an iteration and the position of the U's
and D's refer to the coupling-parameter range sampled.
<p>
The program calculates first order quantum corrections to
the intermolecular free energy with the method of
<A HREF="#Powles">Powles and Rickayzen</A>.
This calculation is based on the computed
configurational average of the force and torque components and on
the mass and moment of inertia of the molecule involved. The
solute contribution to the quantum correction is also computed and
printed, along with the computed force and torque component
squares, molecular mass and moments of inertia.
<p>
The various distribution functions are printed only
optionally and they are self-explanatory. Each distribution is
followed by a summary locating the extrema and other pertinent
characteristics.
<p>
Two more pieces of information relevant to the convergence
of the run are given near the end of the output: the closest
solvent-solvent distance and the number of steps elapsed during
which all molecules were moved at least once.
<p>
Proximity analysis results are printed by the atoms on which
the proximity analysis was requested.
The output may consist of several parts (depending on key
<B><A HREF="#PXPR">PXPR</A></B> and, of course, on the keys
invoking the calculation of the various propeties).
<UL>
<LI>Listing of the calulated distribution functions,
followed by the position and value of their extrema.<BR>
<LI>Tables of various calculated averages,
listed both in the original order of solute atoms and grouped by
residues
and listed by functional groups (either chemical or user-defined
- see key <B><A HREF="#FCGD">FCGD</A></B>)
The table entries for each solute atom (one line) are as follows:
<UL>
<LI>Atom numbers in the current order<BR>
<LI>Atom numbers in the original order<BR>
<LI>Residue number<BR>
<LI>Atom name (read as <B>labslt(1)</B> with
key <B><A HREF="#SLTA">SLTA</A></B>)<BR>
<LI>Residue name
(read as <B>labslt(2)</B> with
key <B><A HREF="#SLTA">SLTA</A></B>)<BR>
<LI>Chemical symbol<BR>
First shell properties:
<UL>
<LI>First shell radius, <b>rfs</b>
(specified by key <B><A HREF="#RFSL">RFSL</A></B>)
in <BR>
<LI>Volume of the first shell, <b>vfs</B>, calculated as the sum of
 
volume elements for shells within <b>rfs</b>
in <sup>3</sup>
(see key <B><A HREF="#GENV">GENV</A></B>)<BR>
<LI>The volume of the first two shells, <b>v2fs</b>,
defined as the volume of the proximity region extending to a
distance
<b>rfs</b>+<b>solvdia</b>, where <b>solvdia</b> is the L-J sigma of
the
<b>islvrep</b>-th (see key <B><A HREF="#SLTA">SLTA</A></B>)
solvent atom
in <sup>3</sup><BR>
<LI>First shell coordination number, <b>&lt;K&gt;</b>, calculated
as the
average number of solvents in the proximity region within
<b>rfs</b><BR>
<LI>First shell density, <b>&lt;K&gt;</b>/<b>vfs</B>, converted to
g/ml<BR>
<LI>First two shell coordination number, <b>&lt;2K&gt;</b>,
calculated as the
average number of solvents in the proximity region within
<b>rfs</b>
<b>solvdia</b><BR>
<LI>First shell binding energy, <b>&lt;sltbe&gt;</b>, the sum of
interactions
of solvents in the first shell with the (whole) solute
(see key <B><A HREF="#PXBE">PXBE</A></B>)<BR>
<LI>First shell pair energy, <b>&lt;sltpe&gt;</b>, calculated as
<b>&lt;sltbe&gt;</b>/<b>&lt;K&gt;</b>
</UL>
Total primary region properties:
<UL>
<LI>Total coordination number, <b>&lt;K&gt;</b>, calculated as the
total number of solvents in the proximity region<BR>
<LI>Total binding energy, <b>&lt;sltbe&gt;</b>, the sum of
interactions
of all solvents in the proximity region
</UL>
First shell solvent-solvent properties:
<UL>
<LI>Coordination number, <b>&lt;Kw&gt;</b> - see key
<B><A HREF="#PXWW">PXWW</A></B><BR>
<LI>Pair energy, <b>&lt;nnwwpe&gt;</b> - see key
<B><A HREF="#PXWW">PXWW</A></B><BR>
<LI>Binding energy, <b>&lt;bewwe&gt;</b> - see key
<B><A HREF="#PXWW">PXWW</A></B>
</UL>
</UL>
<LI>Tables of the calculated extrema of the proximity g(r),
total g(r), and mean dipole
(see key <B><A HREF="#PXDP">PXDP</A></B>) function.
</UL>
<p>
The output concludes with averages of the total energy and
solute binding energy over the run, as well as the coupling
parameter range sampled in the run, if the variable coupling
parameter option was used.
<p>
Throughout the output, lines with periods are used to
delineate certain segments of the output.
Warning messages (indicating potential problem with the input)
are preceded by 5 dashes,
strong warnings (likely, but not certain problems)
are preceded by 5 equal signs,
and error messages are preceded by 5 asterisks.
Messages announcing some
specific action (not taken all the time) are preceded by 5 plus
signs.
Messages announcing that an input setting is overriden by the
program are preceded with five 'less-than' signs.
 
Before stopping the program prints the run time and the number of
overrides, warnings, strong warnings and errors encountered.
It is always a good practice to find
these messages in the output file
to make sure the program was running as intended.
<!-- HardPage --><p>
Minimum critical values for the run test:
<pre>
  n2:  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
 n1  ----------------------------------------------------------
  2 |                                2  2  2  2  2  2  2  2  2 | 2
  3 |              2  2  2  2  2  2  2  2  2  3  3  3  3  3  3 | 3
  4 |           2  2  2  3  3  3  3  3  3  3  3  4  4  4  4  4 | 4
  5 |        2  2  3  3  3  3  3  4  4  4  4  4  4  4  5  5  5 | 5
  6 |     2  2  3  3  3  3  4  4  4  4  5  5  5  5  5  5  6  6 | 6
  7 |     2  2  3  3  3  4  4  5  5  5  5  5  6  6  6  6  6  6 | 7
  8 |     2  3  3  3  4  4  5  5  5  6  6  6  6  6  7  7  7  7 | 8
  9 |     2  3  3  4  4  5  5  5  6  6  6  7  7  7  7  8  8  8 | 9
 10 |     2  3  3  4  5  5  5  6  6  7  7  7  7  8  8  8  8  9 | 10
 11 |     2  3  4  4  5  5  6  6  7  7  7  8  8  8  9  9  9  9 | 11
 12 |  2  2  3  4  4  5  6  6  7  7  7  8  8  8  9  9  9 10 10 | 12
 13 |  2  2  3  4  5  5  6  6  7  7  8  8  9  9  9 10 10 10 10 | 13
 14 |  2  2  3  4  5  5  6  7  7  8  8  9  9  9 10 10 10 11 11 | 14
 15 |  2  3  3  4  5  6  6  7  7  8  8  9  9 10 10 11 11 11 12 | 15
 16 |  2  3  4  4  5  6  6  7  8  8  9  9 10 10 11 11 11 12 12 | 16
 17 |  2  3  4  4  5  6  7  7  8  9  9 10 10 11 11 11 12 12 13 | 17
 18 |  2  3  4  5  5  6  7  8  8  9  9 10 10 11 11 12 12 13 13 | 18
 19 |  2  3  4  5  6  6  7  8  8  9 10 10 11 11 12 12 13 13 13 | 19
 20 |  2  3  4  5  6  6  7  8  9  9 10 10 11 12 12 13 13 13 14 | 20
     ----------------------------------------------------------
       2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
</pre>
<p>
Maximum critical values for the run test:
<pre>
  n2:  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
 n1  ----------------------------------------------------------
  2 |                                                          | 2
  3 |                                                          | 3
  4 |           9  9                                           | 4
  5 |        9 10 10 11 11                                     | 5
  6 |        9 10 11 12 12 13 13 13 13                         | 6
  7 |          11 12 12 12 14 14 14 14 15 15 15                | 7
  8 |          11 12 13 13 14 15 15 16 16 16 16 17 17 17 17 17 | 8
  9 |             13 14 14 15 16 16 16 17 17 18 18 18 18 18 18 | 9
 10 |             13 14 15 16 16 17 17 18 18 18 19 19 19 20 20 | 10
 11 |             13 14 15 16 17 17 18 19 19 19 20 20 20 21 21 | 11
 12 |             13 14 16 16 17 18 19 19 20 20 21 21 21 22 22 | 12
 13 |                15 16 17 18 19 19 20 20 21 21 22 22 23 23 | 13
 14 |                15 16 17 18 19 20 20 21 22 23 23 23 23 24 | 14
 15 |                15 16 18 18 19 20 21 22 22 23 23 24 24 25 | 15
 16 |                   17 18 19 29 21 21 22 23 23 24 25 25 25 | 16
 17 |                   17 18 19 20 21 22 23 23 24 25 25 26 26 | 17
 18 |                   17 18 19 20 21 22 23 24 25 25 26 26 27 | 18
 19 |                   17 18 29 21 22 23 23 24 25 26 26 27 27 | 19
 20 |                   17 18 20 21 22 23 25 25 26 26 27 27 28 | 20
     ----------------------------------------------------------
       2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
</pre>
<!-- HardPage --><p>
<A NAME = "Compilation">VI.</A> COMPILATION OF THE PROGRAM.
<p>
VI.1. Preprocessor
<p>
The program source file <b>mmc.for</b> contains a number of special
symbols allowing the customization of the program.
To produce a valid Fortran source code <b>mmc.f</b>
a preprocessor (also a Fortran program) <b>pre.f</b>
is provided.
The preprocessor performs two functions:
<UL>
<LI> It replaces the special symbols in <b>mmc.for</b>
related to the size of the problem (dimension of arrays, etc.);
quantities that depend on other values
(e.g., the maximum number of atoms)
are calculated by the preprocessor to ensure consistency.
<LI>It selectively activates segments of code depending on
the computational environment (e.g., timing routines),
and/or features planned to be used (to save computation time
by eliminating operations superfluous in certain context).
</UL>
If the generated <b>mmc.f</b> is inadequate for a particular input,
mmc will abort the run and give specific instructions what to change.
<p>
The preprocessor has to be compiled and executed
(after which you can compile the program):
<UL>
<LI>To compile, type <b>f77 -o pre.bin pre.f</b>
(replace f77 with the command invoking the Fortran compiler on
your system, if different)
<LI>To run, type <b>pre.bin</b>,
followed by three &lt;ENTER&gt;'s to accept the default
filename <b>mmc</b>, default choices and values
and to give only a brief list of the options used.
</UL>
The prepocessor as provided has default choices and values that
will create a valid source file. There are three ways to change the defaults:
<UL>
<LI>Running the preprocessor interactively
(by responding with 'y' to the second prompt of <b>pre.f</b>)
<LI>Editing the source file to change the defaults.
This involves
<UL>
<LI>Changing the initialization of selected variables
named iopdef*** (specifying the activation or inactivation of
<A HREF="#PREP">selected functionalities</A>) from 'T' to 'F' (or vice versa)
<LI>Changing the initialization of
selected variables id<size name>
(specifying the size of arrays corresponding to
the selection) to a new value.
The notation <size name> refers to the the name assigned to each
size-related symbol. The names and symbols used are lietse below.
The two quantities most likely to be modified are
<B>maxmol</B> and <B>maxslt</B>.
</UL>
<LI>If there is a running version of MMC the key
<B><A HREF="#PRCO">PRCO</A> SAVE</B> can print a data statement that
- copied to the end of pre.f - will initialize the defaults to be the same
as those of the running version.
</UL>
Instructions for changing the source code of <b>pre.f</b>
are at the beginning of that file.
The part of the prepocessor that may be changed by the user is between the
comment lines 'START Customization' and 'END Customization'.
In the first segment, calls to the subroutine <i>setopt</i>
select functionalities to enable/disable.
In the second (longer) segment,
calls to the subroutine <i>setdim</i> set maximum values for the various
aspects of the system size. Some array sizes are generated automatically
depending on the code segments activated.
<P>
The following list gives the special symbols related to array sizes
appearing in <b>mmc.for</b> with the names used in the preprocessor
and the quantity they limit.
<UL>
<LI><TT>#MO</TT> <B>maxmol</B>: Solvent molecules +1
(has to be even) (see the key <B><A HREF="#NSLV">NSLV</A></B>)
<LI><TT>#MA</TT> <B>maxmolslt</B>: Number of atoms per single
solute molecule
<LI><TT>#SX</TT> <B>mxpxslt</B>: Number solute atoms for proximity
analysis
<LI><TT>#MM</TT> <B>maxsltmol</B>: Number of molecules in the
solute.
<LI><TT>#UW</TT> <B>maxwnnu</B>: Bit map for the neighbors of a
solute atom
<LI><TT>#TN</TT> <B>maxnst</B>: Solute nuclear centers
<LI><TT>#VN</TT> <B>maxnsv</B>: Solvent nuclear centers
<LI><TT>#TE</TT> <B>maxest</B>: Solute electron centers
<LI><TT>#VE</TT> <B>maxesv</B>: Solvent electron centers
<LI><TT>#VW</TT> <B>maxwnnv</B>: Bit map for the neighbors of a molecule
(maxwnnv/(Number of integer word bits -1) + 1.
The integer word size is 32 on most machines,
46 on the CRAY).
<LI><TT>#ST</TT> <B>maxslt</B>: Solute centers (see the key
<B><A HREF="#SLTA">SLTA</A></B>)
<LI><TT>#GR</TT> <B>maxgslt</B>: Solute groups
<LI><TT>#TA</TT> <B>maxtslt</B>: Torsion-affected atoms' list
<LI><TT>#SV</TT> <B>maxslv</B>:  Solvent centers
<LI><TT>#SV</TT> <B>maxslv4</B>:  <B>maxslv</B>-4
<LI><TT>#VT</TT> <B>maxss</B>: Greater of <TT>#SV</TT> and <TT>#ST</TT>
<LI><TT>#NA</TT> <B>maxat</B>: Atoms
<LI><TT>#TL</TT> <B>maxtrgrgr</B>: Torsion-affected group' list
<LI><TT>#GT</TT> <B>maxstg</B>: Solute centers for general
solvent case
<LI><TT>#GV</TT> <B>maxsvg</B>: Solvent centers for general
solvent case
<LI><TT>#DT</TT> <B>maxsst</B>: Solute centers for parameter
derivative
(see key <B><A HREF="#SENS">SENS</A></B>)
and energy decomposition
(see key <B><A HREF="#ENGL">ENGL</A></B>)
calc.
<LI><TT>#DM</TT> <B>maxmst</B>: Molecules for parameter
derivative calculation
<LI><TT>#RG</TT> <B>maxgrid</B>: Radial distribution function
gridpoints (&lt; 505 !)
<LI><TT>#PG</TT> <B>maxpfgr</B>: Preferential weight list length
(RMAX/RI)+1
(RI from RECTYPE 32 - see key</TT> <B><A HREF="#DSTC">DSTC</A></B>)
<LI><TT>#WG</TT> <B>maxcggr</B>: US weighing function gridpoints
<LI><TT>#OR</TT> <B>maxorgr</B>: Number of grid points in the
energy distribution function for overlap ratio method calculations
<LI><TT>#GX</TT> <B>maxxgr</B>: Number of x cavity gridpoints
(&lt;128, even)
<LI><TT>#GY</TT> <B>maxygr</B>: Number of y cavity gridpoints
(&lt;128, even)
<LI><TT>#GZ</TT> <B>maxzgr</B>: Number of z cavity gridpoints
(&gt;31, &lt;128, even)
<LI><TT>#CV</TT> <B>maxcav</B>: Number of 'free' gridpoints for
cavity-biased (if less then #GX*#GY*#GZ
then some test calculations can't be done)
<LI><TT>#W2</TT> <B>maxlin</B>: Work space for n-step optimized
matchings
<LI><TT>#WS</TT> <B>maxausp</B>: Storage space size in adaptive US
<LI><TT>#WI</TT> <B>maxauit</B>: Number of adaptive US iterations + 1
<LI><TT>#MI</TT> <B>maxavit</B>: Number of block average entries
<LI><TT>#TR</TT> <B>maxtors</B>: Torsion list
<LI><TT>#AT</TT> <B>maxatyp</B>: Number of different atomtypes
used
<LI><TT>#UU</TT> <B>maxatypu</B>: Maximum no of atom types in a
given solute
<LI><TT>#UV</TT> <B>maxstmol</B>: Greater of #MO and #ST
<LI><TT>#TG</TT> <B>maxtgrid</B>: Number of total g(r) grid points
<LI><TT>#VG</TT> <B>maxwrgrid</B>: Number of grids for
solvent-solvent g(r)
<LI><TT>#ND</TT> <B>maxgvv</B>: Number of solvent-solvent g(r)'s
<LI><TT>#DG</TT> <B>maxdrgrid</B>: Number of radial grids for
dipole correl. QCDF
<LI><TT>#LG</TT> <B>maxdagrid</B>: Number of angular grids for
dipole correl. QCDF
<LI><TT>#GE</TT> <B>maxpegrid</B>: Number of energy grids for
solute-solvent PE QCDF
<LI><TT>#GQ</TT> <B>mxpxgslt</B>: Number different RDF's and
QCDF's
<LI><TT>#PP</TT> <B>maxcavps</B>: Number of 'free' gridpoints for
cavity-biased
<LI><TT>#PS</TT> <B>maxpfsum</B>: Preferential weight partial sum
list length
<LI><TT>#WM</TT> <B>maxmatch</B>: Number of AUS iterations to
match.
<LI><TT>#TD</TT> <B>maxtgrid</B>: Number of torsion angle
distribution grids
<LI><TT>#FE</TT> <B>mxfeslt</B>: Number of free-energy solute
atoms (all copies)
<LI><TT>#MH</TT> <B>maxhunmat</B>: Number of sites for generic
solvation site calculations
<LI><TT>#NH</TT> <B>maxhmneig</B>: Number of neighbors for
establising a complete match
<LI><TT>#LT</TT> <B>mxtorloop</B>: Number of torsion loops
<LI><TT>#LS</TT> <B>maxloopslt</B>:When torsion loops moves are active,
<b>maxslt</b> otherwise 1
<LI><TT>#MD</TT> <B>mxdiffmol</B>: Number of solvent molecules+1 for
diffusion constant calculations
<LI><TT>#MR</TT> <B>mxrescmol</B>: Number of solvent molecules+1 for
residence time calculations
<LI><TT>#DC</TT> <B>mxdiffcr</B>: Number of structures to use
in diffusion calc
<LI><TT>#RC</TT> <B>mxrescr</B>: Number of structures to use
in residence time calc
<LI><TT>#MW</TT> <B>mxwidslt</B>: Number of Widom solute molecules
<LI><TT>#MS</TT> <B>mxphsmol</B>: Number of solvent molecules+1 for
<LI><TT>#MG</TT> <B>maxmolfg</B>: Number of solvent molecules+1 to
calculate field gradients for
<B><A HREF="#PBCN">PHS </A></B> calculations
<LI><TT>#HA</TT> <B>maxath</B>: Number of atoms for representing
all sites (maxslt+maxslv*maxhunsite)
<LI><TT>#GM</TT> <B>maxmapgrid</B>: Number of gridpoints in the energy map
<LI><TT>#GH</TT> <B>maxhbgrid</B>: Number of gridpoints in the hydrogen-bond map
<LI><TT>#NE</TT> <B>maxmap</B>: Number of atoms to save for minimum energy
configuration
<LI><TT>#AU</TT> <B>maxtorauc</B>: Number of angles to save for torsion angle
autocorrelation calculation
<LI><TT>#GC</TT> <B>maxgrdclst</B>: The larger of the number of grid clusters fo
cavity/pocket calculations and of the number of sites used for the various site
analysis (key <B><A HREF="#GSAN">GSAN</A></B>) calculations
<LI><TT>#RN</TT> <B>maxrandinp</B>: The maximum number of random numbers
that can be read
<LI><TT>#NL</TT> <B>maxnnlist</B>: The maximum number of 1st + 2nd + 3rd neighbo
</UL>
<p>
Since the increased memory usage taxes the
computer resources (both memory and disk-space),
it is usually worth recompiling the program
separately for significantly different system sizes
(see the key <B><A HREF="#PRCO">PRCO</A> SVMN</B> for
preparing a minimum-sized executable).
Remember,
however, that the checkpoint file (<B>.ckp</B>) is only readable by
a program that was compiled with the same dimensioning!
<p>
<A NAME="PREP"></A>
Code segment that can be selectively activated contains lines with
the comment <b>C@**</b>.
The selection and removal of such lines is controlled by a number of
character variables in pre.f that can be set during the customization
of the preprocessor. Except for the variable <B>SYSTEM</B>, the value is
either <B>'T'</B> or <B>'F'</B>.
The following variables can be set:
<UL>
<LI><TT><B>SYSTEM</B> {<B>'IRIX'</B>}</TT> - Choices:
  <UL>
   <LI><TT><B>'IRIX'</B> - C@UG</TT> (SGI Irix)
   <LI><TT><B>'AIX '</B> - C@UR</TT> (IBM RS600)
   <LI><TT><B>'HP  '</B> - C@HP</TT> (Hewlett-Packard)
   <LI><TT><B>'MPDM'</B> - C@DM</TT> (MPI - Distributed Memory),
   <LI><TT><B>'G77 '</B> - C@G7</TT> (Gnu Fortran-77),
   <LI><TT><B>'G95 '</B> - C@G9</TT> (Gnu Fortran-95),
   <LI><TT><B>'EFC '</B> - C@EF</TT> (Intel Fortran),
   <LI><TT><B>'ABSF'</B> - C@EF</TT> (Absoft Fortran90),
   <LI><TT><B>'UNIX'</B> - C@UX</TT> (Generic Unix)
  </UL>
<LI><TT><B>CB</B> {<B>'T'</B>} - C@CB</TT>:
Cavity-biased (T,V,mu) ensemble code is active (inactive when 'F')
- see key <B><A HREF="#GCEN">GCEN</A></B>
<LI><TT><B>IB</B> {<B>'T'</B>} - C@IB</TT>: (T,P,N) ensemble code is
active
- see key <B><A HREF="#IBEN">IBEN</A></B>
<LI><TT><B>FE</B> {<B>'T'</B>} - #FE </TT>: Free-energy simulation arrays are
active
- see key <B><A HREF="#FREE">FREE</A></B>
<LI><TT><B>MS</B> {<B>'T'</B>}</TT>: Any intersolute move is
active - see key <B><A HREF="#PARD">PARD</A></B>
or <B><A HREF="#PART">PART</A></B>
<LI><TT><B>TR</B> {<B>'T'</B>} - #TR </TT>: Torsion move arrays are
active
- see key <B><A HREF="#PART">PART</A></B>
<LI><TT><B>LO</B> {<B>'T'</B>} - #LT </TT>: Torsion loop move arrays are
active - see key <B><A HREF="#LOOP">LOOP</A></B>
<LI><TT><B>PX</B> {<B>'T'</B>} - #SX </TT>: Proximity analysis code is
active - see key <B><A HREF="#PXCR">PXCR</A></B>
<LI><TT><B>FG</B> {<B>'F'</B>} - C@FG</TT>: Field-gradient calculation code is
inactive - see keys <B><A HREF="#PRTG">PRTG</A></B>
<LI><TT><B>PG</B> {<B>'F'</B>} - C@PG</TT>: Cavity grid analysis code is
inactive - see keys <B><A HREF="#PRTG">PRTG</A></B>
or <B><A HREF="#STVG">STVG</A></B>
<LI><TT><B>HU</B> {<B>'F'</B>} - #MH </TT>: Generic solvation site code is
inactive - see key <B><A HREF="#GENS">GENS</A></B>
<LI><TT><B>RF</B> {<B>'F'</B>} - C@RF</TT>: Reaction field correction code is
inactive
<LI><TT><B>DD</B> {<B>'F'</B>} - C@DD</TT>: Distance-dependent dielectric
code is inactive
<LI><TT><B>DD</B> {<B>'F'</B>} - C@1R</TT>: eps(r)=r dielectric
code is inactive
<LI><TT><B>PH</B> {<B>'F'</B>} - #MS </TT>: Primary hydration shell
(<B><A HREF="#PBCN">PHS </A></B>) code is inactive (active when 'T')
- see key <B><A HREF="#RFCR">RFCR</A></B>
<LI><TT><B>ED</B> {<B>'F'</B>} - #DT </TT>: Energy decomposition
(<B><A HREF="#ENGL">ENGL</A></B>) key arrays are inactive (active when 'T')
<LI><TT><B>DM</B> {<B>'F'</B>} - C@DM </TT>: MPI communication library
calls are inactive (active when 'T') - for now only a very limited
set of functions are available.
<LI><TT><B>DR</B> {<B>'F'</B>} - #MD </TT>:
Diffusion constant and residence time calculation code is inactive
(active when 'T') - see keys <B><A HREF="#DIFC">DIFC</A></B>
or <B><A HREF="#RTIM">RTIM</A></B>
<LI><TT><B>FR</B> {<B>'T'</B>} - C@FR</TT>: Solvent force calculation code is
active (inactive when 'F') (used for force-biasing and pressure
calculation) - see key <B><A HREF="#SAMP">SAMP</A></B>.
Calculation of forces can be inhibited
to save time when only proximity analysis is
run or when force-biasing is not used and the pressure is of no
interest.
<LI><TT><B>TS</B> {<B>'T'</B>} - C@TS</TT>: Solute torque calculation code is
active (inactive when 'F') (for whole solute rot)
If the solute is not going to be rotated or no force bias
is applied on the solute (<B><A HREF="#NFBU">NFBU</A></B>) then
the solute torque calculation is superfluous
and can be left out, resulting in some savings (significant for
large solutes) in computer time.
<LI><TT><B>NN</B> {<B>'T'</B>} - C@NN</TT>: Solvent-solvent near-neighbor map is
active
Generally, the near-neighbor code has to be left out is when (1)
the solvent-solvent cutoff is close to the inside sphere radius
of the central cell, or (2) minimum image convention is used for
solvent-solvent solvent interactions
or (3) there are no or few solvent molecules.
<LI><TT><B>TN</B> {<B>'T'</B>} - C@TN</TT>: Solute-solvent near-neighbor map is
active
It can be set to 'F' when the solute is completely rigid
(i.e., neither <B><A HREF="#PARD">PARD</A></B>
nor <B><A HREF="#PART">PART</A></B>) are active).
<LI><TT><B>NA</B> {<B>'T'</B>} - C@NA</TT>: Near-neighbor map arithmetic code
(logical code when 'F') is active.<BR>
The bitmap operations used both for the near-neighbour table
and for keeping track of intramolecular interactions to be used
can be performed either by logical or arithmetis operations.
The relative efficiency (when there is a choice) varies
from machine to machine.
Also, the code involving logical operations uses equivalence statements
that are planned to be phased out in futore Fortran versions.
Furthermore, using logical operations works only if the compiler performs
the logical operation on all bits.
If it is not the case (e.g., the Silicon Graphics),
<B>NA</B> should be set to <B>'T'</B>.
<LI><TT><B>VC</B> {<B>'T'</B>} - C@VC</TT>: Vectorizable (scalar when F)
routine call code is active
<LI><TT><B>I2</B> {<B>'F'</B>} - C@I2</TT>:
Integer*2 array use for cavity grids is inactive
- see key <B><A HREF="#GCEN">GCEN</A></B>.
For grand-canonical ensemble calculations requiring a
relatively small number of free grids
(&lt; 2<sup>15</sup>-1 ~ 30,000) but a
large number of gridpoints setting <BI2</b> to <B>'T'</B> can
save a few megabytes of memory.
<LI><TT><B>PS</B> {<B>'F'</B>} - C@PS</TT>:
Code amenable to autoparallelization
is active (so far, if failed to yield speedup).
<LI><TT><B>QP</B> {<B>'F'</B>} - #TN </TT>: QPEN/EPEN potential arrays are
inactive.
</UL>
Activating some of the codes requires deactivating a complementary code.
Such code pairs are:<BR>
<TT><B>C@NL</B> -  <B>C@NA</B></TT><BR>
<TT><B>C@VC</B> -  <B>C@NC</B></TT><BR>
<TT><B>C@DM</B> -  <B>C@ND</B></TT><BR>
<TT><B>C@1R</B> -  <B>C@NR</B></TT><BR>
<TT><B>C@RF</B> -  <B>C@1R</B></TT><BR>
The preprocessor takes care of it automatically.
<p>
Besides the activating and inactivating lines prefixed with C@**,
the preprocessor provides the option to change all
double precision (real*8) variables to quadruple precision variables (real*16).
This option can be useful for calculations using the
<B><A HREF="#LOOP">LOOP</A></B> key.
Note, however, that not all compilers support
quadruple precision variables.
<p>
The defaults for variables <B>FG</B>, <B>PG</B>, <B>HU</B>, and <B>DR</B>
are set to <B>'F'</B> because the corresponding code requires
large arrays and <B>RF</B> is set to <B>'F'</B>
because it adds code to the energy calculations.
<p>
The key <B><A HREF="#PRCO">PRCO</A></B> produces an annotated list
of the values used for the currently executing version as well as
of the code segment selection choices made.
A quick listing on the terminal can be obtained by typing
<b>mmc.bin</b>, followed by the command
<B><A HREF="#PRCO">PRCO</A></B>
at the MMC&gt; prompt (followed by <B><A HREF="#STOP">STOP</A></B>
at the next prompt for a graceful exit).
<p>
If a compilation is needed to match a given checkpoint file
the  command <B><A HREF="#PRCO">PRCO</A> SAVE</B>
will read the information of the current run's checkpoint file
and write Fortran data statements to be inserted into
<b>pre.f</b>
<p>
VI.2. Compilation
<p>
Once the <b>mmc.f</b> file is prepared it should be compiled
with the appropriate Fortran compilation
command, to produce an executable, say <b>mmc.bin</b>, e.g.,
<p>
<b>f77 -o mmc.bin mmc.f</B>.
<p>
Compiler options:
<UL>
<LI>For test runs use the <b>-C</b>
(to inhibit optimization and enable index boundary check)
<LI>For production runs, high(est?) level optimization should be invoked,
e.g., <b>-Ofast</b> (SGI); <b>-O4</b> (Compaq Alpha)
<LI>For compatibility of binary trajectories note that the default
byte order on Linux and SGI is "<b>big endian</b>".
if your compiler's deafault 
 is "<b>little endian</b>" you can compile an executable with the
Linux/SGI byte order using the added directive "<b>-byteswapio</b>" on Linux .
<LI>For runs requiring the MPI communication library, the compilation should
include linking instructions to the library and running the program
generally requires the use of a system-dependent script
- consult your system manager for the apropriate syntax.
</UL>
<p>
Possible problems:
<p>
For some systems, the source code is too big to compile it in one step.
In this case, the program <b>splitmmc.f</b> (part of the distribution)
should be compiled and run.
This will result in five files: <b>mmc1.f</b>, ..., <b>mmc5.f</b>.
Each one should be compiled separately with the <b>-c</b> option, i.e.,<BR>
&gt; <b>f77 -c mmc1.f</b><BR>
followed by <BR>
&gt; <b>f77 -o mmc.bin mmc*.o</b>
<p>
Some compilers fail due to a so-called 'relocation error' when optimizing
at levels higher than one is asked.
When using the Intel Fortran compiler (ifort), adding the compiler directives<br>
<b>-mcmodel=medium -share_intel</b><br>
solved the problem. With some of the other compiler (but not the GNU compiler)
the compilation key
<b> -fpic</b> was found to solve the problem.
<p>
VI.3. Parallelization
<p>
There are a number of limitations in the parallelized version of MMC.
In particular, it is extremely inefficient (due to latency issues)
on distributed memory systems
so it should be run either on shared-memory systems, or
limit the number of CPU's to the nmumber of procesors on a single node
and make sure the the queuing system places each job on the same node.
<P>
The following has been parallelized: solvent-solvent energy calculations;
solute-solvent energy calculations; Widom-insertion free-energy calculations,
and grand-canonical ensemble (GCE) simulations.
However, force-calculations have to be disabled.
Also, use any solute move (keys
<B><A HREF="#PARD">PARD</A></B> or <B><A HREF="#PART">PART</A></B>)
would reduce the parallel efficiency.
<p>
<A NAME = "Testing">VIII.</A> TESTING THE PROGRAM
<p>
A test suite has been compiled to allow a systematic check on the
integrity of the program.
The tests are located in the directory <B>testsuite</B>.
There is a c-shell script <b>test.csh</b>.
It runs all the tests and examines the output.
For each test, it prints different messages
if there were input errors, inconcistency errors or no errors.
For runs with special solute moves, it also prints the acceptance
rates.
The program is considered passing the test if there are no errors
and the acceptance rates printed are all positive.
Furthermore, if there is a subdirectory oldout containing
reference output files, the new output will be compared to the
old one. Differences should only show in CPU times and dates.
Clearly, this is only meaningful if the reference output was
generated by the same compiler and compilation options.
The table below gives a summary of the features tested by each test
system.
<PRE>
 Nmolslt Nwat  PARD PART SWAP TICA PRM PRM GCEN IBEN SUUCm CLON PMLI PROX WIDO
                                       Ft  Dr
1   1     215
2   1     215
3   3      0     x   x     x
4   7      0     x         x
5   2      15                   x
6   2     484    x   x
7   1      0         x
8   4     484                            x
9   6     484                            x
10  1      9                                 x
11  6      0     x   x     x                                 x
12  6      0     x                                           x
13  2      0         x
14  2      0         x          x
15  2     484    x   x
16  3     484    x         x
17  1      4                                 x
18  3     484    x
19  3     693                        x
20  1                x
21  2      50    x                           x
22  2      50    x                                x
23  2      50                                     x
24  2      50    x   x                            x
25  3      0     x   x     x                           x
26  3     484    x   x                     x
27  3     500                                                    x
28  2     484                                                         x
29  4     215    x   x         x
30  1      0     x   x
31  1     190        x
32  1     107                                                              x
  Nmlslt Nwat  PARD PART SWAP TICA PRM PRM GCEN IBEN SUUCm CLON PMLI PROX WIDO
                                       Ft  Dr
</PRE>
 
<!-- HardPage --><p>
<A NAME = "References">IX.</A> REFERENCES
<p>
<A NAME="Adams">Grand-canonical ensemble formalism:
D.J. Adams</A>, Mol. Phys, <B>29</B>,  307 (1975).
<p>
<A NAME="Ben-Naim">QCDF formalism:
A. Ben-Naim</A>, Water and aqueous
solutions, Plenum, New York (1974).
<p>
<A NAME="Jacucci">Overlap ratio method:
Jacucci</A> and N. Quirke, Mol. Phys., <B>40</B>,
1005 (1980)
<p>
<A NAME="Marchese">EPEN/QPEN potential:
F. Marchese</A>, P.K. Mehrotra and D.L.
Beveridge, J. Phys. Chem., <B>85</B>, 1 (1981).
<p>
<A NAME="Mehrotra">Proximity criterion formalism:
P.K. Mehrotra</A> and D.L. Beveridge, J. Am.
Chem. Soc., <B>102</B>,4287 (1980).
<p>
<A NAME="Metropolis">Original description of the Metropolis method:
N. Metropolis</A> et al., J. Chem. Phys.,
<B>21</B>, 1087 (1953),
<p>
<A NAME="Mezei_81">Shuffled cyclic selection of molecules to be
moved:
M. Mezei</A>, J. Comp. Phys., <B>39</B>, 128 (1981)
<BR><CENTER>Click <A
HREF="https://mezeim01.u.hpc.mssm.edu/ms/cv23mmc.pdf">HERE</A>
to download a copy</CENTER>
<p>
<A NAME="Mezei_82">Cavity-biased insertion - original formalism:
M. Mezei</A>, Mol. Phys., <B>47</B>, 1307 (1982)
<p>
<A NAME="Mezei_83">Virial-biased volume change in the (T,P,N)
ensemble:
M. Mezei</A>, Mol. Phys., <B>48</B>, 1075 (1983)
<p>
<A NAME="Mezei_86">Description of self tests:
par_all27_prot_na_crb.acly.crbn.prmM. Mezei</A>, CCP5 Newsletter, Daresbury
Laboratory, No. 23, (1986)
<BR><CENTER>Click <A
HREF="https://mezeim01.u.hpc.mssm.edu/ms/cv57ccp5.pdf">HERE</A>
to download a copy</CENTER>
<p>
<A NAME="Mezei_86_NYAS">Free energy methodology review:
M. Mezei</A> and D.L. Beveridge, Ann. Acad.
NY Sci., <B>482</B>, 1, (1986).
<BR><CENTER>Click <A HREF="https://mezeim01.u.hpc.mssm.edu/ms/cv54mmc.pdf">HERE</A>
to download a copy</CENTER>
<p>
<A NAME="Mezei_87">Cavity-biased insertion - grid-based formalism:
M. Mezei</A>, Mol. Phys., <B>61</B>, 565-582
(1987); Erratum, <B>67</B>,1207-1208 (1989).
<BR><CENTER>Click <A
HREF="https://mezeim01.u.hpc.mssm.edu/ms/cv58mmc.pdf">HERE</A>
to download a copy</CENTER>
<p>
<A NAME="Mezei_87_AUS">Adaptive Umbrella sampling:
M. Mezei</A>, J. Comp. Phys. <B>68</B>, 237 (1987)
<BR><CENTER>Click <A
HREF="https://mezeim01.u.hpc.mssm.edu/ms/cv53mmc.pdf">HERE</A>
to download a copy</CENTER>
<p>
<A NAME="Mezei_87_FDTI">Finite-difference thermodynamic
integration:
M. Mezei</A>, J. Chem. Phys. <B>86</B>, 7084 (1987)
<BR><CENTER>Click <A
HREF="https://mezeim01.u.hpc.mssm.edu/ms/cv60mmc.pdf">HERE</A>
to download a copy</CENTER>
<p>
<A NAME="Mezei_88">Radical-plane based proximity criterion:
M. Mezei</A>, Molecular Simulation, <B>1</B>,
327 (1988).
<BR><CENTER>Click <A
HREF="https://mezeim01.u.hpc.mssm.edu/ms/cv64mmc.pdf">HERE</A>
to download a copy</CENTER>
<p>
<A NAME="Mezei_88_BM">Using a bitmap for keeping neighbour lists:
M. Mezei</A>, Molecular Simulation, <B>1</B>,
169 (1988).
<BR><CENTER>Click <A HREF="https://mezeim01.u.hpc.mssm.edu/ms/cv63mmc.pdf">HERE</A>
to download a copy</CENTER>
<p>
<A NAME="Mezei_90">Scaled force bias method:
M. Mezei</A>, Molecular Simulation, <B>5</B>,
405 (1990)
<BR><CENTER>Click <A
HREF="https://mezeim01.u.hpc.mssm.edu/ms/cv72mmc.pdf">HERE</A>
to download a copy</CENTER>
<p>
<A NAME="Mezei_92">Polynomial Thermodynamic Integration
for the solvation free energy of liquid water:
M. Mezei</A>, J. Comput. Chem., <B>13</B>,
(1992) 651;
<P>
<A NAME="Mezei_94">Iso-energy cutoff for ion-ion PMF:
M. Mezei</A>, Int. J. Quantum Chem., <B>52</B>,
147 (1994).
<p>
<A NAME="Owicki">Preferential sampling near the solute:
J.C. Owicki</A>, Computer Modelling of Matter,
P.G. Lykos, ed., ACS, Washington, D.C., (1978), p 159.
<p>
<A NAME="Pangali">Force-bias sampling:
C. Pangali</A>, Rao and B.J. Berne, Chem. Phys.
Lett., <B>55</B>, 413 (1978), Mol. Phys., <B>37</B>, 1773 (1979)
<p>
<A NAME="Powles">Quantum corrections:
Powles</A> and Rickayzen, Mol. Phys., <B>38</B>,
1875 (1979).
<P>
<A NAME="Resat">Polynomial TI
for the solvation free energy of the alanine dipeptede:
H. Resat</A> and M. Mezei, J. Chem. Phys.,
<B>99</B>, (1993) 6052.
<BR><CENTER>Click <A
HREF="https://mezeim01.u.hpc.mssm.edu/ms/cv85.pdf">HERE</A>
to download a copy</CENTER>
<p>
<A NAME="Rossky">Smart MC (a force-biasing variant):
P.J. Rossky</A>, J.D. Doll and H.L. Friedman, J.
Chem. Phys., <B>69</B>, 4628 (1978).
<p>
<A NAME="Gensite">Generic solvation sites</a>:
M. Mezei and D.L. Beveridge,
J. Comput. Chem., <B>5</B>, 523 (1984).
(1992) 651;
<BR><CENTER>Click <A HREF="https://mezeim01.u.hpc.mssm.edu/ms/cv43mmc.pdf">HERE</A>
to download a copy</CENTER>
<p>
<A NAME="PHSref">Primary Hydration Shell</a>:
D. Beglov and B. Roux, Biopolymers., <B>35,</b> 171 (1995).
<BR>
A. Kentsis, M. Mezei, and R. Osman,
Biophys. J, <b>84,</B> 805-815 (2003).
<p>
<A NAME="TSALLIS">Sampling with Tsallis statistics</a>:
I. Andricioaei and J.E. Straub, <B>53,</b> 3055 (1996).
<p>
<A NAME="WIDOref">Cavity-biased Widom insertions</a>:
P. Jedlovszky, and M. Mezei, J. Am. Chem. Soc.,
<B>122,</B> 5125-5131 (2000).
<p>
<A NAME="EXBIref">Extension-biased torsion moves</a>:
P. Jedlovszky, and M. Mezei, J. Chem. Phys.,
<B>111,</B> 10770-10773 (1999).
<BR><CENTER>Click <A HREF="https://mezeim01.u.hpc.mssm.edu/ms/cv106mmc.pdf">HERE</A>
to download a copy</CENTER>
<p>
<A NAME="LOOPref">Local torsion moves</a>:
M. Mezei, J. Chem. Phys.,
<B>118,</b> 3874-3879 (2003).
<p>
<A NAME="BTUNEref">Automatic tuning of the B-parameter</a>:
J.A. Speidel, J.R. Banfelder, and M. Mezei, J. Chem. Theory and Comp.,
<B>,</b>  (2006).
<p>
<A NAME="STUNEref">Automatic tuning of the stepsize parameters</a>:
J.A. Speidel, J.R. Banfelder, and M. Mezei,
Algorithms, <B>2,</B> 215-226 (2009).
<p>
<A NAME="SIMref">Simulaid: a simulation facilitator and analysis program</a>
J. Comput. Chem., <B>31,</B> 2658-2668 (2010).
<p>
<A HREF="#Contents">Return to contents page</A><P>
</BODY>
</HTML>
 
